# Manipulating Tree with Data {#chapter2}


```{r include=FALSE}
library(knitr)
opts_chunk$set(message=FALSE, warning=FALSE, eval=TRUE, echo=TRUE, cache=TRUE)
```



```{r echo=FALSE, results="hide", message=FALSE}
library("ape")
library("tidytree")
library(treeio)
```

## Manipulating tree data using tidy interface {#tidytree}

All the tree data parsed/merged
by [treeio](https://bioconductor.org/packages/treeio/) [@wang_treeio_2020] can be converted to tidy
data frame using the [tidytree](https://cran.r-project.org/package=tidytree)
package. The [tidytree](https://cran.r-project.org/package=tidytree) package
provides tidy interfaces to manipulate tree with associated data. For instances,
external data can be linked to phylogeny or evolutionary data obtained from
different sources can be merged using tidyverse verbs. After the tree data was
manipulated, it can be converted back to `treedata` object and exported to [a
single tree file](#chapter3), further analyzed in R or visualized using [ggtree](https://bioconductor.org/packages/ggtree/) [@yu_ggtree:_2017].


### The `phylo` object


The `phylo` class defined in the [ape](https://cran.r-project.org/package=ape) package [@paradis_ape_2004] is
fundamental for phylogenetic analysis in R. Most of the R packages
in [this field](https://CRAN.R-project.org/view=Phylogenetics) rely
extensively on the `phylo` object. The `r CRANpkg("tidytree")` package provides `as_tibble`
method to convert the `phylo` object to tidy data frame, a `tbl_tree` object.

```{r}
library(ape)
set.seed(2017)
tree <- rtree(4)
tree
x <- as_tibble(tree)
x
```

The `tbl_tree` object can be converted back to a `phylo` object using the `as.phylo()` method.


```{r}
as.phylo(x)
```


Using `tbl_tree` object makes tree and data manipulation more effective and
easier (see also example in [FAQ](#bind-tip)). For example, we can link evolutionary trait to phylogeny using the `r CRANpkg("dplyr")` verbs `full_join`:

```{r}
d <- tibble(label = paste0('t', 1:4),
            trait = rnorm(4))

y <- full_join(x, d, by = 'label')
y
```

### The `treedata` object


The `r CRANpkg("tidytree")` package defines `treedata` class to store phylogenetic tree with
associated data. After mapping external data to the tree structure, the
`tbl_tree` object can be converted to a `treedata` object.


```{r}
as.treedata(y)
```

The `treedata` class is also used
in the [treeio](https://bioconductor.org/packages/treeio/) package [@wang_treeio_2020] to store evolutionary evidences inferred by commonly used software (BEAST, EPA, HYPHY,
MrBayes, PAML, PHYLDOG, pplacer, r8s, RAxML and RevBayes, etc.) (see details in [Chapter 1](#chapter1)).

The `r CRANpkg("tidytree")` package also provides the `as_tibble()` method to convert a `treedata` object to a tidy data frame. The phylogenetic tree structure and the evolutionary
inferences were stored in the `tbl_tree` object, making it consistent and easier
for manipulating evolutionary statistics inferred by different software as well
as linking external data to the same tree structure.


```{r}
y %>% as.treedata %>% as_tibble
```



### Access related nodes {#accesor-tidytree}

The [dplyr](https://cran.r-project.org/package=dplyr) verbs can be applied to `tbl_tree` directly to manipulate tree data. In addition, `r CRANpkg("tidytree")` provides several verbs to filter related nodes, including
`child`, `parent`, `offspring`, `ancestor`, `sibling` and `MRCA`.

These verbs accept a `tbl_tree` and a selected node which can be node number or label.


```{r}
child(y, 5)
parent(y, 2)
offspring(y, 5)
ancestor(y, 2)
sibling(y, 2)
MRCA(y, 2, 3)
```

All these methods also implemented in `r Biocpkg("treeio")` for working with `phylo` and `treedata` objects. You can try accessing related nodes using the tree object. For instance, the following command will output child nodes of the selected internal node `5`:

```{r}
child(tree, 5)
```

Beware that the methods work for tree objects output related node numbers, while the methods implemented for `tbl_tree` object output a `tibble` object that contains related information.


## Data Integration

### Combining tree data {#merge-tree}


The [treeio](https://bioconductor.org/packages/treeio/) package [@wang_treeio_2020] serves as an
infrastructure that enables various types of phylogenetic data inferred from
common analysis programs to be imported and used in R. For instance *d~N~/d~S~*
or ancestral sequences estimated
by [CODEML](http://abacus.gene.ucl.ac.uk/software/paml.html),
and clade support values (posterior) inferred
by [BEAST](http://beast2.org/)/[MrBayes](http://nbisweden.github.io/MrBayes/).
In addition, [treeio](https://bioconductor.org/packages/treeio/) supports linking external data to phylogeny. It brings these external
phylogenetic data (either from software output or exteranl sources) to the R
community and make it available for further analysis in R.
Furthermore, [treeio](https://bioconductor.org/packages/treeio/) can combine
multiple phylogenetic trees together into one with their node/branch-specific
attribute data. Essentially, as a result, one such attribute (*e.g.*,
substitution rate) can be mapped to another attribute (*e.g.*, *d~N~/d~S~*) of
the same node/branch for comparison and further computations [@yu_ggtree:_2017].


A previously published data set, seventy-six H3 hemagglutinin gene sequences of
a lineage containing swine and human influenza A viruses
[@liang_expansion_2014], was here to demonstrate the utilities of comparing
evolutionary statistics inferred by different software. The dataset was
re-analyzed by [BEAST](http://beast2.org/) for timescale estimation
and [CODEML](http://abacus.gene.ucl.ac.uk/software/paml.html) for synonymous and
non-synonymous substitution estimation. In this example, we first parsed the
outputs from [BEAST](http://beast2.org/) using the `read.beast()` function and
from [CODEML](http://abacus.gene.ucl.ac.uk/software/paml.html) using
the `read.codeml()` function into two `treedata` objects. Then these two objects containing separate sets of node/branch-specific data were merged via the `merge_tree()` function.



```{r}
beast_file <- system.file("examples/MCC_FluA_H3.tree", package="ggtree")
rst_file <- system.file("examples/rst", package="ggtree")
mlc_file <- system.file("examples/mlc", package="ggtree")
beast_tree <- read.beast(beast_file)
codeml_tree <- read.codeml(rst_file, mlc_file)

merged_tree <- merge_tree(beast_tree, codeml_tree)
merged_tree
```

After merging the `beast_tree` and `codeml_tree` objects, all
node/branch-specific data imported from [BEAST](http://beast2.org/)
and [CODEML](http://abacus.gene.ucl.ac.uk/software/paml.html) output files are
all available in the `merged_tree` object. The tree object was converted to a
tidy data frame using the [tidytree](https://cran.r-project.org/package=tidytree)
package and visualized as hexbin scatterplot of *d~N~/d~S~*, *d~N~* and *d~S~* inferred
by [CODEML](http://abacus.gene.ucl.ac.uk/software/paml.html) versus *rate*
(substitution rate in unit of substitutions/site/year) inferred
by [BEAST](http://beast2.org/) on the same branches.


(ref:correlationscap) Correlation of *d~N~/d~S~*, *d~N~* and *d~S~* versus substitution rate.

(ref:correlationcap) **Correlation of *d~N~/d~S~*, *d~N~* and *d~S~* versus substitution rate.** After merging the *BEAST* and *CodeML* outputs, the branch-specific estimates (substitution rate, *d~N~/d~S~* , *d~N~* and *d~S~*) from the two analysis programs are compared on the same branch basis. The associations of *d~N~/d~S~*, *d~N~* and *d~S~* vs. *rate* are visualized in hexbin scatter plots.


```{r correlations, fig.width=9, fig.height=3, warning=FALSE, fig.cap="(ref:correlationcap)", fig.scap="(ref:correlationscap)", out.extra='', out.width="100%"}
library(dplyr)
df <- merged_tree %>% 
  as_tibble() %>%
  select(dN_vs_dS, dN, dS, rate) %>%
  subset(dN_vs_dS >=0 & dN_vs_dS <= 1.5) %>%
  tidyr::gather(type, value, dN_vs_dS:dS)
df$type[df$type == 'dN_vs_dS'] <- 'dN/dS'
df$type <- factor(df$type, levels=c("dN/dS", "dN", "dS"))
ggplot(df, aes(rate, value)) + geom_hex() + 
  facet_wrap(~type, scale='free_y') 
```


The output is illustrated in Fig. \@ref(fig:correlations). We can then test the association of these node/branch-specific data using Pearson correlation, which in this case showed that *d~N~* and *d~S~*, but not *d~N~/d~S~*\index{d\textsubscript{N}/d\textsubscript{S}}  are significantly (*p*-values) associated with *rate*.


Using the `merge_tree()` function, we are able to compare analysis results using identical
model from different software packages or different models using different or
identical software. It also allows users to integrate different analysis finding
from different software packages. Merging tree data is not restricted to
software findings, associating external data to analysis findings is also
granted. The `merge_tree()` function is chainable and allows several tree objects
to be merged into one.


```{r}
phylo <- as.phylo(beast_tree)
N <- Nnode2(phylo)
d <- tibble(node = 1:N, fake_trait = rnorm(N), another_trait = runif(N))
fake_tree <- treedata(phylo = phylo, data = d)
triple_tree <- merge_tree(merged_tree, fake_tree)
triple_tree
```

The `triple_tree` object showed above contains analysis results obtained from [BEAST](http://beast2.org/)
and [CODEML](http://abacus.gene.ucl.ac.uk/software/paml.html), and evolutionary
trait from external sources. All these information can be used to annotate the
tree using [ggtree](https://bioconductor.org/packages/ggtree/) [@yu_ggtree:_2017].




### Linking external data to phylogeny {#link-external-data}


In addition to analysis findings that are associated with the tree as we showed
above, there is a wide range of heterogeneous data, including phenotypic data,
experimental data and clinical data *etc.*, that need to be integrated and
linked to phylogeny. For example, in the study of viral evolution, tree nodes may
associated with epidemiological information, such as location, age and subtype.
Functional annotations may need to be mapped on gene trees for comparative
genomics studies. To facilitate data
integration, [treeio](https://bioconductor.org/packages/treeio) provides
`full_join` method to link external data to phylogeny and stored in either `phylo` or `treedata` object. Beware that linking external data to a `phylo` object will produce a `treedata` object to store the input `phylo` with associated data. The `full_join` methods can also be used at tidy data frame level (*i.e.* `tbl_tree` object described previously) and at `ggtree` level (described in [session 7.1](#attach-operator)) [@yu_two_2018].


The following example calculated bootstrap values and merging that values with the tree (a `phylo` object) by matching their node numbers.

```{r apeBoot, message=FALSE}
library(ape)
data(woodmouse)
d <- dist.dna(woodmouse)
tr <- nj(d)
bp <- boot.phylo(tr, woodmouse, function(x) nj(dist.dna(x)))
bp2 <- tibble(node=1:Nnode(tr) + Ntip(tr), bootstrap = bp)
full_join(tr, bp2, by="node")
```

Another example demonstrates merging evolutionary trait with the tree (a `treedata` object) by matching their tip labels.


```{r}
file <- system.file("extdata/BEAST", "beast_mcc.tree", package="treeio")
beast <- read.beast(file)
x <- tibble(label = as.phylo(beast)$tip.label, trait = rnorm(Ntip(beast)))
full_join(beast, x, by="label")
```

Manipulating tree object is frustrated with the fragmented functions available
for working with `phylo` object, not to mention linking external data to the
phylogeny structure. With the `r Biocpkg("treeio")` package [@wang_treeio_2020], it is easy to combine tree data from various sources. 
In addition, with `r CRANpkg("tidytree")` package ([session 2.1](#tidytree)), manipulating tree is more easier using 
[tidy data principles](https://www.jstatsoft.org/article/view/v059i10) and 
consistent with tools already in wide use, including
[dplyr](https://cran.r-project.org/package=dplyr),
[tidyr](https://cran.r-project.org/package=tidyr),
[ggplot2](https://cran.r-project.org/package=ggplot2)
and [ggtree](https://bioconductor.org/packages/ggtree/) [@yu_ggtree:_2017].


### Grouping taxa

The `groupOTU()` and `groupClade()` methods are designed for adding taxa grouping
information to the input tree object. The methods were implemented in `r CRANpkg("tidytree")`,
`r Biocpkg("treeio")` and `r Biocpkg("ggtree")` respectively to support adding grouping information at the 
`tbl_tree`, `phylo` and `treedata`, and `ggtree` levels. These grouping information can be
used directly in tree visualization (*e.g.* [coloring tree based on grouping](#group-taxa-vis))
with [ggtree](https://bioconductor.org/packages/ggtree/) (Figure \@ref(fig:groupOTU)).



#### groupClade

The `groupClade()` method accepts an internal node or a vector of internal nodes
to add grouping information of selected clade/clades.



```{r}
nwk <- '(((((((A:4,B:4):6,C:5):8,D:6):3,E:21):10,((F:4,G:12):14,H:8):13):13,((I:5,J:2):30,(K:11,L:11):2):17):4,M:56);'
tree <- read.tree(text=nwk)

groupClade(as_tibble(tree), c(17, 21))
```


#### groupOTU


```{r}
set.seed(2017)
tr <- rtree(4)
x <- as_tibble(tr)
## the input nodes can be node ID or label
groupOTU(x, c('t1', 't4'), group_name = "fake_group")
```

Both `groupClade()` and `groupOTU()` work with the `tbl_tree`, `phylo` and `treedata`, and `ggtree` objects. Here is an example of using `groupOTU()` with a `phylo` tree object.

```{r}
groupOTU(tr, c('t2', 't4'), group_name = "fake_group") %>%
  as_tibble
```

Another example of working with the `ggtree` object can be found in [session 6.5](#group-taxa-vis).


The `groupOTU` will trace back from input nodes to most recent common ancestor.
In this example, nodes 1, 4, 5 and 6 are
grouping together (`4 (t2) -> 6 -> 5` and `1 (t4) -> 5`).

Related OTUs are grouping together and they are not necessarily within a clade.
They can be monophyletic (clade), polyphyletic or paraphyletic.

```{r}
cls <- list(c1=c("A", "B", "C", "D", "E"),
            c2=c("F", "G", "H"),
            c3=c("L", "K", "I", "J"),
            c4="M")

as_tibble(tree) %>% groupOTU(cls)
```

If there are conflicts when tracing back to most recent common ancestor, user can set `overlap`
parameter to "origin" (the first one counts), "overwrite" (default, the last one
counts) or "abandon" (un-selected for grouping)^[<https://groups.google.com/forum/#!msg/bioc-ggtree/Q4LnwoTf1DM/uqYdYB_VBAAJ>].


## Rerooting tree {#reroot-treeio}

Rerooting is a frequently used operation in various tree editors, such as [Figtree](https://github.com/rambaut/figtree), [TreeGraph 2](http://treegraph.bioinfweb.info/) have this function. The [root](https://rdrr.io/cran/ape/man/root.html) function of [ape](https://cran.r-project.org/web/packages/ape/index.html) allows you to reroot your tree in R.

(ref:rerootingscap) Rerooting tree in R.

(ref:rerootingscap) **Rerooting tree with root of ape.** The tree rerooted by [root](https://rdrr.io/cran/ape/man/root.html) function

```{r rerooting, fig.width=12, fig.height=4.5, message=F, echo=T, fig.cap="(ref:rerootingscap)", fig.scap="(ref:rerootingscap)", out.extra='', out.width="100%"}
set.seed(2020)
library(ape)
library(ggtree)

tree <- rtree(8)

# you can reroot with a leaf
tree_root <- root(tree, "t3")

# you can reroot with a outgroup
tree_root <- root(tree, c("t3","t8"))

ggtree(tree_root) + geom_tiplab()
```

But one user found that this result is somewhat different from the one shown by Figtree in this [question](https://groups.google.com/g/bioc-ggtree/c/f8TeyYUD-J4/m/fboQOXlLAQAJ). We can export the above tree to a file and see what it looks like in Figtree.

```{r}
write.tree(tree,file = "tree.nwk")
```

(ref:rootFigtreescap) Rerooting tree in FigTree.

(ref:rootFigtreescap) **Rerooting tree in FigTree.** Rerooting tree in `FigTree` is easy, just select a branch and click `Reroot` in top left.

```{r rootFigtree, fig.width=15.1, fig.height=11.4, echo=FALSE, fig.cap="(ref:rootFigtreescap)", fig.scap="(ref:rootFigtreescap)", out.width="100%"}
p = magick::image_read("img/figtree_reroot.png")
ggplotify::as.ggplot(p)
```
The tree graph after the [root](https://rdrr.io/cran/ape/man/root.html) function reroot loses the branch length before outgroup (Figure \@ref(fig:rerootingscap)), This branch length exists in Figtree (Figure \@ref(fig:rootFigtreescap)) and equal to another branch after root node. 

Based on the above conclusions, we can optimize the results of [root](https://rdrr.io/cran/ape/man/root.html) function so that the branch length will not lost.

(ref:root2scap) Optimize the results of root function.

(ref:root2scap) **Optimize the results of root function.** The tree rerooted by [root](https://rdrr.io/cran/ape/man/root.html) function and modified the branch length before outgroup. These operations are integrated into the `root2` function.

```{r root2scap, fig.width=12, fig.height=4.5, message=F, echo=T, fig.cap="(ref:root2scap)", fig.scap="(ref:root2scap)", out.extra='', out.width="100%"}
root2 <- function(tree, rootlab) {
  #  tree is tree your need to reroot
  #  rootlab is outgroup or node in function root of ape
  tree_root <- ape::root(tree,rootlab)
  # convert tree to data table
  tree_data <- tidytree::as_tibble(tree_root)
  # get branch.length between root node
  root_num <- dplyr::filter(tree_data, parent == node)$node
  length <- max(dplyr::filter(tree_data, parent == root_num)$branch.length,na.rm = TRUE)
  
  # change branch.length between root node to node 59
  if (length > 0) {
    tree_data_out <- dplyr::mutate(tree_data, branch.length = ifelse(parent == root_num & branch.length == 0, 
                                                        length, branch.length))
  }  else {
    tree_data_out <- tree_data
  }
  # convert data table to tree
  tree_out <- ape::as.phylo(tree_data_out)
  return(tree_out)
}

tree_root2 <- root2(tree,c("t3","t8"))

ggtree(tree_root2) + geom_tiplab()
```

## Rescaling Tree Branches {#rescale-treeio}

Phylogenetic data can be merged for joint analysis (Figure \@ref(fig:correlations)). They can be displayed on the same tree structure as more complex annotation to help visually inspection of their evolutionary patterns. All the numerical data stored in `treedata` object can be used to re-scale tree branches. For example, CodeML infers d~N~/d~S~, d~N~ and d~S~, all these statistics can be used as branch lengths. All these values can also be used to color the tree (session [4.3.4](#color-tree)) and can be project to vertical dimension to create two-dimensional tree or phenogram (session [4.2.2](#layouts-of-phylogenetic-tree) and Figure \@ref(fig:2d) and \@ref(fig:continuousColor)). Instead of modifying branch lengths in the tree object, user can directly specifying a variable as branch length in `ggtree()` as demonstrated in [session 4.3.6](#rescale-tree). 



(ref:rescalescap) Re-scaling tree branches.

(ref:rescalecap) **Re-scaling tree branches.** The tree with branches scaled in time (year from the root) (A). The tree was re-scaled using *d~N~* as branch lengths (B). The tree was re-scaled using substitution rates (C).

```{r rescale, fig.width=12, fig.height=4.5, message=F, echo=T, fig.cap="(ref:rescalecap)", fig.scap="(ref:rescalescap)", out.extra='', out.width="100%"}
p1 <- ggtree(merged_tree) + theme_tree2()
p2 <- ggtree(rescale_tree(merged_tree, 'dN')) + theme_tree2()
p3 <- ggtree(rescale_tree(merged_tree, 'rate')) + theme_tree2()

cowplot::plot_grid(p1, p2, p3, ncol=3, labels = LETTERS[1:3])
```


## Subsetting Tree with Data

### Removing tips in a phylogenetic tree {#remove-tip}

Sometimes we want to remove selected tips from a phylogenetic tree. This is due to several reasons, including low sequence quality, errors in sequence assembly, an alignment error in part of the sequence and an error in phylogenetic inference *etc*.

Let's say that we want to remove three tips (colored by red) from the tree (Figure \@ref(fig:removeTip)A), the `drop.tip()` method removes specified tips and update the tree (Figure \@ref(fig:removeTip)B). All associated data will be maintained in the updated tree. 

(ref:removeTipscap) Removing tips from tree.

(ref:removeTipcap) **Removing tips from tree.** Original tree with three tips (colored by red) to remove (A). Updated tree that removed selected tips (B).


```{r removeTip, fig.width=10, fig.height=6, fig.cap="(ref:removeTipcap)", fig.scap="(ref:removeTipscap)", out.width="100%"}
f <- system.file("extdata/NHX", "phyldog.nhx", package="treeio")
nhx <- read.nhx(f)
to_drop <- c("Physonect_sp_@2066767",
            "Lychnagalma_utricularia@2253871",
            "Kephyes_ovata@2606431")
p1 <- ggtree(nhx) + geom_tiplab(aes(color = label %in% to_drop)) +
  scale_color_manual(values=c("black", "red")) + xlim(0, 0.8)

nhx_reduced <- drop.tip(nhx, to_drop)
p2 <- ggtree(nhx_reduced) + geom_tiplab() + xlim(0, 0.8)  
plot_grid(p1, p2, ncol=2, labels = c("A", "B"))
```


### Subsetting tree by tip label {#subset-tip}

Tree can be large and difficult to look at only the portions of interest. The `tree_subset()` function was created in the `r Biocpkg("treeio")` package [@wang_treeio_2020] to extract a subset of the tree portion while still maintaining the structure of the tree portion. The `beast_tree` in Figure \@ref(fig:subsetTip)A is slightly crowded. Obviously, we can make the figure taller to allow more space for the labels (similar to use "Expansion" slider in `FigTree`) or we can make the text smaller. However, these solutions are not always applicable when you have a lot of tips (*e.g.* hundreds or thousands of tips). In particular, when you are only interested in the portion of the tree around a particular tip, you certainly don't want to explore a large tree to find centain species you are interested in.

Let's say you are interested in tip *A/Swine/HK/168/2012* from the tree (Figure \@ref(fig:subsetTip)A) and you want to look at the immediate relatives of this tip. 

The `tree_subset()` function allows you to look at the portions of the tree that are of interest. By default, the `tree_subset()` function will internally call the [`groupOTU()`](#groupotu) to assign group specified tip from the rest of other tips (Figure \@ref(fig:subsetTip)B). Additionally, the branch lengths and related associated data are maintained after subsetting (Figure \@ref(fig:subsetTip)C). The root of the tree is always anchored at zero for the subset tree by default and all the distances are relative to this root. If you want all the distances are relative to the original root, you can specify the root position (by `root.position` parameter) to the root edge of the subset tree, which is the sum of branch lengths from the original root to the root of the subset tree (Figure \@ref(fig:subsetTip)D and E).

(ref:subsetTipscap) Subsetting tree for specific tip.

(ref:subsetTipcap) **Subsetting tree for specific tip.** The original tree (A). The subset tree (B). Subset tree with data (C). Visualize the subset tree relative to original position, without root edge (D) and with root edge (E).

```{r subsetTip, fig.width=12, fig.height=7, echo=T, fig.cap="(ref:subsetTipcap)", fig.scap="(ref:subsetTipscap)", out.extra='', out.width="100%"}
beast_file <- system.file("examples/MCC_FluA_H3.tree", package="ggtree")
beast_tree <- read.beast(beast_file)

p1 = ggtree(beast_tree) + 
  geom_tiplab() +  xlim(0, 40) + theme_tree2()

tree2 = tree_subset(beast_tree, "A/Swine/HK/168/2012", levels_back=4)  
p2 <- ggtree(tree2, aes(color=group)) +
  scale_color_manual(values = c("black", "red")) +
  geom_tiplab() +  xlim(0, 4) + theme_tree2() 

p3 <- ggtree(tree2, aes(color=group)) +
  geom_tiplab(hjust = -.1) + xlim(0, 5) + 
  geom_point(aes(fill = rate), shape = 21, size = 4) +
  scale_color_manual(values = c("black", "red"), guide = FALSE) +
  scale_fill_continuous(low = 'blue', high = 'red') +
  theme_tree2() + theme(legend.position = 'right')


p4 <- ggtree(tree2, aes(color=group), 
          root.position = as.phylo(tree2)$root.edge) +
  geom_tiplab() + xlim(18, 24) + 
  scale_color_manual(values = c("black", "red")) +
  theme_tree2()

p5 <- ggtree(tree2, aes(color=group), 
          root.position = as.phylo(tree2)$root.edge) +
  geom_rootedge() + geom_tiplab() + xlim(0, 40) + 
  scale_color_manual(values = c("black", "red")) +
  theme_tree2()

plot_grid(p2, p3, p4, p5, ncol=2, labels=LETTERS[2:5]) %>%
  plot_grid(p1, ., ncol=2, labels=c("A", ""), rel_widths=c(.5, 1))
```

### Subsetting tree by internal node number {#subset-node}

If you are interesting at certain clade, you can specify the input node as an internal node number. The `tree_subset()` function will take the clade as a whole and also trace back to particular levels to look at the immediate relatives of the clade (Figure \@ref(fig:subsetNode)A and B). We can use the `tree_subset()` function to zoom in selected portions and plot a whole tree with the portion of it, that is similar to the `ape::zoom()` function to explore very large tree (Figure \@ref(fig:subsetNode)C and D). Users can also use `viewClade()` function to restrict tree visualization at specific clade as demonstrated in [session 6.1](#viewing-selected-clade).


(ref:subsetNodescap) Subsetting tree for specific clade.

(ref:ssubsetNodecap) **Subsetting tree for specific clade.** Extracting a clade (A). Extracting a clade and trace back to look at its immediate relatives (B). Viewing a very large tree (C) and a selected portion of it (D).

```{r subsetNode, fig.width=10, fig.height=10, echo=T, fig.cap="(ref:ssubsetNodecap)", fig.scap="(ref:subsetNodescap)", out.extra='', out.width="100%"}
clade <- tree_subset(beast_tree, node=121, levels_back=0)
clade2 <- tree_subset(beast_tree, node=121, levels_back=2)
p1 <- ggtree(clade) + geom_tiplab() + xlim(0, 5)
p2 <- ggtree(clade2, aes(color=group)) + geom_tiplab() + 
  xlim(0, 8) + scale_color_manual(values=c("black", "red"))


library(ape)
library(tidytree)
library(treeio)

data(chiroptera)

nodes <- grep("Plecotus", chiroptera$tip.label)
chiroptera <- groupOTU(chiroptera, nodes)

clade <- MRCA(chiroptera, nodes)
x <- tree_subset(chiroptera, clade, levels_back = 0)

p3 <- ggtree(chiroptera, aes(colour = group)) + 
  scale_color_manual(values=c("black", "red")) +
  theme(legend.position = "none")
p4 <- ggtree(x) + geom_tiplab() + xlim(0, 5)
plot_grid(p1, p2, p3, p4, 
  ncol=2, labels=LETTERS[1:4])
```  



## Manipulating tree data for visualization {#ggtree-fortify}

Tree visualization is supported by [ggtree](#chapter4) [@yu_ggtree:_2017]. Although `r Biocpkg("ggtree")` implemented several methods for [visual exploration of tree with data](#chapter6), you may want to do something that is not supported directly. In this case, you need to manipulate tree data with node coordination positions that used for visualization. This is quite easy with `r Biocpkg("ggtree")`. User can use the `fortify()` method which internally call `tidytree::as_tibble()` to convert the tree to a tidy data frame and add columns of coordination positions (*i.e.* x, y, branch and angle) that are used to plot the tree. You can also access the data via `ggtree(tree)$data`.

Here is an example to plot two trees face to face that is similar to a `ape::cophyloplot()`.



(ref:cophyloscap) Plot two phylogenetic trees face to face.

(ref:cophylocap) **Plot two phylogenetic trees face to face.** Plotting a tree using `ggtree()` (left hand side) and subsequently add another layer of tree by `geom_tree()` (right hand side). The relative positions of the plotted trees can be manual adjusted and adding layers to each of the tree (*e.g.* tip labels and highlighting clades) is independent.

```{r cophylo, fig.width=8, fig.height=6, message=F, echo=T, fig.cap="(ref:cophylocap)", fig.scap="(ref:cophyloscap)", out.extra='', out.width="100%"}
library(dplyr)
library(ggtree)

set.seed(1024)
x <- rtree(30)
y <- rtree(30)
p1 <- ggtree(x, layout='roundrect') + 
  geom_hilight(
         mapping=aes(subset = node %in% c(38, 48, 58, 36),
                     node = node,
                     fill = as.factor(node)
                     )
     ) +
    labs(fill = "clades for tree in left" )

p2 <- ggtree(y)

d1 <- p1$data
d2 <- p2$data

## reverse x-axis and 
## set offset to make the tree in the right hand side of the first tree
d2$x <- max(d2$x) - d2$x + max(d1$x) + 1

pp <- p1 + geom_tree(data=d2, layout='ellipse') +      
  ggnewscale::new_scale_fill() +
  geom_hilight(
         data = d2, 
         mapping = aes( 
            subset = node %in% c(38, 48, 58),
            node=node,
            fill=as.factor(node))
  ) +
  labs(fill = "clades for tree in right" ) 

dd <- bind_rows(d1, d2) %>% 
  filter(!is.na(label))

pp + geom_line(aes(x, y, group=label), data=dd, color='grey') +
    geom_tiplab(geom = 'shadowtext', bg.colour = alpha('firebrick', .5)) +
    geom_tiplab(data = d2, hjust=1, geom = 'shadowtext', bg.colour = alpha('firebrick', .5))
```

It is quite easy to plot multiple trees and connect taxa in one figure. For instance, plotting trees contructed from all internal gene segments of influenza virus and connecting equivalent strans across the trees [@venkatesh_avian_2018]. 


(ref:cophylo3scap) Plot multiple phylogenetic trees side by side.

(ref:cophylo3cap) **Plot multiple phylogenetic trees side by side.** Plotting a tree using `ggtree()` and subsequently add multiple layers of trees by `geom_tree()`.

```{r cophylo3, fig.width=8, fig.height=6, message=F, echo=T, fig.cap="(ref:cophylo3cap)", fig.scap="(ref:cophylo3scap)", out.extra='', out.width="100%"}
z <- rtree(30)
d2 <- fortify(y)
d3 <- fortify(z)
d2$x <- d2$x + max(d1$x) + 1
d3$x <- d3$x + max(d2$x) + 1

dd = bind_rows(d1, d2, d3) %>% 
  filter(!is.na(label))

p1 + geom_tree(data = d2) + geom_tree(data = d3) + geom_tiplab(data=d3) + 
  geom_line(aes(x, y, group=label, color=node < 15), data=dd, alpha=.3)
```



## Summary {#summary2}

The `r Biocpkg("treeio")` package allows us to import diverse phylogeny associated data into R. However, phylogenetic tree is stored in a way to facilitate computational processing which is not human fridenly and need expertise to manipulate and explore tree data. The `r CRANpkg("tidytree")` package provides tidy interface for exploring tree data, while `r Biocpkg("ggtree")` provides a set of utilitise to visualize and explore tree data using grammar of graphics. This full suit of packages make it easy for ordinary users to interact with tree data, and allow us to integrate phylogeny associated data from different sources (*e.g.* experimental result or analysis finding), which creates the possibility of comparative study. 

