<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>About the Author | Data Integration, Manipulation and Visualization of Phylogenetic Trees</title>
<meta name="author" content="Guangchuang Yu">
<meta name="description" content="Guangchuang Yu (https://yulab-smu.top) is a professor of Bioinformatics and director of the Department of Bioinformatics at Southern Medical University. He earned his Ph.D. from the School of...">
<meta name="generator" content="bookdown 0.27 with bs4_book()">
<meta property="og:title" content="About the Author | Data Integration, Manipulation and Visualization of Phylogenetic Trees">
<meta property="og:type" content="book">
<meta property="og:image" content="/book-cover.png">
<meta property="og:description" content="Guangchuang Yu (https://yulab-smu.top) is a professor of Bioinformatics and director of the Department of Bioinformatics at Southern Medical University. He earned his Ph.D. from the School of...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="About the Author | Data Integration, Manipulation and Visualization of Phylogenetic Trees">
<meta name="twitter:description" content="Guangchuang Yu (https://yulab-smu.top) is a professor of Bioinformatics and director of the Department of Bioinformatics at Southern Medical University. He earned his Ph.D. from the School of...">
<meta name="twitter:image" content="/book-cover.png">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap/bootstrap.bundle.min.js"></script><script src="libs/bs3compat/transition.js"></script><script src="libs/bs3compat/tabs.js"></script><script src="libs/bs3compat/bs3compat.js"></script><link href="libs/bs4_book/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book/bs4_book.js"></script><script src="libs/kePrint/kePrint.js"></script><link href="libs/lightable/lightable.css" rel="stylesheet">
<script src="libs/htmlwidgets/htmlwidgets.js"></script><script src="libs/pathseg/pathseg.js"></script><script src="libs/comic/comic.min.js"></script><script src="libs/comicR-binding/comicR.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<style type="text/css">
    /* Used with Pandoc 2.11+ new --citeproc when CSL is used */
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
        }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Data Integration, Manipulation and Visualization of Phylogenetic Trees</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">Preface</a></li>
<li><a class="active" href="author.html">About the Author</a></li>
<li class="book-part">Part I: Tree data input, output, and manipulation</li>
<li><a class="" href="chapter1.html"><span class="header-section-number">1</span> Importing Tree with Data</a></li>
<li><a class="" href="chapter2.html"><span class="header-section-number">2</span> Manipulating Tree with Data</a></li>
<li><a class="" href="chapter3.html"><span class="header-section-number">3</span> Exporting tree with data</a></li>
<li class="book-part">Part II: Tree data visualization and annotation</li>
<li><a class="" href="chapter4.html"><span class="header-section-number">4</span> Phylogenetic Tree Visualization</a></li>
<li><a class="" href="chapter5.html"><span class="header-section-number">5</span> Phylogenetic Tree Annotation</a></li>
<li><a class="" href="chapter6.html"><span class="header-section-number">6</span> Visual Exploration of Phylogenetic Trees</a></li>
<li><a class="" href="chapter7.html"><span class="header-section-number">7</span> Plotting tree with data</a></li>
<li><a class="" href="chapter8.html"><span class="header-section-number">8</span> Annotating Tree with Silhouette Images and Sub-plots</a></li>
<li><a class="" href="chapter9.html"><span class="header-section-number">9</span> ggtree for Other Tree-like Objects</a></li>
<li class="book-part">Part III: ggtree extensions</li>
<li><a class="" href="chapter10.html"><span class="header-section-number">10</span> ggtreeExtra for Presenting Data on a Circular Layout</a></li>
<li><a class="" href="chapter11.html"><span class="header-section-number">11</span> Other ggtree Extensions</a></li>
<li class="book-part">Part IV: Miscellaneous topics</li>
<li><a class="" href="chapter12.html"><span class="header-section-number">12</span> ggtree Utilities</a></li>
<li><a class="" href="chapter13.html"><span class="header-section-number">13</span> Gallery of Reproducible Examples</a></li>
<li class="book-part">Appendix</li>
<li><a class="" href="faq.html"><span class="header-section-number">A</span> Frequently Asked Questions</a></li>
<li><a class="" href="related-tools.html"><span class="header-section-number">B</span> Related Tools</a></li>
<li><a class="" href="figures-and-tables.html"><span class="header-section-number">C</span> Figures and Tables</a></li>
<li><a class="" href="publications-of-the-ggtree-package-suite.html"><span class="header-section-number">D</span> Publications of the ggtree Package Suite</a></li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="author" class="section level1 unnumbered">
<h1>About the Author<a class="anchor" aria-label="anchor" href="#author"><i class="fas fa-link"></i></a>
</h1>
<p><strong>Guangchuang Yu</strong> (<a href="https://yulab-smu.top" class="uri">https://yulab-smu.top</a>) is a professor of Bioinformatics and director of the Department of Bioinformatics at Southern Medical University. He earned his Ph.D. from the School of Public Health, The University of Hong Kong. As an active R user, he has authored several R packages, such as <a href="https://CRAN.R-project.org/package=aplot"><strong>aplot</strong></a>, <a href="https://CRAN.R-project.org/package=badger"><strong>badger</strong></a>, <a href="http://bioconductor.org/packages/ChIPseeker"><strong>ChIPseeker</strong></a>, <a href="http://bioconductor.org/packages/clusterProfiler"><strong>clusterProfiler</strong></a>, <a href="http://bioconductor.org/packages/DOSE"><strong>DOSE</strong></a>, <a href="https://CRAN.R-project.org/package=emojifont"><strong>emojifont</strong></a>, <a href="http://bioconductor.org/packages/enrichplot"><strong>enrichplot</strong></a>, <a href="https://CRAN.R-project.org/package=ggbreak"><strong>ggbreak</strong></a>, <a href="https://CRAN.R-project.org/package=ggfun"><strong>ggfun</strong></a>, <a href="https://CRAN.R-project.org/package=ggimage"><strong>ggimage</strong></a>, <a href="https://CRAN.R-project.org/package=ggplotify"><strong>ggplotify</strong></a>, <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a>, <a href="http://bioconductor.org/packages/GOSemSim"><strong>GOSemSim</strong></a>, <a href="https://CRAN.R-project.org/package=hexSticker"><strong>hexSticker</strong></a>, <a href="https://CRAN.R-project.org/package=meme"><strong>meme</strong></a>, <a href="http://bioconductor.org/packages/meshes"><strong>meshes</strong></a>, <a href="https://CRAN.R-project.org/package=nCov2019"><strong>nCov2019</strong></a>, <a href="https://CRAN.R-project.org/package=plotbb"><strong>plotbb</strong></a>, <a href="http://bioconductor.org/packages/ReactomePA"><strong>ReactomePA</strong></a>, <a href="https://CRAN.R-project.org/package=scatterpie"><strong>scatterpie</strong></a>, <a href="https://CRAN.R-project.org/package=seqmagick"><strong>seqmagick</strong></a>, <a href="http://bioconductor.org/packages/seqcombo"><strong>seqcombo</strong></a>, <a href="https://CRAN.R-project.org/package=shadowtext"><strong>shadowtext</strong></a>, <a href="https://CRAN.R-project.org/package=tidytree"><strong>tidytree</strong></a> and <a href="http://bioconductor.org/packages/treeio"><strong>treeio</strong></a>. He has supervised post-graduate students to develop a few other packages, including <a href="http://bioconductor.org/packages/ggmsa"><strong>ggmsa</strong></a>, <a href="http://bioconductor.org/packages/ggtreeExtra"><strong>ggtreeExtra</strong></a>, <a href="http://bioconductor.org/packages/MicrobiomeProfiler"><strong>MicrobiomeProfiler</strong></a> and <a href="http://bioconductor.org/packages/MicrobiotaProcess"><strong>MicrobiotaProcess</strong></a>.</p>
<p>His research group aims to generate new insights into human health and disease through the development of new software tools and novel analysis of biomedical data. The software package developed by his research group helps biologists analyze data and reveal biological clues hidden in the data.</p>
<p>He has published several journal articles, including 5 highly cited papers <span class="citation">(<a href="references.html#ref-yu_ggtree:_2017" role="doc-biblioref">Yu et al., 2017</a>, <a href="references.html#ref-yu2012" role="doc-biblioref">2012</a>; <a href="references.html#ref-yu_dose_2015" role="doc-biblioref">Yu, Wang, Yan, et al., 2015</a>; <a href="references.html#ref-yu_chipseeker:_2015" role="doc-biblioref">Yu, Wang, &amp; He, 2015</a>; <a href="references.html#ref-yu_reactomepa_2016" role="doc-biblioref">Yu &amp; He, 2016</a>)</span>. The articles have been cited more than 10,000 times. The ggtree <span class="citation">(<a href="references.html#ref-yu_ggtree:_2017" role="doc-biblioref">Yu et al., 2017</a>)</span> paper was selected as a feature article to celebrate the 10<sup>th</sup> anniversary of the launch of <em>Methods in Ecology and Evolution</em><a class="footnote-ref" tabindex="0" data-toggle="popover" data-content='&lt;p&gt;10th Anniversary Volume 8: Phylogenetic tree visualization with multivariate data: &lt;a href="https://methodsblog.com/2020/11/19/ggtree-tree-visualization/" class="uri"&gt;https://methodsblog.com/2020/11/19/ggtree-tree-visualization/&lt;/a&gt;&lt;/p&gt;'><sup>2</sup></a>. He was one of the 2020 Highly Cited Chinese Researchers (Elsevier-Scopus) in Biomedical Engineering.</p>

</div>



<div class="chapter-nav">
<div class="prev"><a href="index.html">Preface</a></div>
<div class="next"><a href="chapter1.html"><span class="header-section-number">1</span> Importing Tree with Data</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav"><li><a class="nav-link" href="#author">About the Author</a></li></ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>
</div>

  

  

</div>
 <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Data Integration, Manipulation and Visualization of Phylogenetic Trees</strong>" was written by Guangchuang Yu. It was last built on 2022-07-13.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
