<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 2 Manipulating Tree with Data | Data Integration, Manipulation and Visualization of Phylogenetic Trees with R</title>
<meta name="author" content="Guangchuang Yu">
<meta name="description" content="2.1 Manipulating Tree Data Using Tidy Interface All the tree data parsed/merged by treeio (Wang et al., 2020) can be converted to a tidy data frame using the tidytree package. The tidytree package...">
<meta name="generator" content="bookdown 0.25.4 with bs4_book()">
<meta property="og:title" content="Chapter 2 Manipulating Tree with Data | Data Integration, Manipulation and Visualization of Phylogenetic Trees with R">
<meta property="og:type" content="book">
<meta property="og:image" content="/book-cover.png">
<meta property="og:description" content="2.1 Manipulating Tree Data Using Tidy Interface All the tree data parsed/merged by treeio (Wang et al., 2020) can be converted to a tidy data frame using the tidytree package. The tidytree package...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 2 Manipulating Tree with Data | Data Integration, Manipulation and Visualization of Phylogenetic Trees with R">
<meta name="twitter:description" content="2.1 Manipulating Tree Data Using Tidy Interface All the tree data parsed/merged by treeio (Wang et al., 2020) can be converted to a tidy data frame using the tidytree package. The tidytree package...">
<meta name="twitter:image" content="/book-cover.png">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap/bootstrap.bundle.min.js"></script><script src="libs/bs3compat/transition.js"></script><script src="libs/bs3compat/tabs.js"></script><script src="libs/bs3compat/bs3compat.js"></script><link href="libs/bs4_book/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book/bs4_book.js"></script><script src="libs/kePrint/kePrint.js"></script><link href="libs/lightable/lightable.css" rel="stylesheet">
<script src="libs/htmlwidgets/htmlwidgets.js"></script><script src="libs/pathseg/pathseg.js"></script><script src="libs/comic/comic.min.js"></script><script src="libs/comicR-binding/comicR.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<style type="text/css">
    /* Used with Pandoc 2.11+ new --citeproc when CSL is used */
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
        }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Data Integration, Manipulation and Visualization of Phylogenetic Trees with R</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">Preface</a></li>
<li><a class="" href="author.html">About the Author</a></li>
<li class="book-part">Part I: Tree data input, output, and manipulation</li>
<li><a class="" href="chapter1.html"><span class="header-section-number">1</span> Importing Tree with Data</a></li>
<li><a class="active" href="chapter2.html"><span class="header-section-number">2</span> Manipulating Tree with Data</a></li>
<li><a class="" href="chapter3.html"><span class="header-section-number">3</span> Exporting tree with data</a></li>
<li class="book-part">Part II: Tree data visualization and annotation</li>
<li><a class="" href="chapter4.html"><span class="header-section-number">4</span> Phylogenetic Tree Visualization</a></li>
<li><a class="" href="chapter5.html"><span class="header-section-number">5</span> Phylogenetic Tree Annotation</a></li>
<li><a class="" href="chapter6.html"><span class="header-section-number">6</span> Visual Exploration of Phylogenetic Trees</a></li>
<li><a class="" href="chapter7.html"><span class="header-section-number">7</span> Plotting tree with data</a></li>
<li><a class="" href="chapter8.html"><span class="header-section-number">8</span> Annotating Tree with Silhouette Images and Sub-plots</a></li>
<li><a class="" href="chapter9.html"><span class="header-section-number">9</span> ggtree for Other Tree-like Objects</a></li>
<li class="book-part">Part III: ggtree extensions</li>
<li><a class="" href="chapter10.html"><span class="header-section-number">10</span> ggtreeExtra for Presenting Data on a Circular Layout</a></li>
<li><a class="" href="chapter11.html"><span class="header-section-number">11</span> Other ggtree Extensions</a></li>
<li class="book-part">Part IV: Miscellaneous topics</li>
<li><a class="" href="chapter12.html"><span class="header-section-number">12</span> ggtree Utilities</a></li>
<li><a class="" href="chapter13.html"><span class="header-section-number">13</span> Gallery of Reproducible Examples</a></li>
<li class="book-part">Appendix</li>
<li><a class="" href="faq.html"><span class="header-section-number">A</span> Frequently Asked Questions</a></li>
<li><a class="" href="related-tools.html"><span class="header-section-number">B</span> Related Tools</a></li>
<li><a class="" href="figures-and-tables.html"><span class="header-section-number">C</span> Figures and Tables</a></li>
<li><a class="" href="publications-of-the-ggtree-package-suite.html"><span class="header-section-number">D</span> Publications of the ggtree package suite</a></li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="chapter2" class="section level1" number="2">
<h1>
<span class="header-section-number">2</span> Manipulating Tree with Data<a class="anchor" aria-label="anchor" href="#chapter2"><i class="fas fa-link"></i></a>
</h1>
<div id="tidytree" class="section level2" number="2.1">
<h2>
<span class="header-section-number">2.1</span> Manipulating Tree Data Using Tidy Interface<a class="anchor" aria-label="anchor" href="#tidytree"><i class="fas fa-link"></i></a>
</h2>
<p>All the tree data parsed/merged
by <a href="http://bioconductor.org/packages/treeio"><strong>treeio</strong></a> <span class="citation">(<a href="references.html#ref-wang_treeio_2020" role="doc-biblioref">Wang et al., 2020</a>)</span> can be converted to a tidy
data frame using the <a href="https://CRAN.R-project.org/package=tidytree"><strong>tidytree</strong></a>
package. The <a href="https://CRAN.R-project.org/package=tidytree"><strong>tidytree</strong></a> package
provides tidy interfaces to manipulate trees with associated data. For instance,
external data can be linked to phylogeny or evolutionary data obtained from
different sources can be merged using tidyverse verbs. After the tree data was
processed, it can be converted back to a <code>treedata</code> object and exported to <a href="chapter3.html#chapter3">a single tree file</a>, further analyzed in R or visualized using <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> <span class="citation">(<a href="references.html#ref-yu_ggtree:_2017" role="doc-biblioref">Yu et al., 2017</a>)</span> and <a href="http://bioconductor.org/packages/ggtreeExtra"><strong>ggtreeExtra</strong></a> <span class="citation">(<a href="references.html#ref-ggtreeExtra_2021" role="doc-biblioref">Xu, Dai, et al., 2021</a>)</span>.</p>
<div id="the-phylo-object" class="section level3" number="2.1.1">
<h3>
<span class="header-section-number">2.1.1</span> The <code>phylo</code> object<a class="anchor" aria-label="anchor" href="#the-phylo-object"><i class="fas fa-link"></i></a>
</h3>
<p>The <code>phylo</code> class defined in the <a href="https://CRAN.R-project.org/package=ape"><strong>ape</strong></a> package <span class="citation">(<a href="references.html#ref-paradis_ape_2004" role="doc-biblioref">Paradis et al., 2004</a>)</span> is
fundamental for phylogenetic analysis in R. Most of the R packages
in <a href="https://CRAN.R-project.org/view=Phylogenetics">this field</a> rely
extensively on the <code>phylo</code> object. The <a href="https://CRAN.R-project.org/package=tidytree"><strong>tidytree</strong></a> package provides <code>as_tibble</code>
method to convert the <code>phylo</code> object to a tidy data frame, a <code>tbl_tree</code> object.</p>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://ape-package.ird.fr/">ape</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">2017</span><span class="op">)</span>
<span class="va">tree</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ape/man/rtree.html">rtree</a></span><span class="op">(</span><span class="fl">4</span><span class="op">)</span>
<span class="va">tree</span></code></pre></div>
<pre><code>## 
## Phylogenetic tree with 4 tips and 3 internal nodes.
## 
## Tip labels:
##   t4, t1, t3, t2
## 
## Rooted; includes branch lengths.</code></pre>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu">as_tibble</span><span class="op">(</span><span class="va">tree</span><span class="op">)</span>
<span class="va">x</span></code></pre></div>
<pre><code>## # A tibble: 7 × 4
##   parent  node branch.length label
##    &lt;int&gt; &lt;int&gt;         &lt;dbl&gt; &lt;chr&gt;
## 1      5     1       0.435   t4   
## 2      7     2       0.674   t1   
## 3      7     3       0.00202 t3   
## 4      6     4       0.0251  t2   
## 5      5     5      NA       &lt;NA&gt; 
## 6      5     6       0.472   &lt;NA&gt; 
## 7      6     7       0.274   &lt;NA&gt;</code></pre>
<p>The <code>tbl_tree</code> object can be converted back to a <code>phylo</code> object using the <code><a href="https://rdrr.io/pkg/ape/man/as.phylo.html">as.phylo()</a></code> method.</p>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/ape/man/as.phylo.html">as.phylo</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></code></pre></div>
<pre><code>## 
## Phylogenetic tree with 4 tips and 3 internal nodes.
## 
## Tip labels:
##   t4, t1, t3, t2
## 
## Rooted; includes branch lengths.</code></pre>
<p>Using <code>tbl_tree</code> object makes tree and data manipulation more effective and
easier (see also the example in <a href="faq.html#bind-tip">FAQ</a>). For example, we can link evolutionary trait to phylogeny using the <a href="https://CRAN.R-project.org/package=dplyr"><strong>dplyr</strong></a> verbs <code><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">full_join()</a></code>:</p>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">d</span> <span class="op">&lt;-</span> <span class="fu">tibble</span><span class="op">(</span>label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">'t'</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span>,
            trait <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/Normal.html">rnorm</a></span><span class="op">(</span><span class="fl">4</span><span class="op">)</span><span class="op">)</span>

<span class="va">y</span> <span class="op">&lt;-</span> <span class="fu">full_join</span><span class="op">(</span><span class="va">x</span>, <span class="va">d</span>, by <span class="op">=</span> <span class="st">'label'</span><span class="op">)</span>
<span class="va">y</span></code></pre></div>
<pre><code>## # A tibble: 7 × 5
##   parent  node branch.length label  trait
##    &lt;int&gt; &lt;int&gt;         &lt;dbl&gt; &lt;chr&gt;  &lt;dbl&gt;
## 1      5     1       0.435   t4     0.943
## 2      7     2       0.674   t1    -0.171
## 3      7     3       0.00202 t3     0.570
## 4      6     4       0.0251  t2    -0.283
## 5      5     5      NA       &lt;NA&gt;  NA    
## 6      5     6       0.472   &lt;NA&gt;  NA    
## 7      6     7       0.274   &lt;NA&gt;  NA</code></pre>
</div>
<div id="the-treedata-object" class="section level3" number="2.1.2">
<h3>
<span class="header-section-number">2.1.2</span> The <code>treedata</code> object<a class="anchor" aria-label="anchor" href="#the-treedata-object"><i class="fas fa-link"></i></a>
</h3>
<p>The <a href="https://CRAN.R-project.org/package=tidytree"><strong>tidytree</strong></a> package defines <code>treedata</code> class to store a phylogenetic tree with
associated data. After mapping external data to the tree structure, the
<code>tbl_tree</code> object can be converted to a <code>treedata</code> object.</p>
<div class="sourceCode" id="cb64"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">as.treedata</span><span class="op">(</span><span class="va">y</span><span class="op">)</span></code></pre></div>
<pre><code>## 'treedata' S4 object'.
## 
## ...@ phylo:
## 
## Phylogenetic tree with 4 tips and 3 internal nodes.
## 
## Tip labels:
##   t4, t1, t3, t2
## 
## Rooted; includes branch lengths.
## 
## with the following features available:
##   'trait'.</code></pre>
<p>The <code>treedata</code> class is used
in the <a href="https://bioconductor.org/packages/treeio/">treeio</a> package <span class="citation">(<a href="references.html#ref-wang_treeio_2020" role="doc-biblioref">Wang et al., 2020</a>)</span> to store evolutionary evidence inferred by commonly used software (<a href="http://beast2.org/"><strong>BEAST</strong></a>, <a href="http://sco.h-its.org/exelixis/web/software/epa/index.html"><strong>EPA</strong></a>, <a href="https://veg.github.io/hyphy-site/"><strong>HyPhy</strong></a>, <a href="http://nbisweden.github.io/MrBayes/"><strong>MrBayes</strong></a>, <a href="http://abacus.gene.ucl.ac.uk/software/paml.html"><strong>PAML</strong></a>, <a href="http://pbil.univ-lyon1.fr/software/phyldog/"><strong>PHYLDOG</strong></a>, <a href="http://matsen.fhcrc.org/pplacer/"><strong>PPLACER</strong></a>, <a href="http://loco.biosci.arizona.edu/r8s/"><strong>r8s</strong></a>, <a href="http://evomics.org/learning/phylogenetics/raxml/"><strong>RAxML</strong></a>, and <a href="http://revbayes.github.io/intro.html"><strong>RevBayes</strong></a>, etc.) (see details in <a href="chapter1.html#chapter1">Chapter 1</a>).</p>
<p>The <a href="https://CRAN.R-project.org/package=tidytree"><strong>tidytree</strong></a> package also provides the <code><a href="https://tibble.tidyverse.org/reference/as_tibble.html">as_tibble()</a></code> method to convert a <code>treedata</code> object to a tidy data frame. The phylogenetic tree structure and the evolutionary
inferences were stored in the <code>tbl_tree</code> object, making it consistent and easier
for manipulating evolutionary statistics inferred by different software as well
as linking external data to the same tree structure.</p>
<div class="sourceCode" id="cb66"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">y</span> <span class="op">%&gt;%</span> <span class="va">as.treedata</span> <span class="op">%&gt;%</span> <span class="va">as_tibble</span></code></pre></div>
<pre><code>## # A tibble: 7 × 5
##   parent  node branch.length label  trait
##    &lt;int&gt; &lt;int&gt;         &lt;dbl&gt; &lt;chr&gt;  &lt;dbl&gt;
## 1      5     1       0.435   t4     0.943
## 2      7     2       0.674   t1    -0.171
## 3      7     3       0.00202 t3     0.570
## 4      6     4       0.0251  t2    -0.283
## 5      5     5      NA       &lt;NA&gt;  NA    
## 6      5     6       0.472   &lt;NA&gt;  NA    
## 7      6     7       0.274   &lt;NA&gt;  NA</code></pre>
</div>
<div id="accesor-tidytree" class="section level3" number="2.1.3">
<h3>
<span class="header-section-number">2.1.3</span> Access related nodes<a class="anchor" aria-label="anchor" href="#accesor-tidytree"><i class="fas fa-link"></i></a>
</h3>
<p>The <a href="https://CRAN.R-project.org/package=dplyr"><strong>dplyr</strong></a> verbs can be applied to <code>tbl_tree</code> directly to manipulate tree data. In addition, <a href="https://CRAN.R-project.org/package=tidytree"><strong>tidytree</strong></a> provides several verbs to filter related nodes, including
<code><a href="https://rdrr.io/pkg/tidytree/man/child.html">child()</a></code>, <code><a href="https://rdrr.io/pkg/tidytree/man/parent.html">parent()</a></code>, <code><a href="https://rdrr.io/pkg/tidytree/man/offspring.html">offspring()</a></code>, <code><a href="https://rdrr.io/pkg/tidytree/man/ancestor.html">ancestor()</a></code>, <code><a href="https://rdrr.io/pkg/tidytree/man/sibling.html">sibling()</a></code> and <code><a href="https://rdrr.io/pkg/tidytree/man/MRCA.html">MRCA()</a></code>.</p>
<p>These verbs accept a <code>tbl_tree</code> object and a selected node which can be node number or label.</p>
<div class="sourceCode" id="cb68"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">child</span><span class="op">(</span><span class="va">y</span>, <span class="fl">5</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 2 × 5
##   parent  node branch.length label  trait
##    &lt;int&gt; &lt;int&gt;         &lt;dbl&gt; &lt;chr&gt;  &lt;dbl&gt;
## 1      5     1         0.435 t4     0.943
## 2      5     6         0.472 &lt;NA&gt;  NA</code></pre>
<div class="sourceCode" id="cb70"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">parent</span><span class="op">(</span><span class="va">y</span>, <span class="fl">2</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 1 × 5
##   parent  node branch.length label trait
##    &lt;int&gt; &lt;int&gt;         &lt;dbl&gt; &lt;chr&gt; &lt;dbl&gt;
## 1      6     7         0.274 &lt;NA&gt;     NA</code></pre>
<div class="sourceCode" id="cb72"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">offspring</span><span class="op">(</span><span class="va">y</span>, <span class="fl">5</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 6 × 5
##   parent  node branch.length label  trait
##    &lt;int&gt; &lt;int&gt;         &lt;dbl&gt; &lt;chr&gt;  &lt;dbl&gt;
## 1      5     1       0.435   t4     0.943
## 2      7     2       0.674   t1    -0.171
## 3      7     3       0.00202 t3     0.570
## 4      6     4       0.0251  t2    -0.283
## 5      5     6       0.472   &lt;NA&gt;  NA    
## 6      6     7       0.274   &lt;NA&gt;  NA</code></pre>
<div class="sourceCode" id="cb74"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ancestor</span><span class="op">(</span><span class="va">y</span>, <span class="fl">2</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 3 × 5
##   parent  node branch.length label trait
##    &lt;int&gt; &lt;int&gt;         &lt;dbl&gt; &lt;chr&gt; &lt;dbl&gt;
## 1      5     5        NA     &lt;NA&gt;     NA
## 2      5     6         0.472 &lt;NA&gt;     NA
## 3      6     7         0.274 &lt;NA&gt;     NA</code></pre>
<div class="sourceCode" id="cb76"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">sibling</span><span class="op">(</span><span class="va">y</span>, <span class="fl">2</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 1 × 5
##   parent  node branch.length label trait
##    &lt;int&gt; &lt;int&gt;         &lt;dbl&gt; &lt;chr&gt; &lt;dbl&gt;
## 1      7     3       0.00202 t3    0.570</code></pre>
<div class="sourceCode" id="cb78"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">MRCA</span><span class="op">(</span><span class="va">y</span>, <span class="fl">2</span>, <span class="fl">3</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 1 × 5
##   parent  node branch.length label trait
##    &lt;int&gt; &lt;int&gt;         &lt;dbl&gt; &lt;chr&gt; &lt;dbl&gt;
## 1      6     7         0.274 &lt;NA&gt;     NA</code></pre>
<p>All these methods are also implemented in <a href="http://bioconductor.org/packages/treeio"><strong>treeio</strong></a> for working with <code>phylo</code> and <code>treedata</code> objects. You can try accessing related nodes using the tree object. For instance, the following command will output child nodes of the selected internal node <code>5</code>:</p>
<div class="sourceCode" id="cb80"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">child</span><span class="op">(</span><span class="va">tree</span>, <span class="fl">5</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 1 6</code></pre>
<p>Beware that the methods for tree objects output relevant node numbers, while the methods for <code>tbl_tree</code> object output a <code>tibble</code> object that contains related information.</p>
</div>
</div>
<div id="data-integration" class="section level2" number="2.2">
<h2>
<span class="header-section-number">2.2</span> Data Integration<a class="anchor" aria-label="anchor" href="#data-integration"><i class="fas fa-link"></i></a>
</h2>
<div id="merge-tree" class="section level3" number="2.2.1">
<h3>
<span class="header-section-number">2.2.1</span> Combining tree data<a class="anchor" aria-label="anchor" href="#merge-tree"><i class="fas fa-link"></i></a>
</h3>
<p>The <a href="http://bioconductor.org/packages/treeio"><strong>treeio</strong></a> package <span class="citation">(<a href="references.html#ref-wang_treeio_2020" role="doc-biblioref">Wang et al., 2020</a>)</span> serves as an
infrastructure that enables various types of phylogenetic data inferred from
common analysis programs to be imported and used in R. For instance <em>d<sub>N</sub>/d<sub>S</sub></em>
or ancestral sequences estimated by <a href="http://abacus.gene.ucl.ac.uk/software/paml.html"><strong>CODEML</strong></a>,
and clade support values (posterior) inferred by <a href="http://beast2.org/"><strong>BEAST</strong></a>/<a href="http://nbisweden.github.io/MrBayes/"><strong>MrBayes</strong></a>.
In addition, <a href="http://bioconductor.org/packages/treeio"><strong>treeio</strong></a> supports linking external data to phylogeny. It brings these external phylogenetic data (either from software output or external sources) to the R
community and makes it available for further analysis in R.
Furthermore, <a href="http://bioconductor.org/packages/treeio"><strong>treeio</strong></a> can combine
multiple phylogenetic trees into one with their node/branch-specific
attribute data. Essentially, as a result, one such attribute (<em>e.g.</em>,
substitution rate) can be mapped to another attribute (<em>e.g.</em>, <em>d<sub>N</sub>/d<sub>S</sub></em>) of
the same node/branch for comparison and further computations <span class="citation">(<a href="references.html#ref-yu_ggtree:_2017" role="doc-biblioref">Yu et al., 2017</a>; <a href="references.html#ref-yu_two_2018" role="doc-biblioref">Yu et al., 2018</a>)</span>.</p>
<p>A previously published data set, seventy-six H3 hemagglutinin gene sequences of
a lineage containing swine and human influenza A viruses
<span class="citation">(<a href="references.html#ref-liang_expansion_2014" role="doc-biblioref">Liang et al., 2014</a>)</span>, was used here to demonstrate the utilities of comparing
evolutionary statistics inferred by different software. The dataset was
re-analyzed by <a href="http://beast2.org/"><strong>BEAST</strong></a> for timescale estimation
and <a href="http://abacus.gene.ucl.ac.uk/software/paml.html"><strong>CODEML</strong></a> for synonymous and
non-synonymous substitution estimation. In this example, we first parsed the
outputs from <a href="http://beast2.org/"><strong>BEAST</strong></a> using the <code><a href="https://docs.ropensci.org/treeio/reference/beast-parser.html">read.beast()</a></code> function and
from <a href="http://abacus.gene.ucl.ac.uk/software/paml.html"><strong>CODEML</strong></a> using
the <code><a href="https://docs.ropensci.org/treeio/reference/read.codeml.html">read.codeml()</a></code> function into two <code>treedata</code> objects. Then these two objects containing separate sets of node/branch-specific data were merged via the <code><a href="https://docs.ropensci.org/treeio/reference/merge_tree.html">merge_tree()</a></code> function.</p>
<div class="sourceCode" id="cb82"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">beast_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"examples/MCC_FluA_H3.tree"</span>, package<span class="op">=</span><span class="st">"ggtree"</span><span class="op">)</span>
<span class="va">rst_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"examples/rst"</span>, package<span class="op">=</span><span class="st">"ggtree"</span><span class="op">)</span>
<span class="va">mlc_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"examples/mlc"</span>, package<span class="op">=</span><span class="st">"ggtree"</span><span class="op">)</span>
<span class="va">beast_tree</span> <span class="op">&lt;-</span> <span class="fu">read.beast</span><span class="op">(</span><span class="va">beast_file</span><span class="op">)</span>
<span class="va">codeml_tree</span> <span class="op">&lt;-</span> <span class="fu">read.codeml</span><span class="op">(</span><span class="va">rst_file</span>, <span class="va">mlc_file</span><span class="op">)</span>

<span class="va">merged_tree</span> <span class="op">&lt;-</span> <span class="fu">merge_tree</span><span class="op">(</span><span class="va">beast_tree</span>, <span class="va">codeml_tree</span><span class="op">)</span>
<span class="va">merged_tree</span></code></pre></div>
<pre><code>## 'treedata' S4 object that stored information
## of
##  '/home/ygc/R/library/ggtree/examples/MCC_FluA_H3.tree',
##  '/home/ygc/R/library/ggtree/examples/rst',
##  '/home/ygc/R/library/ggtree/examples/mlc'.
## 
## ...@ phylo:
## 
## Phylogenetic tree with 76 tips and 75 internal nodes.
## 
## Tip labels:
##   A/Hokkaido/30-1-a/2013, A/New_York/334/2004,
## A/New_York/463/2005, A/New_York/452/1999,
## A/New_York/238/2005, A/New_York/523/1998, ...
## 
## Rooted; includes branch lengths.
## 
## with the following features available:
##   'height', 'height_0.95_HPD', 'height_median',
## 'height_range', 'length', 'length_0.95_HPD',
## 'length_median', 'length_range', 'posterior', 'rate',
## 'rate_0.95_HPD', 'rate_median', 'rate_range', 'subs',
## 'AA_subs', 't', 'N', 'S', 'dN_vs_dS', 'dN', 'dS',
## 'N_x_dN', 'S_x_dS'.</code></pre>
<p>After merging the <code>beast_tree</code> and <code>codeml_tree</code> objects, all
node/branch-specific data imported from <a href="http://beast2.org/"><strong>BEAST</strong></a>
and <a href="http://abacus.gene.ucl.ac.uk/software/paml.html"><strong>CODEML</strong></a> output files are
all available in the <code>merged_tree</code> object. The tree object was converted to a
tidy data frame using the <a href="https://cran.r-project.org/package=tidytree">tidytree</a>
package and visualized as hexbin scatterplots of <em>d<sub>N</sub>/d<sub>S</sub></em>, <em>d<sub>N</sub></em>, and <em>d<sub>S</sub></em> inferred
by <a href="http://abacus.gene.ucl.ac.uk/software/paml.html"><strong>CODEML</strong></a> vs. <em>rate</em>
(substitution rate in a unit of substitutions/site/year) inferred
by <a href="http://beast2.org/"><strong>BEAST</strong></a> on the same branches.</p>


<div class="sourceCode" id="cb84"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span>
<span class="va">df</span> <span class="op">&lt;-</span> <span class="va">merged_tree</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> 
  <span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html">as_tibble</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">dN_vs_dS</span>, <span class="va">dN</span>, <span class="va">dS</span>, <span class="va">rate</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">dN_vs_dS</span> <span class="op">&gt;=</span><span class="fl">0</span> <span class="op">&amp;</span> <span class="va">dN_vs_dS</span> <span class="op">&lt;=</span> <span class="fl">1.5</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
  <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/gather.html">gather</a></span><span class="op">(</span><span class="va">type</span>, <span class="va">value</span>, <span class="va">dN_vs_dS</span><span class="op">:</span><span class="va">dS</span><span class="op">)</span>
<span class="va">df</span><span class="op">$</span><span class="va">type</span><span class="op">[</span><span class="va">df</span><span class="op">$</span><span class="va">type</span> <span class="op">==</span> <span class="st">'dN_vs_dS'</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">'dN/dS'</span>
<span class="va">df</span><span class="op">$</span><span class="va">type</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">type</span>, levels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"dN/dS"</span>, <span class="st">"dN"</span>, <span class="st">"dS"</span><span class="op">)</span><span class="op">)</span>
<span class="fu">ggplot</span><span class="op">(</span><span class="va">df</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">rate</span>, <span class="va">value</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_hex</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">type</span>, scale<span class="op">=</span><span class="st">'free_y'</span><span class="op">)</span> </code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:correlations"></span>
<img src="treedata_files/figure-html/correlations-1.svg" alt="Correlation of dN/dS, dN, and dS vs. substitution rate. After merging the BEAST and CodeML outputs, the branch-specific estimates (substitution rate, dN/dS , dN, and dS) from the two analysis programs are compared on the same branch basis. The associations of dN/dS, dN, and dS vs. rate are visualized in hexbin scatter plots." width="100%"><p class="caption">
FIGURE 2.1: <strong>Correlation of <em>d<sub>N</sub>/d<sub>S</sub></em>, <em>d<sub>N</sub></em>, and <em>d<sub>S</sub></em> vs. substitution rate.</strong> After merging the <em>BEAST</em> and <em>CodeML</em> outputs, the branch-specific estimates (substitution rate, <em>d<sub>N</sub>/d<sub>S</sub></em> , <em>d<sub>N</sub></em>, and <em>d<sub>S</sub></em>) from the two analysis programs are compared on the same branch basis. The associations of <em>d<sub>N</sub>/d<sub>S</sub></em>, <em>d<sub>N</sub></em>, and <em>d<sub>S</sub></em> vs. <em>rate</em> are visualized in hexbin scatter plots.
</p>
</div>
<p>The output is illustrated in Figure <a href="chapter2.html#fig:correlations">2.1</a>. We can then test the association of these node/branch-specific data using Pearson correlation, which in this case showed that <em>d<sub>N</sub></em> and <em>d<sub>S</sub></em>, but not <em>d<sub>N</sub>/d<sub>S</sub></em> are significantly (<em>p</em>-values) associated with <em>rate</em>.</p>
<p>Using the <code><a href="https://docs.ropensci.org/treeio/reference/merge_tree.html">merge_tree()</a></code> function, we are able to compare analysis results using an identical
model from different software packages or different models using different or
identical software. It also allows users to integrate different analysis findings
from different software packages. Merging tree data is not restricted to
software findings, associating external data to analysis findings is also
granted. The <code><a href="https://docs.ropensci.org/treeio/reference/merge_tree.html">merge_tree()</a></code> function is chainable and allows several tree objects
to be merged into one.</p>
<div class="sourceCode" id="cb85"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">phylo</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ape/man/as.phylo.html">as.phylo</a></span><span class="op">(</span><span class="va">beast_tree</span><span class="op">)</span>
<span class="va">N</span> <span class="op">&lt;-</span> <span class="fu">Nnode2</span><span class="op">(</span><span class="va">phylo</span><span class="op">)</span>
<span class="va">d</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html">tibble</a></span><span class="op">(</span>node <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="va">N</span>, fake_trait <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/Normal.html">rnorm</a></span><span class="op">(</span><span class="va">N</span><span class="op">)</span>, another_trait <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/Uniform.html">runif</a></span><span class="op">(</span><span class="va">N</span><span class="op">)</span><span class="op">)</span>
<span class="va">fake_tree</span> <span class="op">&lt;-</span> <span class="fu">treedata</span><span class="op">(</span>phylo <span class="op">=</span> <span class="va">phylo</span>, data <span class="op">=</span> <span class="va">d</span><span class="op">)</span>
<span class="va">triple_tree</span> <span class="op">&lt;-</span> <span class="fu">merge_tree</span><span class="op">(</span><span class="va">merged_tree</span>, <span class="va">fake_tree</span><span class="op">)</span>
<span class="va">triple_tree</span></code></pre></div>
<pre><code>## 'treedata' S4 object that stored information
## of
##  '/home/ygc/R/library/ggtree/examples/MCC_FluA_H3.tree',
##  '/home/ygc/R/library/ggtree/examples/rst',
##  '/home/ygc/R/library/ggtree/examples/mlc'.
## 
## ...@ phylo:
## 
## Phylogenetic tree with 76 tips and 75 internal nodes.
## 
## Tip labels:
##   A/Hokkaido/30-1-a/2013, A/New_York/334/2004,
## A/New_York/463/2005, A/New_York/452/1999,
## A/New_York/238/2005, A/New_York/523/1998, ...
## 
## Rooted; includes branch lengths.
## 
## with the following features available:
##   'height', 'height_0.95_HPD', 'height_median',
## 'height_range', 'length', 'length_0.95_HPD',
## 'length_median', 'length_range', 'posterior', 'rate',
## 'rate_0.95_HPD', 'rate_median', 'rate_range', 'subs',
## 'AA_subs', 't', 'N', 'S', 'dN_vs_dS', 'dN', 'dS',
## 'N_x_dN', 'S_x_dS', 'fake_trait', 'another_trait'.
## 
## # The associated data tibble abstraction: 151 × 28
## # The 'node', 'label' and 'isTip' are from the phylo tree.
##     node label             isTip height height_0.95_HPD
##    &lt;int&gt; &lt;chr&gt;             &lt;lgl&gt;  &lt;dbl&gt; &lt;list&gt;         
##  1     1 A/Hokkaido/30-1-… TRUE       0 &lt;lgl [1]&gt;      
##  2     2 A/New_York/334/2… TRUE       9 &lt;dbl [2]&gt;      
##  3     3 A/New_York/463/2… TRUE       8 &lt;dbl [2]&gt;      
##  4     4 A/New_York/452/1… TRUE      14 &lt;dbl [2]&gt;      
##  5     5 A/New_York/238/2… TRUE       8 &lt;dbl [2]&gt;      
##  6     6 A/New_York/523/1… TRUE      15 &lt;dbl [2]&gt;      
##  7     7 A/New_York/435/2… TRUE      13 &lt;dbl [2]&gt;      
##  8     8 A/New_York/582/1… TRUE      17 &lt;dbl [2]&gt;      
##  9     9 A/New_York/603/1… TRUE      17 &lt;dbl [2]&gt;      
## 10    10 A/New_York/657/1… TRUE      19 &lt;dbl [2]&gt;      
## # … with 141 more rows, and 23 more variables:
## #   height_median &lt;dbl&gt;, height_range &lt;list&gt;,
## #   length &lt;dbl&gt;, length_0.95_HPD &lt;list&gt;,
## #   length_median &lt;dbl&gt;, length_range &lt;list&gt;,
## #   posterior &lt;dbl&gt;, rate &lt;dbl&gt;, rate_0.95_HPD &lt;list&gt;,
## #   rate_median &lt;dbl&gt;, rate_range &lt;list&gt;, subs &lt;chr&gt;,
## #   AA_subs &lt;chr&gt;, t &lt;dbl&gt;, N &lt;dbl&gt;, S &lt;dbl&gt;, …</code></pre>
<p>The <code>triple_tree</code> object shown above contains analysis results obtained from <a href="http://beast2.org/"><strong>BEAST</strong></a>
and <a href="http://abacus.gene.ucl.ac.uk/software/paml.html"><strong>CODEML</strong></a>, and evolutionary
traits from external sources. All these pieces of information can be used to annotate the
tree using <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> <span class="citation">(<a href="references.html#ref-yu_ggtree:_2017" role="doc-biblioref">Yu et al., 2017</a>)</span> and <a href="http://bioconductor.org/packages/ggtreeExtra"><strong>ggtreeExtra</strong></a> <span class="citation">(<a href="references.html#ref-ggtreeExtra_2021" role="doc-biblioref">Xu, Dai, et al., 2021</a>)</span>.</p>
</div>
<div id="link-external-data" class="section level3" number="2.2.2">
<h3>
<span class="header-section-number">2.2.2</span> Linking external data to phylogeny<a class="anchor" aria-label="anchor" href="#link-external-data"><i class="fas fa-link"></i></a>
</h3>
<p>In addition to analysis findings that are associated with the tree as demonstrated
above, there is a wide range of heterogeneous data, including phenotypic data,
experimental data, and clinical data, <em>etc.</em>, that need to be integrated and
linked to phylogeny. For example, in the study of viral evolution, tree nodes may be
associated with epidemiological information, such as location, age, and subtype.
Functional annotations may need to be mapped onto gene trees for comparative
genomics studies. To facilitate data
integration, <a href="http://bioconductor.org/packages/treeio"><strong>treeio</strong></a> provides
<code><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">full_join()</a></code> methods to link external data to phylogeny and store it in either <code>phylo</code> or <code>treedata</code> object. Beware that linking external data to a <code>phylo</code> object will produce a <code>treedata</code> object to store the input <code>phylo</code> with associated data. The <code>full_join</code> methods can also be used at tidy data frame level (<em>i.e.</em>, <code>tbl_tree</code> object described previously) and at <code>ggtree</code> level (described in <a href="chapter7.html#attach-operator">Chapter 7</a>) <span class="citation">(<a href="references.html#ref-yu_two_2018" role="doc-biblioref">Yu et al., 2018</a>)</span>.</p>
<p>The following example calculated bootstrap values and merged those values with the tree (a <code>phylo</code> object) by matching their node numbers.</p>
<div class="sourceCode" id="cb87"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://ape-package.ird.fr/">ape</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">woodmouse</span><span class="op">)</span>
<span class="va">d</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ape/man/dist.dna.html">dist.dna</a></span><span class="op">(</span><span class="va">woodmouse</span><span class="op">)</span>
<span class="va">tr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ape/man/nj.html">nj</a></span><span class="op">(</span><span class="va">d</span><span class="op">)</span>
<span class="va">bp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ape/man/boot.phylo.html">boot.phylo</a></span><span class="op">(</span><span class="va">tr</span>, <span class="va">woodmouse</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/pkg/ape/man/nj.html">nj</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/dist.dna.html">dist.dna</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## 
Running bootstraps:       100 / 100
## Calculating bootstrap values... done.</code></pre>
<div class="sourceCode" id="cb89"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bp2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html">tibble</a></span><span class="op">(</span>node<span class="op">=</span><span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/summary.phylo.html">Nnode</a></span><span class="op">(</span><span class="va">tr</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ape/man/summary.phylo.html">Ntip</a></span><span class="op">(</span><span class="va">tr</span><span class="op">)</span>, bootstrap <span class="op">=</span> <span class="va">bp</span><span class="op">)</span>
<span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">full_join</a></span><span class="op">(</span><span class="va">tr</span>, <span class="va">bp2</span>, by<span class="op">=</span><span class="st">"node"</span><span class="op">)</span></code></pre></div>
<pre><code>## 'treedata' S4 object'.
## 
## ...@ phylo:
## 
## Phylogenetic tree with 15 tips and 13 internal nodes.
## 
## Tip labels:
##   No305, No304, No306, No0906S, No0908S, No0909S, ...
## 
## Unrooted; includes branch lengths.
## 
## with the following features available:
##   'bootstrap'.
## 
## # The associated data tibble abstraction: 28 × 4
## # The 'node', 'label' and 'isTip' are from the phylo tree.
##     node label   isTip bootstrap
##    &lt;int&gt; &lt;chr&gt;   &lt;lgl&gt;     &lt;int&gt;
##  1     1 No305   TRUE         NA
##  2     2 No304   TRUE         NA
##  3     3 No306   TRUE         NA
##  4     4 No0906S TRUE         NA
##  5     5 No0908S TRUE         NA
##  6     6 No0909S TRUE         NA
##  7     7 No0910S TRUE         NA
##  8     8 No0912S TRUE         NA
##  9     9 No0913S TRUE         NA
## 10    10 No1103S TRUE         NA
## # … with 18 more rows</code></pre>
<p>Another example demonstrates merging evolutionary traits with the tree (a <code>treedata</code> object) by matching their tip labels.</p>
<div class="sourceCode" id="cb91"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata/BEAST"</span>, <span class="st">"beast_mcc.tree"</span>, package<span class="op">=</span><span class="st">"treeio"</span><span class="op">)</span>
<span class="va">beast</span> <span class="op">&lt;-</span> <span class="fu">read.beast</span><span class="op">(</span><span class="va">file</span><span class="op">)</span>
<span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html">tibble</a></span><span class="op">(</span>label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/ape/man/as.phylo.html">as.phylo</a></span><span class="op">(</span><span class="va">beast</span><span class="op">)</span><span class="op">$</span><span class="va">tip.label</span>, trait <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/Normal.html">rnorm</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/summary.phylo.html">Ntip</a></span><span class="op">(</span><span class="va">beast</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">full_join</a></span><span class="op">(</span><span class="va">beast</span>, <span class="va">x</span>, by<span class="op">=</span><span class="st">"label"</span><span class="op">)</span></code></pre></div>
<pre><code>## 'treedata' S4 object that stored information
## of
##  '/home/ygc/R/library/treeio/extdata/BEAST/beast_mcc.tree'.
## 
## ...@ phylo:
## 
## Phylogenetic tree with 15 tips and 14 internal nodes.
## 
## Tip labels:
##   A_1995, B_1996, C_1995, D_1987, E_1996, F_1997, ...
## 
## Rooted; includes branch lengths.
## 
## with the following features available:
##   'height', 'height_0.95_HPD', 'height_median',
## 'height_range', 'length', 'length_0.95_HPD',
## 'length_median', 'length_range', 'posterior', 'rate',
## 'rate_0.95_HPD', 'rate_median', 'rate_range', 'trait'.
## 
## # The associated data tibble abstraction: 29 × 17
## # The 'node', 'label' and 'isTip' are from the phylo tree.
##     node label  isTip height height_0.95_HPD
##    &lt;int&gt; &lt;chr&gt;  &lt;lgl&gt;  &lt;dbl&gt; &lt;list&gt;         
##  1     1 A_1995 TRUE      18 &lt;dbl [2]&gt;      
##  2     2 B_1996 TRUE      17 &lt;dbl [2]&gt;      
##  3     3 C_1995 TRUE      18 &lt;dbl [2]&gt;      
##  4     4 D_1987 TRUE      26 &lt;dbl [2]&gt;      
##  5     5 E_1996 TRUE      17 &lt;dbl [2]&gt;      
##  6     6 F_1997 TRUE      16 &lt;dbl [2]&gt;      
##  7     7 G_1992 TRUE      21 &lt;dbl [2]&gt;      
##  8     8 H_1992 TRUE      21 &lt;dbl [2]&gt;      
##  9     9 I_1994 TRUE      19 &lt;dbl [2]&gt;      
## 10    10 J_1983 TRUE      30 &lt;dbl [2]&gt;      
## # … with 19 more rows, and 12 more variables:
## #   height_median &lt;dbl&gt;, height_range &lt;list&gt;,
## #   length &lt;dbl&gt;, length_0.95_HPD &lt;list&gt;,
## #   length_median &lt;dbl&gt;, length_range &lt;list&gt;,
## #   posterior &lt;dbl&gt;, rate &lt;dbl&gt;, rate_0.95_HPD &lt;list&gt;,
## #   rate_median &lt;dbl&gt;, rate_range &lt;list&gt;, trait &lt;dbl&gt;</code></pre>
<p>Manipulating tree objects is frustrated with the fragmented functions available
for working with <code>phylo</code> objects, not to mention linking external data to the
phylogeny structure. With the <a href="http://bioconductor.org/packages/treeio"><strong>treeio</strong></a> package <span class="citation">(<a href="references.html#ref-wang_treeio_2020" role="doc-biblioref">Wang et al., 2020</a>)</span>, it is easy to combine tree data from various sources.
In addition, with the <a href="https://CRAN.R-project.org/package=tidytree"><strong>tidytree</strong></a> package, manipulating trees is easier using the <a href="https://www.jstatsoft.org/article/view/v059i10">tidy data principles</a> and
consistent with tools already in wide use, including
<a href="https://CRAN.R-project.org/package=dplyr"><strong>dplyr</strong></a>,
<a href="https://CRAN.R-project.org/package=tidyr"><strong>tidyr</strong></a>,
<a href="https://CRAN.R-project.org/package=ggplot2"><strong>ggplot2</strong></a>,
and <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> <span class="citation">(<a href="references.html#ref-yu_ggtree:_2017" role="doc-biblioref">Yu et al., 2017</a>)</span>.</p>
</div>
<div id="grouping-taxa" class="section level3" number="2.2.3">
<h3>
<span class="header-section-number">2.2.3</span> Grouping taxa<a class="anchor" aria-label="anchor" href="#grouping-taxa"><i class="fas fa-link"></i></a>
</h3>
<p>The <code><a href="https://rdrr.io/pkg/tidytree/man/groupOTU.html">groupOTU()</a></code> and <code><a href="https://rdrr.io/pkg/tidytree/man/groupClade.html">groupClade()</a></code> methods are designed for adding taxa grouping
information to the input tree object. The methods were implemented in <a href="https://CRAN.R-project.org/package=tidytree"><strong>tidytree</strong></a>,
<a href="http://bioconductor.org/packages/treeio"><strong>treeio</strong></a>, and <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> respectively to support adding grouping information for the
<code>tbl_tree</code>, <code>phylo</code> and <code>treedata</code>, and <code>ggtree</code> objects. This grouping information can be
used directly in tree visualization (<em>e.g.</em>, <a href="chapter6.html#group-taxa-vis">coloring a tree based on grouping information</a>)
with <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> (Figure <a href="chapter6.html#fig:groupOTU">6.5</a>).</p>
<div id="groupclade" class="section level4" number="2.2.3.1">
<h4>
<span class="header-section-number">2.2.3.1</span> groupClade<a class="anchor" aria-label="anchor" href="#groupclade"><i class="fas fa-link"></i></a>
</h4>
<p>The <code><a href="https://rdrr.io/pkg/tidytree/man/groupClade.html">groupClade()</a></code> method accepts an internal node or a vector of internal nodes
to add grouping information of selected clade/clades.</p>
<div class="sourceCode" id="cb93"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">nwk</span> <span class="op">&lt;-</span> <span class="st">'(((((((A:4,B:4):6,C:5):8,D:6):3,E:21):10,((F:4,G:12):14,H:8):13):
        13,((I:5,J:2):30,(K:11,L:11):2):17):4,M:56);'</span>
<span class="va">tree</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ape/man/read.tree.html">read.tree</a></span><span class="op">(</span>text<span class="op">=</span><span class="va">nwk</span><span class="op">)</span>

<span class="fu">groupClade</span><span class="op">(</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html">as_tibble</a></span><span class="op">(</span><span class="va">tree</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">17</span>, <span class="fl">21</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 25 × 5
##    parent  node branch.length label group
##     &lt;int&gt; &lt;int&gt;         &lt;dbl&gt; &lt;chr&gt; &lt;fct&gt;
##  1     20     1             4 A     1    
##  2     20     2             4 B     1    
##  3     19     3             5 C     1    
##  4     18     4             6 D     1    
##  5     17     5            21 E     1    
##  6     22     6             4 F     2    
##  7     22     7            12 G     2    
##  8     21     8             8 H     2    
##  9     24     9             5 I     0    
## 10     24    10             2 J     0    
## # … with 15 more rows</code></pre>
</div>
<div id="groupotu" class="section level4" number="2.2.3.2">
<h4>
<span class="header-section-number">2.2.3.2</span> groupOTU<a class="anchor" aria-label="anchor" href="#groupotu"><i class="fas fa-link"></i></a>
</h4>
<div class="sourceCode" id="cb95"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">2017</span><span class="op">)</span>
<span class="va">tr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ape/man/rtree.html">rtree</a></span><span class="op">(</span><span class="fl">4</span><span class="op">)</span>
<span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html">as_tibble</a></span><span class="op">(</span><span class="va">tr</span><span class="op">)</span>
<span class="co">## the input nodes can be node ID or label</span>
<span class="fu">groupOTU</span><span class="op">(</span><span class="va">x</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'t1'</span>, <span class="st">'t4'</span><span class="op">)</span>, group_name <span class="op">=</span> <span class="st">"fake_group"</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 7 × 5
##   parent  node branch.length label fake_group
##    &lt;int&gt; &lt;int&gt;         &lt;dbl&gt; &lt;chr&gt; &lt;fct&gt;     
## 1      5     1       0.435   t4    1         
## 2      7     2       0.674   t1    1         
## 3      7     3       0.00202 t3    0         
## 4      6     4       0.0251  t2    0         
## 5      5     5      NA       &lt;NA&gt;  1         
## 6      5     6       0.472   &lt;NA&gt;  1         
## 7      6     7       0.274   &lt;NA&gt;  1</code></pre>
<p>Both <code><a href="https://rdrr.io/pkg/tidytree/man/groupClade.html">groupClade()</a></code> and <code><a href="https://rdrr.io/pkg/tidytree/man/groupOTU.html">groupOTU()</a></code> work with the <code>tbl_tree</code>, <code>phylo</code> and <code>treedata</code>, and <code>ggtree</code> objects. Here is an example of using <code><a href="https://rdrr.io/pkg/tidytree/man/groupOTU.html">groupOTU()</a></code> with a <code>phylo</code> tree object.</p>
<div class="sourceCode" id="cb97"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">groupOTU</span><span class="op">(</span><span class="va">tr</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'t2'</span>, <span class="st">'t4'</span><span class="op">)</span>, group_name <span class="op">=</span> <span class="st">"fake_group"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
  <span class="va">as_tibble</span></code></pre></div>
<pre><code>## # A tibble: 7 × 5
##   parent  node branch.length label fake_group
##    &lt;int&gt; &lt;int&gt;         &lt;dbl&gt; &lt;chr&gt; &lt;fct&gt;     
## 1      5     1       0.435   t4    1         
## 2      7     2       0.674   t1    0         
## 3      7     3       0.00202 t3    0         
## 4      6     4       0.0251  t2    1         
## 5      5     5      NA       &lt;NA&gt;  1         
## 6      5     6       0.472   &lt;NA&gt;  1         
## 7      6     7       0.274   &lt;NA&gt;  0</code></pre>
<p>Another example of working with the <code>ggtree</code> object can be found in <a href="chapter6.html#group-taxa-vis">session 6.4</a>.</p>
<p>The <code>groupOTU</code> will trace back from input nodes to most recent common ancestor.
In this example, nodes 1, 4, 5 and 6 are
grouping together (<code>4 (t2) -&gt; 6 -&gt; 5</code> and <code>1 (t4) -&gt; 5</code>).</p>
<p>Related operational taxonomic units (OTUs) are grouping and they are not necessarily within a clade.
They can be monophyletic (clade), polyphyletic or paraphyletic.</p>
<div class="sourceCode" id="cb99"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cls</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>c1<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"A"</span>, <span class="st">"B"</span>, <span class="st">"C"</span>, <span class="st">"D"</span>, <span class="st">"E"</span><span class="op">)</span>,
            c2<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"F"</span>, <span class="st">"G"</span>, <span class="st">"H"</span><span class="op">)</span>,
            c3<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"L"</span>, <span class="st">"K"</span>, <span class="st">"I"</span>, <span class="st">"J"</span><span class="op">)</span>,
            c4<span class="op">=</span><span class="st">"M"</span><span class="op">)</span>

<span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html">as_tibble</a></span><span class="op">(</span><span class="va">tree</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> <span class="fu">groupOTU</span><span class="op">(</span><span class="va">cls</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 25 × 5
##    parent  node branch.length label group
##     &lt;int&gt; &lt;int&gt;         &lt;dbl&gt; &lt;chr&gt; &lt;fct&gt;
##  1     20     1             4 A     c1   
##  2     20     2             4 B     c1   
##  3     19     3             5 C     c1   
##  4     18     4             6 D     c1   
##  5     17     5            21 E     c1   
##  6     22     6             4 F     c2   
##  7     22     7            12 G     c2   
##  8     21     8             8 H     c2   
##  9     24     9             5 I     c3   
## 10     24    10             2 J     c3   
## # … with 15 more rows</code></pre>
<p>If there are conflicts when tracing back to the most recent common ancestor, users can set <code>overlap</code>
parameter to “origin” (the first one counts), “overwrite” (default, the last one
counts), or “abandon” (un-selected for grouping)<a class="footnote-ref" tabindex="0" data-toggle="popover" data-content='&lt;p&gt;&lt;a href="https://groups.google.com/forum/#!msg/bioc-ggtree/Q4LnwoTf1DM/uqYdYB_VBAAJ" class="uri"&gt;https://groups.google.com/forum/#!msg/bioc-ggtree/Q4LnwoTf1DM/uqYdYB_VBAAJ&lt;/a&gt;&lt;/p&gt;'><sup>8</sup></a>.</p>
</div>
</div>
</div>
<div id="reroot-treeio" class="section level2" number="2.3">
<h2>
<span class="header-section-number">2.3</span> Rerooting tree<a class="anchor" aria-label="anchor" href="#reroot-treeio"><i class="fas fa-link"></i></a>
</h2>
<p>A phylogenetic tree can be rerooted with a specified <code>outgroup</code>. The <a href="https://CRAN.R-project.org/package=ape"><strong>ape</strong></a> package implements a <code><a href="https://rdrr.io/pkg/ape/man/root.html">root()</a></code> method to reroot a tree stored in a <code>phylo</code> object, while the <a href="http://bioconductor.org/packages/treeio"><strong>treeio</strong></a> package provides the <code><a href="https://rdrr.io/pkg/ape/man/root.html">root()</a></code> method for <code>treedata</code> object. This method is designed to re-root a phylogenetic tree with associated data concerning the specified <code>outgroup</code> or at the specified <code>node</code> based on the <code><a href="https://rdrr.io/pkg/ape/man/root.html">root()</a></code> implemented in the <a href="https://CRAN.R-project.org/package=ape"><strong>ape</strong></a> package.</p>
<p>We first linked external data to a tree using <code><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">left_join()</a></code> and stored all the information in a <code>treedata</code> object, <code>trda</code>.</p>
<div class="sourceCode" id="cb101"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/treedata-book/">ggtree</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/YuLab-SMU/treeio">treeio</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/treedata-book/">tidytree</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/YuLab-SMU/TDBook">TDbook</a></span><span class="op">)</span>

<span class="co"># load `tree_boots`, `df_tip_data`, and `df_inode_data` from 'TDbook'</span>

<span class="va">trda</span> <span class="op">&lt;-</span> <span class="va">tree_boots</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> 
        <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">left_join</a></span><span class="op">(</span><span class="va">df_tip_data</span>, by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"label"</span> <span class="op">=</span> <span class="st">"Newick_label"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> 
        <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">left_join</a></span><span class="op">(</span><span class="va">df_inode_data</span>, by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"label"</span> <span class="op">=</span> <span class="st">"newick_label"</span><span class="op">)</span><span class="op">)</span>
<span class="va">trda</span></code></pre></div>
<pre><code>## 'treedata' S4 object'.
## 
## ...@ phylo:
## 
## Phylogenetic tree with 7 tips and 6 internal nodes.
## 
## Tip labels:
##   Rangifer_tarandus, Cervus_elaphus, Bos_taurus,
## Ovis_orientalis, Suricata_suricatta,
## Cystophora_cristata, ...
## Node labels:
##   Mammalia, Artiodactyla, Cervidae, Bovidae,
## Carnivora, Caniformia
## 
## Rooted; includes branch lengths.
## 
## with the following features available:
##   '', 'vernacularName', 'imageURL', 'imageLicense',
## 'imageAuthor', 'infoURL', 'mass_in_kg',
## 'trophic_habit', 'ncbi_taxid', 'rank',
## 'vernacularName.y', 'infoURL.y', 'rank.y',
## 'bootstrap', 'posterior'.
## 
## # The associated data tibble abstraction: 13 × 17
## # The 'node', 'label' and 'isTip' are from the phylo tree.
##     node label            isTip vernacularName imageURL
##    &lt;int&gt; &lt;chr&gt;            &lt;lgl&gt; &lt;chr&gt;          &lt;chr&gt;   
##  1     1 Rangifer_tarand… TRUE  Reindeer       http://…
##  2     2 Cervus_elaphus   TRUE  Red deer       http://…
##  3     3 Bos_taurus       TRUE  Cattle         https:/…
##  4     4 Ovis_orientalis  TRUE  Asiatic moufl… http://…
##  5     5 Suricata_surica… TRUE  Meerkat        http://…
##  6     6 Cystophora_cris… TRUE  Hooded seal    http://…
##  7     7 Mephitis_mephit… TRUE  Striped skunk  http://…
##  8     8 Mammalia         FALSE &lt;NA&gt;           &lt;NA&gt;    
##  9     9 Artiodactyla     FALSE &lt;NA&gt;           &lt;NA&gt;    
## 10    10 Cervidae         FALSE &lt;NA&gt;           &lt;NA&gt;    
## # … with 3 more rows, and 12 more variables:
## #   imageLicense &lt;chr&gt;, imageAuthor &lt;chr&gt;,
## #   infoURL &lt;chr&gt;, mass_in_kg &lt;dbl&gt;,
## #   trophic_habit &lt;chr&gt;, ncbi_taxid &lt;int&gt;, rank &lt;chr&gt;,
## #   vernacularName.y &lt;chr&gt;, infoURL.y &lt;chr&gt;,
## #   rank.y &lt;chr&gt;, bootstrap &lt;int&gt;, posterior &lt;dbl&gt;</code></pre>
<p>Then we can reroot the tree with the associated data mapping to the branches and nodes correctly as demonstrated in Figure <a href="chapter2.html#fig:reroot">2.2</a>. The figure was visualized using <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> (see also Chapters <a href="chapter4.html#chapter4">4</a> and <a href="chapter5.html#chapter5">5</a>).</p>


<div class="sourceCode" id="cb103"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># reroot</span>
<span class="va">trda2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ape/man/root.html">root</a></span><span class="op">(</span><span class="va">trda</span>, outgroup <span class="op">=</span> <span class="st">"Suricata_suricatta"</span>, edgelabel <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co"># The original tree</span>
<span class="va">p1</span> <span class="op">&lt;-</span> <span class="va">trda</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
      <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
      <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_nodelab.html">geom_nodelab</a></span><span class="op">(</span>
        mapping <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>
          x <span class="op">=</span> <span class="va">branch</span>,
          label <span class="op">=</span> <span class="va">bootstrap</span>
        <span class="op">)</span>,
        nudge_y <span class="op">=</span> <span class="fl">0.36</span>
      <span class="op">)</span> <span class="op">+</span>
      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html">xlim</a></span><span class="op">(</span><span class="op">-</span><span class="fl">.1</span>, <span class="fl">4</span><span class="op">)</span> <span class="op">+</span>
      <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tippoint.html">geom_tippoint</a></span><span class="op">(</span>
        mapping <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>
          shape <span class="op">=</span> <span class="va">trophic_habit</span>, 
          color <span class="op">=</span> <span class="va">trophic_habit</span>, 
          size <span class="op">=</span> <span class="va">mass_in_kg</span>
        <span class="op">)</span>
      <span class="op">)</span> <span class="op">+</span>
      <span class="fu">scale_size_continuous</span><span class="op">(</span>range <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">3</span>, <span class="fl">10</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
      <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab</a></span><span class="op">(</span>
        offset <span class="op">=</span> <span class="fl">.14</span>, 
      <span class="op">)</span> <span class="op">+</span>
      <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_nodelab.html">geom_nodelab</a></span><span class="op">(</span>
        mapping <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>
          label <span class="op">=</span> <span class="va">vernacularName.y</span>, 
          fill <span class="op">=</span> <span class="va">posterior</span>
        <span class="op">)</span>,
        geom <span class="op">=</span> <span class="st">"label"</span>
      <span class="op">)</span> <span class="op">+</span> 
      <span class="fu">scale_fill_gradientn</span><span class="op">(</span>colors <span class="op">=</span> <span class="fu">RColorBrewer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html">brewer.pal</a></span><span class="op">(</span><span class="fl">3</span>, <span class="st">"YlGnBu"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"right"</span><span class="op">)</span>  

<span class="co"># after reroot</span>
<span class="va">p2</span> <span class="op">&lt;-</span> <span class="va">trda2</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
      <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
      <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_nodelab.html">geom_nodelab</a></span><span class="op">(</span>
        mapping <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>
          x <span class="op">=</span> <span class="va">branch</span>,
          label <span class="op">=</span> <span class="va">bootstrap</span>
        <span class="op">)</span>,
        nudge_y <span class="op">=</span> <span class="fl">0.36</span>
      <span class="op">)</span> <span class="op">+</span>
      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html">xlim</a></span><span class="op">(</span><span class="op">-</span><span class="fl">.1</span>, <span class="fl">5</span><span class="op">)</span> <span class="op">+</span>
      <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tippoint.html">geom_tippoint</a></span><span class="op">(</span>
        mapping <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>
          shape <span class="op">=</span> <span class="va">trophic_habit</span>,
          color <span class="op">=</span> <span class="va">trophic_habit</span>,
          size <span class="op">=</span> <span class="va">mass_in_kg</span>
        <span class="op">)</span>
      <span class="op">)</span> <span class="op">+</span>
      <span class="fu">scale_size_continuous</span><span class="op">(</span>range <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">3</span>, <span class="fl">10</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
      <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab</a></span><span class="op">(</span>
        offset <span class="op">=</span> <span class="fl">.14</span>,
      <span class="op">)</span> <span class="op">+</span>
      <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_nodelab.html">geom_nodelab</a></span><span class="op">(</span>
        mapping <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>
          label <span class="op">=</span> <span class="va">vernacularName.y</span>,
          fill <span class="op">=</span> <span class="va">posterior</span>
        <span class="op">)</span>,
        geom <span class="op">=</span> <span class="st">"label"</span>
      <span class="op">)</span> <span class="op">+</span>
      <span class="fu">scale_fill_gradientn</span><span class="op">(</span>colors <span class="op">=</span> <span class="fu">RColorBrewer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html">brewer.pal</a></span><span class="op">(</span><span class="fl">3</span>, <span class="st">"YlGnBu"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"right"</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/pkg/aplot/man/plot_list.html">plot_list</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span>, tag_levels<span class="op">=</span><span class="st">'A'</span>, ncol<span class="op">=</span><span class="fl">2</span><span class="op">)</span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:reroot"></span>
<img src="treedata_files/figure-html/reroot-1.png" alt="Reroot a phylogenetic tree with associated data. Original tree (A) and re-rooted tree (B) with associated data mapped to the branches or nodes of the tree correctly. (A) and (B) present before and after rooting on the branch leading to the tip node ‘Suricata_suricatta’, respectively." width="98%"><p class="caption">
FIGURE 2.2: <strong>Reroot a phylogenetic tree with associated data.</strong> Original tree (A) and re-rooted tree (B) with associated data mapped to the branches or nodes of the tree correctly. (A) and (B) present before and after rooting on the branch leading to the tip node ‘Suricata_suricatta’, respectively.
</p>
</div>
<p>The <code>outgroup</code> parameter represents the specific new <code>outgroup</code>, it can be a node label (character) or node number. If it is a “single one” value, meaning using the node below this tip as the new root, if it has multiple values, meaning the most recent common of the values will be used as the new root. Note that, if the node labels should be treated as edge labels, the <code>edgelabel</code> should be set to <code>TRUE</code> to return the correct relationship between the <code>node</code> and <code>associated data</code>. For more details about re-root, including precautions and pitfalls, please refer to the review article <span class="citation">(<a href="references.html#ref-reroot_review" role="doc-biblioref">Czech et al., 2017</a>)</span>.</p>
</div>
<div id="rescale-treeio" class="section level2" number="2.4">
<h2>
<span class="header-section-number">2.4</span> Rescaling Tree Branches<a class="anchor" aria-label="anchor" href="#rescale-treeio"><i class="fas fa-link"></i></a>
</h2>
<p>Phylogenetic data can be merged for joint analysis (Figure <a href="chapter2.html#fig:correlations">2.1</a>). They can be displayed on the same tree structure as a more complex annotation to help visually inspection of their evolutionary patterns. All the numerical data stored in a <code>treedata</code> object can be used to re-scale tree branches. For example, CodeML infers d<sub>N</sub>/d<sub>S</sub>, d<sub>N</sub>, and d<sub>S</sub>, all these statistics can be used as branch lengths (Figure <a href="chapter2.html#fig:rescale">2.3</a>). All these values can also be used to color the tree (session <a href="chapter4.html#color-tree">4.3.4</a>) and can be projected to a vertical dimension to create a two-dimensional tree or phenogram (session <a href="#layouts-of-phylogenetic-tree">4.2.2</a> and Figure <a href="chapter4.html#fig:2d">4.5</a> and <a href="chapter4.html#fig:continuousColor">4.11</a>).</p>


<div class="sourceCode" id="cb104"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">merged_tree</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/theme_tree2.html">theme_tree2</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="fu"><a href="https://docs.ropensci.org/treeio/reference/rescale_tree.html">rescale_tree</a></span><span class="op">(</span><span class="va">merged_tree</span>, <span class="st">'dN'</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/theme_tree2.html">theme_tree2</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">p3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="fu"><a href="https://docs.ropensci.org/treeio/reference/rescale_tree.html">rescale_tree</a></span><span class="op">(</span><span class="va">merged_tree</span>, <span class="st">'rate'</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/theme_tree2.html">theme_tree2</a></span><span class="op">(</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/pkg/aplot/man/plot_list.html">plot_list</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span>, <span class="va">p3</span>, ncol<span class="op">=</span><span class="fl">3</span>, tag_levels<span class="op">=</span><span class="st">'A'</span><span class="op">)</span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:rescale"></span>
<img src="treedata_files/figure-html/rescale-1.svg" alt="Re-scaling tree branches. The tree with branches scaled in time (year from the root) (A). The tree was re-scaled using dN as branch lengths (B). The tree was re-scaled using substitution rates (C)." width="100%"><p class="caption">
FIGURE 2.3: <strong>Re-scaling tree branches.</strong> The tree with branches scaled in time (year from the root) (A). The tree was re-scaled using <em>d<sub>N</sub></em> as branch lengths (B). The tree was re-scaled using substitution rates (C).
</p>
</div>
<p>Modifying branch lengths in the tree object in addtion to using the <code><a href="https://docs.ropensci.org/treeio/reference/rescale_tree.html">rescale_tree()</a></code> function, users can directly specify a variable as branch length in <code><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree()</a></code> as demonstrated in <a href="chapter4.html#rescale-tree">session 4.3.6</a>.</p>
</div>
<div id="subsetting-tree-with-data" class="section level2" number="2.5">
<h2>
<span class="header-section-number">2.5</span> Subsetting Tree with Data<a class="anchor" aria-label="anchor" href="#subsetting-tree-with-data"><i class="fas fa-link"></i></a>
</h2>
<div id="remove-tip" class="section level3" number="2.5.1">
<h3>
<span class="header-section-number">2.5.1</span> Removing tips in a phylogenetic tree<a class="anchor" aria-label="anchor" href="#remove-tip"><i class="fas fa-link"></i></a>
</h3>
<p>Sometimes we want to remove selected tips from a phylogenetic tree. This is due to several reasons, including low sequence quality, errors in sequence assembly, an alignment error in part of the sequence, and an error in phylogenetic inference, <em>etc</em>.</p>
<p>Let’s say that we want to remove three tips (colored red) from the tree (Figure <a href="chapter2.html#fig:removeTip">2.4</a>A), the <code><a href="https://rdrr.io/pkg/ape/man/drop.tip.html">drop.tip()</a></code> method removes specified tips and updates the tree (Figure <a href="chapter2.html#fig:removeTip">2.4</a>B). All associated data will be maintained in the updated tree.</p>


<div class="sourceCode" id="cb105"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">f</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata/NHX"</span>, <span class="st">"phyldog.nhx"</span>, package<span class="op">=</span><span class="st">"treeio"</span><span class="op">)</span>
<span class="va">nhx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://docs.ropensci.org/treeio/reference/read.nhx.html">read.nhx</a></span><span class="op">(</span><span class="va">f</span><span class="op">)</span>
<span class="va">to_drop</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Physonect_sp_@2066767"</span>,
            <span class="st">"Lychnagalma_utricularia@2253871"</span>,
            <span class="st">"Kephyes_ovata@2606431"</span><span class="op">)</span>
<span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">nhx</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>color <span class="op">=</span> <span class="va">label</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">to_drop</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_color_manual</a></span><span class="op">(</span>values<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"black"</span>, <span class="st">"red"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html">xlim</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0.8</span><span class="op">)</span>

<span class="va">nhx_reduced</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://docs.ropensci.org/treeio/reference/drop.tip-methods.html">drop.tip</a></span><span class="op">(</span><span class="va">nhx</span>, <span class="va">to_drop</span><span class="op">)</span>
<span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">nhx_reduced</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html">xlim</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0.8</span><span class="op">)</span>  
<span class="fu"><a href="https://rdrr.io/pkg/aplot/man/plot_list.html">plot_list</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span>, ncol<span class="op">=</span><span class="fl">2</span>, tag_levels <span class="op">=</span> <span class="st">"A"</span><span class="op">)</span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:removeTip"></span>
<img src="treedata_files/figure-html/removeTip-1.png" alt="Removing tips from a tree. Original tree with three tips (colored red) to remove (A). The updated tree removed selected tips (B)." width="100%"><p class="caption">
FIGURE 2.4: <strong>Removing tips from a tree.</strong> Original tree with three tips (colored red) to remove (A). The updated tree removed selected tips (B).
</p>
</div>
</div>
<div id="subset-tip" class="section level3" number="2.5.2">
<h3>
<span class="header-section-number">2.5.2</span> Subsetting tree by tip label<a class="anchor" aria-label="anchor" href="#subset-tip"><i class="fas fa-link"></i></a>
</h3>
<p>Sometimes a tree can be large and difficult to look at only the portions of interest. The <code><a href="https://docs.ropensci.org/treeio/reference/tree_subset.html">tree_subset()</a></code> function was created in the <a href="http://bioconductor.org/packages/treeio"><strong>treeio</strong></a> package <span class="citation">(<a href="references.html#ref-wang_treeio_2020" role="doc-biblioref">Wang et al., 2020</a>)</span> to extract a subset of the tree portion while still maintaining the structure of the tree portion. The <code>beast_tree</code> in Figure <a href="chapter2.html#fig:subsetTip">2.5</a>A is slightly crowded. Obviously, we can make the figure taller to allow more space for the labels (similar to using the “Expansion” slider in <code>FigTree</code>) or we can make the text smaller. However, these solutions are not always applicable when you have a lot of tips (<em>e.g.</em>, hundreds or thousands of tips). In particular, when you are only interested in the portion of the tree around a particular tip, you certainly don’t want to explore a large tree to find the certain species you are interested in.</p>
<p>Let’s say you are interested in tip <em>A/Swine/HK/168/2012</em> from the tree (Figure <a href="chapter2.html#fig:subsetTip">2.5</a>A) and you want to look at the immediate relatives of this tip.</p>
<p>The <code><a href="https://docs.ropensci.org/treeio/reference/tree_subset.html">tree_subset()</a></code> function allows you to look at the portions of the tree that are of interest. By default, the <code><a href="https://docs.ropensci.org/treeio/reference/tree_subset.html">tree_subset()</a></code> function will internally call the <a href="chapter2.html#groupotu"><code>groupOTU()</code></a> to assign the group specified tip from the rest of the other tips (Figure <a href="chapter2.html#fig:subsetTip">2.5</a>B). Additionally, the branch lengths and related associated data are maintained after subsetting (Figure <a href="chapter2.html#fig:subsetTip">2.5</a>C). The root of the tree is always anchored at zero for the subset tree by default and all the distances are relative to this root. If you want all the distances to be relative to the original root, you can specify the root position (by <code>root.position</code> parameter) to the root edge of the subset tree, which is the sum of branch lengths from the original root to the root of the subset tree (Figures <a href="chapter2.html#fig:subsetTip">2.5</a>D and E).</p>


<div class="sourceCode" id="cb106"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">beast_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"examples/MCC_FluA_H3.tree"</span>, package<span class="op">=</span><span class="st">"ggtree"</span><span class="op">)</span>
<span class="va">beast_tree</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://docs.ropensci.org/treeio/reference/beast-parser.html">read.beast</a></span><span class="op">(</span><span class="va">beast_file</span><span class="op">)</span>

<span class="va">p1</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">beast_tree</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html">xlim</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">40</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/theme_tree2.html">theme_tree2</a></span><span class="op">(</span><span class="op">)</span>

<span class="va">tree2</span> <span class="op">=</span> <span class="fu"><a href="https://docs.ropensci.org/treeio/reference/tree_subset.html">tree_subset</a></span><span class="op">(</span><span class="va">beast_tree</span>, <span class="st">"A/Swine/HK/168/2012"</span>, levels_back<span class="op">=</span><span class="fl">4</span><span class="op">)</span>  
<span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree2</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>color<span class="op">=</span><span class="va">group</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_color_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"black"</span>, <span class="st">"red"</span><span class="op">)</span>, guide <span class="op">=</span> <span class="st">'none'</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html">xlim</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">4</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/theme_tree2.html">theme_tree2</a></span><span class="op">(</span><span class="op">)</span> 

<span class="va">p3</span> <span class="op">&lt;-</span> <span class="va">p2</span> <span class="op">+</span>   
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="va">rate</span><span class="op">)</span>, shape <span class="op">=</span> <span class="fl">21</span>, size <span class="op">=</span> <span class="fl">4</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">scale_fill_continuous</span><span class="op">(</span>low <span class="op">=</span> <span class="st">'blue'</span>, high <span class="op">=</span> <span class="st">'red'</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html">xlim</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">5</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">'right'</span><span class="op">)</span>

<span class="va">p4</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree2</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>color<span class="op">=</span><span class="va">group</span><span class="op">)</span>, 
          root.position <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/ape/man/as.phylo.html">as.phylo</a></span><span class="op">(</span><span class="va">tree2</span><span class="op">)</span><span class="op">$</span><span class="va">root.edge</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html">xlim</a></span><span class="op">(</span><span class="fl">18</span>, <span class="fl">24</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_color_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"black"</span>, <span class="st">"red"</span><span class="op">)</span>, guide <span class="op">=</span> <span class="st">'none'</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/theme_tree2.html">theme_tree2</a></span><span class="op">(</span><span class="op">)</span>

<span class="va">p5</span> <span class="op">&lt;-</span> <span class="va">p4</span> <span class="op">+</span> 
  <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_rootedge.html">geom_rootedge</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html">xlim</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">40</span><span class="op">)</span> 

<span class="fu"><a href="https://rdrr.io/pkg/aplot/man/plot_list.html">plot_list</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span>, <span class="va">p3</span>, <span class="va">p4</span>, <span class="va">p5</span>, 
        design<span class="op">=</span><span class="st">"AABBCC\nAADDEE"</span>, tag_levels<span class="op">=</span><span class="st">'A'</span><span class="op">)</span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:subsetTip"></span>
<img src="treedata_files/figure-html/subsetTip-1.png" alt="Subsetting tree for a specific tip. The Original tree (A). The Subset tree (B). Subset tree with data (C). Visualize the subset tree relative to the original position, without root-edge (D) and with root-edge (E)." width="100%"><p class="caption">
FIGURE 2.5: <strong>Subsetting tree for a specific tip.</strong> The Original tree (A). The Subset tree (B). Subset tree with data (C). Visualize the subset tree relative to the original position, without root-edge (D) and with root-edge (E).
</p>
</div>
</div>
<div id="subset-node" class="section level3" number="2.5.3">
<h3>
<span class="header-section-number">2.5.3</span> Subsetting tree by internal node number<a class="anchor" aria-label="anchor" href="#subset-node"><i class="fas fa-link"></i></a>
</h3>
<p>If you are interested in a certain clade, you can specify the input node as an internal node number. The <code><a href="https://docs.ropensci.org/treeio/reference/tree_subset.html">tree_subset()</a></code> function will take the clade as a whole and also trace it back to particular levels to look at the immediate relatives of the clade (Figures <a href="chapter2.html#fig:subsetNode">2.6</a>A and B). We can use the <code><a href="https://docs.ropensci.org/treeio/reference/tree_subset.html">tree_subset()</a></code> function to zoom in selected portions and plot a whole tree with the portion of it, which is similar to the <code><a href="https://rdrr.io/pkg/ape/man/zoom.html">ape::zoom()</a></code> function to explore a very large tree (Figures <a href="chapter2.html#fig:subsetNode">2.6</a>C and D). Users can also use <code><a href="https://rdrr.io/pkg/ggtree/man/viewClade.html">viewClade()</a></code> function to restrict tree visualization at specific clade as demonstrated in <a href="chapter6.html#viewing-selected-clade">session 6.1</a>.</p>


<div class="sourceCode" id="cb107"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">clade</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://docs.ropensci.org/treeio/reference/tree_subset.html">tree_subset</a></span><span class="op">(</span><span class="va">beast_tree</span>, node<span class="op">=</span><span class="fl">121</span>, levels_back<span class="op">=</span><span class="fl">0</span><span class="op">)</span>
<span class="va">clade2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://docs.ropensci.org/treeio/reference/tree_subset.html">tree_subset</a></span><span class="op">(</span><span class="va">beast_tree</span>, node<span class="op">=</span><span class="fl">121</span>, levels_back<span class="op">=</span><span class="fl">2</span><span class="op">)</span>
<span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">clade</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html">xlim</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">5</span><span class="op">)</span>
<span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">clade2</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>color<span class="op">=</span><span class="va">group</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html">xlim</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">8</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_color_manual</a></span><span class="op">(</span>values<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"black"</span>, <span class="st">"red"</span><span class="op">)</span><span class="op">)</span>


<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://ape-package.ird.fr/">ape</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/treedata-book/">tidytree</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/YuLab-SMU/treeio">treeio</a></span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">chiroptera</span><span class="op">)</span>

<span class="va">nodes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span><span class="op">(</span><span class="st">"Plecotus"</span>, <span class="va">chiroptera</span><span class="op">$</span><span class="va">tip.label</span><span class="op">)</span>
<span class="va">chiroptera</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/tidytree/man/groupOTU.html">groupOTU</a></span><span class="op">(</span><span class="va">chiroptera</span>, <span class="va">nodes</span><span class="op">)</span>

<span class="va">clade</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/tidytree/man/MRCA.html">MRCA</a></span><span class="op">(</span><span class="va">chiroptera</span>, <span class="va">nodes</span><span class="op">)</span>
<span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://docs.ropensci.org/treeio/reference/tree_subset.html">tree_subset</a></span><span class="op">(</span><span class="va">chiroptera</span>, <span class="va">clade</span>, levels_back <span class="op">=</span> <span class="fl">0</span><span class="op">)</span>

<span class="va">p3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">chiroptera</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>colour <span class="op">=</span> <span class="va">group</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_color_manual</a></span><span class="op">(</span>values<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"black"</span>, <span class="st">"red"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span>
<span class="va">p4</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html">xlim</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">5</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/aplot/man/plot_list.html">plot_list</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span>, <span class="va">p3</span>, <span class="va">p4</span>, 
  ncol<span class="op">=</span><span class="fl">2</span>, tag_levels <span class="op">=</span> <span class="st">'A'</span><span class="op">)</span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:subsetNode"></span>
<img src="treedata_files/figure-html/subsetNode-1.png" alt="Subsetting tree for the specific clade. Extracting a clade (A). Extracting a clade and tracing it back to look at its immediate relatives (B). Viewing a very large tree (C) and a selected portion of it (D)." width="100%"><p class="caption">
FIGURE 2.6: <strong>Subsetting tree for the specific clade.</strong> Extracting a clade (A). Extracting a clade and tracing it back to look at its immediate relatives (B). Viewing a very large tree (C) and a selected portion of it (D).
</p>
</div>
</div>
</div>
<div id="ggtree-fortify" class="section level2" number="2.6">
<h2>
<span class="header-section-number">2.6</span> Manipulating Tree Data for Visualization<a class="anchor" aria-label="anchor" href="#ggtree-fortify"><i class="fas fa-link"></i></a>
</h2>
<p>Tree visualization is supported by <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> <span class="citation">(<a href="references.html#ref-yu_ggtree:_2017" role="doc-biblioref">Yu et al., 2017</a>)</span>. Although <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> implemented several methods for <a href="chapter6.html#chapter6">visual exploration of trees with data</a>, you may want to do something that is not supported directly. In this case, you need to manipulate tree data with node coordination positions that are used for visualization. This is quite easy with <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a>. Users can use the <code><a href="https://ggplot2.tidyverse.org/reference/fortify.html">fortify()</a></code> method which internally calls <code><a href="https://tibble.tidyverse.org/reference/as_tibble.html">tidytree::as_tibble()</a></code> to convert the tree to a tidy data frame and add columns of coordination positions (<em>i.e.</em>, x, y, branch, and angle) that are used to plot the tree. You can also access the data via <code>ggtree(tree)$data</code>.</p>
<p>Here is an example to plot two trees face-to-face that is similar to a graph produced by the <code><a href="https://rdrr.io/pkg/ape/man/cophyloplot.html">ape::cophyloplot()</a></code> function (Figure <a href="chapter2.html#fig:cophylo">2.7</a>).</p>


<div class="sourceCode" id="cb108"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/treedata-book/">ggtree</a></span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">1024</span><span class="op">)</span>
<span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ape/man/rtree.html">rtree</a></span><span class="op">(</span><span class="fl">30</span><span class="op">)</span>
<span class="va">y</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ape/man/rtree.html">rtree</a></span><span class="op">(</span><span class="fl">30</span><span class="op">)</span>
<span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">x</span>, layout<span class="op">=</span><span class="st">'roundrect'</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom-hilight.html">geom_hilight</a></span><span class="op">(</span>
         mapping<span class="op">=</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>subset <span class="op">=</span> <span class="va">node</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">38</span>, <span class="fl">48</span>, <span class="fl">58</span>, <span class="fl">36</span><span class="op">)</span>,
                     node <span class="op">=</span> <span class="va">node</span>,
                     fill <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span><span class="op">(</span><span class="va">node</span><span class="op">)</span>
                     <span class="op">)</span>
     <span class="op">)</span> <span class="op">+</span>
    <span class="fu">labs</span><span class="op">(</span>fill <span class="op">=</span> <span class="st">"clades for tree in left"</span> <span class="op">)</span>

<span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">y</span><span class="op">)</span>

<span class="va">d1</span> <span class="op">&lt;-</span> <span class="va">p1</span><span class="op">$</span><span class="va">data</span>
<span class="va">d2</span> <span class="op">&lt;-</span> <span class="va">p2</span><span class="op">$</span><span class="va">data</span>

<span class="co">## reverse x-axis and </span>
<span class="co">## set offset to make the tree on the right-hand side of the first tree</span>
<span class="va">d2</span><span class="op">$</span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">d2</span><span class="op">$</span><span class="va">x</span><span class="op">)</span> <span class="op">-</span> <span class="va">d2</span><span class="op">$</span><span class="va">x</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">d1</span><span class="op">$</span><span class="va">x</span><span class="op">)</span> <span class="op">+</span> <span class="fl">1</span>

<span class="va">pp</span> <span class="op">&lt;-</span> <span class="va">p1</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tree.html">geom_tree</a></span><span class="op">(</span>data<span class="op">=</span><span class="va">d2</span>, layout<span class="op">=</span><span class="st">'ellipse'</span><span class="op">)</span> <span class="op">+</span>      
  <span class="fu">ggnewscale</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ggnewscale/man/new_scale.html">new_scale_fill</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom-hilight.html">geom_hilight</a></span><span class="op">(</span>
         data <span class="op">=</span> <span class="va">d2</span>, 
         mapping <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span> 
            subset <span class="op">=</span> <span class="va">node</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">38</span>, <span class="fl">48</span>, <span class="fl">58</span><span class="op">)</span>,
            node<span class="op">=</span><span class="va">node</span>,
            fill<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span><span class="op">(</span><span class="va">node</span><span class="op">)</span><span class="op">)</span>
  <span class="op">)</span> <span class="op">+</span>
  <span class="fu">labs</span><span class="op">(</span>fill <span class="op">=</span> <span class="st">"clades for tree in right"</span> <span class="op">)</span> 

<span class="va">dd</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind.html">bind_rows</a></span><span class="op">(</span><span class="va">d1</span>, <span class="va">d2</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">label</span><span class="op">)</span><span class="op">)</span>

<span class="va">pp</span> <span class="op">+</span> <span class="fu">geom_line</span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span><span class="va">x</span>, <span class="va">y</span>, group<span class="op">=</span><span class="va">label</span><span class="op">)</span>, data<span class="op">=</span><span class="va">dd</span>, color<span class="op">=</span><span class="st">'grey'</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab</a></span><span class="op">(</span>geom <span class="op">=</span> <span class="st">'shadowtext'</span>, bg.colour <span class="op">=</span> <span class="fu">alpha</span><span class="op">(</span><span class="st">'firebrick'</span>, <span class="fl">.5</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">d2</span>, hjust<span class="op">=</span><span class="fl">1</span>, geom <span class="op">=</span> <span class="st">'shadowtext'</span>, 
                bg.colour <span class="op">=</span> <span class="fu">alpha</span><span class="op">(</span><span class="st">'firebrick'</span>, <span class="fl">.5</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:cophylo"></span>
<img src="treedata_files/figure-html/cophylo-1.png" alt="Plot two phylogenetic trees face-to-face. Plotting a tree using ggtree() (left-hand side) and subsequently adding another layer of a tree by geom_tree() (right-hand side). The relative positions of the plotted trees can be manually adjusted and adding layers to each of the trees (e.g., tip labels and highlighting clades) is independent." width="100%"><p class="caption">
FIGURE 2.7: <strong>Plot two phylogenetic trees face-to-face.</strong> Plotting a tree using <code><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree()</a></code> (left-hand side) and subsequently adding another layer of a tree by <code><a href="https://rdrr.io/pkg/ggtree/man/geom_tree.html">geom_tree()</a></code> (right-hand side). The relative positions of the plotted trees can be manually adjusted and adding layers to each of the trees (<em>e.g.</em>, tip labels and highlighting clades) is independent.
</p>
</div>
<p>It is quite easy to plot multiple trees and connect taxa in one figure; For instance, plotting trees constructed from all internal gene segments of influenza virus and connecting equivalent strains across the trees <span class="citation">(<a href="references.html#ref-venkatesh_avian_2018" role="doc-biblioref">Venkatesh et al., 2018</a>)</span>. Figure <a href="chapter2.html#fig:cophylo3">2.8</a> demonstrates the usage of plotting multiple trees by combining multiple layers of <code><a href="https://rdrr.io/pkg/ggtree/man/geom_tree.html">geom_tree()</a></code>.</p>


<div class="sourceCode" id="cb109"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">z</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ape/man/rtree.html">rtree</a></span><span class="op">(</span><span class="fl">30</span><span class="op">)</span>
<span class="va">d2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/fortify.html">fortify</a></span><span class="op">(</span><span class="va">y</span><span class="op">)</span>
<span class="va">d3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/fortify.html">fortify</a></span><span class="op">(</span><span class="va">z</span><span class="op">)</span>
<span class="va">d2</span><span class="op">$</span><span class="va">x</span> <span class="op">&lt;-</span> <span class="va">d2</span><span class="op">$</span><span class="va">x</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">d1</span><span class="op">$</span><span class="va">x</span><span class="op">)</span> <span class="op">+</span> <span class="fl">1</span>
<span class="va">d3</span><span class="op">$</span><span class="va">x</span> <span class="op">&lt;-</span> <span class="va">d3</span><span class="op">$</span><span class="va">x</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">d2</span><span class="op">$</span><span class="va">x</span><span class="op">)</span> <span class="op">+</span> <span class="fl">1</span>

<span class="va">dd</span> <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind.html">bind_rows</a></span><span class="op">(</span><span class="va">d1</span>, <span class="va">d2</span>, <span class="va">d3</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">label</span><span class="op">)</span><span class="op">)</span>

<span class="va">p1</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tree.html">geom_tree</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">d2</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tree.html">geom_tree</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">d3</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab</a></span><span class="op">(</span>data<span class="op">=</span><span class="va">d3</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">geom_line</span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span><span class="va">x</span>, <span class="va">y</span>, group<span class="op">=</span><span class="va">label</span>, color<span class="op">=</span><span class="va">node</span> <span class="op">&lt;</span> <span class="fl">15</span><span class="op">)</span>, data<span class="op">=</span><span class="va">dd</span>, alpha<span class="op">=</span><span class="fl">.3</span><span class="op">)</span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:cophylo3"></span>
<img src="treedata_files/figure-html/cophylo3-1.svg" alt="Plot multiple phylogenetic trees side-by-side. Plotting a tree using ggtree() and subsequently adding multiple layers of trees by geom_tree()." width="100%"><p class="caption">
FIGURE 2.8: <strong>Plot multiple phylogenetic trees side-by-side.</strong> Plotting a tree using <code><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree()</a></code> and subsequently adding multiple layers of trees by <code><a href="https://rdrr.io/pkg/ggtree/man/geom_tree.html">geom_tree()</a></code>.
</p>
</div>
</div>
<div id="summary2" class="section level2" number="2.7">
<h2>
<span class="header-section-number">2.7</span> Summary<a class="anchor" aria-label="anchor" href="#summary2"><i class="fas fa-link"></i></a>
</h2>
<p>The <a href="http://bioconductor.org/packages/treeio"><strong>treeio</strong></a> package allows us to import diverse phylogeny associated data into R. However, a phylogenetic tree is stored in a way to facilitate computational processing which is not human friendly and needs the expertise to manipulate and explore tree data. The <a href="https://CRAN.R-project.org/package=tidytree"><strong>tidytree</strong></a> package provides a tidy interface for exploring tree data, while <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> provides a set of utilities to visualize and explore tree data using the grammar of graphics. This full suite of packages makes it easy for ordinary users to interact with tree data and allows us to integrate phylogeny associated data from different sources (<em>e.g.</em>, experimental results or analysis findings), which creates the possibilities of integrative and comparative study. Moreover, this package suite brings phylogenetic analysis into the tidyverse and certainly takes us to the next level of processing phylogenetic data.</p>

<div style="page-break-after: always;"></div>
</div>
</div>

  <div class="chapter-nav">
<div class="prev"><a href="chapter1.html"><span class="header-section-number">1</span> Importing Tree with Data</a></div>
<div class="next"><a href="chapter3.html"><span class="header-section-number">3</span> Exporting tree with data</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#chapter2"><span class="header-section-number">2</span> Manipulating Tree with Data</a></li>
<li>
<a class="nav-link" href="#tidytree"><span class="header-section-number">2.1</span> Manipulating Tree Data Using Tidy Interface</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#the-phylo-object"><span class="header-section-number">2.1.1</span> The phylo object</a></li>
<li><a class="nav-link" href="#the-treedata-object"><span class="header-section-number">2.1.2</span> The treedata object</a></li>
<li><a class="nav-link" href="#accesor-tidytree"><span class="header-section-number">2.1.3</span> Access related nodes</a></li>
</ul>
</li>
<li>
<a class="nav-link" href="#data-integration"><span class="header-section-number">2.2</span> Data Integration</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#merge-tree"><span class="header-section-number">2.2.1</span> Combining tree data</a></li>
<li><a class="nav-link" href="#link-external-data"><span class="header-section-number">2.2.2</span> Linking external data to phylogeny</a></li>
<li><a class="nav-link" href="#grouping-taxa"><span class="header-section-number">2.2.3</span> Grouping taxa</a></li>
</ul>
</li>
<li><a class="nav-link" href="#reroot-treeio"><span class="header-section-number">2.3</span> Rerooting tree</a></li>
<li><a class="nav-link" href="#rescale-treeio"><span class="header-section-number">2.4</span> Rescaling Tree Branches</a></li>
<li>
<a class="nav-link" href="#subsetting-tree-with-data"><span class="header-section-number">2.5</span> Subsetting Tree with Data</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#remove-tip"><span class="header-section-number">2.5.1</span> Removing tips in a phylogenetic tree</a></li>
<li><a class="nav-link" href="#subset-tip"><span class="header-section-number">2.5.2</span> Subsetting tree by tip label</a></li>
<li><a class="nav-link" href="#subset-node"><span class="header-section-number">2.5.3</span> Subsetting tree by internal node number</a></li>
</ul>
</li>
<li><a class="nav-link" href="#ggtree-fortify"><span class="header-section-number">2.6</span> Manipulating Tree Data for Visualization</a></li>
<li><a class="nav-link" href="#summary2"><span class="header-section-number">2.7</span> Summary</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Data Integration, Manipulation and Visualization of Phylogenetic Trees with R</strong>" was written by Guangchuang Yu. It was last built on 2022-04-11.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
