<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 4 Phylogenetic Tree Visualization | Data Integration, Manipulation and Visualization of Phylogenetic Trees</title>
<meta name="author" content="Guangchuang Yu">
<meta name="description" content="4.1 Introduction There are many software packages and web tools that are designed for displaying phylogenetic trees, such as TreeView (Page, 2002), FigTree, TreeDyn (Chevenet et al., 2006),...">
<meta name="generator" content="bookdown 0.26 with bs4_book()">
<meta property="og:title" content="Chapter 4 Phylogenetic Tree Visualization | Data Integration, Manipulation and Visualization of Phylogenetic Trees">
<meta property="og:type" content="book">
<meta property="og:image" content="/book-cover.png">
<meta property="og:description" content="4.1 Introduction There are many software packages and web tools that are designed for displaying phylogenetic trees, such as TreeView (Page, 2002), FigTree, TreeDyn (Chevenet et al., 2006),...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 4 Phylogenetic Tree Visualization | Data Integration, Manipulation and Visualization of Phylogenetic Trees">
<meta name="twitter:description" content="4.1 Introduction There are many software packages and web tools that are designed for displaying phylogenetic trees, such as TreeView (Page, 2002), FigTree, TreeDyn (Chevenet et al., 2006),...">
<meta name="twitter:image" content="/book-cover.png">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap/bootstrap.bundle.min.js"></script><script src="libs/bs3compat/transition.js"></script><script src="libs/bs3compat/tabs.js"></script><script src="libs/bs3compat/bs3compat.js"></script><link href="libs/bs4_book/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book/bs4_book.js"></script><script src="libs/kePrint/kePrint.js"></script><link href="libs/lightable/lightable.css" rel="stylesheet">
<script src="libs/htmlwidgets/htmlwidgets.js"></script><script src="libs/pathseg/pathseg.js"></script><script src="libs/comic/comic.min.js"></script><script src="libs/comicR-binding/comicR.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<style type="text/css">
    /* Used with Pandoc 2.11+ new --citeproc when CSL is used */
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
        }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Data Integration, Manipulation and Visualization of Phylogenetic Trees</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">Preface</a></li>
<li><a class="" href="author.html">About the Author</a></li>
<li class="book-part">Part I: Tree data input, output, and manipulation</li>
<li><a class="" href="chapter1.html"><span class="header-section-number">1</span> Importing Tree with Data</a></li>
<li><a class="" href="chapter2.html"><span class="header-section-number">2</span> Manipulating Tree with Data</a></li>
<li><a class="" href="chapter3.html"><span class="header-section-number">3</span> Exporting tree with data</a></li>
<li class="book-part">Part II: Tree data visualization and annotation</li>
<li><a class="active" href="chapter4.html"><span class="header-section-number">4</span> Phylogenetic Tree Visualization</a></li>
<li><a class="" href="chapter5.html"><span class="header-section-number">5</span> Phylogenetic Tree Annotation</a></li>
<li><a class="" href="chapter6.html"><span class="header-section-number">6</span> Visual Exploration of Phylogenetic Trees</a></li>
<li><a class="" href="chapter7.html"><span class="header-section-number">7</span> Plotting tree with data</a></li>
<li><a class="" href="chapter8.html"><span class="header-section-number">8</span> Annotating Tree with Silhouette Images and Sub-plots</a></li>
<li><a class="" href="chapter9.html"><span class="header-section-number">9</span> ggtree for Other Tree-like Objects</a></li>
<li class="book-part">Part III: ggtree extensions</li>
<li><a class="" href="chapter10.html"><span class="header-section-number">10</span> ggtreeExtra for Presenting Data on a Circular Layout</a></li>
<li><a class="" href="chapter11.html"><span class="header-section-number">11</span> Other ggtree Extensions</a></li>
<li class="book-part">Part IV: Miscellaneous topics</li>
<li><a class="" href="chapter12.html"><span class="header-section-number">12</span> ggtree Utilities</a></li>
<li><a class="" href="chapter13.html"><span class="header-section-number">13</span> Gallery of Reproducible Examples</a></li>
<li class="book-part">Appendix</li>
<li><a class="" href="faq.html"><span class="header-section-number">A</span> Frequently Asked Questions</a></li>
<li><a class="" href="related-tools.html"><span class="header-section-number">B</span> Related Tools</a></li>
<li><a class="" href="figures-and-tables.html"><span class="header-section-number">C</span> Figures and Tables</a></li>
<li><a class="" href="publications-of-the-ggtree-package-suite.html"><span class="header-section-number">D</span> Publications of the ggtree Package Suite</a></li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="chapter4" class="section level1" number="4">
<h1>
<span class="header-section-number">4</span> Phylogenetic Tree Visualization<a class="anchor" aria-label="anchor" href="#chapter4"><i class="fas fa-link"></i></a>
</h1>
<div id="introduction-1" class="section level2" number="4.1">
<h2>
<span class="header-section-number">4.1</span> Introduction<a class="anchor" aria-label="anchor" href="#introduction-1"><i class="fas fa-link"></i></a>
</h2>
<p>There are many software packages and web tools that are designed for displaying phylogenetic trees, such as <a href="http://en.bio-soft.net/tree/TreeView.html"><strong>TreeView</strong></a> <span class="citation">(<a href="references.html#ref-page_visualizing_2002" role="doc-biblioref">Page, 2002</a>)</span>, <a href="http://beast.community/figtree"><strong>FigTree</strong></a>, <a href="http://www.treedyn.org/"><strong>TreeDyn</strong></a> <span class="citation">(<a href="references.html#ref-chevenet_treedyn:_2006" role="doc-biblioref">Chevenet et al., 2006</a>)</span>, <em>Dendroscope</em> <span class="citation">(<a href="references.html#ref-huson_dendroscope_2012" role="doc-biblioref">Huson &amp; Scornavacca, 2012</a>)</span>, <a href="https://www.evolgenius.info/evolview/"><strong>EvolView</strong></a> <span class="citation">(<a href="references.html#ref-he_evolview_2016" role="doc-biblioref">He et al., 2016</a>)</span>, and <a href="https://itol.embl.de/"><strong>iTOL</strong></a> <span class="citation">(<a href="references.html#ref-letunic_interactive_2007" role="doc-biblioref">Letunic &amp; Bork, 2007</a>)</span>, <em>etc</em>. Only a few of them, such as <a href="http://beast.community/figtree"><strong>FigTree</strong></a>, <a href="http://www.treedyn.org/"><strong>TreeDyn</strong></a> and <a href="https://itol.embl.de/"><strong>iTOL</strong></a>, allow users to annotate the trees with colored branches, highlighted clades with tree features. However, their pre-defined annotating functions are usually limited to some specific phylogenetic data. As phylogenetic trees are becoming more widely used in multidisciplinary studies, there is an increasing need to incorporate various types of phylogenetic covariates and other associated data from different sources into the trees for visualizations and further analyses. For instance, the influenza virus has a wide host range, diverse and dynamic genotypes, and characteristic transmission behaviors that are mostly associated with the virus’s evolution and essentially among themselves. Therefore, in addition to standalone applications that focus on each of the specific analysis and data types, researchers studying molecular evolution need a robust and programmable platform that allows the high levels of integration and visualization of many of these different aspects of data (raw or from other primary analyses) over the phylogenetic trees to identify their associations and patterns.</p>
<p>To fill this gap, we developed <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> <span class="citation">(<a href="references.html#ref-yu_ggtree:_2017" role="doc-biblioref">Yu et al., 2017</a>)</span>, a package for the R programming language <span class="citation">(<a href="references.html#ref-rstats" role="doc-biblioref">R Core Team, 2016</a>)</span> released under the Bioconductor project <span class="citation">(<a href="references.html#ref-gentleman_bioconductor_2004" role="doc-biblioref">Gentleman et al., 2004</a>)</span>. The <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> is built to work with <code>treedata</code> objects (see Chapters <a href="chapter1.html#chapter1">1</a> and <a href="chapter9.html#chapter9">9</a>), and display tree graphics with the <a href="https://CRAN.R-project.org/package=ggplot2"><strong>ggplot2</strong></a> package <span class="citation">(<a href="references.html#ref-wickham_ggplot2_2016" role="doc-biblioref">Wickham, 2016</a>)</span> that was based on the grammar of graphics <span class="citation">(<a href="references.html#ref-wilkinson_grammar_2005" role="doc-biblioref">Wilkinson et al., 2005</a>)</span>.</p>
<p>The R language is increasingly used in phylogenetics. However, a comprehensive package, designed for viewing and annotating phylogenetic trees, particularly with complex data integration, is not yet available. Most of the R packages in phylogenetics focus on specific statistical analyses rather than viewing and annotating the trees with more generalized phylogeny-associated data. Some packages, including <a href="https://CRAN.R-project.org/package=ape"><strong>ape</strong></a> <span class="citation">(<a href="references.html#ref-paradis_ape_2004" role="doc-biblioref">Paradis et al., 2004</a>)</span> and <a href="https://CRAN.R-project.org/package=phytools"><strong>phytools</strong></a> <span class="citation">(<a href="references.html#ref-revell_phytools_2012" role="doc-biblioref">Revell, 2012</a>)</span>, which are capable of displaying and annotating trees, are developed using the base graphics system of R. In particular, <a href="https://CRAN.R-project.org/package=ape"><strong>ape</strong></a> is one of the fundamental packages for phylogenetic analysis and data processing. However, the base graphics system is relatively difficult to extend and limits the complexity of the tree figure to be displayed. <a href="https://CRAN.R-project.org/package=OutbreakTools"><strong>OutbreakTools</strong></a> <span class="citation">(<a href="references.html#ref-jombart_outbreaktools_2014" role="doc-biblioref">Jombart et al., 2014</a>)</span> and <a href="http://bioconductor.org/packages/phyloseq"><strong>phyloseq</strong></a> <span class="citation">(<a href="references.html#ref-mcmurdie_phyloseq_2013" role="doc-biblioref">McMurdie &amp; Holmes, 2013</a>)</span> extended <a href="https://CRAN.R-project.org/package=ggplot2"><strong>ggplot2</strong></a> to plot phylogenetic trees. The <a href="https://CRAN.R-project.org/package=ggplot2"><strong>ggplot2</strong></a> system of graphics allows rapid customization and exploration of design solutions. However, these packages were designed for epidemiology and microbiome data respectively and did not aim to provide a general solution for tree visualization and annotation. The <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> package also inherits versatile properties of <a href="https://CRAN.R-project.org/package=ggplot2"><strong>ggplot2</strong></a>, and more importantly allows constructing complex tree figures by freely combining multiple layers of annotations (see also <a href="chapter5.html#chapter5">Chapter 5</a>) using the tree associated data imported from different sources (see detailed in <a href="chapter1.html#chapter1">Chapter 1</a> and <span class="citation">(<a href="references.html#ref-wang_treeio_2020" role="doc-biblioref">Wang et al., 2020</a>)</span>).</p>
</div>
<div id="visualizing-phylogenetic-tree-with-ggtree" class="section level2" number="4.2">
<h2>
<span class="header-section-number">4.2</span> Visualizing Phylogenetic Tree with <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a><a class="anchor" aria-label="anchor" href="#visualizing-phylogenetic-tree-with-ggtree"><i class="fas fa-link"></i></a>
</h2>
<p>The <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> package <span class="citation">(<a href="references.html#ref-yu_ggtree:_2017" role="doc-biblioref">Yu et al., 2017</a>)</span> is designed for annotating phylogenetic trees with their associated data of different types and from various sources. These data could come from users or analysis programs and might include evolutionary rates, ancestral sequences, <em>etc.</em> that are associated with the taxa from real samples, or with the internal nodes representing hypothetic ancestor strain/species, or with the tree branches indicating evolutionary time courses <span class="citation">(<a href="references.html#ref-wang_treeio_2020" role="doc-biblioref">Wang et al., 2020</a>)</span>. For instance, the data could be the geographic positions of the sampled avian influenza viruses (informed by the survey locations) and the ancestral nodes (by phylogeographic inference) in the viral gene tree <span class="citation">(<a href="references.html#ref-lam_phylodynamics_2012" role="doc-biblioref">Lam et al., 2012</a>)</span>.</p>
<p>The <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> supports <a href="https://CRAN.R-project.org/package=ggplot2"><strong>ggplot2</strong></a>’s graphical language, which allows a high level of customization, is intuitive and flexible. Notably, <a href="https://CRAN.R-project.org/package=ggplot2"><strong>ggplot2</strong></a> itself does not provide low-level geometric objects or other support for tree-like structures, and hence <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> is a useful extension in that regard. Even though the other two phylogenetics-related R packages, <a href="https://CRAN.R-project.org/package=OutbreakTools"><strong>OutbreakTools</strong></a>, and <a href="http://bioconductor.org/packages/phyloseq"><strong>phyloseq</strong></a>, are developed based on <a href="https://CRAN.R-project.org/package=ggplot2"><strong>ggplot2</strong></a>, the most valuable part of the <a href="https://CRAN.R-project.org/package=ggplot2"><strong>ggplot2</strong></a> syntax - adding layers of annotations - is not supported in these packages. For example, if we have plotted a tree without taxa labels, <a href="https://CRAN.R-project.org/package=OutbreakTools"><strong>OutbreakTools</strong></a> and <a href="http://bioconductor.org/packages/phyloseq"><strong>phyloseq</strong></a> provide no easy way for general <code>R</code> users, who have little knowledge about the infrastructures of these packages, to add a layer of taxa labels. The <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> extends <a href="https://CRAN.R-project.org/package=ggplot2"><strong>ggplot2</strong></a> to support tree objects by implementing a geometric layer, <code><a href="https://rdrr.io/pkg/ggtree/man/geom_tree.html">geom_tree()</a></code>, to support visualizing tree structure. In <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a>, viewing a phylogenetic tree is relatively easy, via the command <code>ggplot(tree_object) + geom_tree() + theme_tree()</code> or <code>ggtree(tree_object)</code> for short. Layers of annotations can be added one-by-one via the <code>+</code> operator. To facilitate tree visualization, <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> provides several geometric layers, including <code><a href="https://rdrr.io/pkg/ggtree/man/geom_treescale.html">geom_treescale()</a></code> for adding legend of tree branch scale (genetic distance, divergence time, <em>etc.</em>), <code><a href="https://rdrr.io/pkg/ggtree/man/geom_range.html">geom_range()</a></code> for displaying uncertainty of branch lengths (confidence interval or range, <em>etc.</em>), <code><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab()</a></code> for adding taxa label, <code><a href="https://rdrr.io/pkg/ggtree/man/geom_tippoint.html">geom_tippoint()</a></code> and <code><a href="https://rdrr.io/pkg/ggtree/man/geom_nodepoint.html">geom_nodepoint()</a></code> for adding symbols of tips and internal nodes, <code><a href="https://rdrr.io/pkg/ggtree/man/geom-hilight.html">geom_hilight()</a></code> for highlighting clades with rectangle, and <code><a href="https://rdrr.io/pkg/ggtree/man/geom_cladelab.html">geom_cladelab()</a></code> for annotating selected clades with bar and text label, <em>etc.</em>. A full list of geometric layers provided by <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> is summarized in Table <a href="chapter5.html#tab:geoms">5.1</a>.</p>
<p>To view a phylogenetic tree, we first need to parse the tree file into <code>R</code>.
The <a href="http://bioconductor.org/packages/treeio"><strong>treeio</strong></a> package is able to parse diverse annotation data from different software outputs into <code>S4</code> phylogenetic data objects (see also <a href="chapter1.html#chapter1">Chapter 1</a>). The <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> package mainly utilizes these <code>S4</code> objects to display and annotate the tree. Other R packages defined <code>S3</code>/<code>S4</code> classes to store phylogenetic trees with domain-specific associated data, including <code>phylo4</code> and <code>phylo4d</code> defined in the <a href="https://CRAN.R-project.org/package=phylobase"><strong>phylobase</strong></a> package, <code>obkdata</code> defined in the <a href="https://CRAN.R-project.org/package=OutbreakTools"><strong>OutbreakTools</strong></a> package, and <code>phyloseq</code> defined in the <a href="http://bioconductor.org/packages/phyloseq"><strong>phyloseq</strong></a> package, <em>etc</em>. All these tree objects are also supported in <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> and their specific annotation data can be used to annotate the tree directly in <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> (see also <a href="chapter9.html#chapter9">Chapter 9</a>). Such compatibility of <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> facilitates the integration of data and analysis results. In addition, <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> also supports other tree-like structures, including <a href="chapter9.html#dendrogram">dendrogram</a> and <a href="chapter9.html#igraph">tree graphs</a>.</p>
<div id="basic-tree-visualization" class="section level3" number="4.2.1">
<h3>
<span class="header-section-number">4.2.1</span> Basic Tree Visualization<a class="anchor" aria-label="anchor" href="#basic-tree-visualization"><i class="fas fa-link"></i></a>
</h3>
<p>The <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> package extends <a href="https://CRAN.R-project.org/package=ggplot2"><strong>ggplot2</strong></a> <span class="citation">(<a href="references.html#ref-wickham_ggplot2_2016" role="doc-biblioref">Wickham, 2016</a>)</span> package to support viewing a phylogenetic tree.
It implements <code><a href="https://rdrr.io/pkg/ggtree/man/geom_tree.html">geom_tree()</a></code> layer for displaying a phylogenetic tree, as shown below in Figure <a href="chapter4.html#fig:basicviz">4.1</a>A.</p>
<div class="sourceCode" id="cb131"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/YuLab-SMU/treeio">"treeio"</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://yulab-smu.top/treedata-book/">"ggtree"</a></span><span class="op">)</span>

<span class="va">nwk</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"sample.nwk"</span>, package<span class="op">=</span><span class="st">"treeio"</span><span class="op">)</span>
<span class="va">tree</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ape/man/read.tree.html">read.tree</a></span><span class="op">(</span><span class="va">nwk</span><span class="op">)</span>

<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">tree</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span><span class="va">x</span>, <span class="va">y</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tree.html">geom_tree</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/theme_tree.html">theme_tree</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p>The function, <code><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree()</a></code>, was implemented as a shortcut to visualize a tree, and it works exactly the same as shown above.</p>
<p>The <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> package takes all the advantages of <a href="https://CRAN.R-project.org/package=ggplot2"><strong>ggplot2</strong></a>. For example, we can change the color, size, and type of the lines as we do with <a href="https://CRAN.R-project.org/package=ggplot2"><strong>ggplot2</strong></a> (Figure <a href="chapter4.html#fig:basicviz">4.1</a>B).</p>
<div class="sourceCode" id="cb132"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree</span>, color<span class="op">=</span><span class="st">"firebrick"</span>, size<span class="op">=</span><span class="fl">2</span>, linetype<span class="op">=</span><span class="st">"dotted"</span><span class="op">)</span></code></pre></div>
<p>By default, the tree is viewed in ladderize form, user can set the parameter <code>ladderize = FALSE</code> to disable it (Figure <a href="chapter4.html#fig:basicviz">4.1</a>C, see also <a href="faq.html#branch-setting">FAQ A.5</a>).</p>
<div class="sourceCode" id="cb133"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree</span>, ladderize<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<p>The <code>branch.length</code> is used to scale the edge, user can set the parameter <code>branch.length = "none"</code> to only view the tree topology (cladogram, Figure <a href="chapter4.html#fig:basicviz">4.1</a>D) or other numerical variables to scale the tree (<em>e.g.</em>, <em>d<sub>N</sub>/d<sub>S</sub></em>, see also in <a href="chapter5.html#chapter5">Chapter 5</a>).</p>
<div class="sourceCode" id="cb134"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree</span>, branch.length<span class="op">=</span><span class="st">"none"</span><span class="op">)</span></code></pre></div>


<div class="figure">
<span style="display:block;" id="fig:basicviz"></span>
<img src="treedata_files/figure-html/basicviz-1.svg" alt="Basic tree visualization. Default ggtree output with ladderized effect (A), non-variable setting (e.g., color, size, line type) (B), non-ladderized tree (C), cladogram that only displays tree topology without branch length information (D)." width="100%"><p class="caption">
FIGURE 4.1: <strong>Basic tree visualization.</strong> Default ggtree output with ladderized effect (A), non-variable setting (<em>e.g.</em>, color, size, line type) (B), non-ladderized tree (C), cladogram that only displays tree topology without branch length information (D).
</p>
</div>
</div>
<div id="tree-layouts" class="section level3" number="4.2.2">
<h3>
<span class="header-section-number">4.2.2</span> Layouts of a phylogenetic tree<a class="anchor" aria-label="anchor" href="#tree-layouts"><i class="fas fa-link"></i></a>
</h3>
<p>Viewing phylogenetic with <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> is quite simple, just pass the tree object to the <code><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree()</a></code> function. We have developed several types of layouts for tree presentation (Figure <a href="chapter4.html#fig:layout">4.2</a>), including <em>rectangular</em> (by default), <em>roundrect</em> (rounded rectangular), <em>ellipse</em>, <em>slanted</em>, <em>circular</em>, <em>fan</em>, <em>unrooted</em> (equal angle and daylight methods), time-scaled, and two-dimensional layouts.</p>
<p>Here are examples of visualizing a tree with different layouts:</p>
<div class="sourceCode" id="cb135"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/treedata-book/">ggtree</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">2017</span><span class="op">-</span><span class="fl">02</span><span class="op">-</span><span class="fl">16</span><span class="op">)</span>
<span class="va">tree</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ape/man/rtree.html">rtree</a></span><span class="op">(</span><span class="fl">50</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree</span>, layout<span class="op">=</span><span class="st">"roundrect"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree</span>, layout<span class="op">=</span><span class="st">"slanted"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree</span>, layout<span class="op">=</span><span class="st">"ellipse"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree</span>, layout<span class="op">=</span><span class="st">"circular"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree</span>, layout<span class="op">=</span><span class="st">"fan"</span>, open.angle<span class="op">=</span><span class="fl">120</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree</span>, layout<span class="op">=</span><span class="st">"equal_angle"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree</span>, layout<span class="op">=</span><span class="st">"daylight"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree</span>, branch.length<span class="op">=</span><span class="st">'none'</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree</span>, layout<span class="op">=</span><span class="st">"ellipse"</span>, branch.length<span class="op">=</span><span class="st">"none"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree</span>, branch.length<span class="op">=</span><span class="st">'none'</span>, layout<span class="op">=</span><span class="st">'circular'</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree</span>, layout<span class="op">=</span><span class="st">"daylight"</span>, branch.length <span class="op">=</span> <span class="st">'none'</span><span class="op">)</span></code></pre></div>


<div class="figure">
<span style="display:block;" id="fig:layout"></span>
<img src="treedata_files/figure-html/layout-1.svg" alt="Tree layouts. Phylogram: rectangular layout (A), rounded rectangular layout (B), slanted layout (C), ellipse layout (D), circular layout (E), and fan layout (F). Unrooted: equal-angle method (G) and daylight method (H). Cladogram: rectangular layout (I), ellipse (J), circular layout (K), and unrooted layout (L). Slanted and fan layouts for cladogram are also supported." width="100%"><p class="caption">
FIGURE 4.2: <strong>Tree layouts.</strong> Phylogram: rectangular layout (A), rounded rectangular layout (B), slanted layout (C), ellipse layout (D), circular layout (E), and fan layout (F). Unrooted: equal-angle method (G) and daylight method (H). Cladogram: rectangular layout (I), ellipse (J), circular layout (K), and unrooted layout (L). Slanted and fan layouts for cladogram are also supported.
</p>
</div>
<p>Other possible layouts that can be drawn by modifying
scales/coordination (Figure <a href="chapter4.html#fig:layout2">4.3</a>).</p>
<!-- for examples, [reverse label of time
scale](https://github.com/GuangchuangYu/ggtree/issues/87), [repropotion
circular/fan tree](
https://groups.google.com/d/msg/bioc-ggtree/UoGQekWHIvw/ZswUUZKSGwAJ), *etc.*. -->
<div class="sourceCode" id="cb136"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_x_reverse</span><span class="op">(</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree</span><span class="op">)</span> <span class="op">+</span> <span class="fu">coord_flip</span><span class="op">(</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/tree-layout.html">layout_dendrogram</a></span><span class="op">(</span><span class="op">)</span>
<span class="fu">ggplotify</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ggplotify/man/as.ggplot.html">as.ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree</span><span class="op">)</span>, angle<span class="op">=</span><span class="op">-</span><span class="fl">30</span>, scale<span class="op">=</span><span class="fl">.9</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree</span>, layout<span class="op">=</span><span class="st">'slanted'</span><span class="op">)</span> <span class="op">+</span> <span class="fu">coord_flip</span><span class="op">(</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree</span>, layout<span class="op">=</span><span class="st">'slanted'</span>, branch.length<span class="op">=</span><span class="st">'none'</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/tree-layout.html">layout_dendrogram</a></span><span class="op">(</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree</span>, layout<span class="op">=</span><span class="st">'circular'</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html">xlim</a></span><span class="op">(</span><span class="op">-</span><span class="fl">10</span>, <span class="cn">NA</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/tree-layout.html">layout_inward_circular</a></span><span class="op">(</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/tree-layout.html">layout_inward_circular</a></span><span class="op">(</span>xlim<span class="op">=</span><span class="fl">15</span><span class="op">)</span></code></pre></div>


<div class="figure">
<span style="display:block;" id="fig:layout2"></span>
<img src="treedata_files/figure-html/layout2-1.svg" alt="Derived Tree layouts. right-to-left rectangular layout (A), bottom-up rectangular layout (B), top-down rectangular layout (Dendrogram) (C), rotated rectangular layout (D), bottom-up slanted layout (E), top-down slanted layout (Cladogram) (F), circular layout (G), circular inward layout (H and I)." width="100%"><p class="caption">
FIGURE 4.3: <strong>Derived Tree layouts.</strong> right-to-left rectangular layout (A), bottom-up rectangular layout (B), top-down rectangular layout (Dendrogram) (C), rotated rectangular layout (D), bottom-up slanted layout (E), top-down slanted layout (Cladogram) (F), circular layout (G), circular inward layout (H and I).
</p>
</div>
<p><strong>Phylogram.</strong> Layouts of <em>rectangular</em>, <em>roundrect</em>, <em>slanted</em>, <em>ellipse</em>, <em>circular</em>, and <em>fan</em> are supported to visualize phylogram (by default, with branch length scaled) as demonstrated in Figures <a href="chapter4.html#fig:layout">4.2</a>A-F.</p>
<p><strong>Unrooted layout.</strong> Unrooted (also called ‘radial’) layout is supported by equal-angle and daylight algorithms; users can specify unrooted layout algorithm by passing “equal_angle” or “daylight” to <code>layout</code> parameter to visualize the tree. The equal-angle method was proposed by Christopher Meacham in <em>PLOTREE</em>, which was incorporated in <a href="https://evolution.genetics.washington.edu/phylip.html"><strong>PHYLIP</strong></a> <span class="citation">(<a href="references.html#ref-retief_phylogenetic_2000" role="doc-biblioref">Retief, 2000</a>)</span>. This method starts from the root of the tree and allocates arcs of angle to each subtree proportional to the number of tips in it. It iterates from root to tips and subdivides the angle allocated to a subtree into angles for its dependent subtrees. This method is fast and was implemented in many software packages. As shown in Figure <a href="chapter4.html#fig:layout">4.2</a>G, the equal angle method has a drawback that tips tend to be clustered together, which will leave many spaces unused. The daylight method starts from an initial tree built by equal angle and iteratively improves it by successively going to each interior node and swinging subtrees so that the arcs of “daylight” are equal (Figure <a href="chapter4.html#fig:layout">4.2</a>H). This method was firstly implemented in <a href="https://paup.phylosolutions.com/"><strong>PAUP*</strong></a> <span class="citation">(<a href="references.html#ref-wilgenbusch_inferring_2003" role="doc-biblioref">Wilgenbusch &amp; Swofford, 2003</a>)</span>.</p>
<p><strong>Cladogram.</strong> To visualize a cladogram that is without branch length scaling and only displays the tree structure, <code>branch.length</code> is set to “none” and it works for all types of layouts (Figures <a href="chapter4.html#fig:layout">4.2</a>I-L).</p>
<p><strong>Timescaled layout.</strong> For a timescaled tree, the most recent sampling date must be specified via the <code>mrsd</code> parameter, and <code><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree()</a></code> will scale the tree by sampling (tip) and divergence (internal node) time, and a timescale axis will be displayed under the tree by default. Users can use the <a href="https://CRAN.R-project.org/package=deeptime"><strong>deeptime</strong></a> package to add geologic timescale (e.g., periods and eras) to a <code><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree()</a></code> plot.</p>


<div class="sourceCode" id="cb137"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">beast_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"examples/MCC_FluA_H3.tree"</span>, 
                          package<span class="op">=</span><span class="st">"ggtree"</span><span class="op">)</span>
<span class="va">beast_tree</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://docs.ropensci.org/treeio/reference/beast-parser.html">read.beast</a></span><span class="op">(</span><span class="va">beast_file</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">beast_tree</span>, mrsd<span class="op">=</span><span class="st">"2013-01-01"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/theme_tree2.html">theme_tree2</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:timescaled"></span>
<img src="treedata_files/figure-html/timescaled-1.svg" alt="Timescaled layout. The x-axis is the timescale (in units of the year). The divergence time in this example was inferred by BEAST using the molecular clock model." width="98%"><p class="caption">
FIGURE 4.4: <strong>Timescaled layout.</strong> The <em>x</em>-axis is the timescale (in units of the year). The divergence time in this example was inferred by <a href="http://beast2.org/"><strong>BEAST</strong></a> using the molecular clock model.
</p>
</div>
<p><strong>Two-dimensional tree layout.</strong> A two-dimensional tree is a projection of the phylogenetic tree in a space defined by the associated phenotype (numerical or categorical trait, on the <em>y</em>-axis) and tree branch scale (<em>e.g.</em>, evolutionary distance, divergent time, on the <em>x</em>-axis). The phenotype can be a measure of certain biological characteristics of the taxa and hypothetical ancestors in the tree. This layout is useful to track the virus phenotypes or other behaviors (<em>y</em>-axis) changing with the virus evolution (<em>x</em>-axis). In fact, the analysis of phenotypes or genotypes over evolutionary time have been widely used for study of influenza virus evolution <span class="citation">(<a href="references.html#ref-neher_prediction_2016" role="doc-biblioref">Neher et al., 2016</a>)</span>, though such analysis diagrams are not tree-like, <em>i.e.</em>, no connection between data points, unlike our two-dimensional tree layout that connects data points with the corresponding tree branches. Therefore, this new layout we provided will make such data analysis easier and more scalable for large sequence datasets.</p>
<p>In this example, we used the previous timescaled tree of H3 human and swine influenza viruses (Figure <a href="chapter4.html#fig:timescaled">4.4</a>; data published in <span class="citation">(<a href="references.html#ref-liang_expansion_2014" role="doc-biblioref">Liang et al., 2014</a>)</span>) and scaled the <em>y</em>-axis based on the predicted <em>N</em>-linked glycosylation sites (NLG) for each of the taxon and ancestral sequences of hemagglutinin proteins. The NLG sites were predicted using the <a href="http://www.cbs.dtu.dk/services/NetNGlyc/">NetNGlyc 1.0 Server</a>. To scale the <em>y</em>-axis, the parameter <code>yscale</code> in the <code><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree()</a></code> function is set to a numerical or categorical variable. If <code>yscale</code> is a categorical variable as in this example, users should specify how the categories are to be mapped to numerical values via the <code>yscale_mapping</code> variables.</p>


<div class="sourceCode" id="cb138"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">NAG_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"examples/NAG_inHA1.txt"</span>, package<span class="op">=</span><span class="st">"ggtree"</span><span class="op">)</span>

<span class="va">NAG.df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span><span class="va">NAG_file</span>, sep<span class="op">=</span><span class="st">"\t"</span>, header<span class="op">=</span><span class="cn">FALSE</span>, 
                     stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">NAG</span> <span class="op">&lt;-</span> <span class="va">NAG.df</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">NAG</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">NAG.df</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span>

<span class="co">## separate the tree by host species</span>
<span class="va">tip</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ape/man/as.phylo.html">as.phylo</a></span><span class="op">(</span><span class="va">beast_tree</span><span class="op">)</span><span class="op">$</span><span class="va">tip.label</span>
<span class="va">beast_tree</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/tidytree/man/groupOTU.html">groupOTU</a></span><span class="op">(</span><span class="va">beast_tree</span>, <span class="va">tip</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span><span class="op">(</span><span class="st">"Swine"</span>, <span class="va">tip</span><span class="op">)</span><span class="op">]</span>, 
                       group_name <span class="op">=</span> <span class="st">"host"</span><span class="op">)</span>

<span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">beast_tree</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>color<span class="op">=</span><span class="va">host</span><span class="op">)</span>, mrsd<span class="op">=</span><span class="st">"2013-01-01"</span>, 
            yscale <span class="op">=</span> <span class="st">"label"</span>, yscale_mapping <span class="op">=</span> <span class="va">NAG</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">theme_classic</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.position<span class="op">=</span><span class="st">'none'</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_color_manual</a></span><span class="op">(</span>values<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"blue"</span>, <span class="st">"red"</span><span class="op">)</span>, 
                     labels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"human"</span>, <span class="st">"swine"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">ylab</span><span class="op">(</span><span class="st">"Number of predicted N-linked glycosylation sites"</span><span class="op">)</span>

<span class="co">## (optional) add more annotations to help interpretation</span>
<span class="va">p</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_nodepoint.html">geom_nodepoint</a></span><span class="op">(</span>color<span class="op">=</span><span class="st">"grey"</span>, size<span class="op">=</span><span class="fl">3</span>, alpha<span class="op">=</span><span class="fl">.8</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_rootpoint.html">geom_rootpoint</a></span><span class="op">(</span>color<span class="op">=</span><span class="st">"black"</span>, size<span class="op">=</span><span class="fl">3</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tippoint.html">geom_tippoint</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">3</span>, alpha<span class="op">=</span><span class="fl">.5</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">annotate</span><span class="op">(</span><span class="st">"point"</span>, <span class="fl">1992</span>, <span class="fl">5.6</span>, size<span class="op">=</span><span class="fl">3</span>, color<span class="op">=</span><span class="st">"black"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">annotate</span><span class="op">(</span><span class="st">"point"</span>, <span class="fl">1992</span>, <span class="fl">5.4</span>, size<span class="op">=</span><span class="fl">3</span>, color<span class="op">=</span><span class="st">"grey"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">annotate</span><span class="op">(</span><span class="st">"point"</span>, <span class="fl">1991.6</span>, <span class="fl">5.2</span>, size<span class="op">=</span><span class="fl">3</span>, color<span class="op">=</span><span class="st">"blue"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">annotate</span><span class="op">(</span><span class="st">"point"</span>, <span class="fl">1992</span>, <span class="fl">5.2</span>, size<span class="op">=</span><span class="fl">3</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">annotate</span><span class="op">(</span><span class="st">"text"</span>, <span class="fl">1992.3</span>, <span class="fl">5.6</span>, hjust<span class="op">=</span><span class="fl">0</span>, size<span class="op">=</span><span class="fl">4</span>, label<span class="op">=</span><span class="st">"Root node"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">annotate</span><span class="op">(</span><span class="st">"text"</span>, <span class="fl">1992.3</span>, <span class="fl">5.4</span>, hjust<span class="op">=</span><span class="fl">0</span>, size<span class="op">=</span><span class="fl">4</span>, 
           label<span class="op">=</span><span class="st">"Internal nodes"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">annotate</span><span class="op">(</span><span class="st">"text"</span>, <span class="fl">1992.3</span>, <span class="fl">5.2</span>, hjust<span class="op">=</span><span class="fl">0</span>, size<span class="op">=</span><span class="fl">4</span>,
           label<span class="op">=</span><span class="st">"Tip nodes (blue: human; red: swine)"</span><span class="op">)</span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:2d"></span>
<img src="treedata_files/figure-html/2d-1.svg" alt="Two-dimensional tree layout. The trunk and other branches are highlighted in red (for swine) and blue (for humans). The x-axis is scaled to the branch length (in units of year) of the timescaled tree. The y-axis is scaled to the node attribute variable, in this case, the number of predicted N-linked glycosylation sites (NLG) on the hemagglutinin protein. Colored circles indicate the different types of tree nodes. Note that nodes assigned the same x- (temporal) and y- (NLG) coordinates are superimposed in this representation and appear as one node, which is shaded based on the colors of all the nodes at that point." width="100%"><p class="caption">
FIGURE 4.5: <strong>Two-dimensional tree layout.</strong> The trunk and other branches are highlighted in red (for swine) and blue (for humans). The <em>x</em>-axis is scaled to the branch length (in units of year) of the timescaled tree. The <em>y</em>-axis is scaled to the node attribute variable, in this case, the number of predicted <em>N</em>-linked glycosylation sites (NLG) on the hemagglutinin protein. Colored circles indicate the different types of tree nodes. Note that nodes assigned the same <em>x</em>- (temporal) and <em>y</em>- (NLG) coordinates are superimposed in this representation and appear as one node, which is shaded based on the colors of all the nodes at that point.
</p>
</div>
<p>As shown in Figure <a href="chapter4.html#fig:2d">4.5</a>, a two-dimensional tree is good at visualizing the change of phenotype over the evolution in the phylogenetic tree. In this example, it is shown that the H3 gene of the human influenza A virus maintained a high level of <em>N</em>-linked glycosylation sites (n=8 to 9) over the last two decades and dropped significantly to 5 or 6 in a separate viral lineage transmitted to swine populations and established there. It was indeed hypothesized that the human influenza virus with a high level of glycosylation on the viral hemagglutinin protein provides a better shielding effect to protect the antigenic sites from exposure to the herd immunity, and thus has a selective advantage in human populations that maintain a high level of herd immunity against the circulating human influenza virus strains. For the viral lineage that newly jumped across the species barrier and transmitted to the swine population, the shielding effect of the high-level surface glycan oppositely imposes selective disadvantage because the receptor-binding domain may also be shielded which greatly affects the viral fitness of the lineage that newly adapted to a new host species. Another example of a two-dimensional tree can be found in Figure <a href="chapter4.html#fig:phenogram">4.12</a>.</p>
</div>
</div>
<div id="displaying-tree-components" class="section level2" number="4.3">
<h2>
<span class="header-section-number">4.3</span> Displaying Tree Components<a class="anchor" aria-label="anchor" href="#displaying-tree-components"><i class="fas fa-link"></i></a>
</h2>
<div id="geom-trescale" class="section level3" number="4.3.1">
<h3>
<span class="header-section-number">4.3.1</span> Displaying treescale (evolution distance)<a class="anchor" aria-label="anchor" href="#geom-trescale"><i class="fas fa-link"></i></a>
</h3>
<p>To show treescale, the user can use <code><a href="https://rdrr.io/pkg/ggtree/man/geom_treescale.html">geom_treescale()</a></code> layer (Figures <a href="chapter4.html#fig:treescale">4.6</a>A-C).</p>
<div class="sourceCode" id="cb139"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_treescale.html">geom_treescale</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><code><a href="https://rdrr.io/pkg/ggtree/man/geom_treescale.html">geom_treescale()</a></code> supports the following parameters:</p>
<ul>
<li>
<em>x</em> and <em>y</em> for treescale position</li>
<li>
<em>width</em> for the length of the treescale</li>
<li>
<em>fontsize</em> for the size of the text</li>
<li>
<em>linesize</em> for the size of the line</li>
<li>
<em>offset</em> for relative position of the line and the text</li>
<li>
<em>color</em> for color of the treescale</li>
</ul>
<div class="sourceCode" id="cb140"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_treescale.html">geom_treescale</a></span><span class="op">(</span>x<span class="op">=</span><span class="fl">0</span>, y<span class="op">=</span><span class="fl">45</span>, width<span class="op">=</span><span class="fl">1</span>, color<span class="op">=</span><span class="st">'red'</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_treescale.html">geom_treescale</a></span><span class="op">(</span>fontsize<span class="op">=</span><span class="fl">6</span>, linesize<span class="op">=</span><span class="fl">2</span>, offset<span class="op">=</span><span class="fl">1</span><span class="op">)</span></code></pre></div>
<p>We can also use <code><a href="https://rdrr.io/pkg/ggtree/man/theme_tree2.html">theme_tree2()</a></code> to display the treescale by adding <em>x axis</em> (Figure <a href="chapter4.html#fig:treescale">4.6</a>D).</p>
<div class="sourceCode" id="cb141"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/theme_tree2.html">theme_tree2</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>


<div class="figure">
<span style="display:block;" id="fig:treescale"></span>
<img src="treedata_files/figure-html/treescale-1.svg" alt="Display treescale. geom_treescale() automatically add a scale bar for evolutionary distance (A). Users can modify the color, width, and position of the scale (B) as well as the size of the scale bar and text and their relative position (C). Another possible solution is to enable the x-axis which is useful for the timescaled tree (D)." width="100%"><p class="caption">
FIGURE 4.6: <strong>Display treescale.</strong> <code><a href="https://rdrr.io/pkg/ggtree/man/geom_treescale.html">geom_treescale()</a></code> automatically add a scale bar for evolutionary distance (A). Users can modify the color, width, and position of the scale (B) as well as the size of the scale bar and text and their relative position (C). Another possible solution is to enable the <em>x</em>-axis which is useful for the timescaled tree (D).
</p>
</div>
<p>Treescale is not restricted to evolution distance, <a href="http://bioconductor.org/packages/treeio"><strong>treeio</strong></a> can rescale the tree with other numerical variables (details described in <a href="chapter2.html#rescale-treeio">session 2.4</a>), and <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> allows users to specify a numerical variable to serve as branch length for visualization (details described in <a href="chapter4.html#rescale-tree">session 4.3</a>).</p>
</div>
<div id="geom-nodepoint" class="section level3" number="4.3.2">
<h3>
<span class="header-section-number">4.3.2</span> Displaying nodes/tips<a class="anchor" aria-label="anchor" href="#geom-nodepoint"><i class="fas fa-link"></i></a>
</h3>
<p>Showing all the internal nodes and tips in the tree can be done by adding a layer of points using <code><a href="https://rdrr.io/pkg/ggtree/man/geom_nodepoint.html">geom_nodepoint()</a></code>, <code><a href="https://rdrr.io/pkg/ggtree/man/geom_tippoint.html">geom_tippoint()</a></code>, or <code><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point()</a></code> (Figure <a href="chapter4.html#fig:nodeTip">4.7</a>).</p>
<div class="sourceCode" id="cb142"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>shape<span class="op">=</span><span class="va">isTip</span>, color<span class="op">=</span><span class="va">isTip</span><span class="op">)</span>, size<span class="op">=</span><span class="fl">3</span><span class="op">)</span>

<span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_nodepoint.html">geom_nodepoint</a></span><span class="op">(</span>color<span class="op">=</span><span class="st">"#b5e521"</span>, alpha<span class="op">=</span><span class="fl">1</span><span class="op">/</span><span class="fl">4</span>, size<span class="op">=</span><span class="fl">10</span><span class="op">)</span> 
<span class="va">p</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tippoint.html">geom_tippoint</a></span><span class="op">(</span>color<span class="op">=</span><span class="st">"#FDAC4F"</span>, shape<span class="op">=</span><span class="fl">8</span>, size<span class="op">=</span><span class="fl">3</span><span class="op">)</span></code></pre></div>


<div class="figure">
<span style="display:block;" id="fig:nodeTip"></span>
<img src="treedata_files/figure-html/nodeTip-1.svg" alt="Display external and internal nodes. geom_point() automatically add symbolic points of all nodes (A). geom_nodepoint() adds symbolic points for internal nodes and geom_tippoint() adds symbolic points for external nodes (B)." width="100%"><p class="caption">
FIGURE 4.7: <strong>Display external and internal nodes.</strong> <code><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point()</a></code> automatically add symbolic points of all nodes (A). <code><a href="https://rdrr.io/pkg/ggtree/man/geom_nodepoint.html">geom_nodepoint()</a></code> adds symbolic points for internal nodes and <code><a href="https://rdrr.io/pkg/ggtree/man/geom_tippoint.html">geom_tippoint()</a></code> adds symbolic points for external nodes (B).
</p>
</div>
</div>
<div id="displaying-labels" class="section level3" number="4.3.3">
<h3>
<span class="header-section-number">4.3.3</span> Displaying labels<a class="anchor" aria-label="anchor" href="#displaying-labels"><i class="fas fa-link"></i></a>
</h3>
<p>Users can use <code><a href="https://ggplot2.tidyverse.org/reference/geom_text.html">geom_text()</a></code> or <code><a href="https://ggplot2.tidyverse.org/reference/geom_text.html">geom_label()</a></code> to display the node (if available) and tip labels simultaneously or <code><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab()</a></code> to only display tip labels (Figure <a href="chapter4.html#fig:tiplab">4.8</a>A).</p>
<div class="sourceCode" id="cb143"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">3</span>, color<span class="op">=</span><span class="st">"purple"</span><span class="op">)</span></code></pre></div>
<p>The <code><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab()</a></code> layer not only supports using <em>text</em> or <em>label</em> geom to display labels, but
it also supports <em>image</em> geom to label tip with image files (see <a href="chapter7.html#chapter7">Chapter 7</a>). A corresponding
geom, <code><a href="https://rdrr.io/pkg/ggtree/man/geom_nodelab.html">geom_nodelab()</a></code> is also provided for displaying node labels.</p>
<p>For <em>circular</em> and <em>unrooted</em> layouts, <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> supports rotating node labels according to the angles of the branches (Figure <a href="chapter4.html#fig:tiplab">4.8</a>B).</p>
<div class="sourceCode" id="cb144"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree</span>, layout<span class="op">=</span><span class="st">"circular"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>angle<span class="op">=</span><span class="va">angle</span><span class="op">)</span>, color<span class="op">=</span><span class="st">'blue'</span><span class="op">)</span></code></pre></div>
<p>For long tip labels, the label may be truncated. There are several ways to solve this issue (see <a href="faq.html#faq-label-truncated">FAQ: Tip label truncated</a>). Another solution to solve this issue is to display tip labels as <em>y</em>-axis labels (Figure <a href="chapter4.html#fig:tiplab">4.8</a>C). However, it only works for rectangular and dendrogram layouts and users need to use <code><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme()</a></code> to adjust tip labels in this case.</p>
<div class="sourceCode" id="cb145"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab</a></span><span class="op">(</span>as_ylab<span class="op">=</span><span class="cn">TRUE</span>, color<span class="op">=</span><span class="st">'firebrick'</span><span class="op">)</span></code></pre></div>


<div class="figure">
<span style="display:block;" id="fig:tiplab"></span>
<img src="treedata_files/figure-html/tiplab-1.png" alt="Display tip labels. geom_tiplab() supports displaying tip labels (A). For the circular, fan, or unrooted tree layouts, the labels can be rotated to fit the angle of the branches (B). For dendrogram/rectangular layout, tip labels can be displayed as y-axis labels (C)." width="100%"><p class="caption">
FIGURE 4.8: <strong>Display tip labels.</strong> <code><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab()</a></code> supports displaying tip labels (A). For the circular, fan, or unrooted tree layouts, the labels can be rotated to fit the angle of the branches (B). For dendrogram/rectangular layout, tip labels can be displayed as <em>y</em>-axis labels (C).
</p>
</div>
<p>By default, the positions to display text are based on the node positions; we can change them to be based on the middle of the branch/edge (by setting <code>aes(x = branch)</code>), which is very useful when annotating transition from the parent node to the child node.</p>
</div>
<div id="displaying-root-edge" class="section level3" number="4.3.4">
<h3>
<span class="header-section-number">4.3.4</span> Displaying root-edge<a class="anchor" aria-label="anchor" href="#displaying-root-edge"><i class="fas fa-link"></i></a>
</h3>
<p>The <code><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree()</a></code> doesn’t plot the root-edge by default. Users can use <code><a href="https://rdrr.io/pkg/ggtree/man/geom_rootedge.html">geom_rootedge()</a></code> to automatically display the root-edge (Figure <a href="chapter4.html#fig:rootedge">4.9</a>A). If there is no root edge information, <code><a href="https://rdrr.io/pkg/ggtree/man/geom_rootedge.html">geom_rootedge()</a></code> will display nothing by default (Figure <a href="chapter4.html#fig:rootedge">4.9</a>B). Users can set the root-edge to the tree (Figure <a href="chapter4.html#fig:rootedge">4.9</a>C) or specify <code>rootedge</code> in <code><a href="https://rdrr.io/pkg/ggtree/man/geom_rootedge.html">geom_rootedge()</a></code> (Figure <a href="chapter4.html#fig:rootedge">4.9</a>D). A long root length is useful to increase readability of the circular tree (see also <a href="faq.html#faq-enlarge-center-space">FAQ: Enlarge center space</a>).</p>
<div class="sourceCode" id="cb146"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## with root-edge = 1</span>
<span class="va">tree1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ape/man/read.tree.html">read.tree</a></span><span class="op">(</span>text<span class="op">=</span><span class="st">'((A:1,B:2):3,C:2):1;'</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree1</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_rootedge.html">geom_rootedge</a></span><span class="op">(</span><span class="op">)</span>

<span class="co">## without root-edge</span>
<span class="va">tree2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ape/man/read.tree.html">read.tree</a></span><span class="op">(</span>text<span class="op">=</span><span class="st">'((A:1,B:2):3,C:2);'</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree2</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_rootedge.html">geom_rootedge</a></span><span class="op">(</span><span class="op">)</span>

<span class="co">## setting root-edge</span>
<span class="va">tree2</span><span class="op">$</span><span class="va">root.edge</span> <span class="op">&lt;-</span> <span class="fl">2</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree2</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_rootedge.html">geom_rootedge</a></span><span class="op">(</span><span class="op">)</span>

<span class="co">## specify the length of root edge for just plotting</span>
<span class="co">## this will ignore tree$root.edge</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree2</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_rootedge.html">geom_rootedge</a></span><span class="op">(</span>rootedge <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></code></pre></div>


<div class="figure">
<span style="display:block;" id="fig:rootedge"></span>
<img src="treedata_files/figure-html/rootedge-1.svg" alt="Display root-edge. geom_rootedge() supports displaying root-edge if the root edge was presented (A). It shows nothing if there is no root-edge (B). In this case, users can manually set the root edge for the tree (C) or just specify the length of the root for plotting (D)." width="98%"><p class="caption">
FIGURE 4.9: <strong>Display root-edge.</strong> <code><a href="https://rdrr.io/pkg/ggtree/man/geom_rootedge.html">geom_rootedge()</a></code> supports displaying root-edge if the root edge was presented (A). It shows nothing if there is no root-edge (B). In this case, users can manually set the root edge for the tree (C) or just specify the length of the root for plotting (D).
</p>
</div>
</div>
<div id="color-tree" class="section level3" number="4.3.5">
<h3>
<span class="header-section-number">4.3.5</span> Color tree<a class="anchor" aria-label="anchor" href="#color-tree"><i class="fas fa-link"></i></a>
</h3>
<p>In <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> <span class="citation">(<a href="references.html#ref-yu_two_2018" role="doc-biblioref">Yu et al., 2018</a>)</span>, coloring phylogenetic tree is easy, by using <code>aes(color=VAR)</code> to map the color of the tree based on a specific variable (both numerical and categorical variables are supported, see Figure <a href="chapter4.html#fig:colortree">4.10</a>).</p>


<div class="sourceCode" id="cb147"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">beast_tree</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>color<span class="op">=</span><span class="va">rate</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu">scale_color_continuous</span><span class="op">(</span>low<span class="op">=</span><span class="st">'darkgreen'</span>, high<span class="op">=</span><span class="st">'red'</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.position<span class="op">=</span><span class="st">"right"</span><span class="op">)</span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:colortree"></span>
<img src="treedata_files/figure-html/colortree-1.svg" alt="Color tree by continuous or discrete feature. Edges are colored by values associated with the child nodes." width="98%"><p class="caption">
FIGURE 4.10: <strong>Color tree by continuous or discrete feature.</strong> Edges are colored by values associated with the child nodes.
</p>
</div>
<p>Users can use any feature (if available), including clade posterior and <em>d<sub>N</sub>/d<sub>S</sub></em>, <em>etc.</em>, to scale the color of the tree. If the feature is a continuous numerical value, <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> provides a <code>continuous</code> parameter to support plotting continuous state transition in edges. Here, we use an example<a class="footnote-ref" tabindex="0" data-toggle="popover" data-content='&lt;p&gt;&lt;a href="http://www.phytools.org/eqg2015/asr.html" class="uri"&gt;http://www.phytools.org/eqg2015/asr.html&lt;/a&gt;&lt;/p&gt;'><sup>9</sup></a> to demonstrate this functionality (Figure <a href="chapter4.html#fig:continuousColor">4.11</a>A). If you want to add a thin black border in tree branches, you can place a tree with black and slightly thicker branches below your tree to emulate edge outlines as demonstrated in Figure <a href="chapter4.html#fig:continuousColor">4.11</a>B.</p>


<div class="sourceCode" id="cb148"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/treedata-book/">ggtree</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/YuLab-SMU/treeio">treeio</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/treedata-book/">tidytree</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/YuLab-SMU/TDBook">TDbook</a></span><span class="op">)</span>
<span class="co">## ref: http://www.phytools.org/eqg2015/asr.html</span>
<span class="co">##</span>
<span class="co">## load `tree_anole` and `df_svl` from 'TDbook'</span>
<span class="va">svl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="va">df_svl</span><span class="op">)</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span>
<span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu">phytools</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phytools/man/fastAnc.html">fastAnc</a></span><span class="op">(</span><span class="va">tree_anole</span>, <span class="va">svl</span>, vars<span class="op">=</span><span class="cn">TRUE</span>, CI<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>

<span class="va">td</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>node <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/tidytree/man/nodeid.html">nodeid</a></span><span class="op">(</span><span class="va">tree_anole</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">svl</span><span class="op">)</span><span class="op">)</span>,
               trait <span class="op">=</span> <span class="va">svl</span><span class="op">)</span>
<span class="va">nd</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>node <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">fit</span><span class="op">$</span><span class="va">ace</span><span class="op">)</span>, trait <span class="op">=</span> <span class="va">fit</span><span class="op">$</span><span class="va">ace</span><span class="op">)</span>

<span class="va">d</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span><span class="op">(</span><span class="va">td</span>, <span class="va">nd</span><span class="op">)</span>
<span class="va">d</span><span class="op">$</span><span class="va">node</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="va">d</span><span class="op">$</span><span class="va">node</span><span class="op">)</span>
<span class="va">tree</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">full_join</a></span><span class="op">(</span><span class="va">tree_anole</span>, <span class="va">d</span>, by <span class="op">=</span> <span class="st">'node'</span><span class="op">)</span>

<span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>color<span class="op">=</span><span class="va">trait</span><span class="op">)</span>, layout <span class="op">=</span> <span class="st">'circular'</span>, 
        ladderize <span class="op">=</span> <span class="cn">FALSE</span>, continuous <span class="op">=</span> <span class="st">'colour'</span>, size<span class="op">=</span><span class="fl">2</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_gradient.html">scale_color_gradientn</a></span><span class="op">(</span>colours<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"red"</span>, <span class="st">'orange'</span>, <span class="st">'green'</span>, <span class="st">'cyan'</span>, <span class="st">'blue'</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab</a></span><span class="op">(</span>hjust <span class="op">=</span> <span class="op">-</span><span class="fl">.1</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html">xlim</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1.2</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">.05</span>, <span class="fl">.85</span><span class="op">)</span><span class="op">)</span> 

<span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree</span>, layout<span class="op">=</span><span class="st">'circular'</span>, ladderize <span class="op">=</span> <span class="cn">FALSE</span>, size<span class="op">=</span><span class="fl">2.8</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tree.html">geom_tree</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>color<span class="op">=</span><span class="va">trait</span><span class="op">)</span>, continuous <span class="op">=</span> <span class="st">'colour'</span>, size<span class="op">=</span><span class="fl">2</span><span class="op">)</span> <span class="op">+</span>  
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_gradient.html">scale_color_gradientn</a></span><span class="op">(</span>colours<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"red"</span>, <span class="st">'orange'</span>, <span class="st">'green'</span>, <span class="st">'cyan'</span>, <span class="st">'blue'</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>color<span class="op">=</span><span class="va">trait</span><span class="op">)</span>, hjust <span class="op">=</span> <span class="op">-</span><span class="fl">.1</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html">xlim</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1.2</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">.05</span>, <span class="fl">.85</span><span class="op">)</span><span class="op">)</span> 

<span class="fu"><a href="https://rdrr.io/pkg/aplot/man/plot_list.html">plot_list</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span>, ncol<span class="op">=</span><span class="fl">2</span>, tag_levels<span class="op">=</span><span class="st">"A"</span><span class="op">)</span>    </code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:continuousColor"></span>
<img src="treedata_files/figure-html/continuousColor-1.png" alt="Continuous state transition in edges. Edges are colored by the values from ancestral trait to offspring." width="100%"><p class="caption">
FIGURE 4.11: <strong>Continuous state transition in edges.</strong> Edges are colored by the values from ancestral trait to offspring.
</p>
</div>
<p>Besides, we can use a two-dimensional tree (as demonstrated in Figure <a href="chapter4.html#fig:2d">4.5</a>) to visualize phenotype on the vertical dimension to create the phenogram Figure <a href="chapter4.html#fig:phenogram">4.12</a>. We can use the <a href="https://CRAN.R-project.org/package=ggrepel"><strong>ggrepel</strong></a> package to repel tip labels to avoid overlapping as demonstrated in Figure <a href="faq.html#fig:repelTip">A.4</a>.</p>


<div class="sourceCode" id="cb149"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>color<span class="op">=</span><span class="va">trait</span><span class="op">)</span>, continuous <span class="op">=</span> <span class="st">'colour'</span>, yscale <span class="op">=</span> <span class="st">"trait"</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_viridis.html">scale_color_viridis_c</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_minimal</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:phenogram"></span>
<img src="treedata_files/figure-html/phenogram-1.svg" alt="Phenogram. Projecting the tree into a space defined by time (or genetic distance) on the horizontal axis and phenotype on the vertical dimension." width="98%"><p class="caption">
FIGURE 4.12: <strong>Phenogram.</strong> Projecting the tree into a space defined by time (or genetic distance) on the horizontal axis and phenotype on the vertical dimension.
</p>
</div>
</div>
<div id="rescale-tree" class="section level3" number="4.3.6">
<h3>
<span class="header-section-number">4.3.6</span> Rescale tree<a class="anchor" aria-label="anchor" href="#rescale-tree"><i class="fas fa-link"></i></a>
</h3>
<p>Most of the phylogenetic trees are scaled by evolutionary distance (substitution/site). In <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a>, users can rescale a phylogenetic tree by any numerical variable inferred by evolutionary analysis (<em>e.g.</em>, <em>d<sub>N</sub>/d<sub>S</sub></em>).</p>
<p>This example displays a time tree (Figure <a href="chapter4.html#fig:rescaleTree">4.13</a>A) and the branches were rescaled by substitution rate inferred by BEAST (Figure <a href="chapter4.html#fig:rescaleTree">4.13</a>B).</p>
<div class="sourceCode" id="cb150"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/YuLab-SMU/treeio">"treeio"</a></span><span class="op">)</span>
<span class="va">beast_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"examples/MCC_FluA_H3.tree"</span>, package<span class="op">=</span><span class="st">"ggtree"</span><span class="op">)</span>
<span class="va">beast_tree</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://docs.ropensci.org/treeio/reference/beast-parser.html">read.beast</a></span><span class="op">(</span><span class="va">beast_file</span><span class="op">)</span>
<span class="va">beast_tree</span></code></pre></div>
<pre><code>## 'treedata' S4 object that stored information
## of
##  '/home/ygc/R/library/ggtree/examples/MCC_FluA_H3.tree'.
## 
## ...@ phylo:
## 
## Phylogenetic tree with 76 tips and 75 internal nodes.
## 
## Tip labels:
##   A/Hokkaido/30-1-a/2013, A/New_York/334/2004,
## A/New_York/463/2005, A/New_York/452/1999,
## A/New_York/238/2005, A/New_York/523/1998, ...
## 
## Rooted; includes branch lengths.
## 
## with the following features available:
##   'height', 'height_0.95_HPD', 'height_median',
## 'height_range', 'length', 'length_0.95_HPD',
## 'length_median', 'length_range', 'posterior', 'rate',
## 'rate_0.95_HPD', 'rate_median', 'rate_range'.</code></pre>
<div class="sourceCode" id="cb152"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">beast_tree</span>, mrsd<span class="op">=</span><span class="st">'2013-01-01'</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/theme_tree2.html">theme_tree2</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>caption<span class="op">=</span><span class="st">"Divergence time"</span><span class="op">)</span>
<span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">beast_tree</span>, branch.length<span class="op">=</span><span class="st">'rate'</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/theme_tree2.html">theme_tree2</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>caption<span class="op">=</span><span class="st">"Substitution rate"</span><span class="op">)</span></code></pre></div>
<p>The following example draws a tree inferred by CodeML (Figure <a href="chapter4.html#fig:rescaleTree">4.13</a>C), and the branches can be rescaled by using <em>d<sub>N</sub>/d<sub>S</sub></em> values (Figure <a href="chapter4.html#fig:rescaleTree">4.13</a>D).</p>
<div class="sourceCode" id="cb153"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mlcfile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata/PAML_Codeml"</span>, <span class="st">"mlc"</span>, package<span class="op">=</span><span class="st">"treeio"</span><span class="op">)</span>
<span class="va">mlc_tree</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://docs.ropensci.org/treeio/reference/read.codeml_mlc.html">read.codeml_mlc</a></span><span class="op">(</span><span class="va">mlcfile</span><span class="op">)</span>
<span class="va">p3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">mlc_tree</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/theme_tree2.html">theme_tree2</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>caption<span class="op">=</span><span class="st">"nucleotide substitutions per codon"</span><span class="op">)</span>
<span class="va">p4</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">mlc_tree</span>, branch.length<span class="op">=</span><span class="st">'dN_vs_dS'</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/theme_tree2.html">theme_tree2</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>caption<span class="op">=</span><span class="st">"dN/dS tree"</span><span class="op">)</span></code></pre></div>


<div class="figure">
<span style="display:block;" id="fig:rescaleTree"></span>
<img src="treedata_files/figure-html/rescaleTree-1.svg" alt="Rescale tree branches. A time-scaled tree inferred by BEAST (A) and its branches were rescaled by substitution rate (B). A tree was inferred by CodeML (C) and the branches were rescaled by dN/dS values (D)." width="100%"><p class="caption">
FIGURE 4.13: <strong>Rescale tree branches.</strong> A time-scaled tree inferred by BEAST (A) and its branches were rescaled by substitution rate (B). A tree was inferred by CodeML (C) and the branches were rescaled by <em>d<sub>N</sub>/d<sub>S</sub></em> values (D).
</p>
</div>
<p>This provides a very convenient way to allow us to explore the relationship between tree associated data and tree structure through visualization.
In addition to specifying <code>branch.length</code> in tree visualization, users can change branch length stored in tree object by using <code><a href="https://docs.ropensci.org/treeio/reference/rescale_tree.html">rescale_tree()</a></code> function provided by the <a href="http://bioconductor.org/packages/treeio"><strong>treeio</strong></a> package <span class="citation">(<a href="references.html#ref-wang_treeio_2020" role="doc-biblioref">Wang et al., 2020</a>)</span>, and the following command will display a tree that is identical to Figure <a href="chapter4.html#fig:rescaleTree">4.13</a>B. The <code><a href="https://docs.ropensci.org/treeio/reference/rescale_tree.html">rescale_tree()</a></code> function was documented in <a href="chapter2.html#rescale-treeio">session 2.4</a>.</p>
<div class="sourceCode" id="cb154"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">beast_tree2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://docs.ropensci.org/treeio/reference/rescale_tree.html">rescale_tree</a></span><span class="op">(</span><span class="va">beast_tree</span>, branch.length<span class="op">=</span><span class="st">'rate'</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">beast_tree2</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/theme_tree2.html">theme_tree2</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
</div>
<div id="modify-components-of-a-theme" class="section level3" number="4.3.7">
<h3>
<span class="header-section-number">4.3.7</span> Modify components of a theme<a class="anchor" aria-label="anchor" href="#modify-components-of-a-theme"><i class="fas fa-link"></i></a>
</h3>
<p>The <code><a href="https://rdrr.io/pkg/ggtree/man/theme_tree.html">theme_tree()</a></code> defined a totally blank canvas, while <code><a href="https://rdrr.io/pkg/ggtree/man/theme_tree2.html">theme_tree2()</a></code> adds
phylogenetic distance (via <em>x</em>-axis). These two themes all accept a parameter of
<code>bgcolor</code> that defined the background color. Users can use any <a href="http://ggplot2.tidyverse.org/reference/theme.html">theme components</a> to the <code><a href="https://rdrr.io/pkg/ggtree/man/theme_tree.html">theme_tree()</a></code> or <code><a href="https://rdrr.io/pkg/ggtree/man/theme_tree2.html">theme_tree2()</a></code> functions to modify them (Figure <a href="chapter4.html#fig:theme">4.14</a>).</p>
<div class="sourceCode" id="cb155"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">2019</span><span class="op">)</span>
<span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ape/man/rtree.html">rtree</a></span><span class="op">(</span><span class="fl">30</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">x</span>, color<span class="op">=</span><span class="st">"#0808E5"</span>, size<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/theme_tree.html">theme_tree</a></span><span class="op">(</span><span class="st">"#FEE4E9"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">x</span>, color<span class="op">=</span><span class="st">"orange"</span>, size<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/theme_tree.html">theme_tree</a></span><span class="op">(</span><span class="st">'grey30'</span><span class="op">)</span></code></pre></div>


<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:theme"></span>
<img src="treedata_files/figure-html/theme-1.svg" alt="Three themes. All ggplot2 theme components can be modified, and all the ggplot2 themes can be applied to ggtree() output." width="100%"><p class="caption">
FIGURE 4.14: <strong>Three themes.</strong> All ggplot2 theme components can be modified, and all the ggplot2 themes can be applied to <code><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree()</a></code> output.
</p>
</div>
<p>Users can also use an image file as the tree background, see example in <a href="related-tools.html#ggimage-bgimage">Appendix B</a>.</p>
</div>
</div>
<div id="visualize-a-list-of-trees" class="section level2" number="4.4">
<h2>
<span class="header-section-number">4.4</span> Visualize a List of Trees<a class="anchor" aria-label="anchor" href="#visualize-a-list-of-trees"><i class="fas fa-link"></i></a>
</h2>
<p>The <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> supports <code>multiPhylo</code> and <code>treedataList</code> objects and a list of trees can be viewed simultaneously. The trees will visualize one on top of another and can be plotted in different panels through <code><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html">facet_wrap()</a></code> or <code><a href="https://ggplot2.tidyverse.org/reference/facet_grid.html">facet_grid()</a></code> functions (Figure <a href="chapter4.html#fig:multiPhylo">4.15</a>).</p>


<div class="sourceCode" id="cb156"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## trees &lt;- lapply(c(10, 20, 40), rtree)</span>
<span class="co">## class(trees) &lt;- "multiPhylo"</span>
<span class="co">## ggtree(trees) + facet_wrap(~.id, scale="free") + geom_tiplab()</span>

<span class="va">f</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata/r8s"</span>, <span class="st">"H3_r8s_output.log"</span>, package<span class="op">=</span><span class="st">"treeio"</span><span class="op">)</span>
<span class="va">r8s</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://docs.ropensci.org/treeio/reference/read.r8s.html">read.r8s</a></span><span class="op">(</span><span class="va">f</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">r8s</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html">facet_wrap</a></span><span class="op">(</span> <span class="op">~</span><span class="va">.id</span>, scale<span class="op">=</span><span class="st">"free"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/theme_tree2.html">theme_tree2</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:multiPhylo"></span>
<img src="treedata_files/figure-html/multiPhylo-1.svg" alt="Visualizing multiPhylo object. The ggtree() function supports visualizing multiple trees stored in the multiPhylo or treedataList objects." width="100%"><p class="caption">
FIGURE 4.15: <strong>Visualizing multiPhylo object.</strong> The <code><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree()</a></code> function supports visualizing multiple trees stored in the <code>multiPhylo</code> or <code>treedataList</code> objects.
</p>
</div>
<p>One hundred bootstrap trees can also be viewed simultaneously (Figure <a href="chapter4.html#fig:bp100">4.16</a>). This allows researchers to explore a large set of phylogenetic trees to find consensus and distinct trees. The consensus tree can be summarized via a density tree (Figure <a href="chapter4.html#fig:densiTree">4.18</a>).</p>


<div class="sourceCode" id="cb157"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">btrees</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ape/man/read.tree.html">read.tree</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata/RAxML"</span>, 
                                <span class="st">"RAxML_bootstrap.H3"</span>, 
                                package<span class="op">=</span><span class="st">"treeio"</span><span class="op">)</span>
                    <span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">btrees</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">.id</span>, ncol<span class="op">=</span><span class="fl">10</span><span class="op">)</span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:bp100"></span>
<img src="treedata_files/figure-html/bp100-1.png" alt="Visualizing one hundred bootstrap trees simultaneously." width="100%"><p class="caption">
FIGURE 4.16: <strong>Visualizing one hundred bootstrap trees simultaneously.</strong>
</p>
</div>
<div id="annotate-one-tree-with-values-from-different-variables" class="section level3" number="4.4.1">
<h3>
<span class="header-section-number">4.4.1</span> Annotate one tree with values from different variables<a class="anchor" aria-label="anchor" href="#annotate-one-tree-with-values-from-different-variables"><i class="fas fa-link"></i></a>
</h3>
<p>To annotate one tree (the same tree) with the values from different variables, one can plot them separately and use <a href="https://CRAN.R-project.org/package=patchwork"><strong>patchwork</strong></a> or <a href="https://CRAN.R-project.org/package=aplot"><strong>aplot</strong></a> to combine them side-by-side.</p>
<p>Another solution is to utilize the ability to plot a list of trees by <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a>, and then add annotation layers for the selected variable at a specific panel via the <a href="chapter12.html#geom2">subset</a> aesthetic mapping supported by <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> or using the <code><a href="https://rdrr.io/pkg/ggtree/man/td_filter.html">td_filter()</a></code> as demonstrated in Figure <a href="chapter4.html#fig:onetreefacet">4.17</a>. The <code>.id</code> is the conserved variable that is internally used to store the IDs of different trees.</p>


<div class="sourceCode" id="cb158"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">2020</span><span class="op">)</span>
<span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ape/man/rtree.html">rtree</a></span><span class="op">(</span><span class="fl">30</span><span class="op">)</span>
<span class="va">d</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>label<span class="op">=</span><span class="va">x</span><span class="op">$</span><span class="va">tip.label</span>, var1<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Normal.html">rnorm</a></span><span class="op">(</span><span class="fl">30</span><span class="op">)</span><span class="op">)</span>, var2<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Normal.html">rnorm</a></span><span class="op">(</span><span class="fl">30</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="va">tree</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">full_join</a></span><span class="op">(</span><span class="va">x</span>, <span class="va">d</span>, by<span class="op">=</span><span class="st">'label'</span><span class="op">)</span>
<span class="va">trs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>TREE1 <span class="op">=</span> <span class="va">tree</span>, TREE2 <span class="op">=</span> <span class="va">tree</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span><span class="op">(</span><span class="va">trs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">'treedataList'</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">trs</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">.id</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tippoint.html">geom_tippoint</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>subset<span class="op">=</span><span class="va">.id</span> <span class="op">==</span> <span class="st">'TREE1'</span>, colour<span class="op">=</span><span class="va">var1</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_gradient.html">scale_colour_gradient</a></span><span class="op">(</span>low<span class="op">=</span><span class="st">'blue'</span>, high<span class="op">=</span><span class="st">'red'</span><span class="op">)</span> <span class="op">+</span>  
  <span class="fu">ggnewscale</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ggnewscale/man/new_scale.html">new_scale_colour</a></span><span class="op">(</span><span class="op">)</span>  <span class="op">+</span> 
  <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tippoint.html">geom_tippoint</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>colour<span class="op">=</span><span class="va">var2</span><span class="op">)</span>, data<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/td_filter.html">td_filter</a></span><span class="op">(</span><span class="va">.id</span> <span class="op">==</span> <span class="st">"TREE2"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_viridis.html">scale_colour_viridis_c</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:onetreefacet"></span>
<img src="treedata_files/figure-html/onetreefacet-1.svg" alt="Annotate one tree with values from different variables. Using subset aesthetic mapping (as in TREE1 panel) or td_filter() (as in TREE2 panel) to filter variables to be displayed on the specific panel." width="100%"><p class="caption">
FIGURE 4.17: <strong>Annotate one tree with values from different variables.</strong> Using <code>subset</code> aesthetic mapping (as in TREE1 panel) or <code><a href="https://rdrr.io/pkg/ggtree/man/td_filter.html">td_filter()</a></code> (as in TREE2 panel) to filter variables to be displayed on the specific panel.
</p>
</div>
</div>
<div id="densitree" class="section level3" number="4.4.2">
<h3>
<span class="header-section-number">4.4.2</span> DensiTree<a class="anchor" aria-label="anchor" href="#densitree"><i class="fas fa-link"></i></a>
</h3>
<p>Another way to view the bootstrap trees is to merge them to form a density tree using <code><a href="https://rdrr.io/pkg/ggtree/man/ggdensitree.html">ggdensitree()</a></code> function (Figure <a href="chapter4.html#fig:densiTree">4.18</a>). This will help us identify consensus and differences among a large set of trees. The trees will be stacked on top of each other and the structures of the trees will be rotated to ensure the consistency of the tip order. The tip order is determined by the <code>tip.order</code> parameter and by default (<code>tip.order = 'mode'</code>) the tips are ordered by the most commonly seen topology. The user can pass in a character vector to specify the tip order, or pass in an integer, N, to order the tips by the order of the tips in the Nth tree. Passing <code>mds</code> to <code>tip.order</code> will order the tips based on MDS (Multidimensional Scaling) of the path length between the tips, or passing <code>mds_dist</code> will order the tips based on MDS of the distance between the tips.</p>


<div class="sourceCode" id="cb159"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggdensitree.html">ggdensitree</a></span><span class="op">(</span><span class="va">btrees</span>, alpha<span class="op">=</span><span class="fl">.3</span>, colour<span class="op">=</span><span class="st">'steelblue'</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">3</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggexpand.html">hexpand</a></span><span class="op">(</span><span class="fl">.35</span><span class="op">)</span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:densiTree"></span>
<img src="treedata_files/figure-html/densiTree-1.png" alt="DensiTree. Trees are stacked on top of each other and the structures of the trees are rotated to ensure the consistency of the tip order." width="100%"><p class="caption">
FIGURE 4.18: <strong>DensiTree.</strong> Trees are stacked on top of each other and the structures of the trees are rotated to ensure the consistency of the tip order.
</p>
</div>
</div>
</div>
<div id="summary4" class="section level2" number="4.5">
<h2>
<span class="header-section-number">4.5</span> Summary<a class="anchor" aria-label="anchor" href="#summary4"><i class="fas fa-link"></i></a>
</h2>
<p>Visualizing phylogenetic trees with <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> is easy by using a single command <code>ggtree(tree)</code>. The <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> package provides several geometric layers to display tree components such as tip labels, symbolic points for both external and internal nodes, root-edge, <em>etc</em>. Associated data can be used to rescale branch lengths, color the tree, and be displayed on the tree. All these can be done by the <a href="https://CRAN.R-project.org/package=ggplot2"><strong>ggplot2</strong></a> grammar of graphic syntax, which makes it very easy to overlay layers and customize the tree graph (via <a href="https://CRAN.R-project.org/package=ggplot2"><strong>ggplot2</strong></a> themes and scales). The <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> package also provides several layers that are specifically designed for tree annotation which will be introduced in <a href="chapter5.html#chapter5">Chapter 5</a>. The <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> package makes the presentation of trees and associated data extremely easy. Simple graphs are easy to generate, while complex graphs are simply superimposed layers and are also easy to generate.</p>

<div style="page-break-after: always;"></div>
</div>
</div>

  <div class="chapter-nav">
<div class="prev"><a href="chapter3.html"><span class="header-section-number">3</span> Exporting tree with data</a></div>
<div class="next"><a href="chapter5.html"><span class="header-section-number">5</span> Phylogenetic Tree Annotation</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#chapter4"><span class="header-section-number">4</span> Phylogenetic Tree Visualization</a></li>
<li><a class="nav-link" href="#introduction-1"><span class="header-section-number">4.1</span> Introduction</a></li>
<li>
<a class="nav-link" href="#visualizing-phylogenetic-tree-with-ggtree"><span class="header-section-number">4.2</span> Visualizing Phylogenetic Tree with ggtree</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#basic-tree-visualization"><span class="header-section-number">4.2.1</span> Basic Tree Visualization</a></li>
<li><a class="nav-link" href="#tree-layouts"><span class="header-section-number">4.2.2</span> Layouts of a phylogenetic tree</a></li>
</ul>
</li>
<li>
<a class="nav-link" href="#displaying-tree-components"><span class="header-section-number">4.3</span> Displaying Tree Components</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#geom-trescale"><span class="header-section-number">4.3.1</span> Displaying treescale (evolution distance)</a></li>
<li><a class="nav-link" href="#geom-nodepoint"><span class="header-section-number">4.3.2</span> Displaying nodes/tips</a></li>
<li><a class="nav-link" href="#displaying-labels"><span class="header-section-number">4.3.3</span> Displaying labels</a></li>
<li><a class="nav-link" href="#displaying-root-edge"><span class="header-section-number">4.3.4</span> Displaying root-edge</a></li>
<li><a class="nav-link" href="#color-tree"><span class="header-section-number">4.3.5</span> Color tree</a></li>
<li><a class="nav-link" href="#rescale-tree"><span class="header-section-number">4.3.6</span> Rescale tree</a></li>
<li><a class="nav-link" href="#modify-components-of-a-theme"><span class="header-section-number">4.3.7</span> Modify components of a theme</a></li>
</ul>
</li>
<li>
<a class="nav-link" href="#visualize-a-list-of-trees"><span class="header-section-number">4.4</span> Visualize a List of Trees</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#annotate-one-tree-with-values-from-different-variables"><span class="header-section-number">4.4.1</span> Annotate one tree with values from different variables</a></li>
<li><a class="nav-link" href="#densitree"><span class="header-section-number">4.4.2</span> DensiTree</a></li>
</ul>
</li>
<li><a class="nav-link" href="#summary4"><span class="header-section-number">4.5</span> Summary</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Data Integration, Manipulation and Visualization of Phylogenetic Trees</strong>" was written by Guangchuang Yu. It was last built on 2022-04-22.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
