<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 9 ggtree for other tree-like objects | Data Integration, Manipulation and Visualization of Phylogenetic Trees</title>
<meta name="author" content="Guangchuang Yu">
<meta name="description" content="9.1 ggtree for phylogenetic tree objects The treeio packages (Wang et al. 2020) allows parsing evolutionary inferences from several software outputs and linking external data to the tree...">
<meta name="generator" content="bookdown 0.24 with bs4_book()">
<meta property="og:title" content="Chapter 9 ggtree for other tree-like objects | Data Integration, Manipulation and Visualization of Phylogenetic Trees">
<meta property="og:type" content="book">
<meta property="og:image" content="/book-cover.png">
<meta property="og:description" content="9.1 ggtree for phylogenetic tree objects The treeio packages (Wang et al. 2020) allows parsing evolutionary inferences from several software outputs and linking external data to the tree...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 9 ggtree for other tree-like objects | Data Integration, Manipulation and Visualization of Phylogenetic Trees">
<meta name="twitter:description" content="9.1 ggtree for phylogenetic tree objects The treeio packages (Wang et al. 2020) allows parsing evolutionary inferences from several software outputs and linking external data to the tree...">
<meta name="twitter:image" content="/book-cover.png">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/header-attrs/header-attrs.js"></script><script src="libs/jquery/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap/bootstrap.bundle.min.js"></script><script src="libs/bs3compat/transition.js"></script><script src="libs/bs3compat/tabs.js"></script><script src="libs/bs3compat/bs3compat.js"></script><link href="libs/bs4_book/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book/bs4_book.js"></script><script src="libs/kePrint/kePrint.js"></script><link href="libs/lightable/lightable.css" rel="stylesheet">
<script src="libs/htmlwidgets/htmlwidgets.js"></script><script src="libs/pathseg/pathseg.js"></script><script src="libs/comic/comic.min.js"></script><script src="libs/comicR-binding/comicR.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS -->
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Data Integration, Manipulation and Visualization of Phylogenetic Trees</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">Preface</a></li>
<li><a class="" href="author.html">About the Author</a></li>
<li class="book-part">Part I: Tree data input, output, and manipulation</li>
<li><a class="" href="chapter1.html"><span class="header-section-number">1</span> Importing Tree with Data</a></li>
<li><a class="" href="chapter2.html"><span class="header-section-number">2</span> Manipulating Tree with Data</a></li>
<li><a class="" href="chapter3.html"><span class="header-section-number">3</span> Exporting tree with data</a></li>
<li class="book-part">Part II: Tree data visualization and annotation</li>
<li><a class="" href="chapter4.html"><span class="header-section-number">4</span> Phylogenetic Tree Visualization</a></li>
<li><a class="" href="chapter5.html"><span class="header-section-number">5</span> Phylogenetic Tree Annotation</a></li>
<li><a class="" href="chapter6.html"><span class="header-section-number">6</span> Visual Exploration of Phylogenetic Tree</a></li>
<li><a class="" href="chapter7.html"><span class="header-section-number">7</span> Plotting tree with data</a></li>
<li><a class="" href="chapter8.html"><span class="header-section-number">8</span> Annotating Tree with Silhouette Images and Sub-plots</a></li>
<li><a class="active" href="chapter9.html"><span class="header-section-number">9</span> ggtree for other tree-like objects</a></li>
<li class="book-part">Part III: ggtree extensions</li>
<li><a class="" href="chapter10.html"><span class="header-section-number">10</span> ggtreeExtra for presenting data on a circular layout</a></li>
<li><a class="" href="chapter11.html"><span class="header-section-number">11</span> Other ggtree extensions</a></li>
<li class="book-part">Part IV: Miscellaneous topics</li>
<li><a class="" href="chapter12.html"><span class="header-section-number">12</span> ggtree utilities</a></li>
<li><a class="" href="chapter13.html"><span class="header-section-number">13</span> Gallery of reproducible examples</a></li>
<li class="book-part">Appendix</li>
<li><a class="" href="faq.html"><span class="header-section-number">A</span> Frequently asked questions</a></li>
<li><a class="" href="related-tools.html"><span class="header-section-number">B</span> Related tools</a></li>
<li><a class="" href="figures-and-tables.html"><span class="header-section-number">C</span> Figures and Tables</a></li>
<li><a class="" href="publications-of-the-ggtree-package-suite.html"><span class="header-section-number">D</span> Publications of the ggtree package suite</a></li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="chapter9" class="section level1" number="9">
<h1>
<span class="header-section-number">9</span> ggtree for other tree-like objects<a class="anchor" aria-label="anchor" href="#chapter9"><i class="fas fa-link"></i></a>
</h1>
<div id="ggtree-for-phylogenetic-tree-objects" class="section level2" number="9.1">
<h2>
<span class="header-section-number">9.1</span> ggtree for phylogenetic tree objects<a class="anchor" aria-label="anchor" href="#ggtree-for-phylogenetic-tree-objects"><i class="fas fa-link"></i></a>
</h2>
<p>The <a href="http://bioconductor.org/packages/treeio"><strong>treeio</strong></a> packages <span class="citation">(<a href="references.html#ref-wang_treeio_2020" role="doc-biblioref">Wang et al. 2020</a>)</span> allows parsing evolutionary inferences from several software outputs and linking external data to the tree structure. It serves as an infrastructure to bring evolutionary data to the R community. The <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> package <span class="citation">(<a href="references.html#ref-yu_ggtree:_2017" role="doc-biblioref">Yu et al. 2017</a>)</span> works seamlessly with <a href="http://bioconductor.org/packages/treeio"><strong>treeio</strong></a> to visualize tree associated data to annotate the tree. The <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> package is a general tool for tree visualization and annotation and it fits the ecosystem of R packages. Most of the S3/S4 tree objects defined by other R packages are also supported by <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a>, including <code>phylo</code>(<a href="chapter4.html#visualizing-phylogenetic-tree-with-ggtree">session 4.2</a>), <code>multiPhylo</code> (<a href="chapter4.html#visualize-a-list-of-trees">session 4.4</a>), <code>phylo4</code>, <code>phylo4d</code>, <code>phyloseq</code>, and <code>obkData</code>. With <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a>, we are able to generate more complex tree graphs which is not possible or easy to do with other packages. For example, the visualization of the <code>phyloseq</code> object in Figure <a href="chapter9.html#fig:phyloseq">9.4</a> is not supported by the <a href="http://bioconductor.org/packages/phyloseq"><strong>phyloseq</strong></a> package. The <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> package also extend the possibility of linking external data to these tree object <span class="citation">(<a href="references.html#ref-yu_two_2018" role="doc-biblioref">Yu et al. 2018</a>)</span>.</p>
<div id="phylobase" class="section level3" number="9.1.1">
<h3>
<span class="header-section-number">9.1.1</span> The phylo4 and phylo4d objects<a class="anchor" aria-label="anchor" href="#phylobase"><i class="fas fa-link"></i></a>
</h3>
<p>The <code>phylo4</code> and <code>phylo4d</code> are defined in the <a href="https://CRAN.R-project.org/package=phylobase"><strong>phylobase</strong></a> package. The <code>phylo4</code> object is an S4 version of <code>phylo</code>, while <code>phylo4d</code> extends <code>phylo4</code> with a data frame that contains trait data. The <a href="https://CRAN.R-project.org/package=phylobase"><strong>phylobase</strong></a> package provides a <code><a href="https://rdrr.io/pkg/phylobase/man/treePlot-methods.html">plot()</a></code> method, which is internally called the <code><a href="https://rdrr.io/pkg/phylobase/man/treePlot-methods.html">treePlot()</a></code> function, to display the tree with the data. However there are some restrictions of the <code><a href="https://rdrr.io/pkg/phylobase/man/treePlot-methods.html">plot()</a></code> method, it can only plot numeric values for tree-associated data as bubbles and cannot generate figure legend. <code>Phylobase</code> doesnâ€™t implement a visualization method to display categorical values. Using associated data as visual characteristics such as color, size, and shape, is also not supported. Although it is possible to color the tree using associated data, it requires users to extract the data and map them to the color vector manually follow by passing the color vector to the <code>plot</code> method. This is tedious and error-prone since the order of the color vector needs to be consistent with the edge list stored in the object.</p>
<p>The <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> package supports <code>phylo4d</code> object and all the associated data stored in the <code>phylo4d</code> object can be used directly to annotate the tree (Fig. A).</p>


<div class="sourceCode" id="cb209"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/fmichonneau/phylobase">phylobase</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">geospiza_raw</span><span class="op">)</span>
<span class="va">g1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phylobase/man/setAs-methods.html">as</a></span><span class="op">(</span><span class="va">geospiza_raw</span><span class="op">$</span><span class="va">tree</span>, <span class="st">"phylo4"</span><span class="op">)</span>
<span class="va">g2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phylobase/man/phylo4d-methods.html">phylo4d</a></span><span class="op">(</span><span class="va">g1</span>, <span class="va">geospiza_raw</span><span class="op">$</span><span class="va">data</span>, missing.data<span class="op">=</span><span class="st">"warn"</span><span class="op">)</span>

<span class="va">d1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">1.1</span>, <span class="fl">2</span>, length.out <span class="op">=</span> <span class="fl">5</span><span class="op">)</span>,
                lab <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/phylobase/man/print-methods.html">names</a></span><span class="op">(</span><span class="va">geospiza_raw</span><span class="op">$</span><span class="va">data</span><span class="op">)</span><span class="op">)</span>

<span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">g2</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tippoint.html">geom_tippoint</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>size <span class="op">=</span> <span class="va">wingL</span><span class="op">)</span>, x <span class="op">=</span> <span class="va">d1</span><span class="op">$</span><span class="va">x</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, shape <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tippoint.html">geom_tippoint</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>size <span class="op">=</span> <span class="va">tarsusL</span><span class="op">)</span>, x <span class="op">=</span> <span class="va">d1</span><span class="op">$</span><span class="va">x</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>, shape <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tippoint.html">geom_tippoint</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>size <span class="op">=</span> <span class="va">culmenL</span><span class="op">)</span>, x <span class="op">=</span> <span class="va">d1</span><span class="op">$</span><span class="va">x</span><span class="op">[</span><span class="fl">3</span><span class="op">]</span>, shape <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tippoint.html">geom_tippoint</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>size <span class="op">=</span> <span class="va">beakD</span><span class="op">)</span>,   x <span class="op">=</span> <span class="va">d1</span><span class="op">$</span><span class="va">x</span><span class="op">[</span><span class="fl">4</span><span class="op">]</span>, shape <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tippoint.html">geom_tippoint</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>size <span class="op">=</span> <span class="va">gonysW</span><span class="op">)</span>,  x <span class="op">=</span> <span class="va">d1</span><span class="op">$</span><span class="va">x</span><span class="op">[</span><span class="fl">5</span><span class="op">]</span>, shape <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_size.html">scale_size_continuous</a></span><span class="op">(</span>range <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">3</span>,<span class="fl">12</span><span class="op">)</span>, name<span class="op">=</span><span class="st">""</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_text.html">geom_text</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">x</span>, y <span class="op">=</span> <span class="fl">0</span>, label <span class="op">=</span> <span class="va">lab</span><span class="op">)</span>, data <span class="op">=</span> <span class="va">d1</span>, angle <span class="op">=</span> <span class="fl">45</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab</a></span><span class="op">(</span>offset <span class="op">=</span> <span class="fl">1.3</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html">xlim</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">3</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">.1</span>, <span class="fl">.75</span><span class="op">)</span><span class="op">)</span>  

<span class="co">## users can use `as.treedata(g2)` to convert `g2` to a `treedata` object</span>
<span class="co">## and use `get_tree_data()` function to extract the associated data </span>

<span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/gheatmap.html">gheatmap</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">g1</span><span class="op">)</span>, data<span class="op">=</span><span class="va">geospiza_raw</span><span class="op">$</span><span class="va">data</span>, colnames_angle<span class="op">=</span><span class="fl">45</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab</a></span><span class="op">(</span>offset<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggexpand.html">hexpand</a></span><span class="op">(</span><span class="fl">.2</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">.1</span>, <span class="fl">.75</span><span class="op">)</span><span class="op">)</span>

<span class="fu">aplot</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/aplot/man/plot_list.html">plot_list</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span>, ncol<span class="op">=</span><span class="fl">2</span>, tag_levels<span class="op">=</span><span class="st">'A'</span><span class="op">)</span>    </code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:fp4d"></span>
<img src="treedata_files/figure-html/fp4d-1.png" alt="Visualizing phylo4d data using ggtree. Reproduce the output of the plot() method provided in the phylobase package (A); Visuzlize the trait data as a heatmap which is not supported in the phylobase package (B)." width="90%"><p class="caption">
FIGURE 9.1: <strong>Visualizing <code>phylo4d</code> data using ggtree.</strong> Reproduce the output of the <code><a href="https://rdrr.io/pkg/phylobase/man/treePlot-methods.html">plot()</a></code> method provided in the <a href="https://CRAN.R-project.org/package=phylobase"><strong>phylobase</strong></a> package (A); Visuzlize the trait data as a heatmap which is not supported in the <a href="https://CRAN.R-project.org/package=phylobase"><strong>phylobase</strong></a> package (B).
</p>
</div>
</div>
<div id="phylog" class="section level3" number="9.1.2">
<h3>
<span class="header-section-number">9.1.2</span> The phylog object<a class="anchor" aria-label="anchor" href="#phylog"><i class="fas fa-link"></i></a>
</h3>
<p>The <code>phylog</code> is defined in the <a href="https://CRAN.R-project.org/package=ade4"><strong>ade4</strong></a> package. The package is designed for analyzing ecological data and provides <code><a href="https://rdrr.io/pkg/ade4/man/newick2phylog.html">newick2phylog()</a></code>, <code><a href="https://rdrr.io/pkg/ade4/man/newick2phylog.html">hclust2phylog()</a></code>, and <code><a href="https://rdrr.io/pkg/ade4/man/newick2phylog.html">taxo2phylog()</a></code> functions to create phylogeny from Newick string, hierarchical clustering result, or a taxonomy (see also the <a href="http://bioconductor.org/packages/MicrobiotaProcess"><strong>MicrobiotaProcess</strong></a> package described in [Chapter 11]](#MicrobiotaProcess-taxonomy)). The <code>phylog</code> object is also supported by <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> as demonstrated in Figure <a href="chapter9.html#fig:phylog">9.2</a>.</p>


<div class="sourceCode" id="cb210"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://pbil.univ-lyon1.fr/ADE-4/">ade4</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">taxo.eg</span><span class="op">)</span>
<span class="va">tax</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ade4/man/as.taxo.html">as.taxo</a></span><span class="op">(</span><span class="va">taxo.eg</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/phylobase/man/print-methods.html">print</a></span><span class="op">(</span><span class="va">tax</span><span class="op">)</span></code></pre></div>
<pre><code>##       genre famille ordre
## esp3     g1    fam1  ORD1
## esp1     g1    fam1  ORD1
## esp2     g1    fam1  ORD1
## esp4     g1    fam1  ORD1
## esp5     g1    fam1  ORD1
## esp6     g1    fam1  ORD1
## esp7     g1    fam1  ORD1
## esp8     g2    fam2  ORD2
## esp9     g3    fam2  ORD2
## esp10    g4    fam3  ORD2
## esp11    g5    fam3  ORD2
## esp12    g6    fam4  ORD2
## esp13    g7    fam4  ORD2
## esp14    g8    fam5  ORD2
## esp15    g8    fam5  ORD2</code></pre>
<div class="sourceCode" id="cb212"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">tax.phy</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ade4/man/newick2phylog.html">taxo2phylog</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/ade4/man/as.taxo.html">as.taxo</a></span><span class="op">(</span><span class="va">taxo.eg</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/phylobase/man/print-methods.html">print</a></span><span class="op">(</span><span class="va">tax.phy</span><span class="op">)</span></code></pre></div>
<pre><code>## Phylogenetic tree with 15 leaves and 16 nodes
## $class: phylog
## $call: taxo2phylog(taxo = as.taxo(taxo.eg[[1]]))
## $tre: ((((esp3,esp1,esp2,esp4,e...15)l1g8)l2fam5)l3ORD2)Root; 
## 
##         class   length
## $leaves numeric 15    
## $nodes  numeric 16    
## $parts  list    16    
## $paths  list    31    
## $droot  numeric 31    
##         content                                     
## $leaves length of the first preceeding adjacent edge
## $nodes  length of the first preceeding adjacent edge
## $parts  subsets of descendant nodes                 
## $paths  path from root to node or leave             
## $droot  distance to root</code></pre>
<div class="sourceCode" id="cb214"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tax.phy</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_nodelab.html">geom_nodelab</a></span><span class="op">(</span>geom<span class="op">=</span><span class="st">'label'</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggexpand.html">hexpand</a></span><span class="op">(</span><span class="fl">.05</span><span class="op">)</span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:phylog"></span>
<img src="treedata_files/figure-html/phylog-1.svg" alt="Visualizing a phylog tree object." width="100%"><p class="caption">
FIGURE 9.2: <strong>Visualizing a <code>phylog</code> tree object.</strong>
</p>
</div>
</div>
<div id="phyloseq" class="section level3" number="9.1.3">
<h3>
<span class="header-section-number">9.1.3</span> The phyloseq object<a class="anchor" aria-label="anchor" href="#phyloseq"><i class="fas fa-link"></i></a>
</h3>
<p>The <code>phyloseq</code> class defined in the <a href="http://bioconductor.org/packages/phyloseq"><strong>phyloseq</strong></a> package was designed for storing microbiome data, including a phylogenetic tree, associated sample data, and taxonomy assignment. It can import data from popular pipelines, such as <a href="http://qiime.org/"><strong>QIIME</strong></a> <span class="citation">(<a href="references.html#ref-kuczynski_using_2011" role="doc-biblioref">Kuczynski et al. 2011</a>)</span>, <a href="http://mothur.org/"><strong>mothur</strong></a> <span class="citation">(<a href="references.html#ref-schloss_introducing_2009" role="doc-biblioref">Schloss et al. 2009</a>)</span>, <a href="http://bioconductor.org/packages/dada2"><strong>dada2</strong></a> <span class="citation">(<a href="references.html#ref-callahan_dada2_2016" role="doc-biblioref">Callahan et al. 2016</a>)</span> and <a href="http://pyrotagger.jgi-psf.org/"><strong>PyroTagger</strong></a> <span class="citation">(<a href="references.html#ref-kunin_pyrotagger_2010" role="doc-biblioref">Kunin and Hugenholtz 2010</a>)</span>, <em>etc</em>. The <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> supports visualizing the phylogenetic tree stored in the <code>phyloseq</code> object and related data can be used to annotate the tree as demonstrated in Figures <a href="chapter9.html#fig:reproducephyloseq">9.3</a> and <a href="chapter9.html#fig:phyloseq">9.4</a>.</p>


<div class="sourceCode" id="cb215"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://dx.plos.org/10.1371/journal.pone.0061217">phyloseq</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://scales.r-lib.org">scales</a></span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">GlobalPatterns</span><span class="op">)</span>
<span class="va">GP</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/prune_taxa-methods.html">prune_taxa</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/taxa_sums.html">taxa_sums</a></span><span class="op">(</span><span class="va">GlobalPatterns</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span>, <span class="va">GlobalPatterns</span><span class="op">)</span>
<span class="va">GP.chl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/subset_taxa-methods.html">subset_taxa</a></span><span class="op">(</span><span class="va">GP</span>, <span class="va">Phylum</span><span class="op">==</span><span class="st">"Chlamydiae"</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">GP.chl</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_text2.html">geom_text2</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>subset<span class="op">=</span><span class="op">!</span><span class="va">isTip</span>, label<span class="op">=</span><span class="va">label</span><span class="op">)</span>, hjust<span class="op">=</span><span class="op">-</span><span class="fl">.2</span>, size<span class="op">=</span><span class="fl">4</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>label<span class="op">=</span><span class="va">Genus</span><span class="op">)</span>, hjust<span class="op">=</span><span class="op">-</span><span class="fl">.3</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">x</span><span class="op">+</span><span class="va">hjust</span>, color<span class="op">=</span><span class="va">SampleType</span>, shape<span class="op">=</span><span class="va">Family</span>, 
                size<span class="op">=</span><span class="va">Abundance</span><span class="op">)</span>, na.rm<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_size.html">scale_size_continuous</a></span><span class="op">(</span>trans<span class="op">=</span><span class="fu"><a href="https://scales.r-lib.org/reference/log_trans.html">log_trans</a></span><span class="op">(</span><span class="fl">5</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.position<span class="op">=</span><span class="st">"right"</span><span class="op">)</span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:reproducephyloseq"></span>
<img src="treedata_files/figure-html/reproducephyloseq-1.png" alt="Visualizing a phyloseq tree object. This example mimics the output of the plot_tree() function provided in the phyloseq package." width="100%"><p class="caption">
FIGURE 9.3: <strong>Visualizing a <code>phyloseq</code> tree object.</strong> This example mimics the output of the <code><a href="https://rdrr.io/pkg/phyloseq/man/plot_tree.html">plot_tree()</a></code> function provided in the <a href="http://bioconductor.org/packages/phyloseq"><strong>phyloseq</strong></a> package.
</p>
</div>
<p>Figure <a href="chapter9.html#fig:reproducephyloseq">9.3</a> reproduces the output of the <code><a href="https://rdrr.io/pkg/phyloseq/man/plot_tree.html">phyloseq::plot_tree()</a></code> function. Users of <a href="http://bioconductor.org/packages/phyloseq"><strong>phyloseq</strong></a> will find <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> useful for visualizing microbiome data and for further annotation since <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> supports high-level annotation using the grammar of graphics and can add tree data layers that are not available in <a href="http://bioconductor.org/packages/phyloseq"><strong>phyloseq</strong></a>.</p>


<div class="sourceCode" id="cb216"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://wilkelab.org/ggridges/">ggridges</a></span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"GlobalPatterns"</span><span class="op">)</span>
<span class="va">GP</span> <span class="op">&lt;-</span> <span class="va">GlobalPatterns</span>
<span class="va">GP</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/prune_taxa-methods.html">prune_taxa</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/taxa_sums.html">taxa_sums</a></span><span class="op">(</span><span class="va">GP</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">600</span>, <span class="va">GP</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html">sample_data</a></span><span class="op">(</span><span class="va">GP</span><span class="op">)</span><span class="op">$</span><span class="va">human</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/get_variable.html">get_variable</a></span><span class="op">(</span><span class="va">GP</span>, <span class="st">"SampleType"</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> 
  <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Feces"</span>, <span class="st">"Skin"</span><span class="op">)</span> 

<span class="va">mergedGP</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/merge_samples-methods.html">merge_samples</a></span><span class="op">(</span><span class="va">GP</span>, <span class="st">"SampleType"</span><span class="op">)</span>
<span class="va">mergedGP</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/rarefy_even_depth.html">rarefy_even_depth</a></span><span class="op">(</span><span class="va">mergedGP</span>,rngseed<span class="op">=</span><span class="fl">394582</span><span class="op">)</span>
<span class="va">mergedGP</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_glom.html">tax_glom</a></span><span class="op">(</span><span class="va">mergedGP</span>,<span class="st">"Order"</span><span class="op">)</span> 

<span class="va">melt_simple</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/psmelt.html">psmelt</a></span><span class="op">(</span><span class="va">mergedGP</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">Abundance</span> <span class="op">&lt;</span> <span class="fl">120</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">OTU</span>, val<span class="op">=</span><span class="va">Abundance</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">mergedGP</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tippoint.html">geom_tippoint</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>color<span class="op">=</span><span class="va">Phylum</span><span class="op">)</span>, size<span class="op">=</span><span class="fl">1.5</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/facet-plot.html">geom_facet</a></span><span class="op">(</span>mapping <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">val</span>,group<span class="op">=</span><span class="va">label</span>, 
                           fill<span class="op">=</span><span class="va">Phylum</span><span class="op">)</span>,
            data <span class="op">=</span> <span class="va">melt_simple</span>, 
            geom <span class="op">=</span> <span class="va">geom_density_ridges</span>,
            panel<span class="op">=</span><span class="st">"Abundance"</span>,  
            color<span class="op">=</span><span class="st">'grey80'</span>, lwd<span class="op">=</span><span class="fl">.3</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guides.html">guides</a></span><span class="op">(</span>color <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guide_legend.html">guide_legend</a></span><span class="op">(</span>ncol<span class="op">=</span><span class="fl">1</span><span class="op">)</span><span class="op">)</span>          </code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:phyloseq"></span>
<img src="treedata_files/figure-html/phyloseq-1.png" alt="Phylogenetic tree with OTU abundance densities. Tips were colored by Phylum and the corresponding abundances across different samples were visualized as density ridgelines and sorted according to the tree structure." width="100%"><p class="caption">
FIGURE 9.4: <strong>Phylogenetic tree with OTU abundance densities.</strong> Tips were colored by Phylum and the corresponding abundances across different samples were visualized as density ridgelines and sorted according to the tree structure.
</p>
</div>
<p>This example uses microbiome data provided in the <a href="http://bioconductor.org/packages/phyloseq"><strong>phyloseq</strong></a> package and density ridgeline is employed to visualize species abundance data. The <code><a href="https://rdrr.io/pkg/ggtree/man/facet-plot.html">geom_facet()</a></code> layer automatically re-arranges the abundance data according to the tree structure, visualizes the data using the specified <code>geom</code> function, <em>i.e.</em> <code><a href="https://wilkelab.org/ggridges/reference/geom_density_ridges.html">geom_density_ridges()</a></code>, and aligns the density curves with the tree as demonstrated in Fig. . Note that data stored in the <code>phyloseq</code> object is visible to <code><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree()</a></code> and can be used directly in tree visualization (<code>Phylum</code> was used to color tips and density ridgelines in this example). The source code of this example was firstly published in the Supplemental File of <span class="citation">(<a href="references.html#ref-yu_two_2018" role="doc-biblioref">Yu et al. 2018</a>)</span>.</p>
<!-- 
### The obkData object {#obkdata}

The `okbData` is defined in the [**OutbreakTools**](https://CRAN.R-project.org/package=OutbreakTools) package to store incidence-based outbreak data, including meta data of sampling and information of infected individuals such as age and onset of symptoms. The `ggtree` supports the `obkData` object and the information can be used to annotate the tree as shown in Figure \@ref(fig:outbreaktools).

(ref:outbreaktoolsscap) Visualizing obkData tree object.

(ref:outbreaktoolscap) **Visualizing obkData tree object.** *x*-axis was scaled by timeline of the outbreak and tips were colored by location of different individuals.


```r outbreaktools, fig.width=6.3, fig.height=7, fig.cap="(ref:outbreaktoolscap)", fig.scap="(ref:outbreaktoolsscap)", message=FALSE, out.extra=''}
library(OutbreakTools)
data(FluH1N1pdm2009)
attach(FluH1N1pdm2009)

x <- new("obkData", 
         individuals = individuals, 
         dna = dna,
         dna.individualID = samples$individualID, 
         dna.date = samples$date,
         trees = FluH1N1pdm2009$trees)
ggtree(x, mrsd="2009-09-30", as.Date=TRUE, right=TRUE) +
    geom_tippoint(aes(color=location), size=3, alpha=.75) +
    scale_color_brewer("location", palette="Spectral") +
    theme_tree2(legend.position='right')
```


-->
</div>
</div>
<div id="dendrogram" class="section level2" number="9.2">
<h2>
<span class="header-section-number">9.2</span> ggtree for dendrograms<a class="anchor" aria-label="anchor" href="#dendrogram"><i class="fas fa-link"></i></a>
</h2>
<p>A dendrogram is a tree diagram to display hierarchical clustering and classification/regression trees. In R, we can calculate a hierarchical clustering using the function <code><a href="https://rdrr.io/r/stats/hclust.html">hclust()</a></code>.</p>
<div class="sourceCode" id="cb217"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">hc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/hclust.html">hclust</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/dist.html">dist</a></span><span class="op">(</span><span class="va">mtcars</span><span class="op">)</span><span class="op">)</span>
<span class="va">hc</span></code></pre></div>
<pre><code>## 
## Call:
## hclust(d = dist(mtcars))
## 
## Cluster method   : complete 
## Distance         : euclidean 
## Number of objects: 32</code></pre>
<p>The <code>hclust</code> object describes the tree produced by the clustering process. It can be converted to <code>dendrogram</code> object, which stores the tree as deeply-nested lists.</p>
<div class="sourceCode" id="cb219"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">den</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/dendrogram.html">as.dendrogram</a></span><span class="op">(</span><span class="va">hc</span><span class="op">)</span>
<span class="va">den</span></code></pre></div>
<pre><code>## 'dendrogram' with 2 branches and 32 members total, at height 425.3</code></pre>
<p>The <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> package supports most of the hierarchical clustering objects defined in the R community, including <code>hclust</code> and <code>dendrogram</code> as well as <code>agnes</code>, <code>diana</code>, and <code>twins</code> that are defined in the <a href="https://CRAN.R-project.org/package=cluster"><strong>cluster</strong></a> package, and the <code>pvclust</code> object defined in the <a href="https://CRAN.R-project.org/package=pvclust"><strong>pvclust</strong></a> package (Table <a href="figures-and-tables.html#tab:tree-objects">C.2</a>). Users can use <code>ggtree(object)</code> to display its tree structure, and use other layers and utilities to customize the graph and of course, add annotations to the tree.</p>
<p>The <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> provides <code><a href="https://rdrr.io/pkg/ggtree/man/tree-layout.html">layout_dendrogram()</a></code> to layout the tree top-down, and <code><a href="https://rdrr.io/pkg/ggtree/man/theme_dendrogram.html">theme_dendrogram()</a></code> to display tree height (similar to <code><a href="https://rdrr.io/pkg/ggtree/man/theme_tree2.html">theme_tree2()</a></code> for phylogenetic tree) as demonstrated in Figure <a href="chapter9.html#fig:ggtreehclust">9.5</a> (see also the example in <span class="citation">(<a href="references.html#ref-yu_cp_2020" role="doc-biblioref">Yu 2020</a>)</span>).</p>


<div class="sourceCode" id="cb221"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">clus</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/cutree.html">cutree</a></span><span class="op">(</span><span class="va">hc</span>, <span class="fl">4</span><span class="op">)</span>
<span class="va">g</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/split.html">split</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phylobase/man/print-methods.html">names</a></span><span class="op">(</span><span class="va">clus</span><span class="op">)</span>, <span class="va">clus</span><span class="op">)</span>

<span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">hc</span>, linetype<span class="op">=</span><span class="st">'dashed'</span><span class="op">)</span>
<span class="va">clades</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">g</span>, <span class="kw">function</span><span class="op">(</span><span class="va">n</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/pkg/phylobase/man/MRCA.html">MRCA</a></span><span class="op">(</span><span class="va">p</span>, <span class="va">n</span><span class="op">)</span><span class="op">)</span>

<span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/tidytree/man/groupClade.html">groupClade</a></span><span class="op">(</span><span class="va">p</span>, <span class="va">clades</span>, group_name<span class="op">=</span><span class="st">'subtree'</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>color<span class="op">=</span><span class="va">subtree</span><span class="op">)</span>

<span class="va">d</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/phylobase/man/print-methods.html">names</a></span><span class="op">(</span><span class="va">clus</span><span class="op">)</span>, 
                  cyl <span class="op">=</span> <span class="va">mtcars</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/phylobase/man/print-methods.html">names</a></span><span class="op">(</span><span class="va">clus</span><span class="op">)</span>, <span class="st">"cyl"</span><span class="op">]</span><span class="op">)</span>

<span class="va">p</span> <span class="op"><a href="https://rdrr.io/pkg/ggtree/man/attacher.html">%&lt;+%</a></span> <span class="va">d</span> <span class="op">+</span> 
  <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/tree-layout.html">layout_dendrogram</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tippoint.html">geom_tippoint</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>fill<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">cyl</span><span class="op">)</span>, x<span class="op">=</span><span class="va">x</span><span class="op">+</span><span class="fl">.5</span><span class="op">)</span>, 
                size<span class="op">=</span><span class="fl">5</span>, shape<span class="op">=</span><span class="fl">21</span>, color<span class="op">=</span><span class="st">'black'</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>label<span class="op">=</span><span class="va">cyl</span><span class="op">)</span>, size<span class="op">=</span><span class="fl">3</span>, hjust<span class="op">=</span><span class="fl">.5</span>, color<span class="op">=</span><span class="st">'black'</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab</a></span><span class="op">(</span>angle<span class="op">=</span><span class="fl">90</span>, hjust<span class="op">=</span><span class="fl">1</span>, offset<span class="op">=</span><span class="op">-</span><span class="fl">10</span>, show.legend<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html">scale_color_brewer</a></span><span class="op">(</span>palette<span class="op">=</span><span class="st">'Set1'</span>, breaks<span class="op">=</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/theme_dendrogram.html">theme_dendrogram</a></span><span class="op">(</span>plot.margin<span class="op">=</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">margin</a></span><span class="op">(</span><span class="fl">6</span>,<span class="fl">6</span>,<span class="fl">80</span>,<span class="fl">6</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.position<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">.9</span>, <span class="fl">.6</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:ggtreehclust"></span>
<img src="treedata_files/figure-html/ggtreehclust-1.png" alt="Visualizing dendrogram. Use cutree() to split the tree into several groups and groupClade() to assign this grouping information. The tree was dispalyed in the classic top-down layout with branches colored by the grouping information and the tips were colored and labeled by the number of cylinders." width="100%"><p class="caption">
FIGURE 9.5: <strong>Visualizing dendrogram.</strong> Use <code><a href="https://rdrr.io/r/stats/cutree.html">cutree()</a></code> to split the tree into several groups and <code><a href="https://rdrr.io/pkg/tidytree/man/groupClade.html">groupClade()</a></code> to assign this grouping information. The tree was dispalyed in the classic top-down layout with branches colored by the grouping information and the tips were colored and labeled by the number of cylinders.
</p>
</div>
</div>
<div id="igraph" class="section level2" number="9.3">
<h2>
<span class="header-section-number">9.3</span> ggtree for tree graph<a class="anchor" aria-label="anchor" href="#igraph"><i class="fas fa-link"></i></a>
</h2>
<p>The tree graph (as an <code>igraph</code> object) can be converted to a <code>phylo</code> object using <code><a href="https://rdrr.io/pkg/ape/man/as.phylo.html">as.phylo()</a></code> method provided in the <a href="http://bioconductor.org/packages/treeio"><strong>treeio</strong></a> package (Table <a href="figures-and-tables.html#tab:tree-objects">C.2</a>). The <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> supports directly visualizing tree graph as demonstrated in Figure <a href="chapter9.html#fig:treeGraph">9.6</a>. Note that currently not all <code>igraph</code> objects can be supported by <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a>. Currently, it can only be supported when it is a tree graph.</p>


<div class="sourceCode" id="cb222"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://igraph.org">igraph</a></span><span class="op">)</span>
<span class="va">g</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/make_tree.html">graph.tree</a></span><span class="op">(</span><span class="fl">40</span>, <span class="fl">3</span><span class="op">)</span>
<span class="va">arrow_size</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/grid/unit.html">unit</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">3</span><span class="op">)</span>, times <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">27</span>, <span class="fl">13</span><span class="op">)</span><span class="op">)</span>, <span class="st">"mm"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">g</span>, layout<span class="op">=</span><span class="st">'slanted'</span>, arrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/arrow.html">arrow</a></span><span class="op">(</span>length<span class="op">=</span><span class="va">arrow_size</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">5</span>, color<span class="op">=</span><span class="st">'steelblue'</span>, alpha<span class="op">=</span><span class="fl">.6</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab</a></span><span class="op">(</span>hjust<span class="op">=</span><span class="fl">.5</span>,vjust<span class="op">=</span><span class="fl">2</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/tree-layout.html">layout_dendrogram</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:treeGraph"></span>
<img src="treedata_files/figure-html/treeGraph-1.svg" alt="Visualizing a tree graph. The lines with arrows indicate the relationship between the parent node and the child node. All nodes were indicated by steelblue circle points." width="100%"><p class="caption">
FIGURE 9.6: <strong>Visualizing a tree graph.</strong> The lines with arrows indicate the relationship between the parent node and the child node. All nodes were indicated by steelblue circle points.
</p>
</div>
</div>
<div id="ggtree-for-other-tree-like-structures" class="section level2" number="9.4">
<h2>
<span class="header-section-number">9.4</span> ggtree for other tree-like structures<a class="anchor" aria-label="anchor" href="#ggtree-for-other-tree-like-structures"><i class="fas fa-link"></i></a>
</h2>
<p>The <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> package can be used to visualize any data in a hierarchical structure. Here, we use the GNI (Gross National Income) numbers in 2014 as an example. After preparing an edge list, that is a matrix or data frame that contains two columns indicating the relationship of parent and child nodes, we can use the <code><a href="https://rdrr.io/pkg/ape/man/as.phylo.html">as.phylo()</a></code> method provided by the <a href="http://bioconductor.org/packages/treeio"><strong>treeio</strong></a> package to convert the edge list to a <code>phylo</code> object. Then it can be visualized using <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> with associated data. In this example, the population was used to scale the size of circle points for each country.</p>


<div class="sourceCode" id="cb223"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/YuLab-SMU/treeio">treeio</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/treedata-book/">ggtree</a></span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"GNI2014"</span>, package<span class="op">=</span><span class="st">"treemap"</span><span class="op">)</span>
<span class="va">n</span> <span class="op">&lt;-</span> <span class="va">GNI2014</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">3</span>,<span class="fl">1</span><span class="op">)</span><span class="op">]</span>
<span class="va">n</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">n</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span><span class="op">)</span>
<span class="va">n</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"\\s\\(.*\\)"</span>, <span class="st">""</span>, <span class="va">n</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span><span class="op">)</span>

<span class="va">w</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">cbind</a></span><span class="op">(</span><span class="st">"World"</span>, <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">n</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">w</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">n</span><span class="op">)</span>
<span class="va">edgelist</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span><span class="op">(</span><span class="va">n</span>, <span class="va">w</span><span class="op">)</span>

<span class="va">y</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ape/man/as.phylo.html">as.phylo</a></span><span class="op">(</span><span class="va">edgelist</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">y</span>, layout<span class="op">=</span><span class="st">'circular'</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/ggtree/man/attacher.html">%&lt;+%</a></span> <span class="va">GNI2014</span> <span class="op">+</span> 
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>color<span class="op">=</span><span class="va">continent</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tippoint.html">geom_tippoint</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>size<span class="op">=</span><span class="va">population</span><span class="op">)</span>, alpha<span class="op">=</span><span class="fl">.6</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>label<span class="op">=</span><span class="va">country</span><span class="op">)</span>, offset<span class="op">=</span><span class="fl">.1</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>plot.margin<span class="op">=</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">margin</a></span><span class="op">(</span><span class="fl">60</span>,<span class="fl">60</span>,<span class="fl">60</span>,<span class="fl">60</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:gni2014"></span>
<img src="treedata_files/figure-html/gni2014-1.png" alt="Visualizing data in any hierarchical structure. Hierarchical data represented as nodes connected by edges can be converted to a phylo object and visualized by ggtree to explore their relationships or other properties that are associated with the relationships." width="100%"><p class="caption">
FIGURE 9.7: <strong>Visualizing data in any hierarchical structure.</strong> Hierarchical data represented as nodes connected by edges can be converted to a <code>phylo</code> object and visualized by <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> to explore their relationships or other properties that are associated with the relationships.
</p>
</div>
</div>
<div id="summary9" class="section level2" number="9.5">
<h2>
<span class="header-section-number">9.5</span> Summary<a class="anchor" aria-label="anchor" href="#summary9"><i class="fas fa-link"></i></a>
</h2>
<p>The <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> supports various tree objects defined in the R language and extension packages, which makes it very easy to integrate <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> into existing pipelines. Moreover, <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> allows external data integration and exploration of these data on the tree, which will greatly promote the data visualization and result interpretation in the downstream analysis of existing pipelines. Most importantly, the support for converting edge list to a tree object enables more tree-like structures to be incorporated into the framework of <a href="http://bioconductor.org/packages/treeio"><strong>treeio</strong></a> and <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a>. This will enable more tree-like structures and related heterogeneous data in different disciplines to be integrated and visualized through <a href="http://bioconductor.org/packages/treeio"><strong>treeio</strong></a> and <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a>, which facilitates integrated analysis and comparative analysis to discover more systematic patterns and insights.</p>

</div>
</div>



<div class="chapter-nav">
<div class="prev"><a href="chapter8.html"><span class="header-section-number">8</span> Annotating Tree with Silhouette Images and Sub-plots</a></div>
<div class="next"><a href="chapter10.html"><span class="header-section-number">10</span> ggtreeExtra for presenting data on a circular layout</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#chapter9"><span class="header-section-number">9</span> ggtree for other tree-like objects</a></li>
<li>
<a class="nav-link" href="#ggtree-for-phylogenetic-tree-objects"><span class="header-section-number">9.1</span> ggtree for phylogenetic tree objects</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#phylobase"><span class="header-section-number">9.1.1</span> The phylo4 and phylo4d objects</a></li>
<li><a class="nav-link" href="#phylog"><span class="header-section-number">9.1.2</span> The phylog object</a></li>
<li><a class="nav-link" href="#phyloseq"><span class="header-section-number">9.1.3</span> The phyloseq object</a></li>
</ul>
</li>
<li><a class="nav-link" href="#dendrogram"><span class="header-section-number">9.2</span> ggtree for dendrograms</a></li>
<li><a class="nav-link" href="#igraph"><span class="header-section-number">9.3</span> ggtree for tree graph</a></li>
<li><a class="nav-link" href="#ggtree-for-other-tree-like-structures"><span class="header-section-number">9.4</span> ggtree for other tree-like structures</a></li>
<li><a class="nav-link" href="#summary9"><span class="header-section-number">9.5</span> Summary</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>
</div>
  

  

</div>
 <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Data Integration, Manipulation and Visualization of Phylogenetic Trees</strong>" was written by Guangchuang Yu. It was last built on 2021-11-29.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
