<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 7 Plotting tree with data | Data Integration, Manipulation and Visualization of Phylogenetic Trees with R</title>
<meta name="author" content="Guangchuang Yu">
<meta name="description" content="Integrating user data to annotate a phylogenetic tree can be done at different levels. The treeio package (Wang et al., 2020) implements full_join() methods to combine tree data to phylogenetic...">
<meta name="generator" content="bookdown 0.25 with bs4_book()">
<meta property="og:title" content="Chapter 7 Plotting tree with data | Data Integration, Manipulation and Visualization of Phylogenetic Trees with R">
<meta property="og:type" content="book">
<meta property="og:image" content="/book-cover.png">
<meta property="og:description" content="Integrating user data to annotate a phylogenetic tree can be done at different levels. The treeio package (Wang et al., 2020) implements full_join() methods to combine tree data to phylogenetic...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 7 Plotting tree with data | Data Integration, Manipulation and Visualization of Phylogenetic Trees with R">
<meta name="twitter:description" content="Integrating user data to annotate a phylogenetic tree can be done at different levels. The treeio package (Wang et al., 2020) implements full_join() methods to combine tree data to phylogenetic...">
<meta name="twitter:image" content="/book-cover.png">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap/bootstrap.bundle.min.js"></script><script src="libs/bs3compat/transition.js"></script><script src="libs/bs3compat/tabs.js"></script><script src="libs/bs3compat/bs3compat.js"></script><link href="libs/bs4_book/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book/bs4_book.js"></script><script src="libs/kePrint/kePrint.js"></script><link href="libs/lightable/lightable.css" rel="stylesheet">
<script src="libs/htmlwidgets/htmlwidgets.js"></script><script src="libs/pathseg/pathseg.js"></script><script src="libs/comic/comic.min.js"></script><script src="libs/comicR-binding/comicR.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<style type="text/css">
    /* Used with Pandoc 2.11+ new --citeproc when CSL is used */
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
        }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Data Integration, Manipulation and Visualization of Phylogenetic Trees with R</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">Preface</a></li>
<li><a class="" href="author.html">About the Author</a></li>
<li class="book-part">Part I: Tree data input, output, and manipulation</li>
<li><a class="" href="chapter1.html"><span class="header-section-number">1</span> Importing Tree with Data</a></li>
<li><a class="" href="chapter2.html"><span class="header-section-number">2</span> Manipulating Tree with Data</a></li>
<li><a class="" href="chapter3.html"><span class="header-section-number">3</span> Exporting tree with data</a></li>
<li class="book-part">Part II: Tree data visualization and annotation</li>
<li><a class="" href="chapter4.html"><span class="header-section-number">4</span> Phylogenetic Tree Visualization</a></li>
<li><a class="" href="chapter5.html"><span class="header-section-number">5</span> Phylogenetic Tree Annotation</a></li>
<li><a class="" href="chapter6.html"><span class="header-section-number">6</span> Visual Exploration of Phylogenetic Trees</a></li>
<li><a class="active" href="chapter7.html"><span class="header-section-number">7</span> Plotting tree with data</a></li>
<li><a class="" href="chapter8.html"><span class="header-section-number">8</span> Annotating Tree with Silhouette Images and Sub-plots</a></li>
<li><a class="" href="chapter9.html"><span class="header-section-number">9</span> ggtree for Other Tree-like Objects</a></li>
<li class="book-part">Part III: ggtree extensions</li>
<li><a class="" href="chapter10.html"><span class="header-section-number">10</span> ggtreeExtra for Presenting Data on a Circular Layout</a></li>
<li><a class="" href="chapter11.html"><span class="header-section-number">11</span> Other ggtree Extensions</a></li>
<li class="book-part">Part IV: Miscellaneous topics</li>
<li><a class="" href="chapter12.html"><span class="header-section-number">12</span> ggtree Utilities</a></li>
<li><a class="" href="chapter13.html"><span class="header-section-number">13</span> Gallery of Reproducible Examples</a></li>
<li class="book-part">Appendix</li>
<li><a class="" href="faq.html"><span class="header-section-number">A</span> Frequently Asked Questions</a></li>
<li><a class="" href="related-tools.html"><span class="header-section-number">B</span> Related Tools</a></li>
<li><a class="" href="figures-and-tables.html"><span class="header-section-number">C</span> Figures and Tables</a></li>
<li><a class="" href="publications-of-the-ggtree-package-suite.html"><span class="header-section-number">D</span> Publications of the ggtree Package Suite</a></li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="chapter7" class="section level1" number="7">
<h1>
<span class="header-section-number">7</span> Plotting tree with data<a class="anchor" aria-label="anchor" href="#chapter7"><i class="fas fa-link"></i></a>
</h1>
<p>Integrating user data to annotate a phylogenetic tree can be done at different
levels. The <a href="http://bioconductor.org/packages/treeio"><strong>treeio</strong></a> package <span class="citation">(<a href="references.html#ref-wang_treeio_2020" role="doc-biblioref">Wang et al., 2020</a>)</span> implements <code><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">full_join()</a></code> methods to combine tree data to phylogenetic tree object.
The <a href="https://CRAN.R-project.org/package=tidytree"><strong>tidytree</strong></a> package supports linking tree data to phylogeny
using tidyverse verbs (see also <a href="chapter2.html#chapter2">Chapter 2</a>).
The <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> package <span class="citation">(<a href="references.html#ref-yu_two_2018" role="doc-biblioref">Yu et al., 2018</a>)</span> supports mapping external data to phylogeny for
visualization and annotation on the fly. Although the feature of linking external data is overlapping among these packages, they have different application scopes. For example, in addition to the <code>treedata</code> object, <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> also supports several other tree objects (see <a href="chapter9.html#chapter9">Chapter 9</a>), including <code>phylo4d</code>, <code>phyloseq</code>, and <code>obkData</code> that were designed to contain domain-specific data. The design of these objects did not consider supporting linking external data to the object (it can not be done at the tree object level). We can visualize trees from these objects using <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> and link external data at the visualization level <span class="citation">(<a href="references.html#ref-yu_two_2018" role="doc-biblioref">Yu et al., 2018</a>)</span>.</p>
<p>The <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> package provides two general methods for mapping and visualizing associated external data on phylogenies. <a href="chapter7.html#attach-operator">Method 1</a> allows external data to be mapped on the tree structure and used as visual characteristics in the tree and data visualization. <a href="chapter7.html#facet_plot">Method 2</a> plots the data with the tree side-by-side using different geometric functions after reordering the data based on the tree structure. These two methods integrate data with phylogeny for further exploration and comparison in the evolutionary biology context. The <a href="http://bioconductor.org/packages/ggtreeExtra"><strong>ggtreeExtra</strong></a> provides a better implementation of the Method 2 proposed in <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> (see also <a href="chapter10.html#chapter10">Chapter 10</a>) and works with both rectangular and circular layouts <span class="citation">(<a href="references.html#ref-ggtreeExtra_2021" role="doc-biblioref">Xu, Dai, et al., 2021</a>)</span>.</p>
<div id="attach-operator" class="section level2" number="7.1">
<h2>
<span class="header-section-number">7.1</span> Mapping Data to The tree Structure<a class="anchor" aria-label="anchor" href="#attach-operator"><i class="fas fa-link"></i></a>
</h2>
<p>In <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a>, we implemented an operator, <code>%&lt;+%</code>, to attach annotation data to a <code>ggtree</code> graphic object. Any data that contains a column of “node” or the first column of taxa labels can be integrated using the <code>%&lt;+%</code> operator. Multiple datasets can be attached progressively. When the data are attached, all the information stored in the data serves as numerical/categorical node attributes and can be directly used to visualize the tree by scaling the attributes as different colors or line sizes, labeling the tree using the original values of the attributes or parsing them as <a href="faq.html#faq-formatting-label">math expression</a>, <a href="chapter8.html#phylomoji">emoji</a> or <a href="chapter8.html#ggimage">silhouette image</a>. The following example uses the <code>%&lt;+%</code> operator to integrate taxon (<code>df_tip_data</code>) and internal node (<code>df_inode_data</code>) information and map the data to different colors or shapes of symbolic points and labels (Figure <a href="chapter7.html#fig:attacher">7.1</a>). The tip data contains <code>imageURL</code> that links to online figures of the species, which can be parsed and used as tip labels in <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> (see <a href="chapter8.html#chapter8">Chapter 8</a>).</p>


<div class="sourceCode" id="cb191"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/GuangchuangYu/ggimage">ggimage</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/treedata-book/">ggtree</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/YuLab-SMU/TDBook">TDbook</a></span><span class="op">)</span>

<span class="co"># load `tree_boots`, `df_tip_data`, and `df_inode_data` from 'TDbook'</span>
<span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree_boots</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/ggtree/man/attacher.html">%&lt;+%</a></span> <span class="va">df_tip_data</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html">xlim</a></span><span class="op">(</span><span class="op">-</span><span class="fl">.1</span>, <span class="fl">4</span><span class="op">)</span>
<span class="va">p2</span> <span class="op">&lt;-</span> <span class="va">p</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab</a></span><span class="op">(</span>offset <span class="op">=</span> <span class="fl">.6</span>, hjust <span class="op">=</span> <span class="fl">.5</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tippoint.html">geom_tippoint</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>shape <span class="op">=</span> <span class="va">trophic_habit</span>, color <span class="op">=</span> <span class="va">trophic_habit</span>, 
                size <span class="op">=</span> <span class="va">mass_in_kg</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"right"</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_size.html">scale_size_continuous</a></span><span class="op">(</span>range <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">3</span>, <span class="fl">10</span><span class="op">)</span><span class="op">)</span>

<span class="va">p2</span> <span class="op"><a href="https://rdrr.io/pkg/ggtree/man/attacher.html">%&lt;+%</a></span> <span class="va">df_inode_data</span> <span class="op">+</span> 
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_text.html">geom_label</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>label <span class="op">=</span> <span class="va">vernacularName.y</span>, fill <span class="op">=</span> <span class="va">posterior</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_gradient.html">scale_fill_gradientn</a></span><span class="op">(</span>colors <span class="op">=</span> <span class="fu">RColorBrewer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html">brewer.pal</a></span><span class="op">(</span><span class="fl">3</span>, <span class="st">"YlGnBu"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:attacher"></span>
<img src="treedata_files/figure-html/attacher-1.svg" alt="Example of attaching multiple datasets. External datasets including tip data (e.g., trophic habit and body weight) and node data (e.g., clade posterior and vernacular name) were attached to the ggtree graphic via the %&lt;+% operator and the data was used to annotate the tree." width="100%"><p class="caption">
FIGURE 7.1: <strong>Example of attaching multiple datasets</strong>. External datasets including tip data (e.g., trophic habit and body weight) and node data (e.g., clade posterior and vernacular name) were attached to the <code>ggtree</code> graphic via the <code>%&lt;+%</code> operator and the data was used to annotate the tree.
</p>
</div>
<p>Although the data integrated by the <code>%&lt;+%</code> operator in <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> is for tree visualization, the data attached to the <code>ggtree</code> graphic object can be converted to <code>treedata</code> object that contains the tree and the attached data (see <a href="#ggtree_object">session 7.5</a>).</p>
</div>
<div id="facet_plot" class="section level2" number="7.2">
<h2>
<span class="header-section-number">7.2</span> Aligning Graph to the Tree Based on the Tree Structure<a class="anchor" aria-label="anchor" href="#facet_plot"><i class="fas fa-link"></i></a>
</h2>
<p>For associating phylogenetic tree with different types of plot produced by user’s data, <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> provides <code><a href="https://rdrr.io/pkg/ggtree/man/facet-plot.html">geom_facet()</a></code> layer and <code><a href="https://rdrr.io/pkg/ggtree/man/facet-plot.html">facet_plot()</a></code> function which accept an input <code>data.frame</code> and a <code>geom</code> layer to draw the input data. The data will be displayed in an additional panel of the plot. The <code><a href="https://rdrr.io/pkg/ggtree/man/facet-plot.html">geom_facet()</a></code> (or <code>facet_plot</code>) is a general solution for linking the graphic layer to a tree. The function internally re-orders the input data based on the tree structure and visualizes the data at the specific panel by the geometric layer. Users are free to visualize several panels to plot different types of data as demonstrated in Figure <a href="chapter9.html#fig:phyloseq">9.4</a> and to use different geometric layers to plot the same dataset (Figure <a href="chapter13.html#fig:jv2017">13.1</a>) or different datasets on the same panel.</p>
<p>The <code><a href="https://rdrr.io/pkg/ggtree/man/facet-plot.html">geom_facet()</a></code> is designed to work with most of the <code>geom</code> layers defined in <a href="https://CRAN.R-project.org/package=ggplot2"><strong>ggplot2</strong></a> and other <a href="https://CRAN.R-project.org/package=ggplot2"><strong>ggplot2</strong></a>-based packages. A list of the geometric layers that work seamlessly with <code><a href="https://rdrr.io/pkg/ggtree/man/facet-plot.html">geom_facet()</a></code> and <code><a href="https://rdrr.io/pkg/ggtree/man/facet-plot.html">facet_plot()</a></code> can be found in Table <a href="figures-and-tables.html#tab:facet-geom">C.1</a>. As the <a href="https://CRAN.R-project.org/package=ggplot2"><strong>ggplot2</strong></a> community keeps expanding and more <code>geom</code> layers will be implemented in either <a href="https://CRAN.R-project.org/package=ggplot2"><strong>ggplot2</strong></a> or other extensions, <code><a href="https://rdrr.io/pkg/ggtree/man/facet-plot.html">geom_facet()</a></code> and <code><a href="https://rdrr.io/pkg/ggtree/man/facet-plot.html">facet_plot()</a></code> will gain more power to present data in the future. Note that different <code>geom</code> layers can be combined to present data on the same panel and the combinations of different <code>geom</code> layers create the possibility to present more complex data with phylogeny (see also Figures <a href="chapter13.html#fig:jv2017">13.1</a> and <a href="chapter13.html#fig:gggenes">13.4</a>). Users can progressively add multiple panels to present and compare different datasets in the evolutionary context (Figure <a href="chapter7.html#fig:plottree">7.2</a>). Detailed descriptions can be found in the <a href="https://github.com/GuangchuangYu/plotting_tree_with_data/">supplemental file</a> of <span class="citation">(<a href="references.html#ref-yu_two_2018" role="doc-biblioref">Yu et al., 2018</a>)</span>.</p>


<div class="sourceCode" id="cb192"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/treedata-book/">ggtree</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/YuLab-SMU/TDBook">TDbook</a></span><span class="op">)</span>

<span class="co">## load `tree_nwk`, `df_info`, `df_alleles`, and `df_bar_data` from 'TDbook'</span>
<span class="va">tree</span> <span class="op">&lt;-</span> <span class="va">tree_nwk</span>
<span class="va">snps</span> <span class="op">&lt;-</span> <span class="va">df_alleles</span>
<span class="va">snps_strainCols</span> <span class="op">&lt;-</span> <span class="va">snps</span><span class="op">[</span><span class="fl">1</span>,<span class="op">]</span> 
<span class="va">snps</span><span class="op">&lt;-</span><span class="va">snps</span><span class="op">[</span><span class="op">-</span><span class="fl">1</span>,<span class="op">]</span> <span class="co"># drop strain names</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">snps</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">snps_strainCols</span>

<span class="va">gapChar</span> <span class="op">&lt;-</span> <span class="st">"?"</span>
<span class="va">snp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span><span class="op">(</span><span class="va">snps</span><span class="op">)</span>
<span class="va">lsnp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span><span class="op">(</span><span class="va">snp</span>, <span class="fl">1</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span>
        <span class="va">x</span> <span class="op">!=</span> <span class="va">snp</span><span class="op">[</span><span class="fl">1</span>,<span class="op">]</span> <span class="op">&amp;</span> <span class="va">x</span> <span class="op">!=</span> <span class="va">gapChar</span> <span class="op">&amp;</span> <span class="va">snp</span><span class="op">[</span><span class="fl">1</span>,<span class="op">]</span> <span class="op">!=</span> <span class="va">gapChar</span>
    <span class="op">}</span><span class="op">)</span>
<span class="va">lsnp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="va">lsnp</span><span class="op">)</span>
<span class="va">lsnp</span><span class="op">$</span><span class="va">pos</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">lsnp</span><span class="op">)</span><span class="op">)</span>
<span class="va">lsnp</span> <span class="op">&lt;-</span> <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/gather.html">gather</a></span><span class="op">(</span><span class="va">lsnp</span>, <span class="va">name</span>, <span class="va">value</span>, <span class="op">-</span><span class="va">pos</span><span class="op">)</span>
<span class="va">snp_data</span> <span class="op">&lt;-</span> <span class="va">lsnp</span><span class="op">[</span><span class="va">lsnp</span><span class="op">$</span><span class="va">value</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"name"</span>, <span class="st">"pos"</span><span class="op">)</span><span class="op">]</span>

<span class="co">## visualize the tree </span>
<span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree</span><span class="op">)</span> 

<span class="co">## attach the sampling information data set </span>
<span class="co">## and add symbols colored by location</span>
<span class="va">p</span> <span class="op">&lt;-</span> <span class="va">p</span> <span class="op"><a href="https://rdrr.io/pkg/ggtree/man/attacher.html">%&lt;+%</a></span> <span class="va">df_info</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tippoint.html">geom_tippoint</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>color<span class="op">=</span><span class="va">location</span><span class="op">)</span><span class="op">)</span>

<span class="co">## visualize SNP and Trait data using dot and bar charts,</span>
<span class="co">## and align them based on tree structure</span>
<span class="va">p</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/facet-plot.html">geom_facet</a></span><span class="op">(</span>panel <span class="op">=</span> <span class="st">"SNP"</span>, data <span class="op">=</span> <span class="va">snp_data</span>, geom <span class="op">=</span> <span class="va">geom_point</span>, 
               mapping<span class="op">=</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">pos</span>, color <span class="op">=</span> <span class="va">location</span><span class="op">)</span>, shape <span class="op">=</span> <span class="st">'|'</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/facet-plot.html">geom_facet</a></span><span class="op">(</span>panel <span class="op">=</span> <span class="st">"Trait"</span>, data <span class="op">=</span> <span class="va">df_bar_data</span>, geom <span class="op">=</span> <span class="va">geom_col</span>, 
                <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">dummy_bar_value</span>, color <span class="op">=</span> <span class="va">location</span>, 
                fill <span class="op">=</span> <span class="va">location</span><span class="op">)</span>, orientation <span class="op">=</span> <span class="st">'y'</span>, width <span class="op">=</span> <span class="fl">.6</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/theme_tree2.html">theme_tree2</a></span><span class="op">(</span>legend.position<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">.05</span>, <span class="fl">.85</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:plottree"></span>
<img src="treedata_files/figure-html/plottree-1.png" alt="Example of plotting SNP and trait data. The ‘location’ information was attached to the tree and used to color tip symbols (Tree panel), and other datasets. SNP and Trait data were visualized as dot chart (SNP panel) and bar chart (Trait panel)." width="100%"><p class="caption">
FIGURE 7.2: <strong>Example of plotting SNP and trait data</strong>. The ‘location’ information was attached to the tree and used to color tip symbols (Tree panel), and other datasets. SNP and Trait data were visualized as dot chart (SNP panel) and bar chart (Trait panel).
</p>
</div>
<p>Companion functions to adjust <a href="chapter12.html#facet_widths">panel widths</a> and <a href="chapter12.html#facet_labeller">rename panel names</a> are described in <a href="chapter12.html#facet-utils">session 12.1</a>. Removing the panel name is also possible and an example was presented in Figure <a href="chapter13.html#fig:gggenes">13.4</a>. We can also use <a href="https://CRAN.R-project.org/package=aplot"><strong>aplot</strong></a> or <a href="https://CRAN.R-project.org/package=patchwork"><strong>patchwork</strong></a> to create composite plots as described in <a href="chapter7.html#composite_plot">session 7.5</a>.</p>
<p>The <code><a href="https://rdrr.io/pkg/ggtree/man/facet-plot.html">geom_facet()</a></code> (or <code><a href="https://rdrr.io/pkg/ggtree/man/facet-plot.html">facet_plot()</a></code>) internally used <code><a href="https://ggplot2.tidyverse.org/reference/facet_grid.html">ggplot2::facet_grid()</a></code> and only works with Cartesian coordinate system. To align the graph to the tree for the polar system (e.g., for circular or fan layouts), we developed another Bioconductor package, <a href="http://bioconductor.org/packages/ggtreeExtra"><strong>ggtreeExtra</strong></a>. The <a href="http://bioconductor.org/packages/ggtreeExtra"><strong>ggtreeExtra</strong></a> package provides the <code>geom_fruit()</code> layer that works similar to <code><a href="https://rdrr.io/pkg/ggtree/man/facet-plot.html">geom_facet()</a></code> (details described in <a href="chapter10.html#chapter10">Chapter 10</a>). The <code>geom_fruit()</code> is a better implementation of the Method 2 proposed in <span class="citation">(<a href="references.html#ref-yu_two_2018" role="doc-biblioref">Yu et al., 2018</a>)</span>.</p>
</div>
<div id="gheatmap" class="section level2" number="7.3">
<h2>
<span class="header-section-number">7.3</span> Visualize a Tree with an Associated Matrix<a class="anchor" aria-label="anchor" href="#gheatmap"><i class="fas fa-link"></i></a>
</h2>
<p>The <code><a href="https://rdrr.io/pkg/ggtree/man/gheatmap.html">gheatmap()</a></code> function is designed to visualize the phylogenetic tree with a heatmap of an associated matrix (either numerical or categorical). The <code><a href="https://rdrr.io/pkg/ggtree/man/facet-plot.html">geom_facet()</a></code> layer is a general solution for plotting data with the tree, including heatmap. The <code><a href="https://rdrr.io/pkg/ggtree/man/gheatmap.html">gheatmap()</a></code> function is specifically designed for plotting heatmap with a tree and provides a shortcut for handling column labels and color palettes. Another difference is that <code><a href="https://rdrr.io/pkg/ggtree/man/facet-plot.html">geom_facet()</a></code> only supports rectangular and slanted tree layouts, while <code><a href="https://rdrr.io/pkg/ggtree/man/gheatmap.html">gheatmap()</a></code> supports rectangular, slanted, and circular (Figure <a href="chapter7.html#fig:mgheatmap">7.4</a>) layouts.</p>
<p>In the following example, we visualized a tree of H3 influenza viruses with their associated genotypes (Figure <a href="chapter7.html#fig:gheatmap">7.3</a>A).</p>
<div class="sourceCode" id="cb193"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">beast_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"examples/MCC_FluA_H3.tree"</span>, package<span class="op">=</span><span class="st">"ggtree"</span><span class="op">)</span>
<span class="va">beast_tree</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://docs.ropensci.org/treeio/reference/beast-parser.html">read.beast</a></span><span class="op">(</span><span class="va">beast_file</span><span class="op">)</span>

<span class="va">genotype_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"examples/Genotype.txt"</span>, package<span class="op">=</span><span class="st">"ggtree"</span><span class="op">)</span>
<span class="va">genotype</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span><span class="va">genotype_file</span>, sep<span class="op">=</span><span class="st">"\t"</span>, stringsAsFactor<span class="op">=</span><span class="cn">F</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">genotype</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">sub</a></span><span class="op">(</span><span class="st">"\\.$"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">genotype</span><span class="op">)</span><span class="op">)</span>
<span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">beast_tree</span>, mrsd<span class="op">=</span><span class="st">"2013-01-01"</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_treescale.html">geom_treescale</a></span><span class="op">(</span>x<span class="op">=</span><span class="fl">2008</span>, y<span class="op">=</span><span class="fl">1</span>, offset<span class="op">=</span><span class="fl">2</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">2</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/gheatmap.html">gheatmap</a></span><span class="op">(</span><span class="va">p</span>, <span class="va">genotype</span>, offset<span class="op">=</span><span class="fl">5</span>, width<span class="op">=</span><span class="fl">0.5</span>, font.size<span class="op">=</span><span class="fl">3</span>, 
        colnames_angle<span class="op">=</span><span class="op">-</span><span class="fl">45</span>, hjust<span class="op">=</span><span class="fl">0</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_fill_manual</a></span><span class="op">(</span>breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"HuH3N2"</span>, <span class="st">"pdm"</span>, <span class="st">"trig"</span><span class="op">)</span>, 
        values<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"steelblue"</span>, <span class="st">"firebrick"</span>, <span class="st">"darkgreen"</span><span class="op">)</span>, name<span class="op">=</span><span class="st">"genotype"</span><span class="op">)</span></code></pre></div>
<p>The <code>width</code> parameter is to control the width of the heatmap. It supports another parameter <code>offset</code> for controlling the distance between the tree and the heatmap, such as allocating space for tip labels.</p>
<p>For a timescaled tree, as in this example, it’s more common to use <em>x</em>-axis by using <code>theme_tree2</code>. But with this solution, the heatmap is just another layer and will change the <em>x</em>-axis. To overcome this issue, we implemented <code><a href="https://rdrr.io/pkg/ggtree/man/scale_x_ggtree.html">scale_x_ggtree()</a></code> to set the <em>x</em>-axis more reasonably (Figure <a href="chapter7.html#fig:gheatmap">7.3</a>B).</p>
<div class="sourceCode" id="cb194"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">beast_tree</span>, mrsd<span class="op">=</span><span class="st">"2013-01-01"</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">2</span>, align<span class="op">=</span><span class="cn">TRUE</span>, linesize<span class="op">=</span><span class="fl">.5</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/theme_tree2.html">theme_tree2</a></span><span class="op">(</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/gheatmap.html">gheatmap</a></span><span class="op">(</span><span class="va">p</span>, <span class="va">genotype</span>, offset<span class="op">=</span><span class="fl">8</span>, width<span class="op">=</span><span class="fl">0.6</span>, 
        colnames<span class="op">=</span><span class="cn">FALSE</span>, legend_title<span class="op">=</span><span class="st">"genotype"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/scale_x_ggtree.html">scale_x_ggtree</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html">scale_y_continuous</a></span><span class="op">(</span>expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0.3</span><span class="op">)</span><span class="op">)</span></code></pre></div>


<div class="figure">
<span style="display:block;" id="fig:gheatmap"></span>
<img src="treedata_files/figure-html/gheatmap-1.png" alt="Example of plotting matrix with gheatmap(). A H3 influenza tree with a genotype table visualized as a heatmap (A). Tips were aligned and with a tailored x-axis for divergence times (tree) and genomic segments (heatmap) (B)." width="90%"><p class="caption">
FIGURE 7.3: <strong>Example of plotting matrix with <code><a href="https://rdrr.io/pkg/ggtree/man/gheatmap.html">gheatmap()</a></code></strong>. A H3 influenza tree with a genotype table visualized as a heatmap (A). Tips were aligned and with a tailored <em>x</em>-axis for divergence times (tree) and genomic segments (heatmap) (B).
</p>
</div>
<div id="gheatmap-ggnewscale" class="section level3" number="7.3.1">
<h3>
<span class="header-section-number">7.3.1</span> Visualize a tree with multiple associated matrices<a class="anchor" aria-label="anchor" href="#gheatmap-ggnewscale"><i class="fas fa-link"></i></a>
</h3>
<p>Of course, we can use multiple <code><a href="https://rdrr.io/pkg/ggtree/man/gheatmap.html">gheatmap()</a></code> function calls to align several associated matrices with the tree. However, <a href="https://CRAN.R-project.org/package=ggplot2"><strong>ggplot2</strong></a> doesn’t allow us to use multiple <code>fill</code> scales<a class="footnote-ref" tabindex="0" data-toggle="popover" data-content='&lt;p&gt;See also discussion in &lt;a href="https://github.com/GuangchuangYu/ggtree/issues/78" class="uri"&gt;https://github.com/GuangchuangYu/ggtree/issues/78&lt;/a&gt; and &lt;a href="https://groups.google.com/d/msg/bioc-ggtree/VQqbF79NAWU/IjIvpQOBGwAJ" class="uri"&gt;https://groups.google.com/d/msg/bioc-ggtree/VQqbF79NAWU/IjIvpQOBGwAJ&lt;/a&gt;&lt;/p&gt;'><sup>11</sup></a>.</p>
<p>To solve this issue, we can use the <a href="https://CRAN.R-project.org/package=ggnewscale"><strong>ggnewscale</strong></a> package to create new <code>fill</code> scales. Here is an example of using <a href="https://CRAN.R-project.org/package=ggnewscale"><strong>ggnewscale</strong></a> with <code><a href="https://rdrr.io/pkg/ggtree/man/gheatmap.html">gheatmap()</a></code>.</p>


<div class="sourceCode" id="cb195"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">nwk</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"sample.nwk"</span>, package<span class="op">=</span><span class="st">"treeio"</span><span class="op">)</span>

<span class="va">tree</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ape/man/read.tree.html">read.tree</a></span><span class="op">(</span><span class="va">nwk</span><span class="op">)</span>
<span class="va">circ</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree</span>, layout <span class="op">=</span> <span class="st">"circular"</span><span class="op">)</span>

<span class="va">df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>first<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"a"</span>, <span class="st">"b"</span>, <span class="st">"a"</span>, <span class="st">"c"</span>, <span class="st">"d"</span>, <span class="st">"d"</span>, <span class="st">"a"</span>, 
                        <span class="st">"b"</span>, <span class="st">"e"</span>, <span class="st">"e"</span>, <span class="st">"f"</span>, <span class="st">"c"</span>, <span class="st">"f"</span><span class="op">)</span>,
                 second<span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"z"</span>, <span class="st">"z"</span>, <span class="st">"z"</span>, <span class="st">"z"</span>, <span class="st">"y"</span>, <span class="st">"y"</span>, 
                        <span class="st">"y"</span>, <span class="st">"y"</span>, <span class="st">"x"</span>, <span class="st">"x"</span>, <span class="st">"x"</span>, <span class="st">"a"</span>, <span class="st">"a"</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">df</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">tree</span><span class="op">$</span><span class="va">tip.label</span>

<span class="va">df2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Normal.html">rnorm</a></span><span class="op">(</span><span class="fl">39</span><span class="op">)</span>, ncol<span class="op">=</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">df2</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">tree</span><span class="op">$</span><span class="va">tip.label</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">df2</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">LETTERS</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span>


<span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/gheatmap.html">gheatmap</a></span><span class="op">(</span><span class="va">circ</span>, <span class="va">df</span>, offset<span class="op">=</span><span class="fl">.8</span>, width<span class="op">=</span><span class="fl">.2</span>,
               colnames_angle<span class="op">=</span><span class="fl">95</span>, colnames_offset_y <span class="op">=</span> <span class="fl">.25</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_viridis.html">scale_fill_viridis_d</a></span><span class="op">(</span>option<span class="op">=</span><span class="st">"D"</span>, name<span class="op">=</span><span class="st">"discrete\nvalue"</span><span class="op">)</span>


<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">ggnewscale</span><span class="op">)</span>
<span class="va">p2</span> <span class="op">&lt;-</span> <span class="va">p1</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggnewscale/man/new_scale.html">new_scale_fill</a></span><span class="op">(</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/gheatmap.html">gheatmap</a></span><span class="op">(</span><span class="va">p2</span>, <span class="va">df2</span>, offset<span class="op">=</span><span class="fl">15</span>, width<span class="op">=</span><span class="fl">.3</span>,
         colnames_angle<span class="op">=</span><span class="fl">90</span>, colnames_offset_y <span class="op">=</span> <span class="fl">.25</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_viridis.html">scale_fill_viridis_c</a></span><span class="op">(</span>option<span class="op">=</span><span class="st">"A"</span>, name<span class="op">=</span><span class="st">"continuous\nvalue"</span><span class="op">)</span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:mgheatmap"></span>
<img src="treedata_files/figure-html/mgheatmap-1.svg" alt="Example of plotting matrix with gheatmap(). A H3 influenza tree with a genotype table visualized as a heatmap (A). Tips were aligned and with a tailored x-axis for divergence times (tree) and genomic segments (heatmap) (B)." width="100%"><p class="caption">
FIGURE 7.4: <strong>Example of plotting matrix with <code><a href="https://rdrr.io/pkg/ggtree/man/gheatmap.html">gheatmap()</a></code></strong>. A H3 influenza tree with a genotype table visualized as a heatmap (A). Tips were aligned and with a tailored <em>x</em>-axis for divergence times (tree) and genomic segments (heatmap) (B).
</p>
</div>
</div>
</div>
<div id="msaplot" class="section level2" number="7.4">
<h2>
<span class="header-section-number">7.4</span> Visualize a Tree with Multiple Sequence Alignments<a class="anchor" aria-label="anchor" href="#msaplot"><i class="fas fa-link"></i></a>
</h2>
<p>The <code><a href="https://rdrr.io/pkg/ggtree/man/msaplot.html">msaplot()</a></code> accepts a tree (output of <code><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree()</a></code>) and a fasta file, then it can visualize the tree with sequence alignment. We can specify the <code>width</code> (relative to the tree) of the alignment and adjust the relative position by <code>offset</code>, which is similar to the <code><a href="https://rdrr.io/pkg/ggtree/man/gheatmap.html">gheatmap()</a></code> function (Figure <a href="chapter7.html#fig:msaplot">7.5</a>A)..</p>
<div class="sourceCode" id="cb196"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/YuLab-SMU/TDBook">TDbook</a></span><span class="op">)</span>

<span class="co"># load `tree_seq_nwk` and `AA_sequence` from 'TDbook'</span>
<span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree_seq_nwk</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">3</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/msaplot.html">msaplot</a></span><span class="op">(</span><span class="va">p</span>, <span class="va">AA_sequence</span>, offset<span class="op">=</span><span class="fl">3</span>, width<span class="op">=</span><span class="fl">2</span><span class="op">)</span></code></pre></div>
<p>A specific slice of the alignment can also be displayed by specifying the <code>window</code> parameter (Figure <a href="chapter7.html#fig:msaplot">7.5</a>B)..</p>
<div class="sourceCode" id="cb197"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree_seq_nwk</span>, layout<span class="op">=</span><span class="st">'circular'</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab</a></span><span class="op">(</span>offset<span class="op">=</span><span class="fl">4</span>, align<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html">xlim</a></span><span class="op">(</span><span class="cn">NA</span>, <span class="fl">12</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/msaplot.html">msaplot</a></span><span class="op">(</span><span class="va">p</span>, <span class="va">AA_sequence</span>, window<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">120</span>, <span class="fl">200</span><span class="op">)</span><span class="op">)</span></code></pre></div>


<div class="figure">
<span style="display:block;" id="fig:msaplot"></span>
<img src="treedata_files/figure-html/msaplot-1.png" alt="Example of plotting multiple sequence alignments with a tree. Whole MSA sequences were visualized with a tree in rectangular layout (A). Circular layout with a slice of alignment window (B)." width="100%"><p class="caption">
FIGURE 7.5: <strong>Example of plotting multiple sequence alignments with a tree</strong>. Whole MSA sequences were visualized with a tree in rectangular layout (A). Circular layout with a slice of alignment window (B).
</p>
</div>
<p>To better support visualizing multiple sequence alignments with a tree and other associated data, we developed the <a href="http://bioconductor.org/packages/ggmsa"><strong>ggmsa</strong></a> package with the ability to label the sequences and color the sequences with different color schemes <span class="citation">(<a href="references.html#ref-yu_cp_2020" role="doc-biblioref">Yu, 2020</a>)</span>. The <code>ggmsa()</code> output is compatible with <code><a href="https://rdrr.io/pkg/ggtree/man/facet-plot.html">geom_facet()</a></code> and <code><a href="https://rdrr.io/pkg/ggtreeExtra/man/geom_fruit.html">ggtreeExtra::geom_fruit()</a></code> and can be used to visualize a tree, multiple sequence alignments, and different types of associated data to explore their underlying linkages/associations.</p>
</div>
<div id="composite_plot" class="section level2" number="7.5">
<h2>
<span class="header-section-number">7.5</span> Composite Plots<a class="anchor" aria-label="anchor" href="#composite_plot"><i class="fas fa-link"></i></a>
</h2>
<p>In addition to aligning graphs to a tree using <code><a href="https://rdrr.io/pkg/ggtree/man/facet-plot.html">geom_facet()</a></code> or <code><a href="https://rdrr.io/pkg/ggtreeExtra/man/geom_fruit.html">ggtreeExtra::geom_fruit()</a></code> and special cases using the <code><a href="https://rdrr.io/pkg/ggtree/man/gheatmap.html">gheatmap()</a></code> and <code><a href="https://rdrr.io/pkg/ggtree/man/msaplot.html">msaplot()</a></code> functions, users can use <a href="https://CRAN.R-project.org/package=cowplot"><strong>cowplot</strong></a>, <a href="https://CRAN.R-project.org/package=patchwork"><strong>patchwork</strong></a>, <a href="https://CRAN.R-project.org/package=gtable"><strong>gtable</strong></a><a class="footnote-ref" tabindex="0" data-toggle="popover" data-content='&lt;p&gt;&lt;a href="https://github.com/YuLab-SMU/ggtree/issues/313" class="uri"&gt;https://github.com/YuLab-SMU/ggtree/issues/313&lt;/a&gt;&lt;/p&gt;'><sup>12</sup></a> or other packages to create composite plots. However, extra efforts need to be done to make sure all the plots are aligned properly. The <a href="chapter12.html#tiporder"><code>ggtree::get_taxa_name()</code></a> function is quite useful for users to re-order their data based on the tree structure. To remove this obstacle, we created an R package <a href="https://CRAN.R-project.org/package=aplot"><strong>aplot</strong></a> that can re-order the internal data of a <code>ggplot</code> object and create composite plots that align properly with a tree.</p>
<p>In the following example, we have a tree with two associated datasets.</p>
<div class="sourceCode" id="cb198"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/treedata-book/">ggtree</a></span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">2019</span><span class="op">-</span><span class="fl">10</span><span class="op">-</span><span class="fl">31</span><span class="op">)</span>
<span class="va">tr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ape/man/rtree.html">rtree</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span>

<span class="va">d1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>
    <span class="co"># only some labels match</span>
    label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">tr</span><span class="op">$</span><span class="va">tip.label</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/sample.html">sample</a></span><span class="op">(</span><span class="fl">5</span>, <span class="fl">5</span><span class="op">)</span><span class="op">]</span>, <span class="st">"A"</span><span class="op">)</span>,
    value <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html">sample</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">6</span>, <span class="fl">6</span><span class="op">)</span><span class="op">)</span>

<span class="va">d2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>
    label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="va">tr</span><span class="op">$</span><span class="va">tip.label</span>, <span class="fl">5</span><span class="op">)</span>,
    category <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="va">LETTERS</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span>, each<span class="op">=</span><span class="fl">10</span><span class="op">)</span>,
    value <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/Normal.html">rnorm</a></span><span class="op">(</span><span class="fl">50</span>, <span class="fl">0</span>, <span class="fl">3</span><span class="op">)</span><span class="op">)</span> 

<span class="va">g</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tr</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab</a></span><span class="op">(</span>align<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggexpand.html">hexpand</a></span><span class="op">(</span><span class="fl">.01</span><span class="op">)</span>

<span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">d1</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span><span class="va">label</span>, <span class="va">value</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html">geom_col</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>fill<span class="op">=</span><span class="va">label</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_text.html">geom_text</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>label<span class="op">=</span><span class="va">label</span>, y<span class="op">=</span> <span class="va">value</span><span class="op">+</span><span class="fl">.1</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_flip.html">coord_flip</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/theme_tree2.html">theme_tree2</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.position<span class="op">=</span><span class="st">'none'</span><span class="op">)</span>
 
<span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">d2</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">category</span>, y<span class="op">=</span><span class="va">label</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_tile.html">geom_tile</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>fill<span class="op">=</span><span class="va">value</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_viridis.html">scale_fill_viridis_c</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_minimal</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span><span class="op">(</span><span class="cn">NULL</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span><span class="op">(</span><span class="cn">NULL</span><span class="op">)</span></code></pre></div>
<!-- 
We can extract coordinates from `ggtree` object using [**dplyr**](https://CRAN.R-project.org/package=dplyr) verbs implemented in [**tidytree**](https://CRAN.R-project.org/package=tidytree), and merge the y-coordinates to our datasets.

``` r
library(dplyr)
library(tidytree) 

d <- filter(g, isTip) %>% select(c(label, y))

dd1 <- left_join(d1, d, by='label')
dd2 <- left_join(d2, d, by='label') 
```

Now we can visualize our datasets using the y-coordinates that match corresponding labels in the `ggtree` object. 

-->
<!-- 
If we combine these two plots with `ggtree` using either [**cowplot**](https://CRAN.R-project.org/package=cowplot) or [**patchwork**](https://github.com/thomasp85/patchwork), they do not align properly. 

To address this issue, [**ggtree**](http://bioconductor.org/packages/ggtree) provides a `ylim2()` function to reconcile y limits^[the implementation was inspired by <https://thackl.github.io/ggtree-composite-plots>]. 

``` r message=TRUE
p1 <- p1 + ylim2(g)  
p2 <- p2 + ylim2(g)
```

Now we can plot our tree and the data side by side, and perfectly aligned.


## library(patchwork) 
## g + p1 + p2 + plot_annotation(tag_levels="A")

library(cowplot) 
plot_grid(g, p1, p2, ncol=3, align='h', 
    labels=LETTERS[1:3], rel_widths = c(1, .5, .8))


The `ylim2()` is not designed specific for [**ggtree**](http://bioconductor.org/packages/ggtree), it works fine with `ggplot` object. Another function `xlim2()` that works for x limits is also provided. See [session 10.5](#axis_align) for more details.

-->
<p>If we align them using <a href="https://CRAN.R-project.org/package=cowplot"><strong>cowplot</strong></a>, the composite plots are not aligned properly as we anticipated (Figure <a href="chapter7.html#fig:composite">7.6</a>A).</p>
<div class="sourceCode" id="cb199"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">cowplot</span><span class="fu">::</span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html">plot_grid</a></span><span class="op">(</span><span class="va">g</span>, <span class="va">p2</span>, <span class="va">p1</span>, ncol<span class="op">=</span><span class="fl">3</span><span class="op">)</span> </code></pre></div>
<p>Using <a href="https://CRAN.R-project.org/package=aplot"><strong>aplot</strong></a>, it will do all the dirty work for us and all the subplots are aligned properly as demonstrated in Figure <a href="chapter7.html#fig:composite">7.6</a>B.</p>
<div class="sourceCode" id="cb200"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/YuLab-SMU/aplot">aplot</a></span><span class="op">)</span>
<span class="va">p2</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/aplot/man/plot-insertion.html">insert_left</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/aplot/man/plot-insertion.html">insert_right</a></span><span class="op">(</span><span class="va">p1</span>, width<span class="op">=</span><span class="fl">.5</span><span class="op">)</span> </code></pre></div>


<div class="figure">
<span style="display:block;" id="fig:composite"></span>
<img src="treedata_files/figure-html/composite-1.svg" alt="Example of aligning tree with data side-by-side to create composite plot. cowplot` just places the subplots together (A), while aplot does extra work to make sure that tree-associated subplots are properly ordered according to the tree structure (B). Note: The ‘A’ category in the bar plot that is not matched with the tree was removed." width="100%"><p class="caption">
FIGURE 7.6: <strong>Example of aligning tree with data side-by-side to create composite plot</strong>. <a href="https://CRAN.R-project.org/package=cowplot"><strong>cowplot</strong></a>` just places the subplots together (A), while <a href="https://CRAN.R-project.org/package=aplot"><strong>aplot</strong></a> does extra work to make sure that tree-associated subplots are properly ordered according to the tree structure (B). Note: The ‘A’ category in the bar plot that is not matched with the tree was removed.
</p>
</div>
<!--

## The `ggtree` object {#ggtree_object}


to be finished.



## Update tree view with a new tree

In previous example, we have a _`p`_ object that stored the tree viewing of 13 tips and internal nodes highlighted with specific colored big dots. If users want to apply this pattern (we can imaging a more complex one) to a new tree, you don't need to build the tree step by step. `ggtree` provides an operator, _`%<%`_, for applying the visualization pattern to a new tree.

For example, the pattern in the _`p`_ object will be applied to a new tree with 50 tips as shown below:
``` r fig.width=3, fig.height=3, fig.align="center"}
p %<% rtree(50)
```
-->
</div>
<div id="summary7" class="section level2" number="7.6">
<h2>
<span class="header-section-number">7.6</span> Summary<a class="anchor" aria-label="anchor" href="#summary7"><i class="fas fa-link"></i></a>
</h2>
<p>Although there are many software packages that support visualizing phylogenetic trees, plotting a tree with data is often missing or with only limited support. Some of the packages define <code>S4</code> classes to store phylogenetic tree with domain-specific data, such as <a href="https://CRAN.R-project.org/package=OutbreakTools"><strong>OutbreakTools</strong></a> <span class="citation">(<a href="references.html#ref-jombart_outbreaktools_2014" role="doc-biblioref">Jombart et al., 2014</a>)</span> defined <code>obkData</code> for storing tree with epidemiology data and <a href="http://bioconductor.org/packages/phyloseq"><strong>phyloseq</strong></a> <span class="citation">(<a href="references.html#ref-mcmurdie_phyloseq_2013" role="doc-biblioref">McMurdie &amp; Holmes, 2013</a>)</span> defines <code>phyloseq</code> for storing tree with microbiome data. These packages are capable of presenting some of the data stored in the object on the tree. However, not all the associated data are supported. For example, species abundance stored in the <code>phyloseq</code> object is not supported to be visualized using the <a href="http://bioconductor.org/packages/phyloseq"><strong>phyloseq</strong></a> package. These packages did not provide any utilities to integrate external data for tree visualization. None of these packages support visualizing external data and aligning the plot to a tree based on the tree structure.</p>
<p>The <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> package provides two general solutions for integrating data. Method 1, the <code>%&lt;+%</code> operator, can integrate external and internal node data and map the data as a visual characteristic to visualize the tree and other datasets used in <code><a href="https://rdrr.io/pkg/ggtree/man/facet-plot.html">geom_facet()</a></code> or <code><a href="https://rdrr.io/pkg/ggtreeExtra/man/geom_fruit.html">ggtreeExtra::geom_fruit()</a></code>. Method 2, the <code>geom_facet</code> layer or <code><a href="https://rdrr.io/pkg/ggtreeExtra/man/geom_fruit.html">ggtreeExtra::geom_fruit()</a></code>, has no restriction of input data as long as there is a <code>geom</code> function available to plot the data (<em>e.g.</em>, species abundance displayed by <code>geom_density_ridges</code> as demonstrated in Figure <a href="chapter9.html#fig:phyloseq">9.4</a>). Users are free to combine different panels and combine different <code>geom</code> layers in the same panel (Figure <a href="chapter13.html#fig:jv2017">13.1</a>).</p>
<div style="page-break-after: always;"></div>
<p>The <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> package has many unique features that cannot be found in other implementations <span class="citation">(<a href="references.html#ref-yu_two_2018" role="doc-biblioref">Yu et al., 2018</a>)</span>:</p>
<ol style="list-style-type: decimal">
<li>Integrating node/edge data to the tree can be mapped to visual characteristics of the tree or other datasets (Figure <a href="chapter7.html#fig:attacher">7.1</a>).</li>
<li>Capable of parsing expressions (math symbols or text formatting), emoji, and image files (<a href="chapter8.html#chapter8">Chapter 8</a>).</li>
<li>No pre-definition of input data types or how the data should be plotted in <code><a href="https://rdrr.io/pkg/ggtree/man/facet-plot.html">geom_facet()</a></code> (Table <a href="figures-and-tables.html#tab:facet-geom">C.1</a>).</li>
<li>Combining different <code>geom</code> functions to visualize associated data is supported (Figure <a href="chapter13.html#fig:jv2017">13.1</a>).</li>
<li>Visualizing different datasets on the same panel is supported.</li>
<li>Data integrated by <code>%&lt;+%</code> can be used in <code><a href="https://rdrr.io/pkg/ggtree/man/facet-plot.html">geom_facet()</a></code> layer.</li>
<li>Able to add further annotations to specific layers.</li>
<li>Modular design by separating tree visualization, data integration (Method 1), and graph alignment (Method 2).</li>
</ol>
<p>Modular design is a unique feature for <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> to stand out from other packages. The tree can be visualized with data stored in the tree object or external data linked by the <code>%&lt;+%</code> operator, and fully annotated with multiple layers of annotations (Figures <a href="chapter7.html#fig:attacher">7.1</a> and <a href="chapter13.html#fig:jv2017">13.1</a>), before passing it to <code><a href="https://rdrr.io/pkg/ggtree/man/facet-plot.html">geom_facet()</a></code> layer. The <code><a href="https://rdrr.io/pkg/ggtree/man/facet-plot.html">geom_facet()</a></code> layer can be called progressively to add multiple panels or multiple layers on the same panels (Figure <a href="chapter13.html#fig:jv2017">13.1</a>). This creates the possibility of plotting a full annotated tree with complex data panels that contain multiple graphic layers.</p>
<p>The <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> package fits the <code>R</code> ecosystem and extends the abilities to integrate and present data with trees to existing phylogenetic packages. As demonstrated in Figure <a href="chapter9.html#fig:phyloseq">9.4</a>, we can plot species abundance distributions with the <code>phyloseq</code> object. This cannot be easily done without <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a>. With <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a>, we are able to attach additional data to tree objects using the <code>%&lt;+%</code> operator and align graphs to a tree using the <code><a href="https://rdrr.io/pkg/ggtree/man/facet-plot.html">geom_facet()</a></code> layer. Integrating <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> into existing workflows will extend the abilities and broaden the applications to present phylogeny-associated data, especially for comparative studies.</p>

<div style="page-break-after: always;"></div>
</div>
</div>

  <div class="chapter-nav">
<div class="prev"><a href="chapter6.html"><span class="header-section-number">6</span> Visual Exploration of Phylogenetic Trees</a></div>
<div class="next"><a href="chapter8.html"><span class="header-section-number">8</span> Annotating Tree with Silhouette Images and Sub-plots</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#chapter7"><span class="header-section-number">7</span> Plotting tree with data</a></li>
<li><a class="nav-link" href="#attach-operator"><span class="header-section-number">7.1</span> Mapping Data to The tree Structure</a></li>
<li><a class="nav-link" href="#facet_plot"><span class="header-section-number">7.2</span> Aligning Graph to the Tree Based on the Tree Structure</a></li>
<li>
<a class="nav-link" href="#gheatmap"><span class="header-section-number">7.3</span> Visualize a Tree with an Associated Matrix</a><ul class="nav navbar-nav"><li><a class="nav-link" href="#gheatmap-ggnewscale"><span class="header-section-number">7.3.1</span> Visualize a tree with multiple associated matrices</a></li></ul>
</li>
<li><a class="nav-link" href="#msaplot"><span class="header-section-number">7.4</span> Visualize a Tree with Multiple Sequence Alignments</a></li>
<li><a class="nav-link" href="#composite_plot"><span class="header-section-number">7.5</span> Composite Plots</a></li>
<li><a class="nav-link" href="#summary7"><span class="header-section-number">7.6</span> Summary</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Data Integration, Manipulation and Visualization of Phylogenetic Trees with R</strong>" was written by Guangchuang Yu. It was last built on 2022-04-15.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
