<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>C Figures and Tables | Data Integration, Manipulation and Visualization of Phylogenetic Trees with R</title>
<meta name="author" content="Guangchuang Yu">
<meta name="description" content="TABLE C.1: Geometric layers that supported by `geom_facet()’  Package  Geom Layer  Description  ggalt  geom_dumbbell  creates dumbbell charts  ggbio  geom_alignment  shows interval data as...">
<meta name="generator" content="bookdown 0.25 with bs4_book()">
<meta property="og:title" content="C Figures and Tables | Data Integration, Manipulation and Visualization of Phylogenetic Trees with R">
<meta property="og:type" content="book">
<meta property="og:image" content="/book-cover.png">
<meta property="og:description" content="TABLE C.1: Geometric layers that supported by `geom_facet()’  Package  Geom Layer  Description  ggalt  geom_dumbbell  creates dumbbell charts  ggbio  geom_alignment  shows interval data as...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="C Figures and Tables | Data Integration, Manipulation and Visualization of Phylogenetic Trees with R">
<meta name="twitter:description" content="TABLE C.1: Geometric layers that supported by `geom_facet()’  Package  Geom Layer  Description  ggalt  geom_dumbbell  creates dumbbell charts  ggbio  geom_alignment  shows interval data as...">
<meta name="twitter:image" content="/book-cover.png">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap/bootstrap.bundle.min.js"></script><script src="libs/bs3compat/transition.js"></script><script src="libs/bs3compat/tabs.js"></script><script src="libs/bs3compat/bs3compat.js"></script><link href="libs/bs4_book/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book/bs4_book.js"></script><script src="libs/kePrint/kePrint.js"></script><link href="libs/lightable/lightable.css" rel="stylesheet">
<script src="libs/htmlwidgets/htmlwidgets.js"></script><script src="libs/pathseg/pathseg.js"></script><script src="libs/comic/comic.min.js"></script><script src="libs/comicR-binding/comicR.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<style type="text/css">
    /* Used with Pandoc 2.11+ new --citeproc when CSL is used */
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
        }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Data Integration, Manipulation and Visualization of Phylogenetic Trees with R</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">Preface</a></li>
<li><a class="" href="author.html">About the Author</a></li>
<li class="book-part">Part I: Tree data input, output, and manipulation</li>
<li><a class="" href="chapter1.html"><span class="header-section-number">1</span> Importing Tree with Data</a></li>
<li><a class="" href="chapter2.html"><span class="header-section-number">2</span> Manipulating Tree with Data</a></li>
<li><a class="" href="chapter3.html"><span class="header-section-number">3</span> Exporting tree with data</a></li>
<li class="book-part">Part II: Tree data visualization and annotation</li>
<li><a class="" href="chapter4.html"><span class="header-section-number">4</span> Phylogenetic Tree Visualization</a></li>
<li><a class="" href="chapter5.html"><span class="header-section-number">5</span> Phylogenetic Tree Annotation</a></li>
<li><a class="" href="chapter6.html"><span class="header-section-number">6</span> Visual Exploration of Phylogenetic Trees</a></li>
<li><a class="" href="chapter7.html"><span class="header-section-number">7</span> Plotting tree with data</a></li>
<li><a class="" href="chapter8.html"><span class="header-section-number">8</span> Annotating Tree with Silhouette Images and Sub-plots</a></li>
<li><a class="" href="chapter9.html"><span class="header-section-number">9</span> ggtree for Other Tree-like Objects</a></li>
<li class="book-part">Part III: ggtree extensions</li>
<li><a class="" href="chapter10.html"><span class="header-section-number">10</span> ggtreeExtra for Presenting Data on a Circular Layout</a></li>
<li><a class="" href="chapter11.html"><span class="header-section-number">11</span> Other ggtree Extensions</a></li>
<li class="book-part">Part IV: Miscellaneous topics</li>
<li><a class="" href="chapter12.html"><span class="header-section-number">12</span> ggtree Utilities</a></li>
<li><a class="" href="chapter13.html"><span class="header-section-number">13</span> Gallery of Reproducible Examples</a></li>
<li class="book-part">Appendix</li>
<li><a class="" href="faq.html"><span class="header-section-number">A</span> Frequently Asked Questions</a></li>
<li><a class="" href="related-tools.html"><span class="header-section-number">B</span> Related Tools</a></li>
<li><a class="active" href="figures-and-tables.html"><span class="header-section-number">C</span> Figures and Tables</a></li>
<li><a class="" href="publications-of-the-ggtree-package-suite.html"><span class="header-section-number">D</span> Publications of the ggtree Package Suite</a></li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="figures-and-tables" class="section level1" number="16">
<h1>
<span class="header-section-number">C</span> Figures and Tables<a class="anchor" aria-label="anchor" href="#figures-and-tables"><i class="fas fa-link"></i></a>
</h1>
<div class="inline-table"><table class="table table-striped table-hover" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:facet-geom">TABLE C.1: </span>Geometric layers that supported by `geom_facet()’
</caption>
<thead><tr>
<th style="text-align:left;">
Package
</th>
<th style="text-align:left;">
Geom Layer
</th>
<th style="text-align:left;">
Description
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
ggalt
</td>
<td style="text-align:left;">
geom_dumbbell
</td>
<td style="text-align:left;">
creates dumbbell charts
</td>
</tr>
<tr>
<td style="text-align:left;">
ggbio
</td>
<td style="text-align:left;">
geom_alignment
</td>
<td style="text-align:left;">
shows interval data as alignment
</td>
</tr>
<tr>
<td style="text-align:left;">
ggfittext
</td>
<td style="text-align:left;">
geom_fit_text
</td>
<td style="text-align:left;">
shrinks, grows, or wraps text to fit inside a defined rectangular area
</td>
</tr>
<tr>
<td style="text-align:left;">
gggenes
</td>
<td style="text-align:left;">
geom_gene_arrow
</td>
<td style="text-align:left;">
draws genes as arrows
</td>
</tr>
<tr>
<td style="text-align:left;">
ggimage
</td>
<td style="text-align:left;">
geom_image
</td>
<td style="text-align:left;">
visualizes image files
</td>
</tr>
<tr>
<td style="text-align:left;">
ggimage
</td>
<td style="text-align:left;">
geom_phylopic
</td>
<td style="text-align:left;">
queries image files from the PhyloPic database and visualizes them
</td>
</tr>
<tr>
<td style="text-align:left;">
ggplot2
</td>
<td style="text-align:left;">
geom_hline
</td>
<td style="text-align:left;">
adds horizontal lines
</td>
</tr>
<tr>
<td style="text-align:left;">
ggplot2
</td>
<td style="text-align:left;">
geom_jitter
</td>
<td style="text-align:left;">
adds a small amount of random variation to the location of each point
</td>
</tr>
<tr>
<td style="text-align:left;">
ggplot2
</td>
<td style="text-align:left;">
geom_label
</td>
<td style="text-align:left;">
draws a rectangle behind the text
</td>
</tr>
<tr>
<td style="text-align:left;">
ggplot2
</td>
<td style="text-align:left;">
geom_point
</td>
<td style="text-align:left;">
creates scatterplots
</td>
</tr>
<tr>
<td style="text-align:left;">
ggplot2
</td>
<td style="text-align:left;">
geom_raster
</td>
<td style="text-align:left;">
a high-performance special case for all the tiles that are the same size
</td>
</tr>
<tr>
<td style="text-align:left;">
ggplot2
</td>
<td style="text-align:left;">
geom_rect
</td>
<td style="text-align:left;">
draws rectangle by using the locations of the four corners
</td>
</tr>
<tr>
<td style="text-align:left;">
ggplot2
</td>
<td style="text-align:left;">
geom_segment
</td>
<td style="text-align:left;">
draws a straight line between points
</td>
</tr>
<tr>
<td style="text-align:left;">
ggplot2
</td>
<td style="text-align:left;">
geom_spoke
</td>
<td style="text-align:left;">
a polar parameterization of ‘geom_segment()’
</td>
</tr>
<tr>
<td style="text-align:left;">
ggplot2
</td>
<td style="text-align:left;">
geom_text
</td>
<td style="text-align:left;">
adds text to the plot
</td>
</tr>
<tr>
<td style="text-align:left;">
ggplot2
</td>
<td style="text-align:left;">
geom_tile
</td>
<td style="text-align:left;">
draws rectangle by using the center of the tile and its size
</td>
</tr>
<tr>
<td style="text-align:left;">
ggplot2
</td>
<td style="text-align:left;">
geom_vline
</td>
<td style="text-align:left;">
adds vertical lines
</td>
</tr>
<tr>
<td style="text-align:left;">
ggrepel
</td>
<td style="text-align:left;">
geom_text_repel
</td>
<td style="text-align:left;">
adds text to the plot. The text labels repel away from each other and away from the data points
</td>
</tr>
<tr>
<td style="text-align:left;">
ggrepel
</td>
<td style="text-align:left;">
geom_label_repel
</td>
<td style="text-align:left;">
draws a rectangle underneath the text. The text labels repel away from each other and away from the data points
</td>
</tr>
<tr>
<td style="text-align:left;">
ggridges
</td>
<td style="text-align:left;">
geom_density_ridges
</td>
<td style="text-align:left;">
arranges multiple density plots in a staggered fashion
</td>
</tr>
<tr>
<td style="text-align:left;">
ggridges
</td>
<td style="text-align:left;">
geom_density_ridges_gradient
</td>
<td style="text-align:left;">
works just like ‘geom_density_ridges’ except that the ‘fill’ aesthetic can vary along the <em>x</em>-axis
</td>
</tr>
<tr>
<td style="text-align:left;">
ggridges
</td>
<td style="text-align:left;">
geom_ridgeline
</td>
<td style="text-align:left;">
plots the sum of the ‘y’ and ‘height’ aesthetics vs. ‘x’, filling the area between ‘y’ and ‘y + height’ with a color
</td>
</tr>
<tr>
<td style="text-align:left;">
ggridges
</td>
<td style="text-align:left;">
geom_ridgeline_gradient
</td>
<td style="text-align:left;">
works just like ‘geom_ridgeline’ except that the ‘fill’ aesthetic can vary along the <em>x</em>-axis
</td>
</tr>
<tr>
<td style="text-align:left;">
ggstance
</td>
<td style="text-align:left;">
geom_barh
</td>
<td style="text-align:left;">
horizontal version of ‘geom_bar()’
</td>
</tr>
<tr>
<td style="text-align:left;">
ggstance
</td>
<td style="text-align:left;">
geom_boxploth
</td>
<td style="text-align:left;">
horizontal version of ‘geom_boxplot()’
</td>
</tr>
<tr>
<td style="text-align:left;">
ggstance
</td>
<td style="text-align:left;">
geom_crossbarh
</td>
<td style="text-align:left;">
horizontal version of ‘geom_crossbar()’
</td>
</tr>
<tr>
<td style="text-align:left;">
ggstance
</td>
<td style="text-align:left;">
geom_errorbarh
</td>
<td style="text-align:left;">
horizontal version of ‘geom_errorbarh()’
</td>
</tr>
<tr>
<td style="text-align:left;">
ggstance
</td>
<td style="text-align:left;">
geom_histogramh
</td>
<td style="text-align:left;">
horizontal version of ‘geom_histogram()’
</td>
</tr>
<tr>
<td style="text-align:left;">
ggstance
</td>
<td style="text-align:left;">
geom_linerangeh
</td>
<td style="text-align:left;">
horizontal version of ‘geom_linerange()’
</td>
</tr>
<tr>
<td style="text-align:left;">
ggstance
</td>
<td style="text-align:left;">
geom_pointrangeh
</td>
<td style="text-align:left;">
horizontal version of ‘geom_pointrange()’
</td>
</tr>
<tr>
<td style="text-align:left;">
ggstance
</td>
<td style="text-align:left;">
geom_violinh
</td>
<td style="text-align:left;">
horizontal version of ‘geom_violin()’
</td>
</tr>
<tr>
<td style="text-align:left;">
ggtree
</td>
<td style="text-align:left;">
geom_motif
</td>
<td style="text-align:left;">
draws aligned motifs
</td>
</tr>
</tbody>
</table></div>
<div class="inline-table"><table class="table table-striped table-hover" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:tree-objects">TABLE C.2: </span>Tree-like objects supported by ggtree
</caption>
<thead><tr>
<th style="text-align:left;">
Package
</th>
<th style="text-align:left;">
Object
</th>
<th style="text-align:left;">
Description
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
ape
</td>
<td style="text-align:left;">
phylo
</td>
<td style="text-align:left;">
basic phylogenetic tree structure
</td>
</tr>
<tr>
<td style="text-align:left;">
ape
</td>
<td style="text-align:left;">
multiPhylo
</td>
<td style="text-align:left;">
list of phylo objects
</td>
</tr>
<tr>
<td style="text-align:left;">
ade4
</td>
<td style="text-align:left;">
phylog
</td>
<td style="text-align:left;">
tree structure for ecological data
</td>
</tr>
<tr>
<td style="text-align:left;">
phylobase
</td>
<td style="text-align:left;">
phylo4
</td>
<td style="text-align:left;">
S4 version of phylo object
</td>
</tr>
<tr>
<td style="text-align:left;">
phylobase
</td>
<td style="text-align:left;">
phylo4d
</td>
<td style="text-align:left;">
extend phylo4 with trait data
</td>
</tr>
<tr>
<td style="text-align:left;">
phyloseq
</td>
<td style="text-align:left;">
phyloseq
</td>
<td style="text-align:left;">
phylogenetic tree with microbiome data
</td>
</tr>
<tr>
<td style="text-align:left;">
tidytree
</td>
<td style="text-align:left;">
tbl_tree
</td>
<td style="text-align:left;">
phylogenetic tree as a tidy data frame
</td>
</tr>
<tr>
<td style="text-align:left;">
treeio
</td>
<td style="text-align:left;">
treedata
</td>
<td style="text-align:left;">
phylogenetic tree with heterogeneous associated data
</td>
</tr>
<tr>
<td style="text-align:left;">
treeio
</td>
<td style="text-align:left;">
jplace
</td>
<td style="text-align:left;">
treedata object with placement information
</td>
</tr>
<tr>
<td style="text-align:left;">
stats
</td>
<td style="text-align:left;">
hclust
</td>
<td style="text-align:left;">
hierarchical cluster result
</td>
</tr>
<tr>
<td style="text-align:left;">
stats
</td>
<td style="text-align:left;">
dendrogram
</td>
<td style="text-align:left;">
hierarchical clustering or classification/regression tree
</td>
</tr>
<tr>
<td style="text-align:left;">
cluster
</td>
<td style="text-align:left;">
agnes
</td>
<td style="text-align:left;">
agglomerative hierarchical clustering
</td>
</tr>
<tr>
<td style="text-align:left;">
cluster
</td>
<td style="text-align:left;">
diana
</td>
<td style="text-align:left;">
divisive hierarchical clustering
</td>
</tr>
<tr>
<td style="text-align:left;">
cluster
</td>
<td style="text-align:left;">
twins
</td>
<td style="text-align:left;">
agglomerative or divisive (polythetic) hierarchical clustering
</td>
</tr>
<tr>
<td style="text-align:left;">
pvclust
</td>
<td style="text-align:left;">
pvclust
</td>
<td style="text-align:left;">
hierarchical clustering with p-values calculated by multiscale bootstrap resampling
</td>
</tr>
<tr>
<td style="text-align:left;">
igraph
</td>
<td style="text-align:left;">
igraph
</td>
<td style="text-align:left;">
network (currently only tree graph supported)
</td>
</tr>
</tbody>
</table></div>
<!--
#``` r treeio, echo=F, message=FALSE}
library(kableExtra)
ff <- matrix(c(
  "read.beast",  "parsing output of BEAST",
  "read.codeml", "parsing output of CodeML (rst and mlc files)",
  "read.codeml_mlc", "parsing mlc file (output of CodeML)",
  "read.hyphy", "parsing output of HYPHY",
  "read.jplace", "parsing jplace file including output of EPA and pplacer",
  "read.nhx", "parsing NHX file including output of PHYLDOG and RevBayes",
  "read.paml_rst", "parsing rst file (output of BaseML or CodeML)",
  "read.phylip", "parsing phylip file",
  "read.phyloT", "parsing output of phyloT (http://phylot.biobyte.de/)",
  "read.r8s", "parsing output of r8s",
  "read.raxml", "parsing output of RAxML",
  "Nnode", "get number of internal nodes",
  "Nnode2", "get number of all nodes (including internal and external nodes)",
  "Ntip", "get number of all tips (external nodes)",
  "as.phylo", "convert tree object to phylo object",
  "as.treedata", "convert tree object to treedata object",
  "drop.tip", "remove tips in a tree",
  "get.fields", "get available tree attributes stored in tree object",
  "get.placements", "get placement result of jplace object",
  "get.subs", "get substitution by comparing parent node and child node",
  "get.tipseq", "get tip sequences",
  "get.treetext", "get newick text of a tree object",
  "groupClade", "add clade grouping information to tree object",
  "groupOTU", "grouping OTUs by tracing back to most recent commond ancestor",
  "merge_tree", "merging tree objects into one",
  "write.jpace", "write tree with associated annotation data to a jplace file"
  ), ncol=2, byrow=TRUE)
ff <- as.data.frame(ff)
colnames(ff) <- c("Function", "Description")
knitr::kable(ff, caption = "Functions defined in treeio.", booktabs = T) #%>%
  #kable_styling(latex_options = c("striped", "hold_position"), full_width = T)
```


--><div style="page-break-after: always;"></div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="related-tools.html"><span class="header-section-number">B</span> Related Tools</a></div>
<div class="next"><a href="publications-of-the-ggtree-package-suite.html"><span class="header-section-number">D</span> Publications of the ggtree Package Suite</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav"><li><a class="nav-link" href="#figures-and-tables"><span class="header-section-number">C</span> Figures and Tables</a></li></ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Data Integration, Manipulation and Visualization of Phylogenetic Trees with R</strong>" was written by Guangchuang Yu. It was last built on 2022-04-20.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
