<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Preface | Data Integration, Manipulation and Visualization of Phylogenetic Trees with R</title>
<meta name="author" content="Guangchuang Yu">
<meta name="description" content="Master ggtree package suite to handle tree with data.">
<meta name="generator" content="bookdown 0.25 with bs4_book()">
<meta property="og:title" content="Preface | Data Integration, Manipulation and Visualization of Phylogenetic Trees with R">
<meta property="og:type" content="book">
<meta property="og:image" content="/book-cover.png">
<meta property="og:description" content="Master ggtree package suite to handle tree with data.">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Preface | Data Integration, Manipulation and Visualization of Phylogenetic Trees with R">
<meta name="twitter:description" content="Master ggtree package suite to handle tree with data.">
<meta name="twitter:image" content="/book-cover.png">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap/bootstrap.bundle.min.js"></script><script src="libs/bs3compat/transition.js"></script><script src="libs/bs3compat/tabs.js"></script><script src="libs/bs3compat/bs3compat.js"></script><link href="libs/bs4_book/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book/bs4_book.js"></script><script src="libs/kePrint/kePrint.js"></script><link href="libs/lightable/lightable.css" rel="stylesheet">
<script src="libs/htmlwidgets/htmlwidgets.js"></script><script src="libs/pathseg/pathseg.js"></script><script src="libs/comic/comic.min.js"></script><script src="libs/comicR-binding/comicR.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<style type="text/css">
    /* Used with Pandoc 2.11+ new --citeproc when CSL is used */
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
        }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Data Integration, Manipulation and Visualization of Phylogenetic Trees with R</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="active" href="index.html">Preface</a></li>
<li><a class="" href="author.html">About the Author</a></li>
<li class="book-part">Part I: Tree data input, output, and manipulation</li>
<li><a class="" href="chapter1.html"><span class="header-section-number">1</span> Importing Tree with Data</a></li>
<li><a class="" href="chapter2.html"><span class="header-section-number">2</span> Manipulating Tree with Data</a></li>
<li><a class="" href="chapter3.html"><span class="header-section-number">3</span> Exporting tree with data</a></li>
<li class="book-part">Part II: Tree data visualization and annotation</li>
<li><a class="" href="chapter4.html"><span class="header-section-number">4</span> Phylogenetic Tree Visualization</a></li>
<li><a class="" href="chapter5.html"><span class="header-section-number">5</span> Phylogenetic Tree Annotation</a></li>
<li><a class="" href="chapter6.html"><span class="header-section-number">6</span> Visual Exploration of Phylogenetic Trees</a></li>
<li><a class="" href="chapter7.html"><span class="header-section-number">7</span> Plotting tree with data</a></li>
<li><a class="" href="chapter8.html"><span class="header-section-number">8</span> Annotating Tree with Silhouette Images and Sub-plots</a></li>
<li><a class="" href="chapter9.html"><span class="header-section-number">9</span> ggtree for Other Tree-like Objects</a></li>
<li class="book-part">Part III: ggtree extensions</li>
<li><a class="" href="chapter10.html"><span class="header-section-number">10</span> ggtreeExtra for Presenting Data on a Circular Layout</a></li>
<li><a class="" href="chapter11.html"><span class="header-section-number">11</span> Other ggtree Extensions</a></li>
<li class="book-part">Part IV: Miscellaneous topics</li>
<li><a class="" href="chapter12.html"><span class="header-section-number">12</span> ggtree Utilities</a></li>
<li><a class="" href="chapter13.html"><span class="header-section-number">13</span> Gallery of Reproducible Examples</a></li>
<li class="book-part">Appendix</li>
<li><a class="" href="faq.html"><span class="header-section-number">A</span> Frequently Asked Questions</a></li>
<li><a class="" href="related-tools.html"><span class="header-section-number">B</span> Related Tools</a></li>
<li><a class="" href="figures-and-tables.html"><span class="header-section-number">C</span> Figures and Tables</a></li>
<li><a class="" href="publications-of-the-ggtree-package-suite.html"><span class="header-section-number">D</span> Publications of the ggtree Package Suite</a></li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><!--bookdown:title:end--><!--bookdown:title:start--><div id="preface" class="section level1 unnumbered">
<h1>Preface<a class="anchor" aria-label="anchor" href="#preface"><i class="fas fa-link"></i></a>
</h1>
<p>I am so excited to have this book published. The book is meant as a guide for data integration, manipulation and visualization of phylogenetic trees using a suite of R packages, <a href="https://CRAN.R-project.org/package=tidytree"><strong>tidytree</strong></a>, <a href="http://bioconductor.org/packages/treeio"><strong>treeio</strong></a>, <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> and <a href="http://bioconductor.org/packages/ggtreeExtra"><strong>ggtreeExtra</strong></a>. Hence, if you are starting to read this book, we assume you have a working knowledge of how to use R and <a href="https://CRAN.R-project.org/package=ggplot2"><strong>ggplot2</strong></a>.</p>
<p>The development of the <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> package started during my PhD study at the University of Hong Kong. I joined the State Key Laboratory of Emerging Infectious Diseases (SKLEID) under the supervision of Yi Guan and Tommy Lam. I was asked to provide assistance to modify the newick tree string to incorporate some additional information, such as amino acid substitutions, in the internal node labels of the phylogeny for visualization. I wrote an R script to do it and soon realized that most phylogenetic tree visualization software can only display one type of data through node labels. Basically, we cannot display two data variables at the same time for comparative analysis. In order to produce tree graphs displaying different types of branch/node associated information, such as bootstrap values and substitutions, people mostly relied on post-processing image software. This situation motivates me to develop <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a>. First of all, I think a good user interface must fully support the <a href="https://CRAN.R-project.org/package=ggplot2"><strong>ggplot2</strong></a> syntax, which allows us to draw graphs by superimposing layers. In this way, simple graphs are simple, and complex graphs are just a combination of simple layers, which are easy to generate.</p>
<p>After several years of development, <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> has evolved into a package suite, including <a href="https://CRAN.R-project.org/package=tidytree"><strong>tidytree</strong></a> for manipulating tree with data using the tidy interface; <a href="http://bioconductor.org/packages/treeio"><strong>treeio</strong></a> for importing and exporting tree with richly annotated data; <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> for tree visualization and annotation and <a href="http://bioconductor.org/packages/ggtreeExtra"><strong>ggtreeExtra</strong></a> for presenting data with a phylogeny side-by-side for a rectangular layout or in outer rings for a circular layout. The <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> is a general tool that supports different types of tree and tree-like structures and can be applied to different disciplines to help researchers presenting and interpreting data in the evolutionary or hierarchical context.</p>
<div id="structure-of-the-book" class="section level2 unnumbered">
<h2>Structure of the book<a class="anchor" aria-label="anchor" href="#structure-of-the-book"><i class="fas fa-link"></i></a>
</h2>
<ul>
<li>Part I (Tree data input, output and manipulation) describes <a href="http://bioconductor.org/packages/treeio"><strong>treeio</strong></a> package for tree data input and output, and <a href="https://CRAN.R-project.org/package=tidytree"><strong>tidytree</strong></a> package for tree data manipulation.</li>
<li>Part II (Tree data visualization and annotation) introduces tree visualization and annotation using the grammar of graphic syntax implemented in the <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> package. It emphasizes presenting tree-associated data on the tree.</li>
<li>Part III (ggtree extensions) introduces <a href="http://bioconductor.org/packages/ggtreeExtra"><strong>ggtreeExtra</strong></a> for presenting data on circular layout trees and other extensions including <a href="http://bioconductor.org/packages/MicrobiotaProcess"><strong>MicrobiotaProcess</strong></a> and <a href="http://bioconductor.org/packages/tanggle"><strong>tanggle</strong></a> <em>etc</em>.</li>
<li>Part IV (Miscellaneous topics) describes utilities provided by the <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> package suite and presents a set of reproducible examples.</li>
</ul>
</div>
<div id="software-information-and-conventions" class="section level2 unnumbered">
<h2>Software information and conventions<a class="anchor" aria-label="anchor" href="#software-information-and-conventions"><i class="fas fa-link"></i></a>
</h2>
<p>The R and core packages information when compiling this book is as
follows:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">R.version.string</span></code></pre></div>
<pre><code>## [1] "R version 4.1.2 (2021-11-01)"</code></pre>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">treedataverse</span><span class="op">)</span></code></pre></div>
<pre><code>##  Attaching packages  treedataverse 0.0.1 

##  ape         5.5            treeio      1.18.1   
##  dplyr       1.0.7          ggtree      3.2.1
##  ggplot2     3.3.5          ggtreeExtra 1.4.1    
##  tidytree    0.3.6</code></pre>
<p>The <a href="https://github.com/YuLab-SMU/treedataverse"><strong>treedataverse</strong></a> is a
meta package to make it easy to install and load core packages for
processing and visualizing tree with data using the packages described
in this book. The installation guide for
<a href="https://github.com/YuLab-SMU/treedataverse"><strong>treedataverse</strong></a> can be
found in FAQ.</p>
<p>The datasets used in this book have three sources:</p>
<ol style="list-style-type: decimal">
<li>Simulation data</li>
<li>Datasets in the R packages</li>
<li>Data downloaded from the Internet</li>
</ol>
<p>In order to make the data downloaded from the Internet more accessible,
we packed the data in an R package,
<a href="https://CRAN.R-project.org/package=TDbook"><strong>TDbook</strong></a>, with detailed
documentation of the original source, including URL, authors, and
citation if the information is available. The
<a href="https://CRAN.R-project.org/package=TDbook"><strong>TDbook</strong></a> is available on
CRAN and can be installed using <code>install.packages("TDbook")</code>.</p>
<p>Package names in this book are formatted as bold text (<em>e.g.</em>,
<a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a>), and function
names are followed by parentheses (<em>e.g.</em>, <code><a href="https://docs.ropensci.org/treeio/reference/beast-parser.html">treeio::read.beast()</a></code>). The
double-colon operator (<code>::</code>) means accessing an object from a package.</p>
</div>
<div id="acknowledgments" class="section level2 unnumbered">
<h2>Acknowledgments<a class="anchor" aria-label="anchor" href="#acknowledgments"><i class="fas fa-link"></i></a>
</h2>
<p>Many people have contributed to this book with spelling and grammar corrections. I’d particularly like to thank Shuangbin Xu, Lin Li and Xiao Luo for their detailed technical reviews of the book, and Tiao You for designing the front cover of the book.</p>
<p>Many others have contributed during the development of the <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> package suite. I would like to thank Hadley Wickham, for creating the <a href="https://CRAN.R-project.org/package=ggplot2"><strong>ggplot2</strong></a> package that <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> relies on; Tommy Tsan-Yuk Lam and Yi Guan for being great advisors and supporting the development of <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> during my PhD; Richard Ree for inviting me to catalysis meeting on phylogenetic tree visualization; William Pearson for inviting me to publish a protocol paper of <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> in the <em>Current Procotols in Bioinformatics</em> journal; Shuangbin Xu, Yonghe Xia, Justin Silverman, Bradley Jones, Watal M. Iwasaki, Ruizhu Huang, Casey Dunn, Tyler Bradley, Konstantinos Geles, Zebulun Arendsee and many others who have contributed source code or given me feedback; and last, but not least, the members of the <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> mailing list<a class="footnote-ref" tabindex="0" data-toggle="popover" data-content='&lt;p&gt;&lt;a href="https://groups.google.com/forum/#!forum/bioc-ggtree" class="uri"&gt;https://groups.google.com/forum/#!forum/bioc-ggtree&lt;/a&gt;&lt;/p&gt;'><sup>1</sup></a>, for providing many challenging problems that have helped improve the <a href="http://bioconductor.org/packages/ggtree"><strong>ggtree</strong></a> package suite.</p>

</div>
</div>

  <div class="chapter-nav">
<div class="empty"></div>
<div class="next"><a href="author.html">About the Author</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#preface">Preface</a></li>
<li><a class="nav-link" href="#structure-of-the-book">Structure of the book</a></li>
<li><a class="nav-link" href="#software-information-and-conventions">Software information and conventions</a></li>
<li><a class="nav-link" href="#acknowledgments">Acknowledgments</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Data Integration, Manipulation and Visualization of Phylogenetic Trees with R</strong>" was written by Guangchuang Yu. It was last built on 2022-04-15.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
