<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>References | Data Integration, Manipulation and Visualization of Phylogenetic Trees</title>
<meta name="author" content="Guangchuang Yu">
<meta name="description" content="Amer, A., Galvin, S., Healy, C. M., &amp; Moran, G. P. (2017). The microbiome of potentially malignant oral leukoplakia exhibits enrichment for fusobacterium, leptotrichia, campylobacter, and rothia...">
<meta name="generator" content="bookdown 0.26 with bs4_book()">
<meta property="og:title" content="References | Data Integration, Manipulation and Visualization of Phylogenetic Trees">
<meta property="og:type" content="book">
<meta property="og:image" content="/book-cover.png">
<meta property="og:description" content="Amer, A., Galvin, S., Healy, C. M., &amp; Moran, G. P. (2017). The microbiome of potentially malignant oral leukoplakia exhibits enrichment for fusobacterium, leptotrichia, campylobacter, and rothia...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="References | Data Integration, Manipulation and Visualization of Phylogenetic Trees">
<meta name="twitter:description" content="Amer, A., Galvin, S., Healy, C. M., &amp; Moran, G. P. (2017). The microbiome of potentially malignant oral leukoplakia exhibits enrichment for fusobacterium, leptotrichia, campylobacter, and rothia...">
<meta name="twitter:image" content="/book-cover.png">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap/bootstrap.bundle.min.js"></script><script src="libs/bs3compat/transition.js"></script><script src="libs/bs3compat/tabs.js"></script><script src="libs/bs3compat/bs3compat.js"></script><link href="libs/bs4_book/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book/bs4_book.js"></script><script src="libs/kePrint/kePrint.js"></script><link href="libs/lightable/lightable.css" rel="stylesheet">
<script src="libs/htmlwidgets/htmlwidgets.js"></script><script src="libs/pathseg/pathseg.js"></script><script src="libs/comic/comic.min.js"></script><script src="libs/comicR-binding/comicR.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<style type="text/css">
    /* Used with Pandoc 2.11+ new --citeproc when CSL is used */
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
        }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Data Integration, Manipulation and Visualization of Phylogenetic Trees</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">Preface</a></li>
<li><a class="" href="author.html">About the Author</a></li>
<li class="book-part">Part I: Tree data input, output, and manipulation</li>
<li><a class="" href="chapter1.html"><span class="header-section-number">1</span> Importing Tree with Data</a></li>
<li><a class="" href="chapter2.html"><span class="header-section-number">2</span> Manipulating Tree with Data</a></li>
<li><a class="" href="chapter3.html"><span class="header-section-number">3</span> Exporting tree with data</a></li>
<li class="book-part">Part II: Tree data visualization and annotation</li>
<li><a class="" href="chapter4.html"><span class="header-section-number">4</span> Phylogenetic Tree Visualization</a></li>
<li><a class="" href="chapter5.html"><span class="header-section-number">5</span> Phylogenetic Tree Annotation</a></li>
<li><a class="" href="chapter6.html"><span class="header-section-number">6</span> Visual Exploration of Phylogenetic Trees</a></li>
<li><a class="" href="chapter7.html"><span class="header-section-number">7</span> Plotting tree with data</a></li>
<li><a class="" href="chapter8.html"><span class="header-section-number">8</span> Annotating Tree with Silhouette Images and Sub-plots</a></li>
<li><a class="" href="chapter9.html"><span class="header-section-number">9</span> ggtree for Other Tree-like Objects</a></li>
<li class="book-part">Part III: ggtree extensions</li>
<li><a class="" href="chapter10.html"><span class="header-section-number">10</span> ggtreeExtra for Presenting Data on a Circular Layout</a></li>
<li><a class="" href="chapter11.html"><span class="header-section-number">11</span> Other ggtree Extensions</a></li>
<li class="book-part">Part IV: Miscellaneous topics</li>
<li><a class="" href="chapter12.html"><span class="header-section-number">12</span> ggtree Utilities</a></li>
<li><a class="" href="chapter13.html"><span class="header-section-number">13</span> Gallery of Reproducible Examples</a></li>
<li class="book-part">Appendix</li>
<li><a class="" href="faq.html"><span class="header-section-number">A</span> Frequently Asked Questions</a></li>
<li><a class="" href="related-tools.html"><span class="header-section-number">B</span> Related Tools</a></li>
<li><a class="" href="figures-and-tables.html"><span class="header-section-number">C</span> Figures and Tables</a></li>
<li><a class="" href="publications-of-the-ggtree-package-suite.html"><span class="header-section-number">D</span> Publications of the ggtree Package Suite</a></li>
<li><a class="active" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="references" class="section level1 unnumbered">
<h1>References<a class="anchor" aria-label="anchor" href="#references"><i class="fas fa-link"></i></a>
</h1>

<div id="refs" class="references csl-bib-body hanging-indent" line-spacing="2">
<div id="ref-amer2017microbiome" class="csl-entry">
Amer, A., Galvin, S., Healy, C. M., &amp; Moran, G. P. (2017). The microbiome of potentially malignant oral leukoplakia exhibits enrichment for fusobacterium, leptotrichia, campylobacter, and rothia species. <em>Frontiers in Microbiology</em>, <em>8</em>, 2391. <a href="https://doi.org/10.3389/fmicb.2017.02391">https://doi.org/10.3389/fmicb.2017.02391</a>
</div>
<div id="ref-arenas_trends_2015" class="csl-entry">
Arenas, M. (2015). Trends in substitution models of molecular evolution. <em>Frontiers in Genetics</em>, <em>6</em>. <a href="https://doi.org/10.3389/fgene.2015.00319">https://doi.org/10.3389/fgene.2015.00319</a>
</div>
<div id="ref-GraPhlAn" class="csl-entry">
Asnicar, F., Weingart, G., Tickle, T. L., Huttenhower, C., &amp; Segata, N. (2015). Compact graphical representation of phylogenetic data and metadata with GraPhlAn. <em>PeerJ</em>, <em>3</em>, e1029. <a href="https://doi.org/10.7717/peerj.1029">https://doi.org/10.7717/peerj.1029</a>
</div>
<div id="ref-barton_broad_2016" class="csl-entry">
Barton, K., Hiener, B., Winckelmann, A., Rasmussen, T. A., Shao, W., Byth, K., Lanfear, R., Solomon, A., McMahon, J., Harrington, S., Buzon, M., Lichterfeld, M., Denton, P. W., Olesen, R., Østergaard, L., Tolstrup, M., Lewin, S. R., Søgaard, O. S., &amp; Palmer, S. (2016). Broad activation of latent <span>HIV</span>-1 in vivo. <em>Nature Communications</em>, <em>7</em>, 12731. <a href="https://doi.org/10.1038/ncomms12731">https://doi.org/10.1038/ncomms12731</a>
</div>
<div id="ref-berger_performance_2011" class="csl-entry">
Berger, S. A., Krompass, D., &amp; Stamatakis, A. (2011). Performance, <span>Accuracy</span>, and <span>Web</span> <span>Server</span> for <span>Evolutionary</span> <span>Placement</span> of <span>Short</span> <span>Sequence</span> <span>Reads</span> under <span>Maximum</span> <span>Likelihood</span>. <em>Systematic Biology</em>, 291–302. <a href="https://doi.org/10.1093/sysbio/syr010">https://doi.org/10.1093/sysbio/syr010</a>
</div>
<div id="ref-bosi_comparative_2016" class="csl-entry">
Bosi, E., Monk, J. M., Aziz, R. K., Fondi, M., Nizet, V., &amp; Palsson, B. Ø. (2016). Comparative genome-scale modelling of <span>Staphylococcus</span> aureus strains identifies strain-specific metabolic capabilities linked to pathogenicity. <em>Proceedings of the National Academy of Sciences of the United States of America</em>, <em>113</em>(26), E3801–E3809. <a href="https://doi.org/10.1073/pnas.1523199113">https://doi.org/10.1073/pnas.1523199113</a>
</div>
<div id="ref-bouckaert_beast_2014" class="csl-entry">
Bouckaert, R., Heled, J., Kühnert, D., Vaughan, T., Wu, C.-H., Xie, D., Suchard, M. A., Rambaut, A., &amp; Drummond, A. J. (2014). <span>BEAST</span> 2: <span>A</span> <span>Software</span> <span>Platform</span> for <span>Bayesian</span> <span>Evolutionary</span> <span>Analysis</span>. <em>PLoS Comput Biol</em>, <em>10</em>(4), e1003537. <a href="https://doi.org/10.1371/journal.pcbi.1003537">https://doi.org/10.1371/journal.pcbi.1003537</a>
</div>
<div id="ref-boussau_genome-scale_2013" class="csl-entry">
Boussau, B., Szöllősi, G. J., Duret, L., Gouy, M., Tannier, E., &amp; Daubin, V. (2013). Genome-scale coestimation of species and gene trees. <em>Genome Research</em>, <em>23</em>(2), 323–330. <a href="https://doi.org/10.1101/gr.141978.112">https://doi.org/10.1101/gr.141978.112</a>
</div>
<div id="ref-callahan_dada2_2016" class="csl-entry">
Callahan, B. J., McMurdie, P. J., Rosen, M. J., Han, A. W., Johnson, A. J. A., &amp; Holmes, S. P. (2016). <span>DADA</span>2: <span>High</span>-resolution sample inference from <span>Illumina</span> amplicon data. <em>Nature Methods</em>, <em>13</em>(7), 581–583. <a href="https://doi.org/10.1038/nmeth.3869">https://doi.org/10.1038/nmeth.3869</a>
</div>
<div id="ref-chen_ancient_2017" class="csl-entry">
Chen, Z., Ho, W. C. S., Boon, S. S., Law, P. T. Y., Chan, M. C. W., DeSalle, R., Burk, R. D., &amp; Chan, P. K. S. (2017). Ancient evolution and dispersion of human papillomavirus 58 variants. <em>Journal of Virology</em>, <em>91</em>(21), e01285–17. <a href="https://doi.org/10.1128/JVI.01285-17">https://doi.org/10.1128/JVI.01285-17</a>
</div>
<div id="ref-chevenet_treedyn:_2006" class="csl-entry">
Chevenet, F., Brun, C., Bañuls, A.-L., Jacq, B., &amp; Christen, R. (2006). <span>TreeDyn</span>: Towards dynamic graphics and annotations for analyses of trees. <em>BMC Bioinformatics</em>, <em>7</em>, 439. <a href="https://doi.org/10.1186/1471-2105-7-439">https://doi.org/10.1186/1471-2105-7-439</a>
</div>
<div id="ref-Chowe:_2020" class="csl-entry">
Chow, N. A., Muñoz, J. F., Gade, L., Berkow, E. L., Li, X., Welsh, R. M., Forsberg, K., Lockhart, S. R., Adam, R., Alanio, A., Alastruey-Izquierdo, A., Althawadi, S., Araúz, A. B., Ben-Ami, R., Bharat, A., Calvo, B., Desnos-Ollivier, M., Escandón, P., Gardam, D., … Cuomo, C. A. (2020). Tracing the evolutionary history and global expansion of candida auris using population genomic analyses. <em>mBio</em>, <em>11</em>(2). <a href="https://doi.org/10.1128/mBio.03364-19">https://doi.org/10.1128/mBio.03364-19</a>
</div>
<div id="ref-reroot_review" class="csl-entry">
Czech, L., Huerta-Cepas, J., &amp; Stamatakis, A. (2017). <span class="nocase">A Critical Review on the Use of Support Values in Tree Viewers and Bioinformatics Toolkits</span>. <em>Molecular Biology and Evolution</em>, <em>34</em>(6), 1535–1542. <a href="https://doi.org/10.1093/molbev/msx055">https://doi.org/10.1093/molbev/msx055</a>
</div>
<div id="ref-escudero_grand_2020" class="csl-entry">
Escudero, M., &amp; Wendel, J. F. (2020). The grand sweep of chromosomal evolution in angiosperms. <em>New Phytologist</em>, <em>228</em>(3), 805–808. <a href="https://doi.org/10.1111/nph.16802">https://doi.org/10.1111/nph.16802</a>
</div>
<div id="ref-felsenstein_cases_1978" class="csl-entry">
Felsenstein, J. (1978). Cases in which <span>Parsimony</span> or <span>Compatibility</span> <span>Methods</span> will be <span>Positively</span> <span>Misleading</span>. <em>Systematic Biology</em>, <em>27</em>(4), 401–410. <a href="https://doi.org/10.1093/sysbio/27.4.401">https://doi.org/10.1093/sysbio/27.4.401</a>
</div>
<div id="ref-felsenstein_evolutionary_1981" class="csl-entry">
Felsenstein, J. (1981). <a href="https://www.ncbi.nlm.nih.gov/pubmed/7288891">Evolutionary trees from <span>DNA</span> sequences: A maximum likelihood approach</a>. <em>Journal of Molecular Evolution</em>, <em>17</em>(6), 368–376.
</div>
<div id="ref-felsenstein_phylip_1989" class="csl-entry">
Felsenstein, J. (1989). <span>PHYLIP</span> - <span>Phylogeny</span> <span>Inference</span> <span>Package</span> (<span>Version</span> 3.2). <em>Cladistics</em>, <em>5</em>, 164–166.
</div>
<div id="ref-fitch_toward_1971" class="csl-entry">
Fitch, W. M. (1971). Toward <span>Defining</span> the <span>Course</span> of <span>Evolution</span>: <span>Minimum</span> <span>Change</span> for a <span>Specific</span> <span>Tree</span> <span>Topology</span>. <em>Systematic Zoology</em>, <em>20</em>(4), 406–416. <a href="https://doi.org/10.2307/2412116">https://doi.org/10.2307/2412116</a>
</div>
<div id="ref-gentleman_bioconductor_2004" class="csl-entry">
Gentleman, R. C., Carey, V. J., Bates, D. M., Bolstad, B., Dettling, M., Dudoit, S., Ellis, B., Gautier, L., Ge, Y., Gentry, J., Hornik, K., Hothorn, T., Huber, W., Iacus, S., Irizarry, R., Leisch, F., Li, C., Maechler, M., Rossini, A. J., … Zhang, J. (2004). Bioconductor: Open software development for computational biology and bioinformatics. <em>Genome Biology</em>, <em>5</em>(10), R80. <a href="https://doi.org/10.1186/gb-2004-5-10-r80">https://doi.org/10.1186/gb-2004-5-10-r80</a>
</div>
<div id="ref-goldman_codon-based_1994" class="csl-entry">
Goldman, N., &amp; Yang, Z. (1994). <a href="https://www.ncbi.nlm.nih.gov/pubmed/7968486">A codon-based model of nucleotide substitution for protein-coding <span>DNA</span> sequences</a>. <em>Molecular Biology and Evolution</em>, <em>11</em>(5), 725–736.
</div>
<div id="ref-grubaugh_genomic_2017" class="csl-entry">
Grubaugh, N. D., Ladner, J. T., Kraemer, M. U. G., Dudas, G., Tan, A. L., Gangavarapu, K., Wiley, M. R., White, S., Thézé, J., Magnani, D. M., Prieto, K., Reyes, D., Bingham, A. M., Paul, L. M., Robles-Sikisaka, R., Oliveira, G., Pronty, D., Barcellona, C. M., Metsky, H. C., … Andersen, K. G. (2017). Genomic epidemiology reveals multiple introductions of zika virus into the united states. <em>Nature</em>, <em>546</em>(7658), 401–405. <a href="https://doi.org/10.1038/nature22400">https://doi.org/10.1038/nature22400</a>
</div>
<div id="ref-gupta_using_2015" class="csl-entry">
Gupta, A., &amp; Sharma, V. K. (2015). Using the taxon-specific genes for the taxonomic classification of bacterial genomes. <em>BMC Genomics</em>, <em>16</em>(1). <a href="https://doi.org/10.1186/s12864-015-1542-0">https://doi.org/10.1186/s12864-015-1542-0</a>
</div>
<div id="ref-he_emergence_2013" class="csl-entry">
He, Y.-Q., Chen, L., Xu, W.-B., Yang, H., Wang, H.-Z., Zong, W.-P., Xian, H.-X., Chen, H.-L., Yao, X.-J., Hu, Z.-L., Luo, M., Zhang, H.-L., Ma, H.-W., Cheng, J.-Q., Feng, Q.-J., &amp; Zhao, D.-J. (2013). Emergence, <span>Circulation</span>, and <span>Spatiotemporal</span> <span>Phylogenetic</span> <span>Analysis</span> of <span>Coxsackievirus</span> <span>A</span>6- and <span>Coxsackievirus</span> <span>A</span>10-<span>Associated</span> <span>Hand</span>, <span>Foot</span>, and <span>Mouth</span> <span>Disease</span> <span>Infections</span> from 2008 to 2012 in <span>Shenzhen</span>, <span>China</span>. <em>Journal of Clinical Microbiology</em>, <em>51</em>(11), 3560–3566. <a href="https://doi.org/10.1128/JCM.01231-13">https://doi.org/10.1128/JCM.01231-13</a>
</div>
<div id="ref-He289" class="csl-entry">
He, Z., Gharaibeh, R. Z., Newsome, R. C., Pope, J. L., Dougherty, M. W., Tomkovich, S., Pons, B., Mirey, G., Vignard, J., Hendrixson, D. R., &amp; Jobin, C. (2019). Campylobacter jejuni promotes colorectal tumorigenesis through the action of cytolethal distending toxin. <em>Gut</em>, <em>68</em>(2), 289–300. <a href="https://doi.org/10.1136/gutjnl-2018-317200">https://doi.org/10.1136/gutjnl-2018-317200</a>
</div>
<div id="ref-he_evolview_2016" class="csl-entry">
He, Z., Zhang, H., Gao, S., Lercher, M. J., Chen, W.-H., &amp; Hu, S. (2016). Evolview v2: An online visualization and management tool for customized and annotated phylogenetic trees. <em>Nucleic Acids Research</em>, <em>44</em>(W1), W236–241. <a href="https://doi.org/10.1093/nar/gkw370">https://doi.org/10.1093/nar/gkw370</a>
</div>
<div id="ref-hohna_probabilistic_2014" class="csl-entry">
Höhna, S., Heath, T. A., Boussau, B., Landis, M. J., Ronquist, F., &amp; Huelsenbeck, J. P. (2014). Probabilistic graphical model representation in phylogenetics. <em>Systematic Biology</em>, <em>63</em>(5), 753–771. <a href="https://doi.org/10.1093/sysbio/syu039">https://doi.org/10.1093/sysbio/syu039</a>
</div>
<div id="ref-hohna_revbayes:_2016" class="csl-entry">
Höhna, S., Landis, M. J., Heath, T. A., Boussau, B., Lartillot, N., Moore, B. R., Huelsenbeck, J. P., &amp; Ronquist, F. (2016). <span>RevBayes</span>: <span>Bayesian</span> <span>Phylogenetic</span> <span>Inference</span> <span>Using</span> <span>Graphical</span> <span>Models</span> and an <span>Interactive</span> <span>Model</span>-<span>Specification</span> <span>Language</span>. <em>Systematic Biology</em>, <em>65</em>(4), 726–736. <a href="https://doi.org/10.1093/sysbio/syw021">https://doi.org/10.1093/sysbio/syw021</a>
</div>
<div id="ref-huelsenbeck_mrbayes_2001" class="csl-entry">
Huelsenbeck, J. P., &amp; Ronquist, F. (2001). <a href="https://www.ncbi.nlm.nih.gov/pubmed/11524383"><span>MRBAYES</span>: <span>Bayesian</span> inference of phylogenetic trees</a>. <em>Bioinformatics (Oxford, England)</em>, <em>17</em>(8), 754–755.
</div>
<div id="ref-huson_dendroscope_2012" class="csl-entry">
Huson, D. H., &amp; Scornavacca, C. (2012). Dendroscope 3: An interactive tool for rooted phylogenetic trees and networks. <em>Systematic Biology</em>, <em>61</em>(6), 1061–1067. <a href="https://doi.org/10.1093/sysbio/sys062">https://doi.org/10.1093/sysbio/sys062</a>
</div>
<div id="ref-jombart_outbreaktools_2014" class="csl-entry">
Jombart, T., Aanensen, D. M., Baguelin, M., Birrell, P., Cauchemez, S., Camacho, A., Colijn, C., Collins, C., Cori, A., Didelot, X., Fraser, C., Frost, S., Hens, N., Hugues, J., Höhle, M., Opatowski, L., Rambaut, A., Ratmann, O., Soubeyrand, S., … Ferguson, N. (2014). <span>OutbreakTools</span>: A new platform for disease outbreak analysis using the <span>R</span> software. <em>Epidemics</em>, <em>7</em>, 28–34. <a href="https://doi.org/10.1016/j.epidem.2014.04.003">https://doi.org/10.1016/j.epidem.2014.04.003</a>
</div>
<div id="ref-kostic2012genomic" class="csl-entry">
Kostic, A. D., Gevers, D., Pedamallu, C. S., Michaud, M., Duke, F., Earl, A. M., Ojesina, A. I., Jung, J., Bass, A. J., Tabernero, J.others. (2012). Genomic analysis identifies association of fusobacterium with colorectal carcinoma. <em>Genome Research</em>, <em>22</em>(2), 292–298. <a href="https://doi.org/10.1101/gr.126573.111">https://doi.org/10.1101/gr.126573.111</a>
</div>
<div id="ref-kuczynski_using_2011" class="csl-entry">
Kuczynski, J., Stombaugh, J., Walters, W. A., González, A., Caporaso, J. G., &amp; Knight, R. (2011). Using <span>QIIME</span> to analyze 16S <span class="nocase">rRNA</span> gene sequences from <span>Microbial</span> <span>Communities</span>. <em>Current Protocols in Bioinformatics / Editoral Board, Andreas D. Baxevanis ... [Et Al.]</em>, <em>CHAPTER</em>, Unit10.7. <a href="https://doi.org/10.1002/0471250953.bi1007s36">https://doi.org/10.1002/0471250953.bi1007s36</a>
</div>
<div id="ref-kumar_mega7_2016" class="csl-entry">
Kumar, S., Stecher, G., &amp; Tamura, K. (2016). <span>MEGA</span>7: Molecular evolutionary genetics analysis version 7.0 for bigger datasets. <em>Molecular Biology and Evolution</em>, <em>33</em>(7), 1870–1874. <a href="https://doi.org/10.1093/molbev/msw054">https://doi.org/10.1093/molbev/msw054</a>
</div>
<div id="ref-kunin_pyrotagger_2010" class="csl-entry">
Kunin, V., &amp; Hugenholtz, P. (2010). <span>PyroTagger</span> : <span>A</span> fast , accurate pipeline for analysis of <span class="nocase">rRNA</span> amplicon pyrosequence data. <em>The Open Journal</em>, 1–8. <a href="http://www.theopenjournal.org/toj_articles/1">http://www.theopenjournal.org/toj_articles/1</a>
</div>
<div id="ref-lam_phylodynamics_2012" class="csl-entry">
Lam, T. T.-Y., Hon, C.-C., Lemey, P., Pybus, O. G., Shi, M., Tun, H. M., Li, J., Jiang, J., Holmes, E. C., &amp; Leung, F. C.-C. (2012). Phylodynamics of <span>H</span>5N1 avian influenza virus in <span>Indonesia</span>. <em>Molecular Ecology</em>, <em>21</em>(12), 3062–3077. <a href="https://doi.org/10.1111/j.1365-294X.2012.05577.x">https://doi.org/10.1111/j.1365-294X.2012.05577.x</a>
</div>
<div id="ref-lam_dissemination_2015" class="csl-entry">
Lam, T. T.-Y., Zhou, B., Wang, J., Chai, Y., Shen, Y., Chen, X., Ma, C., Hong, W., Chen, Y., Zhang, Y., Duan, L., Chen, P., Jiang, J., Zhang, Y., Li, L., Poon, L. L. M., Webby, R. J., Smith, D. K., Leung, G. M., … Zhu, H. (2015). Dissemination, divergence and establishment of <span>H</span>7N9 influenza viruses in <span>China</span>. <em>Nature</em>, <em>522</em>(7554), 102–105. <a href="https://doi.org/10.1038/nature14348">https://doi.org/10.1038/nature14348</a>
</div>
<div id="ref-larsen_identification_2019" class="csl-entry">
Larsen, F. T., Bed’Hom, B., Guldbrandtsen, B., &amp; Dalgaard, T. S. (2019). Identification and tissue-expression profiling of novel chicken c-type lectin-like domain containing proteins as potential targets for carbohydrate-based vaccine strategies. <em>Molecular Immunology</em>, <em>114</em>, 216–225. <a href="https://doi.org/10.1016/j.molimm.2019.07.022">https://doi.org/10.1016/j.molimm.2019.07.022</a>
</div>
<div id="ref-lemmon_importance_2004" class="csl-entry">
Lemmon, A. R., &amp; Moriarty, E. C. (2004). The importance of proper model assumption in bayesian phylogenetics. <em>Systematic Biology</em>, <em>53</em>(2), 265–277. <a href="https://doi.org/10.1080/10635150490423520">https://doi.org/10.1080/10635150490423520</a>
</div>
<div id="ref-letunic_interactive_2007" class="csl-entry">
Letunic, I., &amp; Bork, P. (2007). Interactive <span>Tree</span> <span>Of</span> <span>Life</span> (<span class="nocase">iTOL</span>): An online tool for phylogenetic tree display and annotation. <em>Bioinformatics</em>, <em>23</em>(1), 127–128. <a href="https://doi.org/10.1093/bioinformatics/btl529">https://doi.org/10.1093/bioinformatics/btl529</a>
</div>
<div id="ref-liang_expansion_2014" class="csl-entry">
Liang, H., Lam, T. T.-Y., Fan, X., Chen, X., Zeng, Y., Zhou, J., Duan, L., Tse, M., Chan, C.-H., Li, L., Leung, T.-Y., Yip, C.-H., Cheung, C.-L., Zhou, B., Smith, D. K., Poon, L. L.-M., Peiris, M., Guan, Y., &amp; Zhu, H. (2014). Expansion of genotypic diversity and establishment of 2009 <span>H</span>1N1 pandemic-origin internal genes in pigs in <span>China</span>. <em>Journal of Virology</em>, JVI.01327–14. <a href="https://doi.org/10.1128/JVI.01327-14">https://doi.org/10.1128/JVI.01327-14</a>
</div>
<div id="ref-lott_covenntree:_2015" class="csl-entry">
Lott, S. C., Voß, B., Hess, W. R., &amp; Steglich, C. (2015). <span>CoVennTree</span>: A new method for the comparative analysis of large datasets. <em>Frontiers in Genetics</em>, <em>6</em>, 43. <a href="https://doi.org/10.3389/fgene.2015.00043">https://doi.org/10.3389/fgene.2015.00043</a>
</div>
<div id="ref-maddison_nexus:_1997" class="csl-entry">
Maddison, D. R., Swofford, D. L., Maddison, W. P., &amp; Cannatella, D. (1997). Nexus: <span>An</span> <span>Extensible</span> <span>File</span> <span>Format</span> for <span>Systematic</span> <span>Information</span>. <em>Systematic Biology</em>, <em>46</em>(4), 590–621. <a href="https://doi.org/10.1093/sysbio/46.4.590">https://doi.org/10.1093/sysbio/46.4.590</a>
</div>
<div id="ref-matsen_format_2012" class="csl-entry">
Matsen, F. A., Hoffman, N. G., Gallagher, A., &amp; Stamatakis, A. (2012). A <span>Format</span> for <span>Phylogenetic</span> <span>Placements</span>. <em>PLOS ONE</em>, <em>7</em>(2), e31009. <a href="https://doi.org/10.1371/journal.pone.0031009">https://doi.org/10.1371/journal.pone.0031009</a>
</div>
<div id="ref-matsen_pplacer_2010" class="csl-entry">
Matsen, F. A., Kodner, R. B., &amp; Armbrust, E. V. (2010). Pplacer: Linear time maximum-likelihood and bayesian phylogenetic placement of sequences onto a fixed reference tree. <em><span>BMC</span> Bioinformatics</em>, <em>11</em>(1), 538. <a href="https://doi.org/10.1186/1471-2105-11-538">https://doi.org/10.1186/1471-2105-11-538</a>
</div>
<div id="ref-mcmurdie_phyloseq_2013" class="csl-entry">
McMurdie, P. J., &amp; Holmes, S. (2013). Phyloseq: An <span>R</span> package for reproducible interactive analysis and graphics of microbiome census data. <em>PloS One</em>, <em>8</em>(4), e61217. <a href="https://doi.org/10.1371/journal.pone.0061217">https://doi.org/10.1371/journal.pone.0061217</a>
</div>
<div id="ref-michonneau_rotl:_2016" class="csl-entry">
Michonneau, F., Brown, J. W., &amp; Winter, D. J. (2016). Rotl: An r package to interact with the open tree of life data. <em>Methods in Ecology and Evolution</em>, <em>7</em>(12), 1476–1481. <a href="https://doi.org/10.1111/2041-210X.12593">https://doi.org/10.1111/2041-210X.12593</a>
</div>
<div id="ref-morgan2013HMP" class="csl-entry">
Morgan, X. C., Segata, N., &amp; Huttenhower, C. (2013). Biodiversity and functional genomics in the human microbiome. <em>Trends in Genetics</em>, <em>29</em>(1), 51–58. <a href="https://doi.org/10.1016/J.TIG.2012.09.005">https://doi.org/10.1016/J.TIG.2012.09.005</a>
</div>
<div id="ref-neher_prediction_2016" class="csl-entry">
Neher, R. A., Bedford, T., Daniels, R. S., Russell, C. A., &amp; Shraiman, B. I. (2016). Prediction, dynamics, and visualization of antigenic phenotypes of seasonal influenza viruses. <em>Proceedings of the National Academy of Sciences</em>, <em>113</em>(12), E1701–E1709. <a href="https://doi.org/10.1073/pnas.1525578113">https://doi.org/10.1073/pnas.1525578113</a>
</div>
<div id="ref-page_visualizing_2002" class="csl-entry">
Page, R. D. M. (2002). Visualizing phylogenetic trees using <span>TreeView</span>. <em>Current Protocols in Bioinformatics</em>, <em>Chapter 6</em>, Unit 6.2. <a href="https://doi.org/10.1002/0471250953.bi0602s01">https://doi.org/10.1002/0471250953.bi0602s01</a>
</div>
<div id="ref-paradis_ape_2004" class="csl-entry">
Paradis, E., Claude, J., &amp; Strimmer, K. (2004). <span>APE</span>: <span>Analyses</span> of <span>Phylogenetics</span> and <span>Evolution</span> in <span>R</span> language. <em>Bioinformatics</em>, <em>20</em>(2), 289–290. <a href="https://doi.org/10.1093/bioinformatics/btg412">https://doi.org/10.1093/bioinformatics/btg412</a>
</div>
<div id="ref-pond_hyphy:_2005" class="csl-entry">
Pond, S. L. K., Frost, S. D. W., &amp; Muse, S. V. (2005). <span>HyPhy</span>: Hypothesis testing using phylogenies. <em>Bioinformatics (Oxford, England)</em>, <em>21</em>(5), 676–679. <a href="https://doi.org/10.1093/bioinformatics/bti079">https://doi.org/10.1093/bioinformatics/bti079</a>
</div>
<div id="ref-rstats" class="csl-entry">
R Core Team. (2016). <em>R: A language and environment for statistical computing</em>. R Foundation for Statistical Computing. <a href="https://www.R-project.org/">https://www.R-project.org/</a>
</div>
<div id="ref-rannala_probability_1996" class="csl-entry">
Rannala, B., &amp; Yang, Z. (1996). <a href="https://www.ncbi.nlm.nih.gov/pubmed/8703097">Probability distribution of molecular evolutionary trees: A new method of phylogenetic inference</a>. <em>Journal of Molecular Evolution</em>, <em>43</em>(3), 304–311.
</div>
<div id="ref-retief_phylogenetic_2000" class="csl-entry">
Retief, J. D. (2000). <a href="https://www.ncbi.nlm.nih.gov/pubmed/10547839">Phylogenetic analysis using <span>PHYLIP</span></a>. <em>Methods in Molecular Biology (Clifton, N.J.)</em>, <em>132</em>, 243–258.
</div>
<div id="ref-revell_phytools_2012" class="csl-entry">
Revell, L. J. (2012). Phytools: An <span>R</span> package for phylogenetic comparative biology (and other things). <em>Methods in Ecology and Evolution</em>, <em>3</em>(2), 217–223. <a href="https://doi.org/10.1111/j.2041-210X.2011.00169.x">https://doi.org/10.1111/j.2041-210X.2011.00169.x</a>
</div>
<div id="ref-sanderson_r8s:_2003" class="csl-entry">
Sanderson, M. J. (2003). r8s: Inferring absolute rates of molecular evolution and divergence times in the absence of a molecular clock. <em>Bioinformatics</em>, <em>19</em>(2), 301–302. <a href="https://doi.org/10.1093/bioinformatics/19.2.301">https://doi.org/10.1093/bioinformatics/19.2.301</a>
</div>
<div id="ref-schliep_phangorn_2011" class="csl-entry">
Schliep, K. P. (2011). Phangorn: Phylogenetic analysis in <span>R</span>. <em>Bioinformatics</em>, <em>27</em>(4), 592–593. <a href="https://doi.org/10.1093/bioinformatics/btq706">https://doi.org/10.1093/bioinformatics/btq706</a>
</div>
<div id="ref-schloss_introducing_2009" class="csl-entry">
Schloss, P. D., Westcott, S. L., Ryabin, T., Hall, J. R., Hartmann, M., Hollister, E. B., Lesniewski, R. A., Oakley, B. B., Parks, D. H., Robinson, C. J., Sahl, J. W., Stres, B., Thallinger, G. G., Van Horn, D. J., &amp; Weber, C. F. (2009). Introducing mothur: Open-source, platform-independent, community-supported software for describing and comparing microbial communities. <em>Applied and Environmental Microbiology</em>, <em>75</em>(23), 7537–7541. <a href="https://doi.org/10.1128/AEM.01541-09">https://doi.org/10.1128/AEM.01541-09</a>
</div>
<div id="ref-schmidt_tree-puzzle:_2002" class="csl-entry">
Schmidt, H. A., Strimmer, K., Vingron, M., &amp; Haeseler, A. von. (2002). <a href="https://www.ncbi.nlm.nih.gov/pubmed/11934758"><span>TREE</span>-<span>PUZZLE</span>: Maximum likelihood phylogenetic analysis using quartets and parallel computing</a>. <em>Bioinformatics (Oxford, England)</em>, <em>18</em>(3), 502–504.
</div>
<div id="ref-schon_age_2015" class="csl-entry">
Schön, I., Shearn, R., Martens, K., Koenders, A., &amp; Halse, S. (2015). Age and origin of <span>Australian</span> <span>Bennelongia</span> (<span>Crustacea</span>, <span>Ostracoda</span>). <em>Hydrobiologia</em>, <em>750</em>(1), 125–146. <a href="https://doi.org/10.1007/s10750-014-2159-z">https://doi.org/10.1007/s10750-014-2159-z</a>
</div>
<div id="ref-segata_metagenomic_2011" class="csl-entry">
Segata, N., Izard, J., Waldron, L., Gevers, D., Miropolsky, L., Garrett, W. S., &amp; Huttenhower, C. (2011). Metagenomic biomarker discovery and explanation. <em>Genome Biology</em>, <em>12</em>(6), R60. <a href="https://doi.org/10.1186/gb-2011-12-6-r60">https://doi.org/10.1186/gb-2011-12-6-r60</a>
</div>
<div id="ref-shoemaker_evidence_1989" class="csl-entry">
Shoemaker, J. S., &amp; Fitch, W. M. (1989). <a href="https://www.ncbi.nlm.nih.gov/pubmed/2622335">Evidence from nuclear sequences that invariable sites should be considered when sequence divergence is calculated</a>. <em>Molecular Biology and Evolution</em>, <em>6</em>(3), 270–289.
</div>
<div id="ref-stamatakis_raxml_2014" class="csl-entry">
Stamatakis, A. (2014). <span>RAxML</span> version 8: A tool for phylogenetic analysis and post-analysis of large phylogenies. <em>Bioinformatics</em>, <em>30</em>(9), 1312–1313. <a href="https://doi.org/10.1093/bioinformatics/btu033">https://doi.org/10.1093/bioinformatics/btu033</a>
</div>
<div id="ref-venkatesh_avian_2018" class="csl-entry">
Venkatesh, D., Poen, M. J., Bestebroer, T. M., Scheuer, R. D., Vuong, O., Chkhaidze, M., Machablishvili, A., Mamuchadze, J., Ninua, L., Fedorova, N. B., Halpin, R. A., Lin, X., Ransier, A., Stockwell, T. B., Wentworth, D. E., Kriti, D., Dutta, J., Bakel, H. van, Puranik, A., … Lewis, N. S. (2018). Avian influenza viruses in wild birds: Virus evolution in a multihost ecosystem. <em>Journal of Virology</em>, <em>92</em>(15). <a href="https://doi.org/10.1128/JVI.00433-18">https://doi.org/10.1128/JVI.00433-18</a>
</div>
<div id="ref-vos_nexml:_2012" class="csl-entry">
Vos, R. A., Balhoff, J. P., Caravas, J. A., Holder, M. T., Lapp, H., Maddison, W. P., Midford, P. E., Priyam, A., Sukumaran, J., Xia, X., &amp; Stoltzfus, A. (2012). <span>NeXML</span>: Rich, extensible, and verifiable representation of comparative data and metadata. <em>Systematic Biology</em>, <em>61</em>(4), 675–689. <a href="https://doi.org/10.1093/sysbio/sys025">https://doi.org/10.1093/sysbio/sys025</a>
</div>
<div id="ref-wang_treeio_2020" class="csl-entry">
Wang, L.-G., Lam, T. T.-Y., Xu, S., Dai, Z., Zhou, L., Feng, T., Guo, P., Dunn, C. W., Jones, B. R., Bradley, T., Zhu, H., Guan, Y., Jiang, Y., &amp; Yu, G. (2020). Treeio: An r package for phylogenetic tree input and output with richly annotated and associated data. <em>Molecular Biology and Evolution</em>, <em>37</em>(2), 599–603. <a href="https://doi.org/10.1093/molbev/msz240">https://doi.org/10.1093/molbev/msz240</a>
</div>
<div id="ref-wickham_ggplot2_2016" class="csl-entry">
Wickham, H. (2016). <em>ggplot2: Elegant graphics for data analysis</em>. Springer. <a href="http://ggplot2.org">http://ggplot2.org</a>
</div>
<div id="ref-wilgenbusch_inferring_2003" class="csl-entry">
Wilgenbusch, J. C., &amp; Swofford, D. (2003). Inferring evolutionary trees with <span>PAUP</span>*. <em>Current Protocols in Bioinformatics</em>, <em>Chapter 6</em>, Unit 6.4. <a href="https://doi.org/10.1002/0471250953.bi0604s00">https://doi.org/10.1002/0471250953.bi0604s00</a>
</div>
<div id="ref-wilkinson_grammar_2005" class="csl-entry">
Wilkinson, L., Wills, D., Rope, D., Norton, A., &amp; Dubbs, R. (2005). <em>The <span>Grammar</span> of <span>Graphics</span></em> (2nd edition). Springer.
</div>
<div id="ref-RN46:_2015" class="csl-entry">
Wong, V. K., Baker, S., Pickard, D. J., Parkhill, J., Page, A. J., Feasey, N. A., Kingsley, R. A., Thomson, N. R., Keane, J. A., Weill, F.-X., Edwards, D. J., Hawkey, J., Harris, S. R., Mather, A. E., Cain, A. K., Hadfield, J., Hart, P. J., Thieu, N. T. V., Klemm, E. J., … Dougan, G. (2015). Phylogeographical analysis of the dominant multidrug-resistant H58 clade of salmonella typhi identifies inter- and intracontinental transmission events [Journal Article]. <em>Nature Genetics</em>, <em>47</em>(6), 632–639. <a href="https://doi.org/10.1038/ng.3281">https://doi.org/10.1038/ng.3281</a>
</div>
<div id="ref-wu2013dysbiosis" class="csl-entry">
Wu, N., Yang, X., Zhang, R., Li, J., Xiao, X., Hu, Y., Chen, Y., Yang, F., Lu, N., Wang, Z.others. (2013). Dysbiosis signature of fecal microbiota in colorectal cancer patients. <em>Microbial Ecology</em>, <em>66</em>(2), 462–470. <a href="https://doi.org/10.1007/s00248-013-0245-9">https://doi.org/10.1007/s00248-013-0245-9</a>
</div>
<div id="ref-ggbreak" class="csl-entry">
Xu, S., Chen, M., Feng, T., Li, Z., Lang, Z., &amp; Yu, G. (2021). Use ggbreak to effectively utilize plotting space to deal with large datasets and outliers. <em>Frontiers in Genetics</em>, <em>12</em>, 774846. <a href="https://doi.org/10.3389/fgene.2021.774846">https://doi.org/10.3389/fgene.2021.774846</a>
</div>
<div id="ref-ggtreeExtra_2021" class="csl-entry">
Xu, S., Dai, Z., Guo, P., Fu, X., Liu, S., Zhou, L., Tang, W., Feng, T., Chen, M., Zhan, L., Wu, T., Hu, E., Jiang, Y., Bo, X., &amp; Yu, G. (2021). <span class="nocase">ggtreeExtra: Compact Visualization of Richly Annotated Phylogenetic Data</span>. <em>Molecular Biology and Evolution</em>, <em>38</em>(9), 4039–4042. <a href="https://doi.org/10.1093/molbev/msab166">https://doi.org/10.1093/molbev/msab166</a>
</div>
<div id="ref-yang_paml_2007" class="csl-entry">
Yang, Z. (2007). <span>PAML</span> 4: <span>Phylogenetic</span> <span>Analysis</span> by <span>Maximum</span> <span>Likelihood</span>. <em>Molecular Biology and Evolution</em>, <em>24</em>(8), 1586–1591. <a href="https://doi.org/10.1093/molbev/msm088">https://doi.org/10.1093/molbev/msm088</a>
</div>
<div id="ref-yang_maximum_1994" class="csl-entry">
Yang, Z. (1994). <a href="https://www.ncbi.nlm.nih.gov/pubmed/7932792">Maximum likelihood phylogenetic estimation from <span>DNA</span> sequences with variable rates over sites: Approximate methods</a>. <em>Journal of Molecular Evolution</em>, <em>39</em>(3), 306–314.
</div>
<div id="ref-yu_cp_2020" class="csl-entry">
Yu, G. (2020). Using ggtree to visualize data on tree-like structures. <em>Current Protocols in Bioinformatics</em>, <em>69</em>(1), e96. <a href="https://doi.org/10.1002/cpbi.96">https://doi.org/10.1002/cpbi.96</a>
</div>
<div id="ref-yu_reactomepa_2016" class="csl-entry">
Yu, G., &amp; He, Q.-Y. (2016). <span>ReactomePA</span>: An r/bioconductor package for reactome pathway analysis and visualization. <em>Molecular <span>BioSystems</span></em>, <em>12</em>(2), 477–479. <a href="https://doi.org/10.1039/C5MB00663E">https://doi.org/10.1039/C5MB00663E</a>
</div>
<div id="ref-yu_two_2018" class="csl-entry">
Yu, G., Lam, T. T.-Y., Zhu, H., &amp; Guan, Y. (2018). Two methods for mapping and visualizing associated data on phylogeny using ggtree. <em>Molecular Biology and Evolution</em>, <em>35</em>(12), 3041–3043. <a href="https://doi.org/10.1093/molbev/msy194">https://doi.org/10.1093/molbev/msy194</a>
</div>
<div id="ref-yu_ggtree:_2017" class="csl-entry">
Yu, G., Smith, D. K., Zhu, H., Guan, Y., &amp; Lam, T. T.-Y. (2017). Ggtree: An r package for visualization and annotation of phylogenetic trees with their covariates and other associated data. <em>Methods in Ecology and Evolution</em>, <em>8</em>(1), 28–36. <a href="https://doi.org/10.1111/2041-210X.12628">https://doi.org/10.1111/2041-210X.12628</a>
</div>
<div id="ref-yu2012" class="csl-entry">
Yu, G., Wang, L.-G., Han, Y., &amp; He, Q.-Y. (2012). <span class="nocase">clusterProfiler:</span> An r package for comparing biological themes among gene clusters. <em><span>OMICS:</span> A Journal of Integrative Biology</em>, <em>16</em>(5), 284–287. <a href="https://doi.org/10.1089/omi.2011.0118">https://doi.org/10.1089/omi.2011.0118</a>
</div>
<div id="ref-yu_chipseeker:_2015" class="csl-entry">
Yu, G., Wang, L.-G., &amp; He, Q.-Y. (2015). <span>ChIPseeker</span>: An <span>R</span>/<span>Bioconductor</span> package for <span>ChIP</span> peak annotation, comparison and visualization. <em>Bioinformatics</em>, <em>31</em>(14), 2382–2383. <a href="https://doi.org/10.1093/bioinformatics/btv145">https://doi.org/10.1093/bioinformatics/btv145</a>
</div>
<div id="ref-yu_dose_2015" class="csl-entry">
Yu, G., Wang, L.-G., Yan, G.-R., &amp; He, Q.-Y. (2015). <span>DOSE</span>: An r/bioconductor package for disease ontology semantic and enrichment analysis. <em>Bioinformatics</em>, <em>31</em>(4), 608–609. <a href="https://doi.org/10.1093/bioinformatics/btu684">https://doi.org/10.1093/bioinformatics/btu684</a>
</div>
<div id="ref-zmasek_atv:_2001" class="csl-entry">
Zmasek, C. M., &amp; Eddy, S. R. (2001). <span>ATV</span>: Display and manipulation of annotated phylogenetic trees. <em>Bioinformatics</em>, <em>17</em>(4), 383–384. <a href="https://doi.org/10.1093/bioinformatics/17.4.383">https://doi.org/10.1093/bioinformatics/17.4.383</a>
</div>
</div>
</div>






























  <div class="chapter-nav">
<div class="prev"><a href="publications-of-the-ggtree-package-suite.html"><span class="header-section-number">D</span> Publications of the ggtree Package Suite</a></div>
<div class="empty"></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav"><li><a class="nav-link" href="#references">References</a></li></ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Data Integration, Manipulation and Visualization of Phylogenetic Trees</strong>" was written by Guangchuang Yu. It was last built on 2022-04-22.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
