<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Chapter 4 Phylogenetic Tree Visualization | Data Integration, Manipulation and Visualization of Phylogenetic Trees</title>
  <meta name="description" content="Chapter 4 Phylogenetic Tree Visualization | Data Integration, Manipulation and Visualization of Phylogenetic Trees">
  <meta name="generator" content="bookdown  and GitBook 2.6.7">

  <meta property="og:title" content="Chapter 4 Phylogenetic Tree Visualization | Data Integration, Manipulation and Visualization of Phylogenetic Trees" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 4 Phylogenetic Tree Visualization | Data Integration, Manipulation and Visualization of Phylogenetic Trees" />
  
  
  

<meta name="author" content="Guangchuang Yu School of Basic Medical Sciences, Southern Medical University">



  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="chapter3.html">
<link rel="next" href="chapter5.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />







<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<script src="libs/htmlwidgets-1.3/htmlwidgets.js"></script>
<script src="libs/pathseg-1.0.2/pathseg.js"></script>
<script src="libs/comic-0.961/comic.min.js"></script>
<script src="libs/comicR-binding-0.2/comicR.js"></script>


<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(title);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="part"><span><b>Part I: Tree data input, output and manipulation</b></span></li>
<li class="chapter" data-level="1" data-path="chapter1.html"><a href="chapter1.html"><i class="fa fa-check"></i><b>1</b> Importing Tree with Data</a><ul>
<li class="chapter" data-level="1.1" data-path="chapter1.html"><a href="chapter1.html#pc"><i class="fa fa-check"></i><b>1.1</b> Overview of Phylogenetic Tree Construction</a></li>
<li class="chapter" data-level="1.2" data-path="chapter1.html"><a href="chapter1.html#format"><i class="fa fa-check"></i><b>1.2</b> Phylogenetic Tree Formats</a><ul>
<li class="chapter" data-level="1.2.1" data-path="chapter1.html"><a href="chapter1.html#newick-tree-format"><i class="fa fa-check"></i><b>1.2.1</b> Newick tree format</a></li>
<li class="chapter" data-level="1.2.2" data-path="chapter1.html"><a href="chapter1.html#nexus-tree-format"><i class="fa fa-check"></i><b>1.2.2</b> NEXUS tree format</a></li>
<li class="chapter" data-level="1.2.3" data-path="chapter1.html"><a href="chapter1.html#nhxtext"><i class="fa fa-check"></i><b>1.2.3</b> New Hampshire eXtended format</a></li>
<li class="chapter" data-level="1.2.4" data-path="chapter1.html"><a href="chapter1.html#jplace-format"><i class="fa fa-check"></i><b>1.2.4</b> Jplace format</a></li>
<li class="chapter" data-level="1.2.5" data-path="chapter1.html"><a href="chapter1.html#software-outputs"><i class="fa fa-check"></i><b>1.2.5</b> Software outputs</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="chapter1.html"><a href="chapter1.html#getting-tree-data-with-treeio"><i class="fa fa-check"></i><b>1.3</b> Getting Tree Data with <em>treeio</em></a><ul>
<li class="chapter" data-level="1.3.1" data-path="chapter1.html"><a href="chapter1.html#overview-of-treeio"><i class="fa fa-check"></i><b>1.3.1</b> Overview of <em>treeio</em></a></li>
<li class="chapter" data-level="1.3.2" data-path="chapter1.html"><a href="chapter1.html#function-demonstration"><i class="fa fa-check"></i><b>1.3.2</b> Function Demonstration</a></li>
<li class="chapter" data-level="1.3.3" data-path="chapter1.html"><a href="chapter1.html#getting-information-from-treedata-object"><i class="fa fa-check"></i><b>1.3.3</b> Getting information from <em>treedata</em> object</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="chapter1.html"><a href="chapter1.html#summary"><i class="fa fa-check"></i><b>1.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="chapter2.html"><a href="chapter2.html"><i class="fa fa-check"></i><b>2</b> Manipulating Tree with Data</a><ul>
<li class="chapter" data-level="2.1" data-path="chapter2.html"><a href="chapter2.html#tidytree"><i class="fa fa-check"></i><b>2.1</b> Manipulating tree data using tidy interface</a><ul>
<li class="chapter" data-level="2.1.1" data-path="chapter2.html"><a href="chapter2.html#the-phylo-object"><i class="fa fa-check"></i><b>2.1.1</b> The <code>phylo</code> object</a></li>
<li class="chapter" data-level="2.1.2" data-path="chapter2.html"><a href="chapter2.html#the-treedata-object"><i class="fa fa-check"></i><b>2.1.2</b> The <code>treedata</code> object</a></li>
<li class="chapter" data-level="2.1.3" data-path="chapter2.html"><a href="chapter2.html#accesor-tidytree"><i class="fa fa-check"></i><b>2.1.3</b> Access related nodes</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="chapter2.html"><a href="chapter2.html#data-integration"><i class="fa fa-check"></i><b>2.2</b> Data Integration</a><ul>
<li class="chapter" data-level="2.2.1" data-path="chapter2.html"><a href="chapter2.html#merge-tree"><i class="fa fa-check"></i><b>2.2.1</b> Combining tree data</a></li>
<li class="chapter" data-level="2.2.2" data-path="chapter2.html"><a href="chapter2.html#link-external-data"><i class="fa fa-check"></i><b>2.2.2</b> Linking external data to phylogeny</a></li>
<li class="chapter" data-level="2.2.3" data-path="chapter2.html"><a href="chapter2.html#grouping-taxa"><i class="fa fa-check"></i><b>2.2.3</b> Grouping taxa</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="chapter2.html"><a href="chapter2.html#rescale-treeio"><i class="fa fa-check"></i><b>2.3</b> Rescaling Tree Branches</a></li>
<li class="chapter" data-level="2.4" data-path="chapter2.html"><a href="chapter2.html#subsetting-tree-with-data"><i class="fa fa-check"></i><b>2.4</b> Subsetting Tree with Data</a><ul>
<li class="chapter" data-level="2.4.1" data-path="chapter2.html"><a href="chapter2.html#remove-tip"><i class="fa fa-check"></i><b>2.4.1</b> Removing tips in a phylogenetic tree</a></li>
<li class="chapter" data-level="2.4.2" data-path="chapter2.html"><a href="chapter2.html#subset-tip"><i class="fa fa-check"></i><b>2.4.2</b> Subsetting tree by tip label</a></li>
<li class="chapter" data-level="2.4.3" data-path="chapter2.html"><a href="chapter2.html#subset-node"><i class="fa fa-check"></i><b>2.4.3</b> Subsetting tree by internal node number</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="chapter2.html"><a href="chapter2.html#ggtree-fortify"><i class="fa fa-check"></i><b>2.5</b> Manipulating tree data for visualization</a></li>
<li class="chapter" data-level="2.6" data-path="chapter2.html"><a href="chapter2.html#summary-1"><i class="fa fa-check"></i><b>2.6</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="chapter3.html"><a href="chapter3.html"><i class="fa fa-check"></i><b>3</b> Exporting tree with data</a><ul>
<li class="chapter" data-level="3.1" data-path="chapter3.html"><a href="chapter3.html#introduction"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="chapter3.html"><a href="chapter3.html#exporting-tree-data-to-beast-nexus-format"><i class="fa fa-check"></i><b>3.2</b> Exporting Tree Data to <em>BEAST</em> Nexus Format</a><ul>
<li class="chapter" data-level="3.2.1" data-path="chapter3.html"><a href="chapter3.html#exportingconverting-software-output"><i class="fa fa-check"></i><b>3.2.1</b> Exporting/converting software output</a></li>
<li class="chapter" data-level="3.2.2" data-path="chapter3.html"><a href="chapter3.html#combining-tree-with-external-data"><i class="fa fa-check"></i><b>3.2.2</b> Combining tree with external data</a></li>
<li class="chapter" data-level="3.2.3" data-path="chapter3.html"><a href="chapter3.html#merging-tree-data-from-different-sources"><i class="fa fa-check"></i><b>3.2.3</b> Merging tree data from different sources</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="chapter3.html"><a href="chapter3.html#write-jtree"><i class="fa fa-check"></i><b>3.3</b> Exporting Tree Data to <em>jtree</em> Format</a></li>
<li class="chapter" data-level="3.4" data-path="chapter3.html"><a href="chapter3.html#summary-2"><i class="fa fa-check"></i><b>3.4</b> Summary</a></li>
</ul></li>
<li class="part"><span><b>Part II: Tree data visualization and annotation</b></span></li>
<li class="chapter" data-level="4" data-path="chapter4.html"><a href="chapter4.html"><i class="fa fa-check"></i><b>4</b> Phylogenetic Tree Visualization</a><ul>
<li class="chapter" data-level="4.1" data-path="chapter4.html"><a href="chapter4.html#introduction-1"><i class="fa fa-check"></i><b>4.1</b> Introduction</a></li>
<li class="chapter" data-level="4.2" data-path="chapter4.html"><a href="chapter4.html#visualizing-phylogenetic-tree-with-ggtree"><i class="fa fa-check"></i><b>4.2</b> Visualizing Phylogenetic Tree with <em>ggtree</em></a><ul>
<li class="chapter" data-level="4.2.1" data-path="chapter4.html"><a href="chapter4.html#basic-tree-visualization"><i class="fa fa-check"></i><b>4.2.1</b> Basic Tree Visualization</a></li>
<li class="chapter" data-level="4.2.2" data-path="chapter4.html"><a href="chapter4.html#layouts-of-phylogenetic-tree"><i class="fa fa-check"></i><b>4.2.2</b> Layouts of phylogenetic tree</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="chapter4.html"><a href="chapter4.html#displaying-tree-components"><i class="fa fa-check"></i><b>4.3</b> Displaying Tree Components</a><ul>
<li class="chapter" data-level="4.3.1" data-path="chapter4.html"><a href="chapter4.html#displaying-tree-scale-evolution-distance"><i class="fa fa-check"></i><b>4.3.1</b> Displaying tree scale (evolution distance)</a></li>
<li class="chapter" data-level="4.3.2" data-path="chapter4.html"><a href="chapter4.html#displaying-nodestips"><i class="fa fa-check"></i><b>4.3.2</b> Displaying nodes/tips</a></li>
<li class="chapter" data-level="4.3.3" data-path="chapter4.html"><a href="chapter4.html#displaying-labels"><i class="fa fa-check"></i><b>4.3.3</b> Displaying labels</a></li>
<li class="chapter" data-level="4.3.4" data-path="chapter4.html"><a href="chapter4.html#color-tree"><i class="fa fa-check"></i><b>4.3.4</b> Color tree</a></li>
<li class="chapter" data-level="4.3.5" data-path="chapter4.html"><a href="chapter4.html#rescale-tree"><i class="fa fa-check"></i><b>4.3.5</b> Rescale tree</a></li>
<li class="chapter" data-level="4.3.6" data-path="chapter4.html"><a href="chapter4.html#modify-compenents-of-the-tree-view"><i class="fa fa-check"></i><b>4.3.6</b> Modify compenents of the tree view</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="chapter4.html"><a href="chapter4.html#visualize-a-list-of-trees"><i class="fa fa-check"></i><b>4.4</b> Visualize a list of trees</a></li>
<li class="chapter" data-level="4.5" data-path="chapter4.html"><a href="chapter4.html#summary-3"><i class="fa fa-check"></i><b>4.5</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="chapter5.html"><a href="chapter5.html"><i class="fa fa-check"></i><b>5</b> Phylogenetic Tree Annotation</a><ul>
<li class="chapter" data-level="5.1" data-path="chapter5.html"><a href="chapter5.html#visualizing-and-annotating-tree-using-grammar-of-graphics"><i class="fa fa-check"></i><b>5.1</b> Visualizing and Annotating Tree using Grammar of Graphics</a></li>
<li class="chapter" data-level="5.2" data-path="chapter5.html"><a href="chapter5.html#layers-for-tree-annotation"><i class="fa fa-check"></i><b>5.2</b> Layers for Tree Annotation</a><ul>
<li class="chapter" data-level="5.2.1" data-path="chapter5.html"><a href="chapter5.html#annotate-clades"><i class="fa fa-check"></i><b>5.2.1</b> Annotate clades</a></li>
<li class="chapter" data-level="5.2.2" data-path="chapter5.html"><a href="chapter5.html#labelling-associated-taxa-monophyletic-polyphyletic-or-paraphyletic"><i class="fa fa-check"></i><b>5.2.2</b> Labelling associated taxa (Monophyletic, Polyphyletic or Paraphyletic)</a></li>
<li class="chapter" data-level="5.2.3" data-path="chapter5.html"><a href="chapter5.html#highlight-clades"><i class="fa fa-check"></i><b>5.2.3</b> Highlight clades</a></li>
<li class="chapter" data-level="5.2.4" data-path="chapter5.html"><a href="chapter5.html#taxa-connection"><i class="fa fa-check"></i><b>5.2.4</b> Taxa connection</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="chapter5.html"><a href="chapter5.html#tree-annotation-with-output-from-evolution-software"><i class="fa fa-check"></i><b>5.3</b> Tree annotation with output from evolution software</a><ul>
<li class="chapter" data-level="5.3.1" data-path="chapter5.html"><a href="chapter5.html#tree-annotation-using-data-from-evolutionary-analysis-software"><i class="fa fa-check"></i><b>5.3.1</b> Tree annotation using data from evolutionary analysis software</a></li>
<li class="chapter" data-level="5.3.2" data-path="chapter5.html"><a href="chapter5.html#tree-annotation-based-on-tree-classes-defined-in-other-r-packages"><i class="fa fa-check"></i><b>5.3.2</b> Tree annotation based on tree classes defined in other R packages</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="chapter5.html"><a href="chapter5.html#summary-4"><i class="fa fa-check"></i><b>5.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="chapter6.html"><a href="chapter6.html"><i class="fa fa-check"></i><b>6</b> Visual Exploration of Phylogenetic Tree</a><ul>
<li class="chapter" data-level="6.1" data-path="chapter6.html"><a href="chapter6.html#viewing-selected-clade"><i class="fa fa-check"></i><b>6.1</b> Viewing Selected Clade</a></li>
<li class="chapter" data-level="6.2" data-path="chapter6.html"><a href="chapter6.html#collapsing-and-expanding-clade"><i class="fa fa-check"></i><b>6.2</b> Collapsing and Expanding Clade</a></li>
<li class="chapter" data-level="6.3" data-path="chapter6.html"><a href="chapter6.html#scaling-selected-clade"><i class="fa fa-check"></i><b>6.3</b> Scaling Selected Clade</a></li>
<li class="chapter" data-level="6.4" data-path="chapter6.html"><a href="chapter6.html#group-taxa-vis"><i class="fa fa-check"></i><b>6.4</b> Grouping Taxa</a></li>
<li class="chapter" data-level="6.5" data-path="chapter6.html"><a href="chapter6.html#exploring-tree-structure"><i class="fa fa-check"></i><b>6.5</b> Exploring tree structure</a></li>
<li class="chapter" data-level="6.6" data-path="chapter6.html"><a href="chapter6.html#zoom-on-a-portion-of-tree"><i class="fa fa-check"></i><b>6.6</b> Zoom on a portion of tree</a></li>
<li class="chapter" data-level="6.7" data-path="chapter6.html"><a href="chapter6.html#summary-5"><i class="fa fa-check"></i><b>6.7</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="chapter7.html"><a href="chapter7.html"><i class="fa fa-check"></i><b>7</b> Plotting tree with data</a><ul>
<li class="chapter" data-level="7.1" data-path="chapter7.html"><a href="chapter7.html#attach-operator"><i class="fa fa-check"></i><b>7.1</b> The <code>%&lt;+%</code> operator</a></li>
<li class="chapter" data-level="7.2" data-path="chapter7.html"><a href="chapter7.html#visualize-tree-with-associated-matrix"><i class="fa fa-check"></i><b>7.2</b> Visualize tree with associated matrix</a><ul>
<li class="chapter" data-level="7.2.1" data-path="chapter7.html"><a href="chapter7.html#gheatmap"><i class="fa fa-check"></i><b>7.2.1</b> Visualize tree with multiple associated matrix</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="chapter7.html"><a href="chapter7.html#msaplot"><i class="fa fa-check"></i><b>7.3</b> Visualize tree with multiple sequence alignment</a></li>
<li class="chapter" data-level="7.4" data-path="chapter7.html"><a href="chapter7.html#plot-tree-with-associated-data"><i class="fa fa-check"></i><b>7.4</b> Plot tree with associated data</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="chapter8.html"><a href="chapter8.html"><i class="fa fa-check"></i><b>8</b> Annotating Tree with Silhouette Images and Sub-plots</a><ul>
<li class="chapter" data-level="8.1" data-path="chapter8.html"><a href="chapter8.html#ggimage"><i class="fa fa-check"></i><b>8.1</b> Annotating Tree with Images</a></li>
<li class="chapter" data-level="8.2" data-path="chapter8.html"><a href="chapter8.html#phylopic"><i class="fa fa-check"></i><b>8.2</b> Annotating Tree with Phylopic</a></li>
<li class="chapter" data-level="8.3" data-path="chapter8.html"><a href="chapter8.html#subplots"><i class="fa fa-check"></i><b>8.3</b> Annotating Tree with Sub-plots</a><ul>
<li class="chapter" data-level="8.3.1" data-path="chapter8.html"><a href="chapter8.html#annotate-with-bar-charts"><i class="fa fa-check"></i><b>8.3.1</b> Annotate with bar charts</a></li>
<li class="chapter" data-level="8.3.2" data-path="chapter8.html"><a href="chapter8.html#annotate-with-pie-charts"><i class="fa fa-check"></i><b>8.3.2</b> Annotate with pie charts</a></li>
<li class="chapter" data-level="8.3.3" data-path="chapter8.html"><a href="chapter8.html#annotate-with-other-types-of-charts"><i class="fa fa-check"></i><b>8.3.3</b> Annotate with other types of charts</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="chapter8.html"><a href="chapter8.html#phylomoji"><i class="fa fa-check"></i><b>8.4</b> Have Fun with Phylomoji</a><ul>
<li class="chapter" data-level="8.4.1" data-path="chapter8.html"><a href="chapter8.html#emoji-in-cirularfan-layout-tree"><i class="fa fa-check"></i><b>8.4.1</b> emoji in cirular/fan layout tree</a></li>
<li class="chapter" data-level="8.4.2" data-path="chapter8.html"><a href="chapter8.html#emoji-to-label-clades"><i class="fa fa-check"></i><b>8.4.2</b> emoji to label clades</a></li>
<li class="chapter" data-level="8.4.3" data-path="chapter8.html"><a href="chapter8.html#apple-color-emoji"><i class="fa fa-check"></i><b>8.4.3</b> Apple Color Emoji</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>Part III: Miscellaneous topics</b></span></li>
<li class="chapter" data-level="9" data-path="chapter9.html"><a href="chapter9.html"><i class="fa fa-check"></i><b>9</b> ggtree utilities</a><ul>
<li class="chapter" data-level="9.1" data-path="chapter9.html"><a href="chapter9.html#facet_widths"><i class="fa fa-check"></i><b>9.1</b> facet_widths</a></li>
<li class="chapter" data-level="9.2" data-path="chapter9.html"><a href="chapter9.html#facet_labeller"><i class="fa fa-check"></i><b>9.2</b> facet_labeller</a></li>
<li class="chapter" data-level="9.3" data-path="chapter9.html"><a href="chapter9.html#xlim_expand"><i class="fa fa-check"></i><b>9.3</b> xlim_expand</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="chapter10.html"><a href="chapter10.html"><i class="fa fa-check"></i><b>10</b> ggtree for other tree objects</a></li>
<li class="chapter" data-level="11" data-path="chapter11.html"><a href="chapter11.html"><i class="fa fa-check"></i><b>11</b> ggtree extensions</a></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="faq.html"><a href="faq.html"><i class="fa fa-check"></i><b>A</b> Frequently asked questions</a><ul>
<li class="chapter" data-level="A.1" data-path="faq.html"><a href="faq.html#installation"><i class="fa fa-check"></i><b>A.1</b> Installation</a></li>
<li class="chapter" data-level="A.2" data-path="faq.html"><a href="faq.html#faq-r"><i class="fa fa-check"></i><b>A.2</b> Basic R related</a><ul>
<li class="chapter" data-level="A.2.1" data-path="faq.html"><a href="faq.html#faq-local-file"><i class="fa fa-check"></i><b>A.2.1</b> Use your local file</a></li>
</ul></li>
<li class="chapter" data-level="A.3" data-path="faq.html"><a href="faq.html#faq-aes-mapping"><i class="fa fa-check"></i><b>A.3</b> Aesthetic mapping</a><ul>
<li class="chapter" data-level="A.3.1" data-path="faq.html"><a href="faq.html#faq-inherit-aes"><i class="fa fa-check"></i><b>A.3.1</b> Inherit <em>aes</em></a></li>
<li class="chapter" data-level="A.3.2" data-path="faq.html"><a href="faq.html#faq-dollar-aes"><i class="fa fa-check"></i><b>A.3.2</b> Never use <code>$</code> in aes</a></li>
</ul></li>
<li class="chapter" data-level="A.4" data-path="faq.html"><a href="faq.html#faq-text"><i class="fa fa-check"></i><b>A.4</b> Text &amp; Label</a><ul>
<li class="chapter" data-level="A.4.1" data-path="faq.html"><a href="faq.html#faq-label-truncated"><i class="fa fa-check"></i><b>A.4.1</b> Tip label truncated</a></li>
<li class="chapter" data-level="A.4.2" data-path="faq.html"><a href="faq.html#faq-modify-label"><i class="fa fa-check"></i><b>A.4.2</b> Modify (tip) labels</a></li>
<li class="chapter" data-level="A.4.3" data-path="faq.html"><a href="faq.html#faq-formatting-label"><i class="fa fa-check"></i><b>A.4.3</b> Formatting (tip) labels</a></li>
<li class="chapter" data-level="A.4.4" data-path="faq.html"><a href="faq.html#faq-ggrepel"><i class="fa fa-check"></i><b>A.4.4</b> Avoid overlapping text labels</a></li>
<li class="chapter" data-level="A.4.5" data-path="faq.html"><a href="faq.html#faq-bootstrap"><i class="fa fa-check"></i><b>A.4.5</b> Bootstrap values from newick format</a></li>
</ul></li>
<li class="chapter" data-level="A.5" data-path="faq.html"><a href="faq.html#faq-under-the-tree"><i class="fa fa-check"></i><b>A.5</b> Plot something behind the phylogeny</a></li>
<li class="chapter" data-level="A.6" data-path="faq.html"><a href="faq.html#faq-enlarge-center-space"><i class="fa fa-check"></i><b>A.6</b> Enlarge center space in circular/fan layout tree</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="ggimage-tips.html"><a href="ggimage-tips.html"><i class="fa fa-check"></i><b>B</b> Image processing using magick package</a><ul>
<li class="chapter" data-level="B.1" data-path="ggimage-tips.html"><a href="ggimage-tips.html#example-1-remove-background-of-images"><i class="fa fa-check"></i><b>B.1</b> Example 1: Remove background of images</a></li>
</ul></li>
<li class="chapter" data-level="C" data-path="commicR.html"><a href="commicR.html"><i class="fa fa-check"></i><b>C</b> Comic (xkcd-like) phylogenetic tree</a></li>
<li class="chapter" data-level="D" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>D</b> Session Info</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Data Integration, Manipulation and Visualization of Phylogenetic Trees</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="chapter4" class="section level1">
<h1><span class="header-section-number">Chapter 4</span> Phylogenetic Tree Visualization</h1>
<div id="introduction-1" class="section level2">
<h2><span class="header-section-number">4.1</span> Introduction</h2>
<p>There are many software packages and webtools that are designed for displaying phylogenetic trees, such as <em>TreeView</em> <span class="citation">(Page <a href="#ref-page_visualizing_2002" role="doc-biblioref">2002</a>)</span>, <em>FigTree</em><a href="#fn11" class="footnote-ref" id="fnref11"><sup>11</sup></a>, <em>TreeDyn</em> <span class="citation">(Chevenet et al. <a href="#ref-chevenet_treedyn:_2006" role="doc-biblioref">2006</a>)</span>, <em>Dendroscope</em> <span class="citation">(Huson and Scornavacca <a href="#ref-huson_dendroscope_2012" role="doc-biblioref">2012</a>)</span>, <em>EvolView</em> <span class="citation">(He et al. <a href="#ref-he_evolview_2016" role="doc-biblioref">2016</a>)</span> and <em>iTOL</em> <span class="citation">(Letunic and Bork <a href="#ref-letunic_interactive_2007" role="doc-biblioref">2007</a>)</span>, <em>etc.</em>. Only several of them, such as <em>FigTree</em>, <em>TreeDyn</em> and <em>iTOL</em>, allow users to annotate the trees with coloring branches, highlighted clades and tree features. However, their pre-defined annotating functions are usually limited to some specific phylogenetic data. As phylogenetic trees become more widely used in multidisciplinary studies, there is an increasing need to incorporate various types of the phylogenetic covariates and other associated data from different sources into the trees for visualizations and further analyses. For instance, influenza virus has a wide host range, diverse and dynamic genotypes and characteristic transmission behaviors that are mostly associated with the virus evolution and essentially among themselves. Therefore, in addition to standalone applications that focus on each of the specific analysis and data type, researchers studying molecular evolution need a robust and programmable platform that allows the high levels integration and visualization of many of these different aspects of data (raw or from other primary analyses) over the phylogenetic trees to identify their associations and patterns.</p>
<p>To fill this gap, we developed <em>ggtree</em>, a package for the R programming language <span class="citation">(R Core Team <a href="#ref-rstats" role="doc-biblioref">2016</a>)</span> released under the Bioconductor project <span class="citation">(Gentleman et al. <a href="#ref-gentleman_bioconductor_2004" role="doc-biblioref">2004</a>)</span>. The <em>ggtree</em> is built to work with phylogenetic data object generated by <a href="#chapter-treeio"><em>treeio</em></a>, and display tree graphics with <em>ggplot2</em> package <span class="citation">(Wickham <a href="#ref-wickham_ggplot2_2016" role="doc-biblioref">2016</a>)</span> that was based on the grammar of graphics <span class="citation">(Wilkinson et al. <a href="#ref-wilkinson_grammar_2005" role="doc-biblioref">2005</a>)</span>.</p>
<p>The R language is increasingly being used in phylogenetics. However, a comprehensive package, designed for viewing and annotating phylogenetic trees, particularly with complex data integration, is not yet available. Most of the R packages in phylogenetics focus on specific statistical analyses rather than viewing and annotating the trees with more generalized phylogeny-associated data. Some packages, including <em>ape</em> <span class="citation">(Paradis, Claude, and Strimmer <a href="#ref-paradis_ape_2004" role="doc-biblioref">2004</a>)</span> and <em>phytools</em> <span class="citation">(Revell <a href="#ref-revell_phytools_2012" role="doc-biblioref">2012</a>)</span>, which are capable of displaying and annotating trees, are developed using the base graphics system of R. In particular, <em>ape</em> is one of the fundamental package for phylogenetic analysis and data processing. However, the base graphics system is relatively difficult to extend and limits the complexity of tree figure to be displayed. <em>OutbreakTools</em> <span class="citation">(Jombart et al. <a href="#ref-jombart_outbreaktools_2014" role="doc-biblioref">2014</a>)</span> and <em>phyloseq</em> <span class="citation">(McMurdie and Holmes <a href="#ref-mcmurdie_phyloseq_2013" role="doc-biblioref">2013</a>)</span> extended <em>ggplot2</em> to plot phylogenetic trees. The <em>ggplot2</em> system of graphics allows rapid customization and exploration of design solutions. However these packages were designed for epidemiology and microbiome data respectively and did not aim to provide a general solution for tree visualization and annotation. The <em>ggtree</em> package also inherits versatile properties of <em>ggplot2</em>, and more importantly allows constructing complex tree figures by freely combining multiple layers of annotations using the tree associated data imported from different sources (importation using <em>treeio</em> package is detailed in <a href="chapter1.html#chapter1">Chapter 1</a>).</p>
</div>
<div id="visualizing-phylogenetic-tree-with-ggtree" class="section level2">
<h2><span class="header-section-number">4.2</span> Visualizing Phylogenetic Tree with <em>ggtree</em></h2>
<p>The <em>ggtree</em> package is designed for annotating phylogenetic trees with their associated data of different types and from various sources. These data could come from users or analysis programs, and might include evolutionary rates, ancestral sequences, <em>etc.</em> that are associated with the taxa from real samples, or with the internal nodes representing hypothetic ancestor strain/species, or with the tree branches indicating evolutionary time courses. For instance, the data could be the geographic positions of the sampled avian influenza viruses (informed by the survey locations) and the ancestral nodes (by phylogeographic inference) in the viral gene tree <span class="citation">(Lam et al. <a href="#ref-lam_phylodynamics_2012" role="doc-biblioref">2012</a>)</span>.</p>
<p>The <em>ggtree</em> supports <em>ggplot2</em>’s graphical language, which allows high level of customization, is intuitive and flexible. It is notable that <em>ggplot2</em> itself does not provide low-level geometric objects or other support for tree-like structures, and hence <em>ggtree</em> is a useful extension on that regard. Even though the other two phylogenetics-related R packages, <em>OutbreakTools</em> and <em>phyloseq</em>, are developed based on <em>ggplot2</em>, the most valuable part of <em>ggplot2</em> syntax - adding layers of annotations - is not supported in these packages. For example, if we have plotted a tree without taxa labels, <em>OutbreakTools</em> and <em>phyloseq</em> provide no easy way for general <code>R</code> users, who have little knowledge about the infrastructures of these packages, to add a layer of taxa labels. The <em>ggtree</em> extends <em>ggplot2</em> to support tree objects by implementing a geometric layer, <code>geom_tree</code>, to support visualizing tree structure. In <em>ggtree</em>, viewing a phylogenetic tree is relatively easy, via the command <code>ggplot(tree_object) + geom_tree() + theme_tree()</code> or <code>ggtree(tree_object)</code> for short. Layers of annotations can be added one by one via the <code>+</code> operator. To facilitate tree visualization, <em>ggtree</em> provides several geometric layers, including <code>geom_treescale</code> for adding legend of tree branch scale (genetic distance, divergence time, <em>etc.</em>), <code>geom_range</code> for displaying uncertainty of branch lengths (confidence interval or range, <em>etc.</em>), <code>geom_tiplab</code> for adding taxa label, <code>geom_tipoint</code> and <code>geom_nodepoint</code> for adding symbols of tips and internal nodes, <code>geom_hilight</code> for highlighting a clade with rectangle, and <code>geom_cladelabel</code> for annotating a selected clade with a bar and text label, <em>etc.</em>. A full list of geometric layers provided by <em>ggtree</em> are summarized in Table <a href="chapter5.html#tab:geoms">5.1</a>.</p>
<p>To view a phylogenetic tree, we first need to parse the tree file into <em>R</em>.
<a href="#chapter-treeio"><em>Treeio</em></a> package parses diverse annotation data from different software outputs into <code>S4</code> phylogenetic data objects. The <em>ggtree</em> mainly utilizes these <code>S4</code> objects to display and annotate the tree. There are also other R packages that defined <code>S3</code>/<code>S4</code> classes to store phylogenetic trees with their specific associated data, including <code>phylo4</code> and <code>phylo4d</code> defined in <em>phylobase</em> package, <code>obkdata</code> defined in <em>OutbreakTools</em> package, and <code>phyloseq</code> defined in <em>phyloseq</em> package. All these tree objects are also supported in <em>ggtree</em> and their specific annotation data can be used to annotate the tree in <em>ggtree</em>. Such compatibility of <em>ggtree</em> facilitates the integration of data and analysis results.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">&quot;treeio&quot;</span>)
<span class="kw">library</span>(<span class="st">&quot;ggtree&quot;</span>)

nwk &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;sample.nwk&quot;</span>, <span class="dt">package=</span><span class="st">&quot;treeio&quot;</span>)
tree &lt;-<span class="st"> </span><span class="kw">read.tree</span>(nwk)</code></pre>
<div id="basic-tree-visualization" class="section level3">
<h3><span class="header-section-number">4.2.1</span> Basic Tree Visualization</h3>
<p>The <a href="http://bioconductor.org/packages/ggtree">ggtree</a> package extends <a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a> <span class="citation">(Wickham <a href="#ref-wickham_ggplot2_2009" role="doc-biblioref">2009</a>)</span> package to support viewing phylogenetic tree.
It implements <code>geom_tree</code> layer for displaying phylogenetic tree, as shown below in Figure <a href="chapter4.html#fig:basicviz">4.1</a>A.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggplot</span>(tree, <span class="kw">aes</span>(x, y)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_tree</span>() <span class="op">+</span><span class="st"> </span><span class="kw">theme_tree</span>()</code></pre>
<p>The function, <code>ggtree</code>, was implemented as a short cut to visualize a tree, and it works exactly the same as shown above.</p>
<p><a href="http://bioconductor.org/packages/ggtree">ggtree</a> takes all the advantages of <a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a>. For example, we can change the color, size and type of the lines as we do with <a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a> (Figure <a href="chapter4.html#fig:basicviz">4.1</a>B).</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggtree</span>(tree, <span class="dt">color=</span><span class="st">&quot;firebrick&quot;</span>, <span class="dt">size=</span><span class="dv">1</span>, <span class="dt">linetype=</span><span class="st">&quot;dotted&quot;</span>)</code></pre>
<p>By default, the tree is viewed in ladderize form, user can set the parameter <em>ladderize = FALSE</em> to disable it (Figure <a href="chapter4.html#fig:basicviz">4.1</a>C).</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggtree</span>(tree, <span class="dt">ladderize=</span><span class="ot">FALSE</span>)</code></pre>
<p>The <em>branch.length</em> is used to scale the edge, user can set the parameter <em>branch.length = “none”</em> to only view the tree topology (cladogram, Figure <a href="chapter4.html#fig:basicviz">4.1</a>D) or other numerical variable to scale the tree (<em>e.g.</em> <em>d<sub>N</sub>/d<sub>S</sub></em>, see also in <a href="chapter5.html#chapter5">Chapter 5</a>).</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggtree</span>(tree, <span class="dt">branch.length=</span><span class="st">&quot;none&quot;</span>)</code></pre>


<div class="figure" style="text-align: center"><span id="fig:basicviz"></span>
<img src="treedata_files/figure-html/basicviz-1.png" alt="Basic tree visualization." width="1152" />
<p class="caption">
Figure 4.1: <strong>Basic tree visualization.</strong>
</p>
</div>
</div>
<div id="layouts-of-phylogenetic-tree" class="section level3">
<h3><span class="header-section-number">4.2.2</span> Layouts of phylogenetic tree</h3>
<p>Viewing phylogenetic with ggtree is quite simple, just pass the tree object to <em>ggtree</em> function. We have developed several types of layouts for tree presentation (Figure <a href="chapter4.html#fig:layout">4.2</a>), including rectangular (by default), slanted, circular, fan, unrooted (equal angle and daylight methods), time-scaled and 2-dimensional layouts.</p>
<p>Here are examples of visualizing a tree with different layouts:</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(ggtree)
<span class="kw">set.seed</span>(<span class="dv">2017-02-16</span>)
tree &lt;-<span class="st"> </span><span class="kw">rtree</span>(<span class="dv">50</span>)
<span class="kw">ggtree</span>(tree)
<span class="kw">ggtree</span>(tree, <span class="dt">layout=</span><span class="st">&quot;slanted&quot;</span>) 
<span class="kw">ggtree</span>(tree, <span class="dt">layout=</span><span class="st">&quot;circular&quot;</span>)
<span class="kw">ggtree</span>(tree, <span class="dt">layout=</span><span class="st">&quot;fan&quot;</span>, <span class="dt">open.angle=</span><span class="dv">120</span>)
<span class="kw">ggtree</span>(tree, <span class="dt">layout=</span><span class="st">&quot;equal_angle&quot;</span>)
<span class="kw">ggtree</span>(tree, <span class="dt">layout=</span><span class="st">&quot;daylight&quot;</span>)
<span class="kw">ggtree</span>(tree, <span class="dt">branch.length=</span><span class="st">&#39;none&#39;</span>)
<span class="kw">ggtree</span>(tree, <span class="dt">branch.length=</span><span class="st">&#39;none&#39;</span>, <span class="dt">layout=</span><span class="st">&#39;circular&#39;</span>)
<span class="kw">ggtree</span>(tree, <span class="dt">layout=</span><span class="st">&quot;daylight&quot;</span>, <span class="dt">branch.length =</span> <span class="st">&#39;none&#39;</span>)</code></pre>


<div class="figure"><span id="fig:layout"></span>
<img src="treedata_files/figure-html/layout-1.png" alt="Tree layouts. Phylogram: rectangular layout (A), slanted layout (B), circular layout (C) and fan layout (D). Unrooted: equal-angle method (E) and daylight method (F). Cladogram: rectangular layout (G), circular layout (H) and unrooted layout (I). Slanted and fan layouts for cladogram are also supported." width="768"  />
<p class="caption">
Figure 4.2: <strong>Tree layouts.</strong> Phylogram: rectangular layout (A), slanted layout (B), circular layout (C) and fan layout (D). Unrooted: equal-angle method (E) and daylight method (F). Cladogram: rectangular layout (G), circular layout (H) and unrooted layout (I). Slanted and fan layouts for cladogram are also supported.
</p>
</div>
<p>There are also other possible layouts that can be drawn by modifying
scales/coordination, for examples, <a href="https://github.com/GuangchuangYu/ggtree/issues/87">reverse label of time
scale</a>, <a href="https://groups.google.com/d/msg/bioc-ggtree/UoGQekWHIvw/ZswUUZKSGwAJ">repropotion
circular/fan tree</a>, <em>etc.</em>.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggtree</span>(tree) <span class="op">+</span><span class="st"> </span><span class="kw">scale_x_reverse</span>()
<span class="kw">ggtree</span>(tree) <span class="op">+</span><span class="st"> </span><span class="kw">coord_flip</span>()
<span class="kw">ggtree</span>(tree) <span class="op">+</span><span class="st"> </span><span class="kw">scale_x_reverse</span>() <span class="op">+</span><span class="st"> </span><span class="kw">coord_flip</span>()
<span class="kw">print</span>(<span class="kw">ggtree</span>(tree), <span class="dt">newpage=</span><span class="ot">TRUE</span>, <span class="dt">vp=</span>grid<span class="op">::</span><span class="kw">viewport</span>(<span class="dt">angle=</span><span class="op">-</span><span class="dv">30</span>, <span class="dt">width=</span>.<span class="dv">9</span>, <span class="dt">height=</span>.<span class="dv">9</span>))
<span class="kw">ggtree</span>(tree, <span class="dt">layout=</span><span class="st">&#39;slanted&#39;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">coord_flip</span>()
<span class="kw">ggtree</span>(tree, <span class="dt">layout=</span><span class="st">&#39;slanted&#39;</span>, <span class="dt">branch.length=</span><span class="st">&#39;none&#39;</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">coord_flip</span>() <span class="op">+</span><span class="st"> </span><span class="kw">scale_y_reverse</span>() <span class="op">+</span><span class="kw">scale_x_reverse</span>()
<span class="kw">ggtree</span>(tree, <span class="dt">layout=</span><span class="st">&#39;circular&#39;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">xlim</span>(<span class="op">-</span><span class="dv">10</span>, <span class="ot">NA</span>)
<span class="kw">ggtree</span>(tree) <span class="op">+</span><span class="st"> </span><span class="kw">scale_x_reverse</span>() <span class="op">+</span><span class="st"> </span><span class="kw">coord_polar</span>(<span class="dt">theta=</span><span class="st">&#39;y&#39;</span>)
<span class="kw">ggtree</span>(tree) <span class="op">+</span><span class="st"> </span><span class="kw">scale_x_reverse</span>(<span class="dt">limits=</span><span class="kw">c</span>(<span class="dv">10</span>, <span class="dv">0</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">coord_polar</span>(<span class="dt">theta=</span><span class="st">&#39;y&#39;</span>)</code></pre>


<div class="figure"><span id="fig:layout2"></span>
<img src="treedata_files/figure-html/layout2-1.png" alt="Derived Tree layouts. right-to-left rectangular layout (A), bottom-up rectangular layout (B), top-down rectangular layout (C), rotated rectangular layout (D), bottom-up slanted layout (E), top-down slanted layout (Cladogram) (F), XX layout (G), XX layout (H) and XX layout (I)." width="768" />
<p class="caption">
Figure 4.3: <strong>Derived Tree layouts.</strong> right-to-left rectangular layout (A), bottom-up rectangular layout (B), top-down rectangular layout (C), rotated rectangular layout (D), bottom-up slanted layout (E), top-down slanted layout (Cladogram) (F), XX layout (G), XX layout (H) and XX layout (I).
</p>
</div>
<p><strong>Phylogram.</strong> Layouts of <em>rectangular</em>, <em>slanted</em>, <em>circular</em> and <em>fan</em> are supported to visualize phylogram (by default, with branch length scaled) as demonstrated in Figure <a href="chapter4.html#fig:layout">4.2</a>A, B, C and D.</p>
<p><strong>Unrooted layout.</strong> Unrooted (also called ‘radial’) layout is supported by equal-angle and daylight algorithms, user can specify unrooted layout algorithm by passing “equal_angle” or “daylight” to <code>layout</code> parameter to visualize the tree. Equal-angle method was proposed by Christopher Meacham in <em>PLOTREE</em>, which was incorporated in <em>PHYLIP</em> <span class="citation">(Retief <a href="#ref-retief_phylogenetic_2000" role="doc-biblioref">2000</a>)</span>. This method starts from the root of the tree and allocates arcs of angle to each subtrees proportional to the number of tips in it. It iterates from root to tips and subdivides the angle allocated to a subtree into angles for its dependent subtrees. This method is fast and was implemented in many software packages. As shown in Figure <a href="chapter4.html#fig:layout">4.2</a>E, equal angle method has a drawback that tips are tend to be clustered together and leaving many spaces unused. The daylight method starts from an initial tree built by equal angle and iteratively improves it by successively going to each interior node and swinging subtrees so that the arcs of “daylight” are equal (Figure <a href="chapter4.html#fig:layout">4.2</a>F). This method was firstly implemented in <em>PAUP*</em> <span class="citation">(Wilgenbusch and Swofford <a href="#ref-wilgenbusch_inferring_2003" role="doc-biblioref">2003</a>)</span>.</p>
<p><strong>Cladogram.</strong> To visualize cladogram that without branch length scaling and only display the tree structure, <code>branch.length</code> is set to “none” and it works for all types of layouts (Figure <a href="chapter4.html#fig:layout">4.2</a>G, H and I).</p>
<p><strong>Time-scaled layout.</strong> For time-scaled tree, the most recent sampling date must be specified via the <code>mrsd</code> parameter and <em>ggtree</em> will scaled the tree by sampling (tip) and divergence (internal node) time, and a time scale axis will be displayed under the tree by default.</p>


<pre class="sourceCode r"><code class="sourceCode r">beast_file &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;examples/MCC_FluA_H3.tree&quot;</span>, 
                          <span class="dt">package=</span><span class="st">&quot;ggtree&quot;</span>)
beast_tree &lt;-<span class="st"> </span><span class="kw">read.beast</span>(beast_file)
<span class="kw">ggtree</span>(beast_tree, <span class="dt">mrsd=</span><span class="st">&quot;2013-01-01&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme_tree2</span>()</code></pre>
<div class="figure"><span id="fig:timescaled"></span>
<img src="treedata_files/figure-html/timescaled-1.png" alt="Time-scaled layout. The x-axis is the timescale (in units of year). The divergence time was inferred by BEAST using molecular clock model." width="672"  />
<p class="caption">
Figure 4.4: <strong>Time-scaled layout.</strong> The <em>x-axis</em> is the timescale (in units of year). The divergence time was inferred by <em>BEAST</em> using molecular clock model.
</p>
</div>
<p><strong>Two-dimensional tree layout.</strong> A two-dimensional tree is a projection of the phylogenetic tree in a space defined by the associated phenotype (numerical or categorical trait, on the <em>y</em>-axis) and tree branch scale (<em>e.g.</em>, evolutionary distance, divergent time, on the <em>x</em>-axis). The phenotype can be a measure of certain biological characteristics of the taxa and hypothetical ancestors in the tree. This is a new layout we proposed in <em>ggtree</em>, which is useful to track the virus phenotypes or other behaviors (<em>y</em>-axis) changing with the virus evolution (<em>x</em>-axis). In fact, the analysis of phenotypes or genotypes over evolutionary time have been widely used for study influenza virus evolution <span class="citation">(Neher et al. <a href="#ref-neher_prediction_2016" role="doc-biblioref">2016</a>)</span>, though such analysis diagrams are not tree-like, <em>i.e.</em>, no connection between data points, unlike our two-dimensional tree layout that connect data points with the corresponding tree branches. Therefore, this new layout we provided will make such data analysis easier and more scalable for large sequence data sets of influenza viruses.</p>
<p>In this example, we used the previous time-scaled tree of H3 human and swine influenza viruses (Figure <a href="chapter4.html#fig:timescaled">4.4</a>; data published in <span class="citation">(Liang et al. <a href="#ref-liang_expansion_2014" role="doc-biblioref">2014</a>)</span>) and scaled the <em>y</em>-axis based on the predicted <em>N</em>-linked glycosylation sites (NLG) for each of the taxon and ancestral sequences of hemagglutinin proteins. The NLG sites were predicted using <em>NetNGlyc</em> 1.0 Server<a href="#fn12" class="footnote-ref" id="fnref12"><sup>12</sup></a>. To scaled the <em>y</em>-axis, the parameter <code>yscale</code> in the <code>ggtree()</code> function is set to a numerical or categorical variable. If <code>yscale</code> is a categorical variable as in this example, users should specify how the categories are to be mapped to numerical values via the <code>yscale_mapping</code> variables.</p>


<pre class="sourceCode r"><code class="sourceCode r">NAG_file &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;examples/NAG_inHA1.txt&quot;</span>, <span class="dt">package=</span><span class="st">&quot;ggtree&quot;</span>)

NAG.df &lt;-<span class="st"> </span><span class="kw">read.table</span>(NAG_file, <span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">header=</span><span class="ot">FALSE</span>, 
                     <span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)
NAG &lt;-<span class="st"> </span>NAG.df[,<span class="dv">2</span>]
<span class="kw">names</span>(NAG) &lt;-<span class="st"> </span>NAG.df[,<span class="dv">1</span>]

<span class="co">## separate the tree by host species</span>
tip &lt;-<span class="st"> </span><span class="kw">get.tree</span>(beast_tree)<span class="op">$</span>tip.label
beast_tree &lt;-<span class="st"> </span><span class="kw">groupOTU</span>(beast_tree, tip[<span class="kw">grep</span>(<span class="st">&quot;Swine&quot;</span>, tip)], 
                       <span class="dt">group_name =</span> <span class="st">&quot;host&quot;</span>)

p &lt;-<span class="st"> </span><span class="kw">ggtree</span>(beast_tree, <span class="kw">aes</span>(<span class="dt">color=</span>host), <span class="dt">mrsd=</span><span class="st">&quot;2013-01-01&quot;</span>, 
            <span class="dt">yscale =</span> <span class="st">&quot;label&quot;</span>, <span class="dt">yscale_mapping =</span> NAG) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">theme_classic</span>() <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&#39;none&#39;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">scale_color_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(<span class="st">&quot;blue&quot;</span>, <span class="st">&quot;red&quot;</span>), 
                     <span class="dt">labels=</span><span class="kw">c</span>(<span class="st">&quot;human&quot;</span>, <span class="st">&quot;swine&quot;</span>)) <span class="op">+</span>
<span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Number of predicted N-linked glycoslyation sites&quot;</span>)

<span class="co">## (optional) add more annotations to help interpretation</span>
p <span class="op">+</span><span class="st"> </span><span class="kw">geom_nodepoint</span>(<span class="dt">color=</span><span class="st">&quot;grey&quot;</span>, <span class="dt">size=</span><span class="dv">3</span>, <span class="dt">alpha=</span>.<span class="dv">8</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_rootpoint</span>(<span class="dt">color=</span><span class="st">&quot;black&quot;</span>, <span class="dt">size=</span><span class="dv">3</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_tippoint</span>(<span class="dt">size=</span><span class="dv">3</span>, <span class="dt">alpha=</span>.<span class="dv">5</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">annotate</span>(<span class="st">&quot;point&quot;</span>, <span class="dv">1992</span>, <span class="fl">5.6</span>, <span class="dt">size=</span><span class="dv">3</span>, <span class="dt">color=</span><span class="st">&quot;black&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">annotate</span>(<span class="st">&quot;point&quot;</span>, <span class="dv">1992</span>, <span class="fl">5.4</span>, <span class="dt">size=</span><span class="dv">3</span>, <span class="dt">color=</span><span class="st">&quot;grey&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">annotate</span>(<span class="st">&quot;point&quot;</span>, <span class="fl">1991.6</span>, <span class="fl">5.2</span>, <span class="dt">size=</span><span class="dv">3</span>, <span class="dt">color=</span><span class="st">&quot;blue&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">annotate</span>(<span class="st">&quot;point&quot;</span>, <span class="dv">1992</span>, <span class="fl">5.2</span>, <span class="dt">size=</span><span class="dv">3</span>, <span class="dt">color=</span><span class="st">&quot;red&quot;</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">annotate</span>(<span class="st">&quot;text&quot;</span>, <span class="fl">1992.3</span>, <span class="fl">5.6</span>, <span class="dt">hjust=</span><span class="dv">0</span>, <span class="dt">size=</span><span class="dv">4</span>, <span class="dt">label=</span><span class="st">&quot;Root node&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">annotate</span>(<span class="st">&quot;text&quot;</span>, <span class="fl">1992.3</span>, <span class="fl">5.4</span>, <span class="dt">hjust=</span><span class="dv">0</span>, <span class="dt">size=</span><span class="dv">4</span>, 
           <span class="dt">label=</span><span class="st">&quot;Internal nodes&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">annotate</span>(<span class="st">&quot;text&quot;</span>, <span class="fl">1992.3</span>, <span class="fl">5.2</span>, <span class="dt">hjust=</span><span class="dv">0</span>, <span class="dt">size=</span><span class="dv">4</span>,
           <span class="dt">label=</span><span class="st">&quot;Tip nodes (blue: human; red: swine)&quot;</span>)</code></pre>
<div class="figure"><span id="fig:2d"></span>
<img src="treedata_files/figure-html/2d-1.png" alt="Two-dimensional tree layout. The trunk and other branches highlighted in red (for swine) and blue (for human). The x-axis is scaled to the branch length (in units of year) of the time-scaled tree. The y-axis is scaled to the node attribute variable, in this case the number of predicted N-linked glycosylation site (NLG) on the hemagglutinin protein. Colored circles indicate the different types of tree nodes. Note that nodes assigned the same x- (temporal) and y- (NLG) coordinates are superimposed in this representation and appear as one node, which is shaded based on the colors of all the nodes at that point." width="816"  />
<p class="caption">
Figure 4.5: <strong>Two-dimensional tree layout.</strong> The trunk and other branches highlighted in red (for swine) and blue (for human). The <em>x</em>-axis is scaled to the branch length (in units of year) of the time-scaled tree. The <em>y</em>-axis is scaled to the node attribute variable, in this case the number of predicted <em>N</em>-linked glycosylation site (NLG) on the hemagglutinin protein. Colored circles indicate the different types of tree nodes. Note that nodes assigned the same <em>x</em>- (temporal) and <em>y</em>- (NLG) coordinates are superimposed in this representation and appear as one node, which is shaded based on the colors of all the nodes at that point.
</p>
</div>
<p>As shown in Figure <a href="chapter4.html#fig:2d">4.5</a>, two-dimensional tree good at visualizing the change of phenotype over the evolution in the phylogenetic tree. In this example, it is shown that H3 gene of human influenza A virus maintained high level of <em>N</em>-linked glycosylation sites (n=8 to 9) over last two decades and dropped significantly to 5 or 6 in a separate viral lineage transmitted to swine populations and established there. It was indeed hypothesized that the human influenza virus with high level of glycosylation on the viral hemagglutinin protein provides better shielding effect to protect the antigenic sites from exposure to the herd immunity, and thus has selective advantage in human populations that maintain high level of herd immunity against the circulating human influenza virus strains. For the viral lineage that newly jumped across the species barrier and transmitted to swine population, the shielding effect of the high-level surface glycan oppositely impose selective disadvantage because the receptor-binding domain may also be shielded which greatly affect the viral fitness of the lineage that newly adapted to a new host species.</p>
</div>
</div>
<div id="displaying-tree-components" class="section level2">
<h2><span class="header-section-number">4.3</span> Displaying Tree Components</h2>
<div id="displaying-tree-scale-evolution-distance" class="section level3">
<h3><span class="header-section-number">4.3.1</span> Displaying tree scale (evolution distance)</h3>
<p>To show tree scale, user can use <code>geom_treescale()</code> layer.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggtree</span>(tree) <span class="op">+</span><span class="st"> </span><span class="kw">geom_treescale</span>()</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-62-1.png" width="384" style="display: block; margin: auto;" /></p>
<p><code>geom_treescale()</code> supports the following parameters:</p>
<ul>
<li><em>x</em> and <em>y</em> for tree scale position</li>
<li><em>width</em> for the length of the tree scale</li>
<li><em>fontsize</em> for the size of the text</li>
<li><em>linesize</em> for the size of the line</li>
<li><em>offset</em> for relative position of the line and the text</li>
<li><em>color</em> for color of the tree scale</li>
</ul>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggtree</span>(tree) <span class="op">+</span><span class="st"> </span><span class="kw">geom_treescale</span>(<span class="dt">x=</span><span class="dv">0</span>, <span class="dt">y=</span><span class="dv">45</span>, <span class="dt">width=</span><span class="dv">1</span>, <span class="dt">color=</span><span class="st">&#39;red&#39;</span>)
<span class="kw">ggtree</span>(tree) <span class="op">+</span><span class="st"> </span><span class="kw">geom_treescale</span>(<span class="dt">fontsize=</span><span class="dv">6</span>, <span class="dt">linesize=</span><span class="dv">2</span>, <span class="dt">offset=</span><span class="dv">1</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-64-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>We can also use <code>theme_tree2()</code> to display the tree scale by adding <em>x axis</em>.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggtree</span>(tree) <span class="op">+</span><span class="st"> </span><span class="kw">theme_tree2</span>()</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-65-1.png" width="288" style="display: block; margin: auto;" /></p>
<p>Tree scale is not restricted to evolution distance, <a href="http://bioconductor.org/packages/treeio">treeio</a> can re-scale the tree with other numerical variable (details described in <a href="chapter2.html#rescale-treeio">session 2.3</a>).</p>
</div>
<div id="displaying-nodestips" class="section level3">
<h3><span class="header-section-number">4.3.2</span> Displaying nodes/tips</h3>
<p>Showing all the internal nodes and tips in the tree can be done by adding a layer of points using <code>geom_nodepoint</code>, <code>geom_tippoint</code> or <code>geom_point</code>.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggtree</span>(tree) <span class="op">+</span><span class="st"> </span><span class="kw">geom_point</span>(<span class="kw">aes</span>(<span class="dt">shape=</span>isTip, <span class="dt">color=</span>isTip), <span class="dt">size=</span><span class="dv">3</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-66-1.png" width="288" style="display: block; margin: auto;" /></p>
<pre class="sourceCode r"><code class="sourceCode r">p &lt;-<span class="st"> </span><span class="kw">ggtree</span>(tree) <span class="op">+</span><span class="st"> </span><span class="kw">geom_nodepoint</span>(<span class="dt">color=</span><span class="st">&quot;#b5e521&quot;</span>, <span class="dt">alpha=</span><span class="dv">1</span><span class="op">/</span><span class="dv">4</span>, <span class="dt">size=</span><span class="dv">10</span>)
p <span class="op">+</span><span class="st"> </span><span class="kw">geom_tippoint</span>(<span class="dt">color=</span><span class="st">&quot;#FDAC4F&quot;</span>, <span class="dt">shape=</span><span class="dv">8</span>, <span class="dt">size=</span><span class="dv">3</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-67-1.png" width="288" style="display: block; margin: auto;" /></p>
</div>
<div id="displaying-labels" class="section level3">
<h3><span class="header-section-number">4.3.3</span> Displaying labels</h3>
<p>Users can use <code>geom_text</code> or <code>geom_label</code> to display the node (if available) and tip labels simultaneously or <code>geom_tiplab</code> to only display tip labels:</p>
<pre class="sourceCode r"><code class="sourceCode r">p <span class="op">+</span><span class="st"> </span><span class="kw">geom_tiplab</span>(<span class="dt">size=</span><span class="dv">3</span>, <span class="dt">color=</span><span class="st">&quot;purple&quot;</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-68-1.png" width="288" style="display: block; margin: auto;" /></p>
<p><code>geom_tiplab</code> not only supports using <em>text</em> or <em>label</em> geom to display labels,
it also supports <em>image</em> geom to label tip with image files. A corresponding
geom, <code>geom_nodelab</code> is also provided for displaying node labels. For details of
label nodes with images, please refer to
the vignette,
<a href="https://guangchuangyu.github.io/software/ggtree/vignettes/ggtree-ggimage.html">Annotating phylogenetic tree with images</a>.</p>
<p>For <em>circular</em> and <em>unrooted</em> layout, <a href="http://bioconductor.org/packages/ggtree">ggtree</a> supports rotating node labels according to the angles of the branches.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggtree</span>(tree, <span class="dt">layout=</span><span class="st">&quot;circular&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">geom_tiplab</span>(<span class="kw">aes</span>(<span class="dt">angle=</span>angle), <span class="dt">color=</span><span class="st">&#39;blue&#39;</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-69-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>To make it more readable for human eye, <a href="http://bioconductor.org/packages/ggtree">ggtree</a> provides a <code>geom_tiplab2</code> for <code>circular</code> layout (see post <a href="https://groups.google.com/forum/?utm_medium=email&amp;utm_source=footer#!topic/bioc-ggtree/o35PV3iHO-0">1</a> and <a href="https://groups.google.com/forum/#!topic/bioc-ggtree/p42R5l8J-14">2</a>).</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggtree</span>(tree, <span class="dt">layout=</span><span class="st">&quot;circular&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">geom_tiplab2</span>(<span class="dt">color=</span><span class="st">&#39;blue&#39;</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-70-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>By default, the positions are based on the node positions, we can change them to based on the middle of the branch/edge.</p>
<pre class="sourceCode r"><code class="sourceCode r">p <span class="op">+</span><span class="st"> </span><span class="kw">geom_tiplab</span>(<span class="kw">aes</span>(<span class="dt">x=</span>branch), <span class="dt">size=</span><span class="dv">3</span>, <span class="dt">color=</span><span class="st">&quot;purple&quot;</span>, <span class="dt">vjust=</span><span class="op">-</span><span class="fl">0.3</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-71-1.png" width="384" style="display: block; margin: auto;" /></p>
<p>Based on the middle of branch is very useful when annotating transition from parent node to child node.</p>
</div>
<div id="color-tree" class="section level3">
<h3><span class="header-section-number">4.3.4</span> Color tree</h3>
<p>In <code>ggtree</code>, coloring phylogenetic tree is easy, by using <code>aes(color=VAR)</code> to map the color of tree based on a specific variable (numeric and category are both supported).</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggtree</span>(beast_tree, <span class="kw">aes</span>(<span class="dt">color=</span>rate)) <span class="op">+</span>
<span class="st">    </span><span class="kw">scale_color_continuous</span>(<span class="dt">low=</span><span class="st">&#39;darkgreen&#39;</span>, <span class="dt">high=</span><span class="st">&#39;red&#39;</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&quot;right&quot;</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-72-1.png" width="480" /></p>
<p>User can use any feature (if available), including clade posterior and <em>dN/dS</em> <em>etc.</em>, to scale the color of the tree.</p>
</div>
<div id="rescale-tree" class="section level3">
<h3><span class="header-section-number">4.3.5</span> Rescale tree</h3>
<p>Most of the phylogenetic trees are scaled by evolutionary distance (substitution/site). In <code>ggtree</code>, users can re-scale a phylogenetic tree by any numerical variable inferred by evolutionary analysis (e.g. <em>dN/dS</em>).</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">&quot;treeio&quot;</span>)
beast_file &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;examples/MCC_FluA_H3.tree&quot;</span>, <span class="dt">package=</span><span class="st">&quot;ggtree&quot;</span>)
beast_tree &lt;-<span class="st"> </span><span class="kw">read.beast</span>(beast_file)
beast_tree</code></pre>
<pre><code>## &#39;treedata&#39; S4 object that stored information of
##  &#39;C:/Users/YGC/Library/R/library/ggtree/examples/MCC_FluA_H3.tree&#39;.
## 
## ...@ phylo: 
## Phylogenetic tree with 76 tips and 75 internal nodes.
## 
## Tip labels:
##  A/Hokkaido/30-1-a/2013, A/New_York/334/2004, A/New_York/463/2005, A/New_York/452/1999, A/New_York/238/2005, A/New_York/523/1998, ...
## 
## Rooted; includes branch lengths.
## 
## with the following features available:
##  &#39;height&#39;,   &#39;height_0.95_HPD&#39;,  &#39;height_median&#39;,
##  &#39;height_range&#39;, &#39;length&#39;,   &#39;length_0.95_HPD&#39;,
##  &#39;length_median&#39;,    &#39;length_range&#39;, &#39;posterior&#39;,    &#39;rate&#39;,
##  &#39;rate_0.95_HPD&#39;,    &#39;rate_median&#39;,  &#39;rate_range&#39;.</code></pre>
<pre class="sourceCode r"><code class="sourceCode r">p1 &lt;-<span class="st"> </span><span class="kw">ggtree</span>(beast_tree, <span class="dt">mrsd=</span><span class="st">&#39;2013-01-01&#39;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme_tree2</span>() <span class="op">+</span>
<span class="st">    </span><span class="kw">ggtitle</span>(<span class="st">&quot;Divergence time&quot;</span>)
p2 &lt;-<span class="st"> </span><span class="kw">ggtree</span>(beast_tree, <span class="dt">branch.length=</span><span class="st">&#39;rate&#39;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme_tree2</span>() <span class="op">+</span>
<span class="st">    </span><span class="kw">ggtitle</span>(<span class="st">&quot;Substitution rate&quot;</span>)

<span class="kw">library</span>(cowplot)
<span class="kw">plot_grid</span>(p1, p2, <span class="dt">ncol=</span><span class="dv">2</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-73-1.png" width="960" /></p>
<pre class="sourceCode r"><code class="sourceCode r">mlcfile &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata/PAML_Codeml&quot;</span>, <span class="st">&quot;mlc&quot;</span>, <span class="dt">package=</span><span class="st">&quot;treeio&quot;</span>)
mlc_tree &lt;-<span class="st"> </span><span class="kw">read.codeml_mlc</span>(mlcfile)
p1 &lt;-<span class="st"> </span><span class="kw">ggtree</span>(mlc_tree) <span class="op">+</span><span class="st"> </span><span class="kw">theme_tree2</span>() <span class="op">+</span>
<span class="st">    </span><span class="kw">ggtitle</span>(<span class="st">&quot;nucleotide substitutions per codon&quot;</span>)
p2 &lt;-<span class="st"> </span><span class="kw">ggtree</span>(mlc_tree, <span class="dt">branch.length=</span><span class="st">&#39;dN_vs_dS&#39;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme_tree2</span>() <span class="op">+</span>
<span class="st">    </span><span class="kw">ggtitle</span>(<span class="st">&quot;dN/dS tree&quot;</span>)
<span class="kw">plot_grid</span>(p1, p2, <span class="dt">ncol=</span><span class="dv">2</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-74-1.png" width="960" /></p>
<p>In addition to specify <code>branch.length</code> in tree visualization, users can change branch length stored in tree object by using <code>rescale_tree</code> function.</p>
<pre class="sourceCode r"><code class="sourceCode r">beast_tree2 &lt;-<span class="st"> </span><span class="kw">rescale_tree</span>(beast_tree, <span class="dt">branch.length=</span><span class="st">&#39;rate&#39;</span>)
<span class="kw">ggtree</span>(beast_tree2) <span class="op">+</span><span class="st"> </span><span class="kw">theme_tree2</span>()</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-75-1.png" width="672" /></p>
</div>
<div id="modify-compenents-of-the-tree-view" class="section level3">
<h3><span class="header-section-number">4.3.6</span> Modify compenents of the tree view</h3>
<p>scale &amp; theme</p>
<p><code>theme_tree()</code> defined a totally blank canvas, while <em><code>theme_tree2()</code></em> adds
phylogenetic distance (via x-axis). These two themes all accept a parameter of
<em><code>bgcolor</code></em> that defined the background color. Users can pass
any <a href="http://ggplot2.tidyverse.org/reference/theme.html">theme components</a> to the
<code>theme_tree()</code> function to modify them.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggtree</span>(<span class="kw">rtree</span>(<span class="dv">30</span>), <span class="dt">color=</span><span class="st">&quot;red&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme_tree</span>(<span class="st">&quot;steelblue&quot;</span>)
<span class="kw">ggtree</span>(<span class="kw">rtree</span>(<span class="dv">20</span>), <span class="dt">color=</span><span class="st">&quot;white&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme_tree</span>(<span class="st">&quot;black&quot;</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-77-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="visualize-a-list-of-trees" class="section level2">
<h2><span class="header-section-number">4.4</span> Visualize a list of trees</h2>
<p><code>ggtree</code> supports <code>multiPhylo</code> object and a list of trees can be viewed simultaneously.</p>
<pre class="sourceCode r"><code class="sourceCode r">trees &lt;-<span class="st"> </span><span class="kw">lapply</span>(<span class="kw">c</span>(<span class="dv">10</span>, <span class="dv">20</span>, <span class="dv">40</span>), rtree)
<span class="kw">class</span>(trees) &lt;-<span class="st"> &quot;multiPhylo&quot;</span>
<span class="kw">ggtree</span>(trees) <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(<span class="op">~</span>.id, <span class="dt">scale=</span><span class="st">&quot;free&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">geom_tiplab</span>()</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-78-1.png" width="1152" /></p>
<p>One hundred bootstrap trees can also be view simultaneously.</p>
<pre class="sourceCode r"><code class="sourceCode r">btrees &lt;-<span class="st"> </span><span class="kw">read.tree</span>(<span class="kw">system.file</span>(<span class="st">&quot;extdata/RAxML&quot;</span>, <span class="st">&quot;RAxML_bootstrap.H3&quot;</span>, <span class="dt">package=</span><span class="st">&quot;treeio&quot;</span>))
<span class="kw">ggtree</span>(btrees) <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(<span class="op">~</span>.id, <span class="dt">ncol=</span><span class="dv">10</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-79-1.png" width="1920" /></p>
<p>Another way to view the bootstrap trees is to merge them together to form a density tree. We can add a layer of the best tree on the top of the density tree.</p>
<pre class="sourceCode r"><code class="sourceCode r">p &lt;-<span class="st"> </span><span class="kw">ggtree</span>(btrees, <span class="dt">layout=</span><span class="st">&quot;rectangular&quot;</span>, <span class="dt">color=</span><span class="st">&quot;lightblue&quot;</span>, <span class="dt">alpha=</span>.<span class="dv">3</span>)

best_tree &lt;-<span class="st"> </span><span class="kw">read.tree</span>(<span class="kw">system.file</span>(<span class="st">&quot;extdata/RAxML&quot;</span>, <span class="st">&quot;RAxML_bipartitionsBranchLabels.H3&quot;</span>, <span class="dt">package=</span><span class="st">&quot;treeio&quot;</span>))
df &lt;-<span class="st"> </span><span class="kw">fortify</span>(best_tree, <span class="dt">branch.length=</span><span class="st">&#39;none&#39;</span>)
p<span class="op">+</span><span class="kw">geom_tree</span>(<span class="dt">data=</span>df, <span class="dt">color=</span><span class="st">&#39;firebrick&#39;</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-80-1.png" width="672" /></p>
</div>
<div id="summary-3" class="section level2">
<h2><span class="header-section-number">4.5</span> Summary</h2>

</div>
</div>
<h3>References</h3>
<div id="refs" class="references">
<div id="ref-page_visualizing_2002">
<p>Page, Roderic D. M. 2002. “Visualizing Phylogenetic Trees Using TreeView.” <em>Current Protocols in Bioinformatics</em> Chapter 6 (August): Unit 6.2. <a href="https://doi.org/10.1002/0471250953.bi0602s01">https://doi.org/10.1002/0471250953.bi0602s01</a>.</p>
</div>
<div id="ref-chevenet_treedyn:_2006">
<p>Chevenet, François, Christine Brun, Anne-Laure Bañuls, Bernard Jacq, and Richard Christen. 2006. “TreeDyn: Towards Dynamic Graphics and Annotations for Analyses of Trees.” <em>BMC Bioinformatics</em> 7 (October): 439. <a href="https://doi.org/10.1186/1471-2105-7-439">https://doi.org/10.1186/1471-2105-7-439</a>.</p>
</div>
<div id="ref-huson_dendroscope_2012">
<p>Huson, Daniel H., and Celine Scornavacca. 2012. “Dendroscope 3: An Interactive Tool for Rooted Phylogenetic Trees and Networks.” <em>Systematic Biology</em> 61 (6): 1061–7. <a href="https://doi.org/10.1093/sysbio/sys062">https://doi.org/10.1093/sysbio/sys062</a>.</p>
</div>
<div id="ref-he_evolview_2016">
<p>He, Zilong, Huangkai Zhang, Shenghan Gao, Martin J. Lercher, Wei-Hua Chen, and Songnian Hu. 2016. “Evolview V2: An Online Visualization and Management Tool for Customized and Annotated Phylogenetic Trees.” <em>Nucleic Acids Research</em> 44 (W1): W236–241. <a href="https://doi.org/10.1093/nar/gkw370">https://doi.org/10.1093/nar/gkw370</a>.</p>
</div>
<div id="ref-letunic_interactive_2007">
<p>Letunic, Ivica, and Peer Bork. 2007. “Interactive Tree of Life (iTOL): An Online Tool for Phylogenetic Tree Display and Annotation.” <em>Bioinformatics</em> 23 (1): 127–28. <a href="https://doi.org/10.1093/bioinformatics/btl529">https://doi.org/10.1093/bioinformatics/btl529</a>.</p>
</div>
<div id="ref-rstats">
<p>R Core Team. 2016. <em>R: A Language and Environment for Statistical Computing</em>. Vienna, Austria: R Foundation for Statistical Computing. <a href="https://www.R-project.org/">https://www.R-project.org/</a>.</p>
</div>
<div id="ref-gentleman_bioconductor_2004">
<p>Gentleman, Robert C, Vincent J Carey, Douglas M Bates, Ben Bolstad, Marcel Dettling, Sandrine Dudoit, Byron Ellis, et al. 2004. “Bioconductor: Open Software Development for Computational Biology and Bioinformatics.” <em>Genome Biology</em> 5 (10): R80. <a href="https://doi.org/10.1186/gb-2004-5-10-r80">https://doi.org/10.1186/gb-2004-5-10-r80</a>.</p>
</div>
<div id="ref-wickham_ggplot2_2016">
<p>Wickham, Hadley. 2016. <em>Ggplot2: Elegant Graphics for Data Analysis</em>. Springer. <a href="http://ggplot2.org">http://ggplot2.org</a>.</p>
</div>
<div id="ref-wilkinson_grammar_2005">
<p>Wilkinson, Leland, D. Wills, D. Rope, A. Norton, and R. Dubbs. 2005. <em>The Grammar of Graphics</em>. 2nd edition. New York: Springer.</p>
</div>
<div id="ref-paradis_ape_2004">
<p>Paradis, Emmanuel, Julien Claude, and Korbinian Strimmer. 2004. “APE: Analyses of Phylogenetics and Evolution in R Language.” <em>Bioinformatics</em> 20 (2): 289–90. <a href="https://doi.org/10.1093/bioinformatics/btg412">https://doi.org/10.1093/bioinformatics/btg412</a>.</p>
</div>
<div id="ref-revell_phytools_2012">
<p>Revell, Liam J. 2012. “Phytools: An R Package for Phylogenetic Comparative Biology (and Other Things).” <em>Methods in Ecology and Evolution</em> 3 (2): 217–23. <a href="https://doi.org/10.1111/j.2041-210X.2011.00169.x">https://doi.org/10.1111/j.2041-210X.2011.00169.x</a>.</p>
</div>
<div id="ref-jombart_outbreaktools_2014">
<p>Jombart, Thibaut, David M. Aanensen, Marc Baguelin, Paul Birrell, Simon Cauchemez, Anton Camacho, Caroline Colijn, et al. 2014. “OutbreakTools: A New Platform for Disease Outbreak Analysis Using the R Software.” <em>Epidemics</em> 7 (June): 28–34. <a href="https://doi.org/10.1016/j.epidem.2014.04.003">https://doi.org/10.1016/j.epidem.2014.04.003</a>.</p>
</div>
<div id="ref-mcmurdie_phyloseq_2013">
<p>McMurdie, Paul J., and Susan Holmes. 2013. “Phyloseq: An R Package for Reproducible Interactive Analysis and Graphics of Microbiome Census Data.” <em>PLoS ONE</em> 8 (4): e61217. <a href="https://doi.org/10.1371/journal.pone.0061217">https://doi.org/10.1371/journal.pone.0061217</a>.</p>
</div>
<div id="ref-lam_phylodynamics_2012">
<p>Lam, Tommy Tsan-Yuk, Chung-Chau Hon, Philippe Lemey, Oliver G. Pybus, Mang Shi, Hein Min Tun, Jun Li, Jingwei Jiang, Edward C. Holmes, and Frederick Chi-Ching Leung. 2012. “Phylodynamics of H5N1 Avian Influenza Virus in Indonesia.” <em>Molecular Ecology</em> 21 (12): 3062–77. <a href="https://doi.org/10.1111/j.1365-294X.2012.05577.x">https://doi.org/10.1111/j.1365-294X.2012.05577.x</a>.</p>
</div>
<div id="ref-wickham_ggplot2_2009">
<p>Wickham, Hadley. 2009. <em>Ggplot2: Elegant Graphics for Data Analysis</em>. 1st ed. Springer.</p>
</div>
<div id="ref-retief_phylogenetic_2000">
<p>Retief, J. D. 2000. “Phylogenetic Analysis Using PHYLIP.” <em>Methods in Molecular Biology (Clifton, N.J.)</em> 132: 243–58.</p>
</div>
<div id="ref-wilgenbusch_inferring_2003">
<p>Wilgenbusch, James C., and David Swofford. 2003. “Inferring Evolutionary Trees with PAUP*.” <em>Current Protocols in Bioinformatics</em> Chapter 6 (February): Unit 6.4. <a href="https://doi.org/10.1002/0471250953.bi0604s00">https://doi.org/10.1002/0471250953.bi0604s00</a>.</p>
</div>
<div id="ref-neher_prediction_2016">
<p>Neher, Richard A., Trevor Bedford, Rodney S. Daniels, Colin A. Russell, and Boris I. Shraiman. 2016. “Prediction, Dynamics, and Visualization of Antigenic Phenotypes of Seasonal Influenza Viruses.” <em>Proceedings of the National Academy of Sciences</em> 113 (12): E1701–E1709. <a href="https://doi.org/10.1073/pnas.1525578113">https://doi.org/10.1073/pnas.1525578113</a>.</p>
</div>
<div id="ref-liang_expansion_2014">
<p>Liang, Huyi, Tommy Tsan-Yuk Lam, Xiaohui Fan, Xinchun Chen, Yu Zeng, Ji Zhou, Lian Duan, et al. 2014. “Expansion of Genotypic Diversity and Establishment of 2009 H1N1 Pandemic-Origin Internal Genes in Pigs in China.” <em>Journal of Virology</em>, July, JVI.01327–14. <a href="https://doi.org/10.1128/JVI.01327-14">https://doi.org/10.1128/JVI.01327-14</a>.</p>
</div>
</div>
<div class="footnotes">
<hr />
<ol start="11">
<li id="fn11"><p><a href="http://tree.bio.ed.ac.uk/software/figtree/" class="uri">http://tree.bio.ed.ac.uk/software/figtree/</a><a href="chapter4.html#fnref11" class="footnote-back">↩</a></p></li>
<li id="fn12"><p><a href="http://www.cbs.dtu.dk/services/NetNGlyc/" class="uri">http://www.cbs.dtu.dk/services/NetNGlyc/</a><a href="chapter4.html#fnref12" class="footnote-back">↩</a></p></li>
</ol>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="chapter3.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="chapter5.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"download": ["treedata.pdf"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
