<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 11 Gallery of reproducible examples | Data Integration, Manipulation and Visualization of Phylogenetic Trees</title>
  <meta name="description" content="Chapter 11 Gallery of reproducible examples | Data Integration, Manipulation and Visualization of Phylogenetic Trees" />
  <meta name="generator" content="bookdown 0.14 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 11 Gallery of reproducible examples | Data Integration, Manipulation and Visualization of Phylogenetic Trees" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 11 Gallery of reproducible examples | Data Integration, Manipulation and Visualization of Phylogenetic Trees" />
  
  
  

<meta name="author" content="Guangchuang Yu" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="chapter10.html"/>
<link rel="next" href="chapter12.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<script src="libs/htmlwidgets-1.3/htmlwidgets.js"></script>
<script src="libs/pathseg-1.0.2/pathseg.js"></script>
<script src="libs/comic-0.961/comic.min.js"></script>
<script src="libs/comicR-binding-0.2/comicR.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="part"><span><b>Part I: Tree data input, output and manipulation</b></span></li>
<li class="chapter" data-level="1" data-path="chapter1.html"><a href="chapter1.html"><i class="fa fa-check"></i><b>1</b> Importing Tree with Data</a><ul>
<li class="chapter" data-level="1.1" data-path="chapter1.html"><a href="chapter1.html#pc"><i class="fa fa-check"></i><b>1.1</b> Overview of Phylogenetic Tree Construction</a></li>
<li class="chapter" data-level="1.2" data-path="chapter1.html"><a href="chapter1.html#format"><i class="fa fa-check"></i><b>1.2</b> Phylogenetic Tree Formats</a><ul>
<li class="chapter" data-level="1.2.1" data-path="chapter1.html"><a href="chapter1.html#newick-tree-format"><i class="fa fa-check"></i><b>1.2.1</b> Newick tree format</a></li>
<li class="chapter" data-level="1.2.2" data-path="chapter1.html"><a href="chapter1.html#nexus-tree-format"><i class="fa fa-check"></i><b>1.2.2</b> NEXUS tree format</a></li>
<li class="chapter" data-level="1.2.3" data-path="chapter1.html"><a href="chapter1.html#nhxtext"><i class="fa fa-check"></i><b>1.2.3</b> New Hampshire eXtended format</a></li>
<li class="chapter" data-level="1.2.4" data-path="chapter1.html"><a href="chapter1.html#jplace-format"><i class="fa fa-check"></i><b>1.2.4</b> Jplace format</a></li>
<li class="chapter" data-level="1.2.5" data-path="chapter1.html"><a href="chapter1.html#software-outputs"><i class="fa fa-check"></i><b>1.2.5</b> Software outputs</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="chapter1.html"><a href="chapter1.html#getting-tree-data-with-treeio"><i class="fa fa-check"></i><b>1.3</b> Getting Tree Data with <em>treeio</em></a><ul>
<li class="chapter" data-level="1.3.1" data-path="chapter1.html"><a href="chapter1.html#overview-of-treeio"><i class="fa fa-check"></i><b>1.3.1</b> Overview of <em>treeio</em></a></li>
<li class="chapter" data-level="1.3.2" data-path="chapter1.html"><a href="chapter1.html#function-demonstration"><i class="fa fa-check"></i><b>1.3.2</b> Function Demonstration</a></li>
<li class="chapter" data-level="1.3.3" data-path="chapter1.html"><a href="chapter1.html#as-treedata"><i class="fa fa-check"></i><b>1.3.3</b> Converting other tree object to <code>treedata</code> object</a></li>
<li class="chapter" data-level="1.3.4" data-path="chapter1.html"><a href="chapter1.html#get-treedata-data"><i class="fa fa-check"></i><b>1.3.4</b> Getting information from <code>treedata</code> object</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="chapter1.html"><a href="chapter1.html#summary"><i class="fa fa-check"></i><b>1.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="chapter2.html"><a href="chapter2.html"><i class="fa fa-check"></i><b>2</b> Manipulating Tree with Data</a><ul>
<li class="chapter" data-level="2.1" data-path="chapter2.html"><a href="chapter2.html#tidytree"><i class="fa fa-check"></i><b>2.1</b> Manipulating tree data using tidy interface</a><ul>
<li class="chapter" data-level="2.1.1" data-path="chapter2.html"><a href="chapter2.html#the-phylo-object"><i class="fa fa-check"></i><b>2.1.1</b> The <code>phylo</code> object</a></li>
<li class="chapter" data-level="2.1.2" data-path="chapter2.html"><a href="chapter2.html#the-treedata-object"><i class="fa fa-check"></i><b>2.1.2</b> The <code>treedata</code> object</a></li>
<li class="chapter" data-level="2.1.3" data-path="chapter2.html"><a href="chapter2.html#accesor-tidytree"><i class="fa fa-check"></i><b>2.1.3</b> Access related nodes</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="chapter2.html"><a href="chapter2.html#data-integration"><i class="fa fa-check"></i><b>2.2</b> Data Integration</a><ul>
<li class="chapter" data-level="2.2.1" data-path="chapter2.html"><a href="chapter2.html#merge-tree"><i class="fa fa-check"></i><b>2.2.1</b> Combining tree data</a></li>
<li class="chapter" data-level="2.2.2" data-path="chapter2.html"><a href="chapter2.html#link-external-data"><i class="fa fa-check"></i><b>2.2.2</b> Linking external data to phylogeny</a></li>
<li class="chapter" data-level="2.2.3" data-path="chapter2.html"><a href="chapter2.html#grouping-taxa"><i class="fa fa-check"></i><b>2.2.3</b> Grouping taxa</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="chapter2.html"><a href="chapter2.html#rescale-treeio"><i class="fa fa-check"></i><b>2.3</b> Rescaling Tree Branches</a></li>
<li class="chapter" data-level="2.4" data-path="chapter2.html"><a href="chapter2.html#subsetting-tree-with-data"><i class="fa fa-check"></i><b>2.4</b> Subsetting Tree with Data</a><ul>
<li class="chapter" data-level="2.4.1" data-path="chapter2.html"><a href="chapter2.html#remove-tip"><i class="fa fa-check"></i><b>2.4.1</b> Removing tips in a phylogenetic tree</a></li>
<li class="chapter" data-level="2.4.2" data-path="chapter2.html"><a href="chapter2.html#subset-tip"><i class="fa fa-check"></i><b>2.4.2</b> Subsetting tree by tip label</a></li>
<li class="chapter" data-level="2.4.3" data-path="chapter2.html"><a href="chapter2.html#subset-node"><i class="fa fa-check"></i><b>2.4.3</b> Subsetting tree by internal node number</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="chapter2.html"><a href="chapter2.html#ggtree-fortify"><i class="fa fa-check"></i><b>2.5</b> Manipulating tree data for visualization</a></li>
<li class="chapter" data-level="2.6" data-path="chapter2.html"><a href="chapter2.html#summary-1"><i class="fa fa-check"></i><b>2.6</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="chapter3.html"><a href="chapter3.html"><i class="fa fa-check"></i><b>3</b> Exporting tree with data</a><ul>
<li class="chapter" data-level="3.1" data-path="chapter3.html"><a href="chapter3.html#introduction"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="chapter3.html"><a href="chapter3.html#exporting-tree-data-to-beast-nexus-format"><i class="fa fa-check"></i><b>3.2</b> Exporting Tree Data to <em>BEAST</em> Nexus Format</a><ul>
<li class="chapter" data-level="3.2.1" data-path="chapter3.html"><a href="chapter3.html#exportingconverting-software-output"><i class="fa fa-check"></i><b>3.2.1</b> Exporting/converting software output</a></li>
<li class="chapter" data-level="3.2.2" data-path="chapter3.html"><a href="chapter3.html#combining-tree-with-external-data"><i class="fa fa-check"></i><b>3.2.2</b> Combining tree with external data</a></li>
<li class="chapter" data-level="3.2.3" data-path="chapter3.html"><a href="chapter3.html#merging-tree-data-from-different-sources"><i class="fa fa-check"></i><b>3.2.3</b> Merging tree data from different sources</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="chapter3.html"><a href="chapter3.html#write-jtree"><i class="fa fa-check"></i><b>3.3</b> Exporting Tree Data to <em>jtree</em> Format</a></li>
<li class="chapter" data-level="3.4" data-path="chapter3.html"><a href="chapter3.html#summary-2"><i class="fa fa-check"></i><b>3.4</b> Summary</a></li>
</ul></li>
<li class="part"><span><b>Part II: Tree data visualization and annotation</b></span></li>
<li class="chapter" data-level="4" data-path="chapter4.html"><a href="chapter4.html"><i class="fa fa-check"></i><b>4</b> Phylogenetic Tree Visualization</a><ul>
<li class="chapter" data-level="4.1" data-path="chapter4.html"><a href="chapter4.html#introduction-1"><i class="fa fa-check"></i><b>4.1</b> Introduction</a></li>
<li class="chapter" data-level="4.2" data-path="chapter4.html"><a href="chapter4.html#visualizing-phylogenetic-tree-with-ggtree"><i class="fa fa-check"></i><b>4.2</b> Visualizing Phylogenetic Tree with <em>ggtree</em></a><ul>
<li class="chapter" data-level="4.2.1" data-path="chapter4.html"><a href="chapter4.html#basic-tree-visualization"><i class="fa fa-check"></i><b>4.2.1</b> Basic Tree Visualization</a></li>
<li class="chapter" data-level="4.2.2" data-path="chapter4.html"><a href="chapter4.html#tree-layouts"><i class="fa fa-check"></i><b>4.2.2</b> Layouts of phylogenetic tree</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="chapter4.html"><a href="chapter4.html#displaying-tree-components"><i class="fa fa-check"></i><b>4.3</b> Displaying Tree Components</a><ul>
<li class="chapter" data-level="4.3.1" data-path="chapter4.html"><a href="chapter4.html#geom-trescale"><i class="fa fa-check"></i><b>4.3.1</b> Displaying tree scale (evolution distance)</a></li>
<li class="chapter" data-level="4.3.2" data-path="chapter4.html"><a href="chapter4.html#geom-nodepoint"><i class="fa fa-check"></i><b>4.3.2</b> Displaying nodes/tips</a></li>
<li class="chapter" data-level="4.3.3" data-path="chapter4.html"><a href="chapter4.html#displaying-labels"><i class="fa fa-check"></i><b>4.3.3</b> Displaying labels</a></li>
<li class="chapter" data-level="4.3.4" data-path="chapter4.html"><a href="chapter4.html#displaying-root-edge"><i class="fa fa-check"></i><b>4.3.4</b> Displaying root edge</a></li>
<li class="chapter" data-level="4.3.5" data-path="chapter4.html"><a href="chapter4.html#color-tree"><i class="fa fa-check"></i><b>4.3.5</b> Color tree</a></li>
<li class="chapter" data-level="4.3.6" data-path="chapter4.html"><a href="chapter4.html#rescale-tree"><i class="fa fa-check"></i><b>4.3.6</b> Rescale tree</a></li>
<li class="chapter" data-level="4.3.7" data-path="chapter4.html"><a href="chapter4.html#modify-compenents-of-a-theme"><i class="fa fa-check"></i><b>4.3.7</b> Modify compenents of a theme</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="chapter4.html"><a href="chapter4.html#visualize-a-list-of-trees"><i class="fa fa-check"></i><b>4.4</b> Visualize a list of trees</a></li>
<li class="chapter" data-level="4.5" data-path="chapter4.html"><a href="chapter4.html#summary-3"><i class="fa fa-check"></i><b>4.5</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="chapter5.html"><a href="chapter5.html"><i class="fa fa-check"></i><b>5</b> Phylogenetic Tree Annotation</a><ul>
<li class="chapter" data-level="5.1" data-path="chapter5.html"><a href="chapter5.html#visualizing-and-annotating-tree-using-grammar-of-graphics"><i class="fa fa-check"></i><b>5.1</b> Visualizing and Annotating Tree using Grammar of Graphics</a></li>
<li class="chapter" data-level="5.2" data-path="chapter5.html"><a href="chapter5.html#layers-for-tree-annotation"><i class="fa fa-check"></i><b>5.2</b> Layers for Tree Annotation</a><ul>
<li class="chapter" data-level="5.2.1" data-path="chapter5.html"><a href="chapter5.html#colored-strips"><i class="fa fa-check"></i><b>5.2.1</b> Colored strips</a></li>
<li class="chapter" data-level="5.2.2" data-path="chapter5.html"><a href="chapter5.html#highlight-clades"><i class="fa fa-check"></i><b>5.2.2</b> Highlight clades</a></li>
<li class="chapter" data-level="5.2.3" data-path="chapter5.html"><a href="chapter5.html#taxa-connection"><i class="fa fa-check"></i><b>5.2.3</b> Taxa connection</a></li>
<li class="chapter" data-level="5.2.4" data-path="chapter5.html"><a href="chapter5.html#uncertainty-of-evolutionary-inference"><i class="fa fa-check"></i><b>5.2.4</b> Uncertainty of evolutionary inference</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="chapter5.html"><a href="chapter5.html#tree-annotation-with-output-from-evolution-software"><i class="fa fa-check"></i><b>5.3</b> Tree annotation with output from evolution software</a><ul>
<li class="chapter" data-level="5.3.1" data-path="chapter5.html"><a href="chapter5.html#tree-annotation-using-data-from-evolutionary-analysis-software"><i class="fa fa-check"></i><b>5.3.1</b> Tree annotation using data from evolutionary analysis software</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="chapter5.html"><a href="chapter5.html#summary-4"><i class="fa fa-check"></i><b>5.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="chapter6.html"><a href="chapter6.html"><i class="fa fa-check"></i><b>6</b> Visual Exploration of Phylogenetic Tree</a><ul>
<li class="chapter" data-level="6.1" data-path="chapter6.html"><a href="chapter6.html#viewing-selected-clade"><i class="fa fa-check"></i><b>6.1</b> Viewing Selected Clade</a></li>
<li class="chapter" data-level="6.2" data-path="chapter6.html"><a href="chapter6.html#scale-clade"><i class="fa fa-check"></i><b>6.2</b> Scaling Selected Clade</a></li>
<li class="chapter" data-level="6.3" data-path="chapter6.html"><a href="chapter6.html#collapsing-and-expanding-clade"><i class="fa fa-check"></i><b>6.3</b> Collapsing and Expanding Clade</a></li>
<li class="chapter" data-level="6.4" data-path="chapter6.html"><a href="chapter6.html#group-taxa-vis"><i class="fa fa-check"></i><b>6.4</b> Grouping Taxa</a></li>
<li class="chapter" data-level="6.5" data-path="chapter6.html"><a href="chapter6.html#exploring-tree-structure"><i class="fa fa-check"></i><b>6.5</b> Exploring tree structure</a></li>
<li class="chapter" data-level="6.6" data-path="chapter6.html"><a href="chapter6.html#summary-5"><i class="fa fa-check"></i><b>6.6</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="chapter7.html"><a href="chapter7.html"><i class="fa fa-check"></i><b>7</b> Plotting tree with data</a><ul>
<li class="chapter" data-level="7.1" data-path="chapter7.html"><a href="chapter7.html#attach-operator"><i class="fa fa-check"></i><b>7.1</b> Mapping data to the tree structure</a></li>
<li class="chapter" data-level="7.2" data-path="chapter7.html"><a href="chapter7.html#facet_plot"><i class="fa fa-check"></i><b>7.2</b> Alignning graph to the tree based on tree structure</a></li>
<li class="chapter" data-level="7.3" data-path="chapter7.html"><a href="chapter7.html#gheatmap"><i class="fa fa-check"></i><b>7.3</b> Visualize tree with associated matrix</a><ul>
<li class="chapter" data-level="7.3.1" data-path="chapter7.html"><a href="chapter7.html#gheatmap-ggnewscale"><i class="fa fa-check"></i><b>7.3.1</b> Visualize tree with multiple associated matrix</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="chapter7.html"><a href="chapter7.html#msaplot"><i class="fa fa-check"></i><b>7.4</b> Visualize tree with multiple sequence alignment</a></li>
<li class="chapter" data-level="7.5" data-path="chapter7.html"><a href="chapter7.html#composite_plot"><i class="fa fa-check"></i><b>7.5</b> Composite plots</a></li>
<li class="chapter" data-level="7.6" data-path="chapter7.html"><a href="chapter7.html#ggtree_object"><i class="fa fa-check"></i><b>7.6</b> The <code>ggtree</code> object</a></li>
<li class="chapter" data-level="7.7" data-path="chapter7.html"><a href="chapter7.html#summary-6"><i class="fa fa-check"></i><b>7.7</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="chapter8.html"><a href="chapter8.html"><i class="fa fa-check"></i><b>8</b> Annotating Tree with Silhouette Images and Sub-plots</a><ul>
<li class="chapter" data-level="8.1" data-path="chapter8.html"><a href="chapter8.html#ggimage"><i class="fa fa-check"></i><b>8.1</b> Annotating Tree with Images</a></li>
<li class="chapter" data-level="8.2" data-path="chapter8.html"><a href="chapter8.html#phylopic"><i class="fa fa-check"></i><b>8.2</b> Annotating Tree with Phylopic</a></li>
<li class="chapter" data-level="8.3" data-path="chapter8.html"><a href="chapter8.html#subplots"><i class="fa fa-check"></i><b>8.3</b> Annotating Tree with Sub-plots</a><ul>
<li class="chapter" data-level="8.3.1" data-path="chapter8.html"><a href="chapter8.html#annotate-with-bar-charts"><i class="fa fa-check"></i><b>8.3.1</b> Annotate with bar charts</a></li>
<li class="chapter" data-level="8.3.2" data-path="chapter8.html"><a href="chapter8.html#annotate-with-pie-charts"><i class="fa fa-check"></i><b>8.3.2</b> Annotate with pie charts</a></li>
<li class="chapter" data-level="8.3.3" data-path="chapter8.html"><a href="chapter8.html#annotate-with-other-types-of-charts"><i class="fa fa-check"></i><b>8.3.3</b> Annotate with other types of charts</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="chapter8.html"><a href="chapter8.html#phylomoji"><i class="fa fa-check"></i><b>8.4</b> Have Fun with Phylomoji</a><ul>
<li class="chapter" data-level="8.4.1" data-path="chapter8.html"><a href="chapter8.html#emoji-in-cirularfan-layout-tree"><i class="fa fa-check"></i><b>8.4.1</b> Emoji in cirular/fan layout tree</a></li>
<li class="chapter" data-level="8.4.2" data-path="chapter8.html"><a href="chapter8.html#emoji-to-label-clades"><i class="fa fa-check"></i><b>8.4.2</b> Emoji to label clades</a></li>
<li class="chapter" data-level="8.4.3" data-path="chapter8.html"><a href="chapter8.html#apple-color-emoji"><i class="fa fa-check"></i><b>8.4.3</b> Apple Color Emoji</a></li>
<li class="chapter" data-level="8.4.4" data-path="chapter8.html"><a href="chapter8.html#phylomoji-in-ascii-art"><i class="fa fa-check"></i><b>8.4.4</b> Phylomoji in ASCII art</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="chapter9.html"><a href="chapter9.html"><i class="fa fa-check"></i><b>9</b> ggtree for other tree-like objects</a><ul>
<li class="chapter" data-level="9.1" data-path="chapter9.html"><a href="chapter9.html#ggtree-for-phylogenetic-tree-objects"><i class="fa fa-check"></i><b>9.1</b> ggtree for phylogenetic tree objects</a><ul>
<li class="chapter" data-level="9.1.1" data-path="chapter9.html"><a href="chapter9.html#phylobase"><i class="fa fa-check"></i><b>9.1.1</b> The phylo4 and phylo4d objects</a></li>
<li class="chapter" data-level="9.1.2" data-path="chapter9.html"><a href="chapter9.html#phylog"><i class="fa fa-check"></i><b>9.1.2</b> The phylog object</a></li>
<li class="chapter" data-level="9.1.3" data-path="chapter9.html"><a href="chapter9.html#phyloseq"><i class="fa fa-check"></i><b>9.1.3</b> The phyloseq object</a></li>
<li class="chapter" data-level="9.1.4" data-path="chapter9.html"><a href="chapter9.html#obkdata"><i class="fa fa-check"></i><b>9.1.4</b> The obkData object</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="chapter9.html"><a href="chapter9.html#dendrogram"><i class="fa fa-check"></i><b>9.2</b> ggtree for dendrograms</a></li>
<li class="chapter" data-level="9.3" data-path="chapter9.html"><a href="chapter9.html#igraph"><i class="fa fa-check"></i><b>9.3</b> ggtree for tree graph</a></li>
</ul></li>
<li class="part"><span><b>Part III: Miscellaneous topics</b></span></li>
<li class="chapter" data-level="10" data-path="chapter10.html"><a href="chapter10.html"><i class="fa fa-check"></i><b>10</b> ggtree utilities</a><ul>
<li class="chapter" data-level="10.1" data-path="chapter10.html"><a href="chapter10.html#facet-utilities"><i class="fa fa-check"></i><b>10.1</b> facet utilities</a><ul>
<li class="chapter" data-level="10.1.1" data-path="chapter10.html"><a href="chapter10.html#facet_widths"><i class="fa fa-check"></i><b>10.1.1</b> facet_widths</a></li>
<li class="chapter" data-level="10.1.2" data-path="chapter10.html"><a href="chapter10.html#facet_labeller"><i class="fa fa-check"></i><b>10.1.2</b> facet_labeller</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="chapter10.html"><a href="chapter10.html#geom2"><i class="fa fa-check"></i><b>10.2</b> Geometric layers</a></li>
<li class="chapter" data-level="10.3" data-path="chapter10.html"><a href="chapter10.html#layout-utilities"><i class="fa fa-check"></i><b>10.3</b> Layout utilities</a></li>
<li class="chapter" data-level="10.4" data-path="chapter10.html"><a href="chapter10.html#legend-utilities"><i class="fa fa-check"></i><b>10.4</b> Legend utilities</a></li>
<li class="chapter" data-level="10.5" data-path="chapter10.html"><a href="chapter10.html#scale-utilities"><i class="fa fa-check"></i><b>10.5</b> Scale utilities</a><ul>
<li class="chapter" data-level="10.5.1" data-path="chapter10.html"><a href="chapter10.html#xlim_expand"><i class="fa fa-check"></i><b>10.5.1</b> xlim_expand</a></li>
<li class="chapter" data-level="10.5.2" data-path="chapter10.html"><a href="chapter10.html#axis_align"><i class="fa fa-check"></i><b>10.5.2</b> Reconcile axis limits</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="chapter11.html"><a href="chapter11.html"><i class="fa fa-check"></i><b>11</b> Gallery of reproducible examples</a><ul>
<li class="chapter" data-level="11.1" data-path="chapter11.html"><a href="chapter11.html#visualizing-pairwise-nucleotide-sequence-distance-with-phylogenetic-tree"><i class="fa fa-check"></i><b>11.1</b> Visualizing pairwise nucleotide sequence distance with phylogenetic tree</a></li>
<li class="chapter" data-level="11.2" data-path="chapter11.html"><a href="chapter11.html#displaying-different-symbolic-points-for-bootstrap-values."><i class="fa fa-check"></i><b>11.2</b> Displaying different symbolic points for bootstrap values.</a></li>
<li class="chapter" data-level="11.3" data-path="chapter11.html"><a href="chapter11.html#genome-locus"><i class="fa fa-check"></i><b>11.3</b> Phylogenetic tree with genome locus structure</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="chapter12.html"><a href="chapter12.html"><i class="fa fa-check"></i><b>12</b> ggtree extensions</a></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="faq.html"><a href="faq.html"><i class="fa fa-check"></i><b>A</b> Frequently asked questions</a><ul>
<li class="chapter" data-level="A.1" data-path="faq.html"><a href="faq.html#installation"><i class="fa fa-check"></i><b>A.1</b> Installation</a></li>
<li class="chapter" data-level="A.2" data-path="faq.html"><a href="faq.html#faq-r"><i class="fa fa-check"></i><b>A.2</b> Basic R related</a><ul>
<li class="chapter" data-level="A.2.1" data-path="faq.html"><a href="faq.html#faq-local-file"><i class="fa fa-check"></i><b>A.2.1</b> Use your local file</a></li>
</ul></li>
<li class="chapter" data-level="A.3" data-path="faq.html"><a href="faq.html#faq-aes-mapping"><i class="fa fa-check"></i><b>A.3</b> Aesthetic mapping</a><ul>
<li class="chapter" data-level="A.3.1" data-path="faq.html"><a href="faq.html#faq-inherit-aes"><i class="fa fa-check"></i><b>A.3.1</b> Inherit <em>aes</em></a></li>
<li class="chapter" data-level="A.3.2" data-path="faq.html"><a href="faq.html#faq-dollar-aes"><i class="fa fa-check"></i><b>A.3.2</b> Never use <code>$</code> in aes</a></li>
</ul></li>
<li class="chapter" data-level="A.4" data-path="faq.html"><a href="faq.html#faq-text"><i class="fa fa-check"></i><b>A.4</b> Text &amp; Label</a><ul>
<li class="chapter" data-level="A.4.1" data-path="faq.html"><a href="faq.html#faq-label-truncated"><i class="fa fa-check"></i><b>A.4.1</b> Tip label truncated</a></li>
<li class="chapter" data-level="A.4.2" data-path="faq.html"><a href="faq.html#faq-modify-label"><i class="fa fa-check"></i><b>A.4.2</b> Modify (tip) labels</a></li>
<li class="chapter" data-level="A.4.3" data-path="faq.html"><a href="faq.html#faq-formatting-label"><i class="fa fa-check"></i><b>A.4.3</b> Formatting (tip) labels</a></li>
<li class="chapter" data-level="A.4.4" data-path="faq.html"><a href="faq.html#faq-ggrepel"><i class="fa fa-check"></i><b>A.4.4</b> Avoid overlapping text labels</a></li>
<li class="chapter" data-level="A.4.5" data-path="faq.html"><a href="faq.html#faq-bootstrap"><i class="fa fa-check"></i><b>A.4.5</b> Bootstrap values from newick format</a></li>
</ul></li>
<li class="chapter" data-level="A.5" data-path="faq.html"><a href="faq.html#different-x-labels-for-different-facet-panels"><i class="fa fa-check"></i><b>A.5</b> Different x labels for different facet panels</a></li>
<li class="chapter" data-level="A.6" data-path="faq.html"><a href="faq.html#faq-under-the-tree"><i class="fa fa-check"></i><b>A.6</b> Plot something behind the phylogeny</a></li>
<li class="chapter" data-level="A.7" data-path="faq.html"><a href="faq.html#faq-enlarge-center-space"><i class="fa fa-check"></i><b>A.7</b> Enlarge center space in circular/fan layout tree</a></li>
<li class="chapter" data-level="A.8" data-path="faq.html"><a href="faq.html#use-the-most-distant-tip-from-the-root-as-the-origin-of-the-time-scale"><i class="fa fa-check"></i><b>A.8</b> Use the most distant tip from the root as the origin of the time scale</a></li>
<li class="chapter" data-level="A.9" data-path="faq.html"><a href="faq.html#changing-branch-length-of-outgroup"><i class="fa fa-check"></i><b>A.9</b> Changing branch length of outgroup</a></li>
<li class="chapter" data-level="A.10" data-path="faq.html"><a href="faq.html#export-edit"><i class="fa fa-check"></i><b>A.10</b> Edit tree graphic details</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="ggimage-tips.html"><a href="ggimage-tips.html"><i class="fa fa-check"></i><b>B</b> Tips for using <code>ggtree</code> with <code>ggimage</code></a><ul>
<li class="chapter" data-level="B.1" data-path="ggimage-tips.html"><a href="ggimage-tips.html#ggimage-rm-image-bg"><i class="fa fa-check"></i><b>B.1</b> Example 1: Remove background of images</a></li>
<li class="chapter" data-level="B.2" data-path="ggimage-tips.html"><a href="ggimage-tips.html#ggimage-bgimage"><i class="fa fa-check"></i><b>B.2</b> Example 2: Plot tree on a background image</a></li>
</ul></li>
<li class="chapter" data-level="C" data-path="commicR.html"><a href="commicR.html"><i class="fa fa-check"></i><b>C</b> Comic (xkcd-like) phylogenetic tree</a></li>
<li class="chapter" data-level="D" data-path="ascii-tree.html"><a href="ascii-tree.html"><i class="fa fa-check"></i><b>D</b> Print ASCII-art rooted tree</a></li>
<li class="chapter" data-level="E" data-path="facet-zoom.html"><a href="facet-zoom.html"><i class="fa fa-check"></i><b>E</b> Zoom in selected portion</a></li>
<li class="chapter" data-level="F" data-path="run-ggtree-in-jupyter-notebook.html"><a href="run-ggtree-in-jupyter-notebook.html"><i class="fa fa-check"></i><b>F</b> Run ggtree in Jupyter notebook</a></li>
<li class="chapter" data-level="G" data-path="figures-and-tables.html"><a href="figures-and-tables.html"><i class="fa fa-check"></i><b>G</b> Figures and Tables</a></li>
<li class="chapter" data-level="H" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>H</b> Session Info</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Data Integration, Manipulation and Visualization of Phylogenetic Trees</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="chapter11" class="section level1">
<h1><span class="header-section-number">Chapter 11</span> Gallery of reproducible examples</h1>
<div id="visualizing-pairwise-nucleotide-sequence-distance-with-phylogenetic-tree" class="section level2">
<h2><span class="header-section-number">11.1</span> Visualizing pairwise nucleotide sequence distance with phylogenetic tree</h2>
<p>This example reproduces Fig. 1 of <span class="citation">(Chen et al. <a href="#ref-chen_ancient_2017" role="doc-biblioref">2017</a>)</span>. It extracts accession numbers from tip labels of the HPV58 tree and calculates pairwise nucleotide sequence distances. The distance matrix is visualized as dot and line plots. This example demonstrates the abilities of adding multiple layers to a specific panel. As illustrated on Figure <a href="chapter11.html#fig:jv2017">11.1</a>, the <code>facet_plot</code> function displays sequence distances as a dot plot and then adds a layer of line plot to the same panel, <em>i.e.</em> Sequence Distance. In addition, the tree in <code>facet_plot</code> can be fully annotated with multiple layers (clade labels, bootstrap support values, <em>etc</em>). The source code of this example was firstly published in Supplemental File of <span class="citation">(Yu et al. <a href="#ref-yu_two_2018" role="doc-biblioref">2018</a>)</span>.</p>
<div class="sourceCode" id="cb236"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb236-1"><a href="chapter11.html#cb236-1"></a><span class="kw">library</span>(tibble)</span>
<span id="cb236-2"><a href="chapter11.html#cb236-2"></a><span class="kw">library</span>(tidyr)</span>
<span id="cb236-3"><a href="chapter11.html#cb236-3"></a><span class="kw">library</span>(Biostrings)</span>
<span id="cb236-4"><a href="chapter11.html#cb236-4"></a><span class="kw">library</span>(treeio)</span>
<span id="cb236-5"><a href="chapter11.html#cb236-5"></a><span class="kw">library</span>(ggplot2)</span>
<span id="cb236-6"><a href="chapter11.html#cb236-6"></a><span class="kw">library</span>(ggtree)</span>
<span id="cb236-7"><a href="chapter11.html#cb236-7"></a></span>
<span id="cb236-8"><a href="chapter11.html#cb236-8"></a>tree &lt;-<span class="st"> </span><span class="kw">read.tree</span>(<span class="st">&quot;data/HPV58.tree&quot;</span>)</span>
<span id="cb236-9"><a href="chapter11.html#cb236-9"></a></span>
<span id="cb236-10"><a href="chapter11.html#cb236-10"></a>clade &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">A3 =</span> <span class="dv">92</span>, <span class="dt">A1 =</span> <span class="dv">94</span>, <span class="dt">A2 =</span> <span class="dv">108</span>, <span class="dt">B1 =</span> <span class="dv">156</span>, <span class="dt">B2 =</span> <span class="dv">159</span>, <span class="dt">C =</span> <span class="dv">163</span>, <span class="dt">D1 =</span> <span class="dv">173</span>, <span class="dt">D2 =</span> <span class="dv">176</span>)</span>
<span id="cb236-11"><a href="chapter11.html#cb236-11"></a>tree &lt;-<span class="st"> </span><span class="kw">groupClade</span>(tree, clade)</span>
<span id="cb236-12"><a href="chapter11.html#cb236-12"></a>cols &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">A1 =</span> <span class="st">&quot;#EC762F&quot;</span>, <span class="dt">A2 =</span> <span class="st">&quot;#CA6629&quot;</span>, <span class="dt">A3 =</span> <span class="st">&quot;#894418&quot;</span>, <span class="dt">B1 =</span> <span class="st">&quot;#0923FA&quot;</span>, </span>
<span id="cb236-13"><a href="chapter11.html#cb236-13"></a>         <span class="dt">B2 =</span> <span class="st">&quot;#020D87&quot;</span>, <span class="dt">C =</span> <span class="st">&quot;#000000&quot;</span>, <span class="dt">D1 =</span> <span class="st">&quot;#9ACD32&quot;</span>,<span class="dt">D2 =</span> <span class="st">&quot;#08630A&quot;</span>)</span>
<span id="cb236-14"><a href="chapter11.html#cb236-14"></a></span>
<span id="cb236-15"><a href="chapter11.html#cb236-15"></a><span class="co">## visualize the tree with tip labels and tree scale</span></span>
<span id="cb236-16"><a href="chapter11.html#cb236-16"></a>p &lt;-<span class="st"> </span><span class="kw">ggtree</span>(tree, <span class="kw">aes</span>(<span class="dt">color =</span> group), <span class="dt">ladderize =</span> <span class="ot">FALSE</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">rotate</span>(<span class="kw">rootnode</span>(tree)) <span class="op">+</span><span class="st"> </span></span>
<span id="cb236-17"><a href="chapter11.html#cb236-17"></a><span class="st">    </span><span class="kw">geom_tiplab</span>(<span class="kw">aes</span>(<span class="dt">label =</span> <span class="kw">paste0</span>(<span class="st">&quot;italic(&#39;&quot;</span>, label, <span class="st">&quot;&#39;)&quot;</span>)), <span class="dt">parse =</span> <span class="ot">TRUE</span>, <span class="dt">size =</span> <span class="fl">2.5</span>) <span class="op">+</span></span>
<span id="cb236-18"><a href="chapter11.html#cb236-18"></a><span class="st">    </span><span class="kw">geom_treescale</span>(<span class="dt">x =</span> <span class="dv">0</span>, <span class="dt">y =</span> <span class="dv">1</span>, <span class="dt">width =</span> <span class="fl">0.002</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb236-19"><a href="chapter11.html#cb236-19"></a><span class="st">    </span><span class="kw">scale_color_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(cols, <span class="st">&quot;black&quot;</span>), <span class="dt">na.value =</span> <span class="st">&quot;black&quot;</span>, <span class="dt">name =</span> <span class="st">&quot;Lineage&quot;</span>,</span>
<span id="cb236-20"><a href="chapter11.html#cb236-20"></a>                      <span class="dt">breaks =</span> <span class="kw">c</span>(<span class="st">&quot;A1&quot;</span>, <span class="st">&quot;A2&quot;</span>, <span class="st">&quot;A3&quot;</span>, <span class="st">&quot;B1&quot;</span>, <span class="st">&quot;B2&quot;</span>, <span class="st">&quot;C&quot;</span>, <span class="st">&quot;D1&quot;</span>, <span class="st">&quot;D2&quot;</span>)) <span class="op">+</span></span>
<span id="cb236-21"><a href="chapter11.html#cb236-21"></a><span class="st">    </span><span class="kw">guides</span>(<span class="dt">color =</span> <span class="kw">guide_legend</span>(<span class="dt">override.aes =</span> <span class="kw">list</span>(<span class="dt">size =</span> <span class="dv">5</span>, <span class="dt">shape =</span> <span class="dv">15</span>))) <span class="op">+</span></span>
<span id="cb236-22"><a href="chapter11.html#cb236-22"></a><span class="st">    </span><span class="kw">theme_tree2</span>(<span class="dt">legend.position =</span> <span class="kw">c</span>(.<span class="dv">1</span>, <span class="fl">.88</span>))</span>
<span id="cb236-23"><a href="chapter11.html#cb236-23"></a><span class="co">## Optional</span></span>
<span id="cb236-24"><a href="chapter11.html#cb236-24"></a><span class="co">## add labels for monophyletic (A, C and D) and paraphyletic (B) groups  </span></span>
<span id="cb236-25"><a href="chapter11.html#cb236-25"></a>p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">geom_cladelabel</span>(<span class="dv">94</span>, <span class="st">&quot;italic(A1)&quot;</span>, <span class="dt">color =</span> cols[[<span class="st">&quot;A1&quot;</span>]], <span class="dt">offset =</span> <span class="fl">.003</span>, <span class="dt">align =</span> <span class="ot">TRUE</span>, </span>
<span id="cb236-26"><a href="chapter11.html#cb236-26"></a>                    <span class="dt">offset.text =</span> <span class="fl">-.001</span>, <span class="dt">barsize =</span> <span class="fl">1.2</span>, <span class="dt">extend =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="dt">parse =</span> <span class="ot">TRUE</span>) <span class="op">+</span></span>
<span id="cb236-27"><a href="chapter11.html#cb236-27"></a><span class="st">    </span><span class="kw">geom_cladelabel</span>(<span class="dv">108</span>, <span class="st">&quot;italic(A2)&quot;</span>, <span class="dt">color =</span> cols[[<span class="st">&quot;A2&quot;</span>]], <span class="dt">offset =</span> <span class="fl">.003</span>, <span class="dt">align =</span> <span class="ot">TRUE</span>, </span>
<span id="cb236-28"><a href="chapter11.html#cb236-28"></a>                   <span class="dt">offset.text =</span> <span class="fl">-.001</span>, <span class="dt">barsize =</span> <span class="fl">1.2</span>, <span class="dt">extend =</span> <span class="fl">0.5</span>, <span class="dt">parse =</span> <span class="ot">TRUE</span>) <span class="op">+</span></span>
<span id="cb236-29"><a href="chapter11.html#cb236-29"></a><span class="st">    </span><span class="kw">geom_cladelabel</span>(<span class="dv">131</span>, <span class="st">&quot;italic(A3)&quot;</span>, <span class="dt">color =</span> cols[[<span class="st">&quot;A3&quot;</span>]], <span class="dt">offset =</span> <span class="fl">.003</span>, <span class="dt">align =</span> <span class="ot">TRUE</span>, </span>
<span id="cb236-30"><a href="chapter11.html#cb236-30"></a>                   <span class="dt">offset.text =</span> <span class="fl">-.001</span>, <span class="dt">barsize =</span> <span class="fl">1.2</span>, <span class="dt">extend =</span> <span class="kw">c</span>(<span class="fl">0.5</span>, <span class="dv">0</span>), <span class="dt">parse =</span> <span class="ot">TRUE</span>) <span class="op">+</span></span>
<span id="cb236-31"><a href="chapter11.html#cb236-31"></a><span class="st">    </span><span class="kw">geom_cladelabel</span>(<span class="dv">92</span>, <span class="st">&quot;italic(A)&quot;</span>, <span class="dt">color =</span> <span class="st">&quot;darkgrey&quot;</span>, <span class="dt">offset =</span> <span class="fl">.00315</span>, <span class="dt">align =</span> <span class="ot">TRUE</span>, </span>
<span id="cb236-32"><a href="chapter11.html#cb236-32"></a>                   <span class="dt">offset.text =</span> <span class="fl">0.0002</span>, <span class="dt">barsize =</span> <span class="dv">2</span>, <span class="dt">fontsize =</span> <span class="dv">5</span>, <span class="dt">parse =</span> <span class="ot">TRUE</span>) <span class="op">+</span></span>
<span id="cb236-33"><a href="chapter11.html#cb236-33"></a><span class="st">    </span><span class="kw">geom_cladelabel</span>(<span class="dv">156</span>, <span class="st">&quot;italic(B1)&quot;</span>, <span class="dt">color =</span> cols[[<span class="st">&quot;B1&quot;</span>]], <span class="dt">offset =</span> <span class="fl">.003</span>, <span class="dt">align =</span> <span class="ot">TRUE</span>, </span>
<span id="cb236-34"><a href="chapter11.html#cb236-34"></a>                   <span class="dt">offset.text =</span> <span class="fl">-.001</span>, <span class="dt">barsize =</span> <span class="fl">1.2</span>, <span class="dt">extend =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="dt">parse =</span> <span class="ot">TRUE</span>) <span class="op">+</span></span>
<span id="cb236-35"><a href="chapter11.html#cb236-35"></a><span class="st">    </span><span class="kw">geom_cladelabel</span>(<span class="dv">159</span>, <span class="st">&quot;italic(B2)&quot;</span>, <span class="dt">color =</span> cols[[<span class="st">&quot;B2&quot;</span>]], <span class="dt">offset =</span> <span class="fl">.003</span>, <span class="dt">align =</span> <span class="ot">TRUE</span>, </span>
<span id="cb236-36"><a href="chapter11.html#cb236-36"></a>                   <span class="dt">offset.text =</span> <span class="fl">-.001</span>, <span class="dt">barsize =</span> <span class="fl">1.2</span>, <span class="dt">extend =</span> <span class="kw">c</span>(<span class="fl">0.5</span>, <span class="dv">0</span>), <span class="dt">parse =</span> <span class="ot">TRUE</span>) <span class="op">+</span></span>
<span id="cb236-37"><a href="chapter11.html#cb236-37"></a><span class="st">    </span><span class="kw">geom_strip</span>(<span class="dv">65</span>, <span class="dv">71</span>, <span class="st">&quot;italic(B)&quot;</span>, <span class="dt">color =</span> <span class="st">&quot;darkgrey&quot;</span>, <span class="dt">offset =</span> <span class="fl">0.00315</span>, <span class="dt">align =</span> <span class="ot">TRUE</span>, </span>
<span id="cb236-38"><a href="chapter11.html#cb236-38"></a>                   <span class="dt">offset.text =</span> <span class="fl">0.0002</span>, <span class="dt">barsize =</span> <span class="dv">2</span>, <span class="dt">fontsize =</span> <span class="dv">5</span>, <span class="dt">parse =</span> <span class="ot">TRUE</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb236-39"><a href="chapter11.html#cb236-39"></a><span class="st">    </span><span class="kw">geom_cladelabel</span>(<span class="dv">163</span>, <span class="st">&quot;italic(C)&quot;</span>, <span class="dt">color =</span> <span class="st">&quot;darkgrey&quot;</span>, <span class="dt">offset =</span> <span class="fl">.0031</span>, <span class="dt">align =</span> <span class="ot">TRUE</span>, </span>
<span id="cb236-40"><a href="chapter11.html#cb236-40"></a>                   <span class="dt">offset.text =</span> <span class="fl">0.0002</span>, <span class="dt">barsize =</span> <span class="fl">3.2</span>, <span class="dt">fontsize =</span> <span class="dv">5</span>, <span class="dt">parse =</span> <span class="ot">TRUE</span>) <span class="op">+</span></span>
<span id="cb236-41"><a href="chapter11.html#cb236-41"></a><span class="st">    </span><span class="kw">geom_cladelabel</span>(<span class="dv">173</span>, <span class="st">&quot;italic(D1)&quot;</span>, <span class="dt">color =</span> cols[[<span class="st">&quot;D1&quot;</span>]], <span class="dt">offset =</span> <span class="fl">.003</span>, <span class="dt">align =</span> <span class="ot">TRUE</span>, </span>
<span id="cb236-42"><a href="chapter11.html#cb236-42"></a>                   <span class="dt">offset.text =</span> <span class="fl">-.001</span>, <span class="dt">barsize =</span> <span class="fl">1.2</span>, <span class="dt">extend =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="dt">parse =</span> <span class="ot">TRUE</span>) <span class="op">+</span></span>
<span id="cb236-43"><a href="chapter11.html#cb236-43"></a><span class="st">    </span><span class="kw">geom_cladelabel</span>(<span class="dv">176</span>, <span class="st">&quot;italic(D2)&quot;</span>, <span class="dt">color =</span> cols[[<span class="st">&quot;D2&quot;</span>]], <span class="dt">offset =</span> <span class="fl">.003</span>, <span class="dt">align =</span> <span class="ot">TRUE</span>, </span>
<span id="cb236-44"><a href="chapter11.html#cb236-44"></a>                   <span class="dt">offset.text =</span> <span class="fl">-.001</span>, <span class="dt">barsize =</span> <span class="fl">1.2</span>, <span class="dt">extend =</span> <span class="kw">c</span>(<span class="fl">0.5</span>, <span class="dv">0</span>), <span class="dt">parse =</span> <span class="ot">TRUE</span>) <span class="op">+</span></span>
<span id="cb236-45"><a href="chapter11.html#cb236-45"></a><span class="st">    </span><span class="kw">geom_cladelabel</span>(<span class="dv">172</span>, <span class="st">&quot;italic(D)&quot;</span>, <span class="dt">color =</span> <span class="st">&quot;darkgrey&quot;</span>, <span class="dt">offset =</span> <span class="fl">.00315</span>, <span class="dt">align =</span> <span class="ot">TRUE</span>, </span>
<span id="cb236-46"><a href="chapter11.html#cb236-46"></a>                   <span class="dt">offset.text =</span> <span class="fl">0.0002</span>, <span class="dt">barsize =</span> <span class="dv">2</span>, <span class="dt">fontsize =</span> <span class="dv">5</span>, <span class="dt">parse =</span> <span class="ot">TRUE</span>) </span>
<span id="cb236-47"><a href="chapter11.html#cb236-47"></a><span class="co">## Optional</span></span>
<span id="cb236-48"><a href="chapter11.html#cb236-48"></a><span class="co">## display support values</span></span>
<span id="cb236-49"><a href="chapter11.html#cb236-49"></a>p &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">geom_nodelab</span>(<span class="kw">aes</span>(<span class="dt">subset =</span> (node <span class="op">==</span><span class="st"> </span><span class="dv">92</span>), <span class="dt">label =</span> <span class="st">&quot;*&quot;</span>), </span>
<span id="cb236-50"><a href="chapter11.html#cb236-50"></a>                    <span class="dt">color =</span> <span class="st">&quot;black&quot;</span>, <span class="dt">nudge_x =</span> <span class="fl">-.001</span>, <span class="dt">nudge_y =</span> <span class="dv">1</span>) <span class="op">+</span></span>
<span id="cb236-51"><a href="chapter11.html#cb236-51"></a><span class="st">    </span><span class="kw">geom_nodelab</span>(<span class="kw">aes</span>(<span class="dt">subset =</span> (node <span class="op">==</span><span class="st"> </span><span class="dv">155</span>), <span class="dt">label =</span> <span class="st">&quot;*&quot;</span>), </span>
<span id="cb236-52"><a href="chapter11.html#cb236-52"></a>                    <span class="dt">color =</span> <span class="st">&quot;black&quot;</span>, <span class="dt">nudge_x =</span> <span class="fl">-.0003</span>, <span class="dt">nudge_y =</span> <span class="dv">-1</span>) <span class="op">+</span></span>
<span id="cb236-53"><a href="chapter11.html#cb236-53"></a><span class="st">    </span><span class="kw">geom_nodelab</span>(<span class="kw">aes</span>(<span class="dt">subset =</span> (node <span class="op">==</span><span class="st"> </span><span class="dv">158</span>), <span class="dt">label =</span> <span class="st">&quot;95/92/1.00&quot;</span>), </span>
<span id="cb236-54"><a href="chapter11.html#cb236-54"></a>                    <span class="dt">color =</span> <span class="st">&quot;black&quot;</span>, <span class="dt">nudge_x =</span> <span class="fl">-0.0001</span>, <span class="dt">nudge_y =</span> <span class="dv">-1</span>, <span class="dt">hjust =</span> <span class="dv">1</span>) <span class="op">+</span></span>
<span id="cb236-55"><a href="chapter11.html#cb236-55"></a><span class="st">    </span><span class="kw">geom_nodelab</span>(<span class="kw">aes</span>(<span class="dt">subset =</span> (node <span class="op">==</span><span class="st"> </span><span class="dv">162</span>), <span class="dt">label =</span> <span class="st">&quot;98/97/1.00&quot;</span>), </span>
<span id="cb236-56"><a href="chapter11.html#cb236-56"></a>                    <span class="dt">color =</span> <span class="st">&quot;black&quot;</span>, <span class="dt">nudge_x =</span> <span class="fl">-0.0001</span>, <span class="dt">nudge_y =</span> <span class="dv">-1</span>, <span class="dt">hjust =</span> <span class="dv">1</span>) <span class="op">+</span></span>
<span id="cb236-57"><a href="chapter11.html#cb236-57"></a><span class="st">    </span><span class="kw">geom_nodelab</span>(<span class="kw">aes</span>(<span class="dt">subset =</span> (node <span class="op">==</span><span class="st"> </span><span class="dv">172</span>), <span class="dt">label =</span> <span class="st">&quot;*&quot;</span>), </span>
<span id="cb236-58"><a href="chapter11.html#cb236-58"></a>                    <span class="dt">color =</span> <span class="st">&quot;black&quot;</span>, <span class="dt">nudge_x =</span> <span class="fl">-.0003</span>, <span class="dt">nudge_y =</span> <span class="dv">-1</span>) </span></code></pre></div>
<div class="sourceCode" id="cb237"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb237-1"><a href="chapter11.html#cb237-1"></a><span class="co">## extract accession numbers from tip labels</span></span>
<span id="cb237-2"><a href="chapter11.html#cb237-2"></a>tl &lt;-<span class="st"> </span>tree<span class="op">$</span>tip.label</span>
<span id="cb237-3"><a href="chapter11.html#cb237-3"></a>acc &lt;-<span class="st"> </span><span class="kw">sub</span>(<span class="st">&quot;</span><span class="ch">\\</span><span class="st">w+</span><span class="ch">\\</span><span class="st">|&quot;</span>, <span class="st">&quot;&quot;</span>, tl)</span>
<span id="cb237-4"><a href="chapter11.html#cb237-4"></a><span class="kw">names</span>(tl) &lt;-<span class="st"> </span>acc</span>
<span id="cb237-5"><a href="chapter11.html#cb237-5"></a></span>
<span id="cb237-6"><a href="chapter11.html#cb237-6"></a><span class="co">## read sequences from GenBank directly into R</span></span>
<span id="cb237-7"><a href="chapter11.html#cb237-7"></a><span class="co">## and convert the object to DNAStringSet</span></span>
<span id="cb237-8"><a href="chapter11.html#cb237-8"></a>tipseq &lt;-<span class="st"> </span>ape<span class="op">::</span><span class="kw">read.GenBank</span>(acc) <span class="op">%&gt;%</span><span class="st"> </span>as.character <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb237-9"><a href="chapter11.html#cb237-9"></a><span class="st">    </span><span class="kw">lapply</span>(., paste0, <span class="dt">collapse =</span> <span class="st">&quot;&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span>unlist <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb237-10"><a href="chapter11.html#cb237-10"></a><span class="st">    </span>DNAStringSet</span>
<span id="cb237-11"><a href="chapter11.html#cb237-11"></a><span class="co">## align the sequences using muscle</span></span>
<span id="cb237-12"><a href="chapter11.html#cb237-12"></a>tipseq_aln &lt;-<span class="st"> </span>muscle<span class="op">::</span><span class="kw">muscle</span>(tipseq)</span>
<span id="cb237-13"><a href="chapter11.html#cb237-13"></a>tipseq_aln &lt;-<span class="st"> </span><span class="kw">DNAStringSet</span>(tipseq_aln)</span></code></pre></div>


<div class="sourceCode" id="cb238"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb238-1"><a href="chapter11.html#cb238-1"></a><span class="co">## calculate pairwise hamming distances among sequences</span></span>
<span id="cb238-2"><a href="chapter11.html#cb238-2"></a>tipseq_dist &lt;-<span class="st"> </span><span class="kw">stringDist</span>(tipseq_aln, <span class="dt">method =</span> <span class="st">&quot;hamming&quot;</span>)</span>
<span id="cb238-3"><a href="chapter11.html#cb238-3"></a></span>
<span id="cb238-4"><a href="chapter11.html#cb238-4"></a><span class="co">## calculate percentage of differences</span></span>
<span id="cb238-5"><a href="chapter11.html#cb238-5"></a>tipseq_d &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(tipseq_dist) <span class="op">/</span><span class="st"> </span><span class="kw">width</span>(tipseq_aln[<span class="dv">1</span>]) <span class="op">*</span><span class="st"> </span><span class="dv">100</span></span>
<span id="cb238-6"><a href="chapter11.html#cb238-6"></a></span>
<span id="cb238-7"><a href="chapter11.html#cb238-7"></a><span class="co">## convert the matrix to tidy data frame for facet_plot</span></span>
<span id="cb238-8"><a href="chapter11.html#cb238-8"></a>dd &lt;-<span class="st"> </span><span class="kw">as_tibble</span>(tipseq_d)</span>
<span id="cb238-9"><a href="chapter11.html#cb238-9"></a>dd<span class="op">$</span>seq1 &lt;-<span class="st"> </span><span class="kw">rownames</span>(tipseq_d)</span>
<span id="cb238-10"><a href="chapter11.html#cb238-10"></a>td &lt;-<span class="st"> </span><span class="kw">gather</span>(dd,seq2, dist, <span class="op">-</span>seq1)</span>
<span id="cb238-11"><a href="chapter11.html#cb238-11"></a>td<span class="op">$</span>seq1 &lt;-<span class="st"> </span>tl[td<span class="op">$</span>seq1]</span>
<span id="cb238-12"><a href="chapter11.html#cb238-12"></a>td<span class="op">$</span>seq2 &lt;-<span class="st"> </span>tl[td<span class="op">$</span>seq2]</span>
<span id="cb238-13"><a href="chapter11.html#cb238-13"></a></span>
<span id="cb238-14"><a href="chapter11.html#cb238-14"></a>g &lt;-<span class="st"> </span>p<span class="op">$</span>data<span class="op">$</span>group</span>
<span id="cb238-15"><a href="chapter11.html#cb238-15"></a><span class="kw">names</span>(g) &lt;-<span class="st"> </span>p<span class="op">$</span>data<span class="op">$</span>label</span>
<span id="cb238-16"><a href="chapter11.html#cb238-16"></a>td<span class="op">$</span>clade &lt;-<span class="st"> </span>g[td<span class="op">$</span>seq2] </span>
<span id="cb238-17"><a href="chapter11.html#cb238-17"></a></span>
<span id="cb238-18"><a href="chapter11.html#cb238-18"></a><span class="co">## visualize the sequence differences using dot plot and line plot</span></span>
<span id="cb238-19"><a href="chapter11.html#cb238-19"></a><span class="co">## and align the sequence difference plot to the tree using facet_plot</span></span>
<span id="cb238-20"><a href="chapter11.html#cb238-20"></a>p2 &lt;-<span class="st"> </span><span class="kw">facet_plot</span>(p, <span class="dt">panel =</span> <span class="st">&quot;Sequence Distance&quot;</span>, <span class="dt">data =</span> td, geom_point, </span>
<span id="cb238-21"><a href="chapter11.html#cb238-21"></a>            <span class="dt">mapping =</span> <span class="kw">aes</span>(<span class="dt">x =</span> dist, <span class="dt">color =</span> clade, <span class="dt">shape =</span> clade), <span class="dt">alpha =</span> <span class="fl">.6</span>) <span class="op">%&gt;%</span></span>
<span id="cb238-22"><a href="chapter11.html#cb238-22"></a><span class="st">    </span><span class="kw">facet_plot</span>(<span class="dt">panel =</span> <span class="st">&quot;Sequence Distance&quot;</span>, <span class="dt">data =</span> td, <span class="dt">geom =</span> geom_path, </span>
<span id="cb238-23"><a href="chapter11.html#cb238-23"></a>            <span class="dt">mapping=</span><span class="kw">aes</span>(<span class="dt">x =</span> dist, <span class="dt">group =</span> seq2, <span class="dt">color =</span> clade), <span class="dt">alpha =</span> <span class="fl">.6</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb238-24"><a href="chapter11.html#cb238-24"></a><span class="st">    </span><span class="kw">scale_shape_manual</span>(<span class="dt">values =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">8</span>, <span class="dt">guide =</span> <span class="ot">FALSE</span>) </span>
<span id="cb238-25"><a href="chapter11.html#cb238-25"></a></span>
<span id="cb238-26"><a href="chapter11.html#cb238-26"></a><span class="kw">print</span>(p2)</span></code></pre></div>
<div class="figure"><span id="fig:jv2017"></span>
<img src="treedata_files/figure-html/jv2017-1.png" alt="Phylogeny of HPV58 complete genomes with dot and line plots of pairwise nucleotide sequence distances." width="1152" />
<p class="caption">
Figure 11.1: <strong>Phylogeny of HPV58 complete genomes with dot and line plots of pairwise nucleotide sequence distances</strong>.
</p>
</div>
</div>
<div id="displaying-different-symbolic-points-for-bootstrap-values." class="section level2">
<h2><span class="header-section-number">11.2</span> Displaying different symbolic points for bootstrap values.</h2>
<p>We can cut the bootstrap values into several intervals, <em>e.g.</em> to indicate whether the clade is high, moderate or low support. Then we can use these intervals as categorical variable to set different color or shape of symbolic points to indicate the bootstrap values belong to which category.</p>


<div class="sourceCode" id="cb239"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb239-1"><a href="chapter11.html#cb239-1"></a><span class="kw">library</span>(treeio)</span>
<span id="cb239-2"><a href="chapter11.html#cb239-2"></a><span class="kw">library</span>(ggplot2)</span>
<span id="cb239-3"><a href="chapter11.html#cb239-3"></a><span class="kw">library</span>(ggtree)</span>
<span id="cb239-4"><a href="chapter11.html#cb239-4"></a></span>
<span id="cb239-5"><a href="chapter11.html#cb239-5"></a>tree &lt;-<span class="st"> </span><span class="kw">read.newick</span>(<span class="st">&quot;data/RMI.phy_phyml_tree_rooted_labeled.txt&quot;</span>, <span class="dt">node.label=</span><span class="st">&#39;support&#39;</span>) </span>
<span id="cb239-6"><a href="chapter11.html#cb239-6"></a>root &lt;-<span class="st"> </span><span class="kw">rootnode</span>(tree)  </span>
<span id="cb239-7"><a href="chapter11.html#cb239-7"></a><span class="kw">ggtree</span>(tree, <span class="dt">color=</span><span class="st">&quot;black&quot;</span>, <span class="dt">size=</span><span class="fl">1.5</span>, <span class="dt">linetype=</span><span class="dv">1</span>,  <span class="dt">right=</span><span class="ot">TRUE</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb239-8"><a href="chapter11.html#cb239-8"></a><span class="st">    </span><span class="kw">geom_tiplab</span>(<span class="dt">size=</span><span class="fl">4.5</span>, <span class="dt">hjust =</span> <span class="fl">-0.060</span>, <span class="dt">fontface=</span><span class="st">&quot;bold&quot;</span>) <span class="op">+</span><span class="st">  </span><span class="kw">xlim</span>(<span class="dv">0</span>, <span class="fl">0.09</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb239-9"><a href="chapter11.html#cb239-9"></a><span class="st">    </span><span class="kw">geom_point2</span>(<span class="kw">aes</span>(<span class="dt">subset=</span><span class="op">!</span>isTip <span class="op">&amp;</span><span class="st"> </span>node <span class="op">!=</span><span class="st"> </span>root, </span>
<span id="cb239-10"><a href="chapter11.html#cb239-10"></a>                    <span class="dt">fill=</span><span class="kw">cut</span>(support, <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">700</span>, <span class="dv">900</span>, <span class="dv">1000</span>))), </span>
<span id="cb239-11"><a href="chapter11.html#cb239-11"></a>                    <span class="dt">shape=</span><span class="dv">21</span>, <span class="dt">size=</span><span class="dv">4</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb239-12"><a href="chapter11.html#cb239-12"></a><span class="st">    </span><span class="kw">theme_tree</span>(<span class="dt">legend.position=</span><span class="kw">c</span>(<span class="fl">0.2</span>, <span class="fl">0.2</span>)) <span class="op">+</span><span class="st"> </span></span>
<span id="cb239-13"><a href="chapter11.html#cb239-13"></a><span class="st">    </span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(<span class="st">&quot;white&quot;</span>, <span class="st">&quot;grey&quot;</span>, <span class="st">&quot;black&quot;</span>), <span class="dt">guide=</span><span class="st">&#39;legend&#39;</span>, </span>
<span id="cb239-14"><a href="chapter11.html#cb239-14"></a>                    <span class="dt">name=</span><span class="st">&#39;Bootstrap Percentage(BP)&#39;</span>, </span>
<span id="cb239-15"><a href="chapter11.html#cb239-15"></a>                    <span class="dt">breaks=</span><span class="kw">c</span>(<span class="st">&#39;(900,1e+03]&#39;</span>, <span class="st">&#39;(700,900]&#39;</span>, <span class="st">&#39;(0,700]&#39;</span>), </span>
<span id="cb239-16"><a href="chapter11.html#cb239-16"></a>                    <span class="dt">labels=</span><span class="kw">expression</span>(BP<span class="op">&gt;=</span><span class="dv">90</span>,<span class="dv">70</span> <span class="op">&lt;=</span><span class="st"> </span>BP <span class="op">*</span><span class="st"> &quot; &lt; 90&quot;</span>, BP <span class="op">&lt;</span><span class="st"> </span><span class="dv">70</span>))</span></code></pre></div>
<div class="figure"><span id="fig:bpinterval"></span>
<img src="treedata_files/figure-html/bpinterval-1.jpeg" alt="Partitioning bootstrap values. Bootstrap values were divided into three categories and this information was used to color circle points." width="720" />
<p class="caption">
Figure 11.2: <strong>Partitioning bootstrap values</strong>. Bootstrap values were divided into three categories and this information was used to color circle points.
</p>
</div>
</div>
<div id="genome-locus" class="section level2">
<h2><span class="header-section-number">11.3</span> Phylogenetic tree with genome locus structure</h2>
<p>The <code>geom_motif</code> is defined in <code>ggtree</code> and it is a wrapper layer of <code>gggenes::geom_gene_arrow</code>. The <code>geom_motif</code> can automatically adjust genomic alignment by selective gene (via the <code>on</code> parameter) and can label genes via the <code>label</code> parameter. In the following example, we use <code>example_genes</code> dataset provided by <a href="https://CRAN.R-project.org/package=gggenes">gggenes</a>. As the dataset only provide genomic coordinations of a set of genes, a phylogeny for the genomes need to be firstly constructed. We calculate jaccard similarity based on the ratio of overlapping genes among genomes and correspondingly determine genome distance. BioNJ algorithm was applied to construct the tree (Figure <a href="chapter11.html#fig:gggenes">11.3</a>).</p>


<div class="sourceCode" id="cb240"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb240-1"><a href="chapter11.html#cb240-1"></a><span class="kw">library</span>(dplyr)</span>
<span id="cb240-2"><a href="chapter11.html#cb240-2"></a><span class="kw">library</span>(ggplot2)</span>
<span id="cb240-3"><a href="chapter11.html#cb240-3"></a><span class="kw">library</span>(gggenes)</span>
<span id="cb240-4"><a href="chapter11.html#cb240-4"></a><span class="kw">library</span>(ggtree)</span>
<span id="cb240-5"><a href="chapter11.html#cb240-5"></a></span>
<span id="cb240-6"><a href="chapter11.html#cb240-6"></a>get_genes &lt;-<span class="st"> </span><span class="cf">function</span>(data, genome) {</span>
<span id="cb240-7"><a href="chapter11.html#cb240-7"></a>    <span class="kw">filter</span>(data, molecule <span class="op">==</span><span class="st"> </span>genome) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">pull</span>(gene)</span>
<span id="cb240-8"><a href="chapter11.html#cb240-8"></a>}</span>
<span id="cb240-9"><a href="chapter11.html#cb240-9"></a></span>
<span id="cb240-10"><a href="chapter11.html#cb240-10"></a>g &lt;-<span class="st"> </span><span class="kw">unique</span>(example_genes[,<span class="dv">1</span>])</span>
<span id="cb240-11"><a href="chapter11.html#cb240-11"></a>n &lt;-<span class="st"> </span><span class="kw">length</span>(g)</span>
<span id="cb240-12"><a href="chapter11.html#cb240-12"></a>d &lt;-<span class="st"> </span><span class="kw">matrix</span>(<span class="dt">nrow =</span> n, <span class="dt">ncol =</span> n)</span>
<span id="cb240-13"><a href="chapter11.html#cb240-13"></a><span class="kw">rownames</span>(d) &lt;-<span class="st"> </span><span class="kw">colnames</span>(d) &lt;-<span class="st"> </span>g</span>
<span id="cb240-14"><a href="chapter11.html#cb240-14"></a>genes &lt;-<span class="st"> </span><span class="kw">lapply</span>(g, get_genes, <span class="dt">data =</span> example_genes)</span>
<span id="cb240-15"><a href="chapter11.html#cb240-15"></a></span>
<span id="cb240-16"><a href="chapter11.html#cb240-16"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span>n) {</span>
<span id="cb240-17"><a href="chapter11.html#cb240-17"></a>    <span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span>i) {</span>
<span id="cb240-18"><a href="chapter11.html#cb240-18"></a>        jaccard_sim &lt;-<span class="st"> </span><span class="kw">length</span>(<span class="kw">intersect</span>(genes[[i]], genes[[j]])) <span class="op">/</span><span class="st"> </span></span>
<span id="cb240-19"><a href="chapter11.html#cb240-19"></a><span class="st">                       </span><span class="kw">length</span>(<span class="kw">union</span>(genes[[i]], genes[[j]]))</span>
<span id="cb240-20"><a href="chapter11.html#cb240-20"></a>        d[j, i] &lt;-<span class="st"> </span>d[i, j] &lt;-<span class="st"> </span><span class="dv">1</span> <span class="op">-</span><span class="st"> </span>jaccard_sim</span>
<span id="cb240-21"><a href="chapter11.html#cb240-21"></a>    }</span>
<span id="cb240-22"><a href="chapter11.html#cb240-22"></a>}</span>
<span id="cb240-23"><a href="chapter11.html#cb240-23"></a></span>
<span id="cb240-24"><a href="chapter11.html#cb240-24"></a>tree &lt;-<span class="st"> </span>ape<span class="op">::</span><span class="kw">bionj</span>(d) </span>
<span id="cb240-25"><a href="chapter11.html#cb240-25"></a></span>
<span id="cb240-26"><a href="chapter11.html#cb240-26"></a>p &lt;-<span class="st"> </span><span class="kw">ggtree</span>(tree, <span class="dt">branch.length=</span><span class="st">&#39;none&#39;</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb240-27"><a href="chapter11.html#cb240-27"></a><span class="st">    </span><span class="kw">geom_tiplab</span>() <span class="op">+</span><span class="st"> </span><span class="kw">xlim_tree</span>(<span class="fl">5.5</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb240-28"><a href="chapter11.html#cb240-28"></a><span class="st">    </span><span class="kw">geom_facet</span>(<span class="dt">mapping =</span> <span class="kw">aes</span>(<span class="dt">xmin =</span> start, <span class="dt">xmax =</span> end, <span class="dt">fill =</span> gene),</span>
<span id="cb240-29"><a href="chapter11.html#cb240-29"></a>               <span class="dt">data =</span> example_genes, <span class="dt">geom =</span> geom_motif, <span class="dt">panel =</span> <span class="st">&#39;Alignment&#39;</span>,</span>
<span id="cb240-30"><a href="chapter11.html#cb240-30"></a>               <span class="dt">on =</span> <span class="st">&#39;genE&#39;</span>, <span class="dt">label =</span> <span class="st">&#39;gene&#39;</span>, <span class="dt">align =</span> <span class="st">&#39;left&#39;</span>) <span class="op">+</span></span>
<span id="cb240-31"><a href="chapter11.html#cb240-31"></a><span class="st">    </span><span class="kw">scale_fill_brewer</span>(<span class="dt">palette =</span> <span class="st">&quot;Set3&quot;</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb240-32"><a href="chapter11.html#cb240-32"></a><span class="st">    </span><span class="kw">scale_x_continuous</span>(<span class="dt">expand=</span><span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="op">+</span></span>
<span id="cb240-33"><a href="chapter11.html#cb240-33"></a><span class="st">    </span><span class="kw">theme</span>(<span class="dt">strip.text=</span><span class="kw">element_blank</span>(),</span>
<span id="cb240-34"><a href="chapter11.html#cb240-34"></a>        <span class="dt">panel.spacing=</span><span class="kw">unit</span>(<span class="dv">0</span>, <span class="st">&#39;cm&#39;</span>))</span>
<span id="cb240-35"><a href="chapter11.html#cb240-35"></a></span>
<span id="cb240-36"><a href="chapter11.html#cb240-36"></a><span class="kw">facet_widths</span>(p, <span class="dt">widths=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span></code></pre></div>
<div class="figure"><span id="fig:gggenes"></span>
<img src="treedata_files/figure-html/gggenes-1.png" alt="Genomic features with phylogenetic tree." width="864" />
<p class="caption">
Figure 11.3: <strong>Genomic features with phylogenetic tree.</strong>
</p>
</div>

</div>
</div>
<h3>References</h3>
<div id="refs" class="references">
<div id="ref-chen_ancient_2017">
<p>Chen, Zigui, Wendy C. S. Ho, Siaw Shi Boon, Priscilla T. Y. Law, Martin C. W. Chan, Rob DeSalle, Robert D. Burk, and Paul K. S. Chan. 2017. Ancient Evolution and Dispersion of Human Papillomavirus 58 Variants. <em>Journal of Virology</em> 91 (21): e0128517. <a href="https://doi.org/10.1128/JVI.01285-17">https://doi.org/10.1128/JVI.01285-17</a>.</p>
</div>
<div id="ref-yu_two_2018">
<p>Yu, Guangchuang, Tommy Tsan-Yuk Lam, Huachen Zhu, and Yi Guan. 2018. Two Methods for Mapping and Visualizing Associated Data on Phylogeny Using Ggtree. <em>Molecular Biology and Evolution</em> 35 (12): 30413. <a href="https://doi.org/10.1093/molbev/msy194">https://doi.org/10.1093/molbev/msy194</a>.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="chapter10.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="chapter12.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"download": ["treedata.pdf", "treedata.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
