<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 Phylogenetic Tree Annotation | Data Integration, Manipulation and Visualization of Phylogenetic Trees</title>
  <meta name="description" content="Chapter 5 Phylogenetic Tree Annotation | Data Integration, Manipulation and Visualization of Phylogenetic Trees" />
  <meta name="generator" content="bookdown 0.13 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 Phylogenetic Tree Annotation | Data Integration, Manipulation and Visualization of Phylogenetic Trees" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Phylogenetic Tree Annotation | Data Integration, Manipulation and Visualization of Phylogenetic Trees" />
  
  
  

<meta name="author" content="Guangchuang Yu School of Basic Medical Sciences, Southern Medical University" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="chapter4.html"/>
<link rel="next" href="chapter6.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />







<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<script src="libs/htmlwidgets-1.3/htmlwidgets.js"></script>
<script src="libs/pathseg-1.0.2/pathseg.js"></script>
<script src="libs/comic-0.961/comic.min.js"></script>
<script src="libs/comicR-binding-0.2/comicR.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="part"><span><b>Part I: Tree data input, output and manipulation</b></span></li>
<li class="chapter" data-level="1" data-path="chapter1.html"><a href="chapter1.html"><i class="fa fa-check"></i><b>1</b> Importing Tree with Data</a><ul>
<li class="chapter" data-level="1.1" data-path="chapter1.html"><a href="chapter1.html#pc"><i class="fa fa-check"></i><b>1.1</b> Overview of Phylogenetic Tree Construction</a></li>
<li class="chapter" data-level="1.2" data-path="chapter1.html"><a href="chapter1.html#format"><i class="fa fa-check"></i><b>1.2</b> Phylogenetic Tree Formats</a><ul>
<li class="chapter" data-level="1.2.1" data-path="chapter1.html"><a href="chapter1.html#newick-tree-format"><i class="fa fa-check"></i><b>1.2.1</b> Newick tree format</a></li>
<li class="chapter" data-level="1.2.2" data-path="chapter1.html"><a href="chapter1.html#nexus-tree-format"><i class="fa fa-check"></i><b>1.2.2</b> NEXUS tree format</a></li>
<li class="chapter" data-level="1.2.3" data-path="chapter1.html"><a href="chapter1.html#nhxtext"><i class="fa fa-check"></i><b>1.2.3</b> New Hampshire eXtended format</a></li>
<li class="chapter" data-level="1.2.4" data-path="chapter1.html"><a href="chapter1.html#jplace-format"><i class="fa fa-check"></i><b>1.2.4</b> Jplace format</a></li>
<li class="chapter" data-level="1.2.5" data-path="chapter1.html"><a href="chapter1.html#software-outputs"><i class="fa fa-check"></i><b>1.2.5</b> Software outputs</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="chapter1.html"><a href="chapter1.html#getting-tree-data-with-treeio"><i class="fa fa-check"></i><b>1.3</b> Getting Tree Data with <em>treeio</em></a><ul>
<li class="chapter" data-level="1.3.1" data-path="chapter1.html"><a href="chapter1.html#overview-of-treeio"><i class="fa fa-check"></i><b>1.3.1</b> Overview of <em>treeio</em></a></li>
<li class="chapter" data-level="1.3.2" data-path="chapter1.html"><a href="chapter1.html#function-demonstration"><i class="fa fa-check"></i><b>1.3.2</b> Function Demonstration</a></li>
<li class="chapter" data-level="1.3.3" data-path="chapter1.html"><a href="chapter1.html#getting-information-from-treedata-object"><i class="fa fa-check"></i><b>1.3.3</b> Getting information from <em>treedata</em> object</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="chapter1.html"><a href="chapter1.html#summary"><i class="fa fa-check"></i><b>1.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="chapter2.html"><a href="chapter2.html"><i class="fa fa-check"></i><b>2</b> Manipulating Tree with Data</a><ul>
<li class="chapter" data-level="2.1" data-path="chapter2.html"><a href="chapter2.html#tidytree"><i class="fa fa-check"></i><b>2.1</b> Manipulating tree data using tidy interface</a><ul>
<li class="chapter" data-level="2.1.1" data-path="chapter2.html"><a href="chapter2.html#the-phylo-object"><i class="fa fa-check"></i><b>2.1.1</b> The <code>phylo</code> object</a></li>
<li class="chapter" data-level="2.1.2" data-path="chapter2.html"><a href="chapter2.html#the-treedata-object"><i class="fa fa-check"></i><b>2.1.2</b> The <code>treedata</code> object</a></li>
<li class="chapter" data-level="2.1.3" data-path="chapter2.html"><a href="chapter2.html#accesor-tidytree"><i class="fa fa-check"></i><b>2.1.3</b> Access related nodes</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="chapter2.html"><a href="chapter2.html#data-integration"><i class="fa fa-check"></i><b>2.2</b> Data Integration</a><ul>
<li class="chapter" data-level="2.2.1" data-path="chapter2.html"><a href="chapter2.html#merge-tree"><i class="fa fa-check"></i><b>2.2.1</b> Combining tree data</a></li>
<li class="chapter" data-level="2.2.2" data-path="chapter2.html"><a href="chapter2.html#link-external-data"><i class="fa fa-check"></i><b>2.2.2</b> Linking external data to phylogeny</a></li>
<li class="chapter" data-level="2.2.3" data-path="chapter2.html"><a href="chapter2.html#grouping-taxa"><i class="fa fa-check"></i><b>2.2.3</b> Grouping taxa</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="chapter2.html"><a href="chapter2.html#rescale-treeio"><i class="fa fa-check"></i><b>2.3</b> Rescaling Tree Branches</a></li>
<li class="chapter" data-level="2.4" data-path="chapter2.html"><a href="chapter2.html#subsetting-tree-with-data"><i class="fa fa-check"></i><b>2.4</b> Subsetting Tree with Data</a><ul>
<li class="chapter" data-level="2.4.1" data-path="chapter2.html"><a href="chapter2.html#remove-tip"><i class="fa fa-check"></i><b>2.4.1</b> Removing tips in a phylogenetic tree</a></li>
<li class="chapter" data-level="2.4.2" data-path="chapter2.html"><a href="chapter2.html#subset-tip"><i class="fa fa-check"></i><b>2.4.2</b> Subsetting tree by tip label</a></li>
<li class="chapter" data-level="2.4.3" data-path="chapter2.html"><a href="chapter2.html#subset-node"><i class="fa fa-check"></i><b>2.4.3</b> Subsetting tree by internal node number</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="chapter2.html"><a href="chapter2.html#ggtree-fortify"><i class="fa fa-check"></i><b>2.5</b> Manipulating tree data for visualization</a></li>
<li class="chapter" data-level="2.6" data-path="chapter2.html"><a href="chapter2.html#summary-1"><i class="fa fa-check"></i><b>2.6</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="chapter3.html"><a href="chapter3.html"><i class="fa fa-check"></i><b>3</b> Exporting tree with data</a><ul>
<li class="chapter" data-level="3.1" data-path="chapter3.html"><a href="chapter3.html#introduction"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="chapter3.html"><a href="chapter3.html#exporting-tree-data-to-beast-nexus-format"><i class="fa fa-check"></i><b>3.2</b> Exporting Tree Data to <em>BEAST</em> Nexus Format</a><ul>
<li class="chapter" data-level="3.2.1" data-path="chapter3.html"><a href="chapter3.html#exportingconverting-software-output"><i class="fa fa-check"></i><b>3.2.1</b> Exporting/converting software output</a></li>
<li class="chapter" data-level="3.2.2" data-path="chapter3.html"><a href="chapter3.html#combining-tree-with-external-data"><i class="fa fa-check"></i><b>3.2.2</b> Combining tree with external data</a></li>
<li class="chapter" data-level="3.2.3" data-path="chapter3.html"><a href="chapter3.html#merging-tree-data-from-different-sources"><i class="fa fa-check"></i><b>3.2.3</b> Merging tree data from different sources</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="chapter3.html"><a href="chapter3.html#write-jtree"><i class="fa fa-check"></i><b>3.3</b> Exporting Tree Data to <em>jtree</em> Format</a></li>
<li class="chapter" data-level="3.4" data-path="chapter3.html"><a href="chapter3.html#summary-2"><i class="fa fa-check"></i><b>3.4</b> Summary</a></li>
</ul></li>
<li class="part"><span><b>Part II: Tree data visualization and annotation</b></span></li>
<li class="chapter" data-level="4" data-path="chapter4.html"><a href="chapter4.html"><i class="fa fa-check"></i><b>4</b> Phylogenetic Tree Visualization</a><ul>
<li class="chapter" data-level="4.1" data-path="chapter4.html"><a href="chapter4.html#introduction-1"><i class="fa fa-check"></i><b>4.1</b> Introduction</a></li>
<li class="chapter" data-level="4.2" data-path="chapter4.html"><a href="chapter4.html#visualizing-phylogenetic-tree-with-ggtree"><i class="fa fa-check"></i><b>4.2</b> Visualizing Phylogenetic Tree with <em>ggtree</em></a><ul>
<li class="chapter" data-level="4.2.1" data-path="chapter4.html"><a href="chapter4.html#basic-tree-visualization"><i class="fa fa-check"></i><b>4.2.1</b> Basic Tree Visualization</a></li>
<li class="chapter" data-level="4.2.2" data-path="chapter4.html"><a href="chapter4.html#tree-layouts"><i class="fa fa-check"></i><b>4.2.2</b> Layouts of phylogenetic tree</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="chapter4.html"><a href="chapter4.html#displaying-tree-components"><i class="fa fa-check"></i><b>4.3</b> Displaying Tree Components</a><ul>
<li class="chapter" data-level="4.3.1" data-path="chapter4.html"><a href="chapter4.html#geom-trescale"><i class="fa fa-check"></i><b>4.3.1</b> Displaying tree scale (evolution distance)</a></li>
<li class="chapter" data-level="4.3.2" data-path="chapter4.html"><a href="chapter4.html#geom-nodepoint"><i class="fa fa-check"></i><b>4.3.2</b> Displaying nodes/tips</a></li>
<li class="chapter" data-level="4.3.3" data-path="chapter4.html"><a href="chapter4.html#displaying-labels"><i class="fa fa-check"></i><b>4.3.3</b> Displaying labels</a></li>
<li class="chapter" data-level="4.3.4" data-path="chapter4.html"><a href="chapter4.html#displaying-root-edge"><i class="fa fa-check"></i><b>4.3.4</b> Displaying root edge</a></li>
<li class="chapter" data-level="4.3.5" data-path="chapter4.html"><a href="chapter4.html#color-tree"><i class="fa fa-check"></i><b>4.3.5</b> Color tree</a></li>
<li class="chapter" data-level="4.3.6" data-path="chapter4.html"><a href="chapter4.html#rescale-tree"><i class="fa fa-check"></i><b>4.3.6</b> Rescale tree</a></li>
<li class="chapter" data-level="4.3.7" data-path="chapter4.html"><a href="chapter4.html#modify-compenents-of-a-theme"><i class="fa fa-check"></i><b>4.3.7</b> Modify compenents of a theme</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="chapter4.html"><a href="chapter4.html#visualize-a-list-of-trees"><i class="fa fa-check"></i><b>4.4</b> Visualize a list of trees</a></li>
<li class="chapter" data-level="4.5" data-path="chapter4.html"><a href="chapter4.html#summary-3"><i class="fa fa-check"></i><b>4.5</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="chapter5.html"><a href="chapter5.html"><i class="fa fa-check"></i><b>5</b> Phylogenetic Tree Annotation</a><ul>
<li class="chapter" data-level="5.1" data-path="chapter5.html"><a href="chapter5.html#visualizing-and-annotating-tree-using-grammar-of-graphics"><i class="fa fa-check"></i><b>5.1</b> Visualizing and Annotating Tree using Grammar of Graphics</a></li>
<li class="chapter" data-level="5.2" data-path="chapter5.html"><a href="chapter5.html#layers-for-tree-annotation"><i class="fa fa-check"></i><b>5.2</b> Layers for Tree Annotation</a><ul>
<li class="chapter" data-level="5.2.1" data-path="chapter5.html"><a href="chapter5.html#colored-strips"><i class="fa fa-check"></i><b>5.2.1</b> Colored strips</a></li>
<li class="chapter" data-level="5.2.2" data-path="chapter5.html"><a href="chapter5.html#highlight-clades"><i class="fa fa-check"></i><b>5.2.2</b> Highlight clades</a></li>
<li class="chapter" data-level="5.2.3" data-path="chapter5.html"><a href="chapter5.html#taxa-connection"><i class="fa fa-check"></i><b>5.2.3</b> Taxa connection</a></li>
<li class="chapter" data-level="5.2.4" data-path="chapter5.html"><a href="chapter5.html#uncertainty-of-evolutionary-inference"><i class="fa fa-check"></i><b>5.2.4</b> Uncertainty of evolutionary inference</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="chapter5.html"><a href="chapter5.html#tree-annotation-with-output-from-evolution-software"><i class="fa fa-check"></i><b>5.3</b> Tree annotation with output from evolution software</a><ul>
<li class="chapter" data-level="5.3.1" data-path="chapter5.html"><a href="chapter5.html#tree-annotation-using-data-from-evolutionary-analysis-software"><i class="fa fa-check"></i><b>5.3.1</b> Tree annotation using data from evolutionary analysis software</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="chapter5.html"><a href="chapter5.html#summary-4"><i class="fa fa-check"></i><b>5.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="chapter6.html"><a href="chapter6.html"><i class="fa fa-check"></i><b>6</b> Visual Exploration of Phylogenetic Tree</a><ul>
<li class="chapter" data-level="6.1" data-path="chapter6.html"><a href="chapter6.html#viewing-selected-clade"><i class="fa fa-check"></i><b>6.1</b> Viewing Selected Clade</a></li>
<li class="chapter" data-level="6.2" data-path="chapter6.html"><a href="chapter6.html#scale-clade"><i class="fa fa-check"></i><b>6.2</b> Scaling Selected Clade</a></li>
<li class="chapter" data-level="6.3" data-path="chapter6.html"><a href="chapter6.html#collapsing-and-expanding-clade"><i class="fa fa-check"></i><b>6.3</b> Collapsing and Expanding Clade</a></li>
<li class="chapter" data-level="6.4" data-path="chapter6.html"><a href="chapter6.html#group-taxa-vis"><i class="fa fa-check"></i><b>6.4</b> Grouping Taxa</a></li>
<li class="chapter" data-level="6.5" data-path="chapter6.html"><a href="chapter6.html#exploring-tree-structure"><i class="fa fa-check"></i><b>6.5</b> Exploring tree structure</a></li>
<li class="chapter" data-level="6.6" data-path="chapter6.html"><a href="chapter6.html#summary-5"><i class="fa fa-check"></i><b>6.6</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="chapter7.html"><a href="chapter7.html"><i class="fa fa-check"></i><b>7</b> Plotting tree with data</a><ul>
<li class="chapter" data-level="7.1" data-path="chapter7.html"><a href="chapter7.html#attach-operator"><i class="fa fa-check"></i><b>7.1</b> Mapping data to the tree structure</a></li>
<li class="chapter" data-level="7.2" data-path="chapter7.html"><a href="chapter7.html#facet_plot"><i class="fa fa-check"></i><b>7.2</b> Alignning graph to the tree based on tree structure</a></li>
<li class="chapter" data-level="7.3" data-path="chapter7.html"><a href="chapter7.html#gheatmap"><i class="fa fa-check"></i><b>7.3</b> Visualize tree with associated matrix</a><ul>
<li class="chapter" data-level="7.3.1" data-path="chapter7.html"><a href="chapter7.html#gheatmap-ggnewscale"><i class="fa fa-check"></i><b>7.3.1</b> Visualize tree with multiple associated matrix</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="chapter7.html"><a href="chapter7.html#msaplot"><i class="fa fa-check"></i><b>7.4</b> Visualize tree with multiple sequence alignment</a></li>
<li class="chapter" data-level="7.5" data-path="chapter7.html"><a href="chapter7.html#ggtree_object"><i class="fa fa-check"></i><b>7.5</b> The <code>ggtree</code> object</a></li>
<li class="chapter" data-level="7.6" data-path="chapter7.html"><a href="chapter7.html#summary-6"><i class="fa fa-check"></i><b>7.6</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="chapter8.html"><a href="chapter8.html"><i class="fa fa-check"></i><b>8</b> Annotating Tree with Silhouette Images and Sub-plots</a><ul>
<li class="chapter" data-level="8.1" data-path="chapter8.html"><a href="chapter8.html#ggimage"><i class="fa fa-check"></i><b>8.1</b> Annotating Tree with Images</a></li>
<li class="chapter" data-level="8.2" data-path="chapter8.html"><a href="chapter8.html#phylopic"><i class="fa fa-check"></i><b>8.2</b> Annotating Tree with Phylopic</a></li>
<li class="chapter" data-level="8.3" data-path="chapter8.html"><a href="chapter8.html#subplots"><i class="fa fa-check"></i><b>8.3</b> Annotating Tree with Sub-plots</a><ul>
<li class="chapter" data-level="8.3.1" data-path="chapter8.html"><a href="chapter8.html#annotate-with-bar-charts"><i class="fa fa-check"></i><b>8.3.1</b> Annotate with bar charts</a></li>
<li class="chapter" data-level="8.3.2" data-path="chapter8.html"><a href="chapter8.html#annotate-with-pie-charts"><i class="fa fa-check"></i><b>8.3.2</b> Annotate with pie charts</a></li>
<li class="chapter" data-level="8.3.3" data-path="chapter8.html"><a href="chapter8.html#annotate-with-other-types-of-charts"><i class="fa fa-check"></i><b>8.3.3</b> Annotate with other types of charts</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="chapter8.html"><a href="chapter8.html#phylomoji"><i class="fa fa-check"></i><b>8.4</b> Have Fun with Phylomoji</a><ul>
<li class="chapter" data-level="8.4.1" data-path="chapter8.html"><a href="chapter8.html#emoji-in-cirularfan-layout-tree"><i class="fa fa-check"></i><b>8.4.1</b> emoji in cirular/fan layout tree</a></li>
<li class="chapter" data-level="8.4.2" data-path="chapter8.html"><a href="chapter8.html#emoji-to-label-clades"><i class="fa fa-check"></i><b>8.4.2</b> emoji to label clades</a></li>
<li class="chapter" data-level="8.4.3" data-path="chapter8.html"><a href="chapter8.html#apple-color-emoji"><i class="fa fa-check"></i><b>8.4.3</b> Apple Color Emoji</a></li>
<li class="chapter" data-level="8.4.4" data-path="chapter8.html"><a href="chapter8.html#phylomoji-in-ascii-art"><i class="fa fa-check"></i><b>8.4.4</b> Phylomoji in ASCII art</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="chapter9.html"><a href="chapter9.html"><i class="fa fa-check"></i><b>9</b> ggtree for other tree-like objects</a><ul>
<li class="chapter" data-level="9.1" data-path="chapter9.html"><a href="chapter9.html#ggtree-for-phylogenetic-tree-objects"><i class="fa fa-check"></i><b>9.1</b> ggtree for phylogenetic tree objects</a><ul>
<li class="chapter" data-level="9.1.1" data-path="chapter9.html"><a href="chapter9.html#phylobase"><i class="fa fa-check"></i><b>9.1.1</b> The phylo4 and phylo4d objects</a></li>
<li class="chapter" data-level="9.1.2" data-path="chapter9.html"><a href="chapter9.html#phylog"><i class="fa fa-check"></i><b>9.1.2</b> The phylog object</a></li>
<li class="chapter" data-level="9.1.3" data-path="chapter9.html"><a href="chapter9.html#phyloseq"><i class="fa fa-check"></i><b>9.1.3</b> The phyloseq object</a></li>
<li class="chapter" data-level="9.1.4" data-path="chapter9.html"><a href="chapter9.html#obkdata"><i class="fa fa-check"></i><b>9.1.4</b> The obkData object</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="chapter9.html"><a href="chapter9.html#dendrogram"><i class="fa fa-check"></i><b>9.2</b> ggtree for dendrograms</a></li>
</ul></li>
<li class="part"><span><b>Part III: Miscellaneous topics</b></span></li>
<li class="chapter" data-level="10" data-path="chapter10.html"><a href="chapter10.html"><i class="fa fa-check"></i><b>10</b> ggtree utilities</a><ul>
<li class="chapter" data-level="10.1" data-path="chapter10.html"><a href="chapter10.html#facet-utilities"><i class="fa fa-check"></i><b>10.1</b> facet utilities</a><ul>
<li class="chapter" data-level="10.1.1" data-path="chapter10.html"><a href="chapter10.html#facet_widths"><i class="fa fa-check"></i><b>10.1.1</b> facet_widths</a></li>
<li class="chapter" data-level="10.1.2" data-path="chapter10.html"><a href="chapter10.html#facet_labeller"><i class="fa fa-check"></i><b>10.1.2</b> facet_labeller</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="chapter10.html"><a href="chapter10.html#layout-utilities"><i class="fa fa-check"></i><b>10.2</b> layout utilities</a></li>
<li class="chapter" data-level="10.3" data-path="chapter10.html"><a href="chapter10.html#legend-utilities"><i class="fa fa-check"></i><b>10.3</b> legend utilities</a></li>
<li class="chapter" data-level="10.4" data-path="chapter10.html"><a href="chapter10.html#scale-utilities"><i class="fa fa-check"></i><b>10.4</b> scale utilities</a><ul>
<li class="chapter" data-level="10.4.1" data-path="chapter10.html"><a href="chapter10.html#xlim_expand"><i class="fa fa-check"></i><b>10.4.1</b> xlim_expand</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="chapter11.html"><a href="chapter11.html"><i class="fa fa-check"></i><b>11</b> Gallery of reproducible examples</a><ul>
<li class="chapter" data-level="11.1" data-path="chapter11.html"><a href="chapter11.html#visualizing-pairwise-nucleotide-sequence-distance-with-phylogenetic-tree"><i class="fa fa-check"></i><b>11.1</b> Visualizing pairwise nucleotide sequence distance with phylogenetic tree</a></li>
<li class="chapter" data-level="11.2" data-path="chapter11.html"><a href="chapter11.html#displaying-different-symbolic-points-for-bootstrap-values."><i class="fa fa-check"></i><b>11.2</b> Displaying different symbolic points for bootstrap values.</a></li>
<li class="chapter" data-level="11.3" data-path="chapter11.html"><a href="chapter11.html#genome-locus"><i class="fa fa-check"></i><b>11.3</b> Phylogenetic tree with genome locus structure</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="chapter12.html"><a href="chapter12.html"><i class="fa fa-check"></i><b>12</b> ggtree extensions</a></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="faq.html"><a href="faq.html"><i class="fa fa-check"></i><b>A</b> Frequently asked questions</a><ul>
<li class="chapter" data-level="A.1" data-path="faq.html"><a href="faq.html#installation"><i class="fa fa-check"></i><b>A.1</b> Installation</a></li>
<li class="chapter" data-level="A.2" data-path="faq.html"><a href="faq.html#faq-r"><i class="fa fa-check"></i><b>A.2</b> Basic R related</a><ul>
<li class="chapter" data-level="A.2.1" data-path="faq.html"><a href="faq.html#faq-local-file"><i class="fa fa-check"></i><b>A.2.1</b> Use your local file</a></li>
</ul></li>
<li class="chapter" data-level="A.3" data-path="faq.html"><a href="faq.html#faq-aes-mapping"><i class="fa fa-check"></i><b>A.3</b> Aesthetic mapping</a><ul>
<li class="chapter" data-level="A.3.1" data-path="faq.html"><a href="faq.html#faq-inherit-aes"><i class="fa fa-check"></i><b>A.3.1</b> Inherit <em>aes</em></a></li>
<li class="chapter" data-level="A.3.2" data-path="faq.html"><a href="faq.html#faq-dollar-aes"><i class="fa fa-check"></i><b>A.3.2</b> Never use <code>$</code> in aes</a></li>
</ul></li>
<li class="chapter" data-level="A.4" data-path="faq.html"><a href="faq.html#faq-text"><i class="fa fa-check"></i><b>A.4</b> Text &amp; Label</a><ul>
<li class="chapter" data-level="A.4.1" data-path="faq.html"><a href="faq.html#faq-label-truncated"><i class="fa fa-check"></i><b>A.4.1</b> Tip label truncated</a></li>
<li class="chapter" data-level="A.4.2" data-path="faq.html"><a href="faq.html#faq-modify-label"><i class="fa fa-check"></i><b>A.4.2</b> Modify (tip) labels</a></li>
<li class="chapter" data-level="A.4.3" data-path="faq.html"><a href="faq.html#faq-formatting-label"><i class="fa fa-check"></i><b>A.4.3</b> Formatting (tip) labels</a></li>
<li class="chapter" data-level="A.4.4" data-path="faq.html"><a href="faq.html#faq-ggrepel"><i class="fa fa-check"></i><b>A.4.4</b> Avoid overlapping text labels</a></li>
<li class="chapter" data-level="A.4.5" data-path="faq.html"><a href="faq.html#faq-bootstrap"><i class="fa fa-check"></i><b>A.4.5</b> Bootstrap values from newick format</a></li>
</ul></li>
<li class="chapter" data-level="A.5" data-path="faq.html"><a href="faq.html#faq-under-the-tree"><i class="fa fa-check"></i><b>A.5</b> Plot something behind the phylogeny</a></li>
<li class="chapter" data-level="A.6" data-path="faq.html"><a href="faq.html#faq-enlarge-center-space"><i class="fa fa-check"></i><b>A.6</b> Enlarge center space in circular/fan layout tree</a></li>
<li class="chapter" data-level="A.7" data-path="faq.html"><a href="faq.html#use-the-most-distant-tip-from-the-root-as-the-origin-of-the-time-scale"><i class="fa fa-check"></i><b>A.7</b> Use the most distant tip from the root as the origin of the time scale</a></li>
<li class="chapter" data-level="A.8" data-path="faq.html"><a href="faq.html#export-edit"><i class="fa fa-check"></i><b>A.8</b> Edit tree graphic details</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="ggimage-tips.html"><a href="ggimage-tips.html"><i class="fa fa-check"></i><b>B</b> Tips for using <code>ggtree</code> with <code>ggimage</code></a><ul>
<li class="chapter" data-level="B.1" data-path="ggimage-tips.html"><a href="ggimage-tips.html#ggimage-rm-image-bg"><i class="fa fa-check"></i><b>B.1</b> Example 1: Remove background of images</a></li>
<li class="chapter" data-level="B.2" data-path="ggimage-tips.html"><a href="ggimage-tips.html#ggimage-bgimage"><i class="fa fa-check"></i><b>B.2</b> Example 2: Plot tree on a background image</a></li>
</ul></li>
<li class="chapter" data-level="C" data-path="commicR.html"><a href="commicR.html"><i class="fa fa-check"></i><b>C</b> Comic (xkcd-like) phylogenetic tree</a></li>
<li class="chapter" data-level="D" data-path="ascii-tree.html"><a href="ascii-tree.html"><i class="fa fa-check"></i><b>D</b> Print ASCII-art rooted tree</a></li>
<li class="chapter" data-level="E" data-path="facet-zoom.html"><a href="facet-zoom.html"><i class="fa fa-check"></i><b>E</b> Zoom in selected portion</a></li>
<li class="chapter" data-level="F" data-path="run-ggtree-in-jupyter-notebook.html"><a href="run-ggtree-in-jupyter-notebook.html"><i class="fa fa-check"></i><b>F</b> Run ggtree in Jupyter notebook</a></li>
<li class="chapter" data-level="G" data-path="figures-and-tables.html"><a href="figures-and-tables.html"><i class="fa fa-check"></i><b>G</b> Figures and Tables</a></li>
<li class="chapter" data-level="H" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>H</b> Session Info</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Data Integration, Manipulation and Visualization of Phylogenetic Trees</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="chapter5" class="section level1">
<h1><span class="header-section-number">Chapter 5</span> Phylogenetic Tree Annotation</h1>
<div id="visualizing-and-annotating-tree-using-grammar-of-graphics" class="section level2">
<h2><span class="header-section-number">5.1</span> Visualizing and Annotating Tree using Grammar of Graphics</h2>
<p>The <em>ggtree</em> is designed for more general purpose or specific type of tree visualization and annotation. It supports grammar of graphics implemented in <em>ggplot2</em> and users can freely visualize/annotate a tree by combining several annotation layers.</p>



<div class="sourceCode" id="cb154"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb154-1"><a href="chapter5.html#cb154-1"></a><span class="kw">library</span>(ggtree)</span>
<span id="cb154-2"><a href="chapter5.html#cb154-2"></a>treetext =<span class="st"> &quot;(((ADH2:0.1[&amp;&amp;NHX:S=human], ADH1:0.11[&amp;&amp;NHX:S=human]):</span></span>
<span id="cb154-3"><a href="chapter5.html#cb154-3"></a><span class="st">0.05 [&amp;&amp;NHX:S=primates:D=Y:B=100],ADHY:</span></span>
<span id="cb154-4"><a href="chapter5.html#cb154-4"></a><span class="st">0.1[&amp;&amp;NHX:S=nematode],ADHX:0.12 [&amp;&amp;NHX:S=insect]):</span></span>
<span id="cb154-5"><a href="chapter5.html#cb154-5"></a><span class="st">0.1[&amp;&amp;NHX:S=metazoa:D=N],(ADH4:0.09[&amp;&amp;NHX:S=yeast],</span></span>
<span id="cb154-6"><a href="chapter5.html#cb154-6"></a><span class="st">ADH3:0.13[&amp;&amp;NHX:S=yeast], ADH2:0.12[&amp;&amp;NHX:S=yeast],</span></span>
<span id="cb154-7"><a href="chapter5.html#cb154-7"></a><span class="st">ADH1:0.11[&amp;&amp;NHX:S=yeast]):0.1[&amp;&amp;NHX:S=Fungi])[&amp;&amp;NHX:D=N];&quot;</span></span>
<span id="cb154-8"><a href="chapter5.html#cb154-8"></a>tree &lt;-<span class="st"> </span><span class="kw">read.nhx</span>(<span class="kw">textConnection</span>(treetext))</span>
<span id="cb154-9"><a href="chapter5.html#cb154-9"></a><span class="kw">ggtree</span>(tree) <span class="op">+</span><span class="st"> </span><span class="kw">geom_tiplab</span>() <span class="op">+</span><span class="st"> </span></span>
<span id="cb154-10"><a href="chapter5.html#cb154-10"></a><span class="st">  </span><span class="kw">geom_label</span>(<span class="kw">aes</span>(<span class="dt">x=</span>branch, <span class="dt">label=</span>S), <span class="dt">fill=</span><span class="st">&#39;lightgreen&#39;</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb154-11"><a href="chapter5.html#cb154-11"></a><span class="st">  </span><span class="kw">geom_label</span>(<span class="kw">aes</span>(<span class="dt">label=</span>D), <span class="dt">fill=</span><span class="st">&#39;steelblue&#39;</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb154-12"><a href="chapter5.html#cb154-12"></a><span class="st">  </span><span class="kw">geom_text</span>(<span class="kw">aes</span>(<span class="dt">label=</span>B), <span class="dt">hjust=</span><span class="op">-</span>.<span class="dv">5</span>)</span></code></pre></div>
<div class="figure"><span id="fig:ggtreeNHX"></span>
<img src="treedata_files/figure-html/ggtreeNHX-1.png" alt="Annotating tree using grammar of graphics. The NHX tree was annotated using grammar of graphic syntax by combining different layers using + operator. Species information were labelled on the middle of the branches, Duplication events were shown on most recent common ancestor and clade bootstrap value were dispalyed near to it." width="672"  />
<p class="caption">
Figure 5.1: <strong>Annotating tree using grammar of graphics.</strong> The NHX tree was annotated using grammar of graphic syntax by combining different layers using <code>+</code> operator. Species information were labelled on the middle of the branches, Duplication events were shown on most recent common ancestor and clade bootstrap value were dispalyed near to it.
</p>
</div>

<p>Here, as an example, we visualized the tree with several layers to display annotation stored in NHX tags, including a layer of <code>geom_tiplab</code> to display tip labels (gene name in this case), a layer using <code>geom_label</code> to show species information (<code>S</code> tag) colored by lightgreen, a layer of duplication event information (<code>D</code> tag) colored by steelblue and another layer using <em>geom_text</em> to show bootstrap value (<code>B</code> tag).</p>
<p>Layers defined in <em>ggplot2</em> can be applied to <em>ggtree</em> directly as demonstrated in Figure <a href="chapter5.html#fig:ggtreeNHX">5.1</a> of using <em>geom_label</em> and <em>geom_text</em>. But <em>ggplot2</em> does not provide graphic layers that are specific designed for phylogenetic tree annotation. For instance, layers for tip labels, tree branch scale legend, highlight or labeling clade are all unavailable. To make tree annotation more flexible, a number of layers have been implemented in <em>ggtree</em> (Table <a href="chapter5.html#tab:geoms">5.1</a>), enabling different ways of annotation on various parts/components of a phylogenetic tree.</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:geoms">Table 5.1: </span>Geom layers defined in ggtree.
</caption>
<thead>
<tr>
<th style="text-align:left;">
Layer
</th>
<th style="text-align:left;">
Description
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
geom_balance
</td>
<td style="text-align:left;">
highlights the two direct descendant clades of an internal node
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_cladelabel
</td>
<td style="text-align:left;">
annotate a clade with bar and text label
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_facet
</td>
<td style="text-align:left;">
plot associated data in specific panel (facet) and align the plot with the tree
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_hilight
</td>
<td style="text-align:left;">
highlight a clade with rectangle
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_inset
</td>
<td style="text-align:left;">
add insets (subplots) to tree nodes
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_label2
</td>
<td style="text-align:left;">
modified version of geom_label, with subsetting supported
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_nodepoint
</td>
<td style="text-align:left;">
annotate internal nodes with symbolic points
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_point2
</td>
<td style="text-align:left;">
modified version of geom_point, with subsetting supported
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_range
</td>
<td style="text-align:left;">
bar layer to present uncertainty of evolutionary inference
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_rootpoint
</td>
<td style="text-align:left;">
annotate root node with symbolic point
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_rootedge
</td>
<td style="text-align:left;">
add root edge to a tree
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_segment2
</td>
<td style="text-align:left;">
modified version of geom_segment, with subsetting supported
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_strip
</td>
<td style="text-align:left;">
annotate associated taxa with bar and (optional) text label
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_taxalink
</td>
<td style="text-align:left;">
associate two related taxa by linking them with a curve
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_text2
</td>
<td style="text-align:left;">
modified version of geom_text, with subsetting supported
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_tiplab
</td>
<td style="text-align:left;">
layer of tip labels
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_tiplab2
</td>
<td style="text-align:left;">
layer of tip labels for circular layout
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_tippoint
</td>
<td style="text-align:left;">
annotate external nodes with symbolic points
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_tree
</td>
<td style="text-align:left;">
tree structure layer, with multiple layout supported
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_treescale
</td>
<td style="text-align:left;">
tree branch scale legend
</td>
</tr>
</tbody>
</table>
</div>
<div id="layers-for-tree-annotation" class="section level2">
<h2><span class="header-section-number">5.2</span> Layers for Tree Annotation</h2>
<div id="colored-strips" class="section level3">
<h3><span class="header-section-number">5.2.1</span> Colored strips</h3>
<p><a href="http://bioconductor.org/packages/ggtree">ggtree</a> <span class="citation">(Yu et al. <a href="#ref-yu_ggtree:_2017" role="doc-biblioref">2017</a>)</span> implements <em><code>geom_cladelabel</code></em> layer to annotate a selected clade with a bar indicating the clade with a corresponding label.</p>
<p>The <em><code>geom_cladelabel</code></em> layer accepts a selected internal node number and label corresponding clade automatically (Figure <a href="chapter5.html#fig:cladelabel">5.2</a>A). To get the internal node number, please refer to <a href="chapter2.html#accesor-tidytree">Chapter 2</a>.</p>
<div class="sourceCode" id="cb155"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb155-1"><a href="chapter5.html#cb155-1"></a><span class="kw">set.seed</span>(<span class="dv">2015-12-21</span>)</span>
<span id="cb155-2"><a href="chapter5.html#cb155-2"></a>tree &lt;-<span class="st"> </span><span class="kw">rtree</span>(<span class="dv">30</span>)</span>
<span id="cb155-3"><a href="chapter5.html#cb155-3"></a>p &lt;-<span class="st"> </span><span class="kw">ggtree</span>(tree) <span class="op">+</span><span class="st"> </span><span class="kw">xlim</span>(<span class="ot">NA</span>, <span class="dv">6</span>)</span>
<span id="cb155-4"><a href="chapter5.html#cb155-4"></a></span>
<span id="cb155-5"><a href="chapter5.html#cb155-5"></a>p <span class="op">+</span><span class="st"> </span><span class="kw">geom_cladelabel</span>(<span class="dt">node=</span><span class="dv">45</span>, <span class="dt">label=</span><span class="st">&quot;test label&quot;</span>) <span class="op">+</span></span>
<span id="cb155-6"><a href="chapter5.html#cb155-6"></a><span class="st">    </span><span class="kw">geom_cladelabel</span>(<span class="dt">node=</span><span class="dv">34</span>, <span class="dt">label=</span><span class="st">&quot;another clade&quot;</span>)</span></code></pre></div>
<p>Users can set the parameter, <code>align = TRUE</code>, to align the clade label, <code>offset</code>, to adjust the position and color to set the color of bar and label text <em>etc</em> (Figure <a href="chapter5.html#fig:cladelabel">5.2</a>B).</p>
<div class="sourceCode" id="cb156"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb156-1"><a href="chapter5.html#cb156-1"></a>p <span class="op">+</span><span class="st"> </span><span class="kw">geom_cladelabel</span>(<span class="dt">node=</span><span class="dv">45</span>, <span class="dt">label=</span><span class="st">&quot;test label&quot;</span>, <span class="dt">align=</span><span class="ot">TRUE</span>,  <span class="dt">offset =</span> <span class="fl">.2</span>, <span class="dt">color=</span><span class="st">&#39;red&#39;</span>) <span class="op">+</span></span>
<span id="cb156-2"><a href="chapter5.html#cb156-2"></a><span class="st">    </span><span class="kw">geom_cladelabel</span>(<span class="dt">node=</span><span class="dv">34</span>, <span class="dt">label=</span><span class="st">&quot;another clade&quot;</span>, <span class="dt">align=</span><span class="ot">TRUE</span>, <span class="dt">offset =</span> <span class="fl">.2</span>, <span class="dt">color=</span><span class="st">&#39;blue&#39;</span>)</span></code></pre></div>
<p>Users can change the <code>angle</code> of the clade label text and relative position from text to bar via the parameter <code>offset.text</code>. The size of the bar and text can be changed via the parameters <code>barsize</code> and <code>fontsize</code> respectively (Figure <a href="chapter5.html#fig:cladelabel">5.2</a>C).</p>
<div class="sourceCode" id="cb157"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb157-1"><a href="chapter5.html#cb157-1"></a>p <span class="op">+</span><span class="st"> </span><span class="kw">geom_cladelabel</span>(<span class="dt">node=</span><span class="dv">45</span>, <span class="dt">label=</span><span class="st">&quot;test label&quot;</span>, <span class="dt">align=</span>T, <span class="dt">angle=</span><span class="dv">270</span>, <span class="dt">hjust=</span><span class="st">&#39;center&#39;</span>, <span class="dt">offset.text=</span>.<span class="dv">5</span>, <span class="dt">barsize=</span><span class="fl">1.5</span>) <span class="op">+</span></span>
<span id="cb157-2"><a href="chapter5.html#cb157-2"></a><span class="st">    </span><span class="kw">geom_cladelabel</span>(<span class="dt">node=</span><span class="dv">34</span>, <span class="dt">label=</span><span class="st">&quot;another clade&quot;</span>, <span class="dt">align=</span>T, <span class="dt">angle=</span><span class="dv">45</span>, <span class="dt">fontsize=</span><span class="dv">8</span>)</span></code></pre></div>
<p>Users can also use <code>geom_label</code> to label the text and can set the background color by <code>fill</code> parameter (Figure <a href="chapter5.html#fig:cladelabel">5.2</a>D).</p>
<div class="sourceCode" id="cb158"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb158-1"><a href="chapter5.html#cb158-1"></a>p <span class="op">+</span><span class="st"> </span><span class="kw">geom_cladelabel</span>(<span class="dt">node=</span><span class="dv">34</span>, <span class="dt">label=</span><span class="st">&quot;another clade&quot;</span>, <span class="dt">align=</span>T, <span class="dt">geom=</span><span class="st">&#39;label&#39;</span>, <span class="dt">fill=</span><span class="st">&#39;lightblue&#39;</span>)</span></code></pre></div>


<div class="figure"><span id="fig:cladelabel"></span>
<img src="treedata_files/figure-html/cladelabel-1.png" alt="Labeling clades." width="1152" />
<p class="caption">
Figure 5.2: <strong>Labeling clades.</strong>
</p>
</div>
<p><code>geom_cladelabel</code> also supports unrooted tree layouts (Figure <a href="chapter5.html#fig:striplabel">5.3</a>A).</p>
<div class="sourceCode" id="cb159"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb159-1"><a href="chapter5.html#cb159-1"></a><span class="kw">ggtree</span>(tree, <span class="dt">layout=</span><span class="st">&quot;daylight&quot;</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb159-2"><a href="chapter5.html#cb159-2"></a><span class="st">  </span><span class="kw">geom_cladelabel</span>(<span class="dt">node=</span><span class="dv">35</span>, <span class="dt">label=</span><span class="st">&quot;test label&quot;</span>, <span class="dt">angle=</span><span class="dv">0</span>, </span>
<span id="cb159-3"><a href="chapter5.html#cb159-3"></a>                  <span class="dt">fontsize=</span><span class="dv">8</span>, <span class="dt">offset=</span>.<span class="dv">5</span>, <span class="dt">vjust=</span>.<span class="dv">5</span>)  <span class="op">+</span><span class="st"> </span></span>
<span id="cb159-4"><a href="chapter5.html#cb159-4"></a><span class="st">  </span><span class="kw">geom_cladelabel</span>(<span class="dt">node=</span><span class="dv">55</span>, <span class="dt">label=</span><span class="st">&#39;another clade&#39;</span>, </span>
<span id="cb159-5"><a href="chapter5.html#cb159-5"></a>                  <span class="dt">angle=</span><span class="op">-</span><span class="dv">95</span>, <span class="dt">hjust=</span>.<span class="dv">5</span>, <span class="dt">fontsize=</span><span class="dv">8</span>)</span></code></pre></div>
<p><code>geom_cladelabel</code> is designed for labeling Monophyletic (Clade) while there are related taxa that are not form a clade. <code>ggtree</code> provides <code>geom_strip</code> to add a strip/bar to indicate the association with optional label for Polyphyletic or Paraphyletic (Figure <a href="chapter5.html#fig:striplabel">5.3</a>B).</p>
<div class="sourceCode" id="cb160"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb160-1"><a href="chapter5.html#cb160-1"></a>p <span class="op">+</span><span class="st"> </span><span class="kw">geom_tiplab</span>() <span class="op">+</span><span class="st"> </span></span>
<span id="cb160-2"><a href="chapter5.html#cb160-2"></a><span class="st">  </span><span class="kw">geom_strip</span>(<span class="st">&#39;t10&#39;</span>, <span class="st">&#39;t30&#39;</span>, <span class="dt">barsize=</span><span class="dv">2</span>, <span class="dt">color=</span><span class="st">&#39;red&#39;</span>, </span>
<span id="cb160-3"><a href="chapter5.html#cb160-3"></a>            <span class="dt">label=</span><span class="st">&quot;associated taxa&quot;</span>, <span class="dt">offset.text=</span>.<span class="dv">1</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb160-4"><a href="chapter5.html#cb160-4"></a><span class="st">  </span><span class="kw">geom_strip</span>(<span class="st">&#39;t1&#39;</span>, <span class="st">&#39;t18&#39;</span>, <span class="dt">barsize=</span><span class="dv">2</span>, <span class="dt">color=</span><span class="st">&#39;blue&#39;</span>, </span>
<span id="cb160-5"><a href="chapter5.html#cb160-5"></a>            <span class="dt">label =</span> <span class="st">&quot;another label&quot;</span>, <span class="dt">offset.text=</span>.<span class="dv">1</span>)</span></code></pre></div>


<div class="figure"><span id="fig:striplabel"></span>
<img src="treedata_files/figure-html/striplabel-1.png" alt="Labeling associated taxa. geom_cladelabel is for labeling Monophyletic and it also supports unrooted layout (A). geom_strip is designed for labeling associated taxa (Monophyletic, Polyphyletic or Paraphyletic) (B)." width="1296" />
<p class="caption">
Figure 5.3: <strong>Labeling associated taxa.</strong> <code>geom_cladelabel</code> is for labeling Monophyletic and it also supports unrooted layout (A). <code>geom_strip</code> is designed for labeling associated taxa (Monophyletic, Polyphyletic or Paraphyletic) (B).
</p>
</div>
</div>
<div id="highlight-clades" class="section level3">
<h3><span class="header-section-number">5.2.2</span> Highlight clades</h3>
<p><code>ggtree</code> implements <em><code>geom_hilight</code></em> layer, that accepts an internal node number and add a layer of rectangle to highlight the selected clade Figure (<a href="chapter5.html#fig:hilight">5.4</a>).<a href="references.html#fn13" class="footnote-ref" id="fnref13"><sup>13</sup></a></p>
<div class="sourceCode" id="cb161"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb161-1"><a href="chapter5.html#cb161-1"></a>nwk &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;sample.nwk&quot;</span>, <span class="dt">package=</span><span class="st">&quot;treeio&quot;</span>)</span>
<span id="cb161-2"><a href="chapter5.html#cb161-2"></a>tree &lt;-<span class="st"> </span><span class="kw">read.tree</span>(nwk)</span>
<span id="cb161-3"><a href="chapter5.html#cb161-3"></a><span class="kw">ggtree</span>(tree) <span class="op">+</span><span class="st"> </span><span class="kw">geom_hilight</span>(<span class="dt">node=</span><span class="dv">21</span>, <span class="dt">fill=</span><span class="st">&quot;steelblue&quot;</span>, <span class="dt">alpha=</span>.<span class="dv">6</span>) <span class="op">+</span></span>
<span id="cb161-4"><a href="chapter5.html#cb161-4"></a><span class="st">    </span><span class="kw">geom_hilight</span>(<span class="dt">node=</span><span class="dv">17</span>, <span class="dt">fill=</span><span class="st">&quot;darkgreen&quot;</span>, <span class="dt">alpha=</span>.<span class="dv">6</span>) </span></code></pre></div>
<div class="sourceCode" id="cb162"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb162-1"><a href="chapter5.html#cb162-1"></a><span class="kw">ggtree</span>(tree, <span class="dt">layout=</span><span class="st">&quot;circular&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">geom_hilight</span>(<span class="dt">node=</span><span class="dv">21</span>, <span class="dt">fill=</span><span class="st">&quot;steelblue&quot;</span>, <span class="dt">alpha=</span>.<span class="dv">6</span>) <span class="op">+</span></span>
<span id="cb162-2"><a href="chapter5.html#cb162-2"></a><span class="st">    </span><span class="kw">geom_hilight</span>(<span class="dt">node=</span><span class="dv">23</span>, <span class="dt">fill=</span><span class="st">&quot;darkgreen&quot;</span>, <span class="dt">alpha=</span>.<span class="dv">6</span>)</span></code></pre></div>
<p>The <code>geom_hilight</code> layer also support highlighting clades for unrooted layout trees (Figure <a href="chapter5.html#fig:hilight">5.4</a>C).</p>
<div class="sourceCode" id="cb163"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb163-1"><a href="chapter5.html#cb163-1"></a>pg <span class="op">+</span><span class="st"> </span><span class="kw">geom_hilight</span>(<span class="dt">node=</span><span class="dv">55</span>) <span class="op">+</span><span class="st"> </span><span class="kw">geom_hilight</span>(<span class="dt">node=</span><span class="dv">35</span>, <span class="dt">fill=</span><span class="st">&#39;darkgreen&#39;</span>)</span></code></pre></div>
<p>Another way to highlight selected clades is setting the clades with different colors and/or line types as demonstrated in Figure <a href="chapter6.html#fig:scaleClade">6.2</a>.</p>
<p>In addition to <code>geom_hilight</code>, <code>ggtree</code> also implements <code>geom_balance</code>
which is designed to highlight neighboring subclades of a given internal node (Figure <a href="chapter5.html#fig:hilight">5.4</a>D).</p>
<div class="sourceCode" id="cb164"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb164-1"><a href="chapter5.html#cb164-1"></a><span class="kw">ggtree</span>(tree) <span class="op">+</span></span>
<span id="cb164-2"><a href="chapter5.html#cb164-2"></a><span class="st">  </span><span class="kw">geom_balance</span>(<span class="dt">node=</span><span class="dv">16</span>, <span class="dt">fill=</span><span class="st">&#39;steelblue&#39;</span>, <span class="dt">color=</span><span class="st">&#39;white&#39;</span>, <span class="dt">alpha=</span><span class="fl">0.6</span>, <span class="dt">extend=</span><span class="dv">1</span>) <span class="op">+</span></span>
<span id="cb164-3"><a href="chapter5.html#cb164-3"></a><span class="st">  </span><span class="kw">geom_balance</span>(<span class="dt">node=</span><span class="dv">19</span>, <span class="dt">fill=</span><span class="st">&#39;darkgreen&#39;</span>, <span class="dt">color=</span><span class="st">&#39;white&#39;</span>, <span class="dt">alpha=</span><span class="fl">0.6</span>, <span class="dt">extend=</span><span class="dv">1</span>) </span></code></pre></div>


<div class="figure"><span id="fig:hilight"></span>
<img src="treedata_files/figure-html/hilight-1.png" alt="Highlight selected clades. Rectangular layout (A), circular/fan (B) and unrooted layouts. Highlight neighboring subclades simultaneously (D)." width="960" />
<p class="caption">
Figure 5.4: <strong>Highlight selected clades.</strong> Rectangular layout (A), circular/fan (B) and unrooted layouts. Highlight neighboring subclades simultaneously (D).
</p>
</div>
</div>
<div id="taxa-connection" class="section level3">
<h3><span class="header-section-number">5.2.3</span> Taxa connection</h3>
<p>Some evolutionary events (e.g.reassortment, horizontal gene transfer) cannot be modeled by a simple tree. <a href="http://bioconductor.org/packages/ggtree">ggtree</a> provides <code>geom_taxalink</code> layer that allows drawing straight or curved lines between any of two nodes in the tree, allow it to represent evolutionary events by connecting taxa.</p>


<div class="sourceCode" id="cb165"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb165-1"><a href="chapter5.html#cb165-1"></a><span class="kw">ggtree</span>(tree) <span class="op">+</span><span class="st"> </span><span class="kw">geom_tiplab</span>() <span class="op">+</span><span class="st"> </span><span class="kw">geom_taxalink</span>(<span class="st">&#39;A&#39;</span>, <span class="st">&#39;E&#39;</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb165-2"><a href="chapter5.html#cb165-2"></a><span class="st">  </span><span class="kw">geom_taxalink</span>(<span class="st">&#39;F&#39;</span>, <span class="st">&#39;K&#39;</span>, <span class="dt">color=</span><span class="st">&#39;red&#39;</span>, <span class="dt">linetype =</span> <span class="st">&#39;dashed&#39;</span>,</span>
<span id="cb165-3"><a href="chapter5.html#cb165-3"></a>    <span class="dt">arrow=</span>grid<span class="op">::</span><span class="kw">arrow</span>(<span class="dt">length=</span>grid<span class="op">::</span><span class="kw">unit</span>(<span class="fl">0.02</span>, <span class="st">&quot;npc&quot;</span>)))</span></code></pre></div>
<div class="figure"><span id="fig:taxalink"></span>
<img src="treedata_files/figure-html/taxalink-1.png" alt="Linking related taxa. This can be used to indicate evolutionary events such as reassortment and horozontal gene transfer." width="480" />
<p class="caption">
Figure 5.5: <strong>Linking related taxa.</strong> This can be used to indicate evolutionary events such as reassortment and horozontal gene transfer.
</p>
</div>
</div>
<div id="uncertainty-of-evolutionary-inference" class="section level3">
<h3><span class="header-section-number">5.2.4</span> Uncertainty of evolutionary inference</h3>
<p>The <code>geom_range</code> layer supports displaying interval (highest posterior density, confidence interval, range) as horizontal bars on tree nodes. The center of the interval will anchor to corresponding node. The center by default is the mean value of the interval (Figure <a href="chapter5.html#fig:geomRange">5.6</a>A). We can set the <code>center</code> to estimated mean or median value (Figure <a href="chapter5.html#fig:geomRange">5.6</a>B), or observed value. As the tree branch and the interval may not be in the same scale, <a href="http://bioconductor.org/packages/ggtree">ggtree</a> provides <code>scale_x_range</code> to add second x axis for the range (Figure <a href="chapter5.html#fig:geomRange">5.6</a>C). Note that x axis is disable by default theme and we need to enable it if we want to dispaly it (<em>e.g.</em> <code>theme_tree2</code>).</p>


<pre eval="FALSE"><code>file &lt;- system.file(&quot;extdata/MEGA7&quot;, &quot;mtCDNA_timetree.nex&quot;, package = &quot;treeio&quot;)
x &lt;- read.mega(file)
p1 &lt;- ggtree(x) + geom_range(&#39;reltime_0.95_CI&#39;, color=&#39;red&#39;, size=3, alpha=.3)
p2 &lt;- ggtree(x) + geom_range(&#39;reltime_0.95_CI&#39;, color=&#39;red&#39;, size=3, alpha=.3, center=&#39;reltime&#39;)  
p3 &lt;- p2 + scale_x_range() + theme_tree2()</code></pre>
<div class="figure"><span id="fig:geomRange"></span>
<img src="treedata_files/figure-html/geomRange-1.png" alt="Displaying uncertainty of evolutoinary inference. The center (mean value of the range (A) or estimated value (B)) is anchor to the tree nodes. A second x axis was used for range scaling (C)." width="1152" />
<p class="caption">
Figure 5.6: <strong>Displaying uncertainty of evolutoinary inference.</strong> The center (mean value of the range (A) or estimated value (B)) is anchor to the tree nodes. A second x axis was used for range scaling (C).
</p>
</div>
</div>
</div>
<div id="tree-annotation-with-output-from-evolution-software" class="section level2">
<h2><span class="header-section-number">5.3</span> Tree annotation with output from evolution software</h2>
<div id="tree-annotation-using-data-from-evolutionary-analysis-software" class="section level3">
<h3><span class="header-section-number">5.3.1</span> Tree annotation using data from evolutionary analysis software</h3>
<p><a href="#chapter">Chapter 1</a> introduced using <a href="http://bioconductor.org/packages/treeio">treeio</a> packages to parse different tree formats and commonly used software outputs to obtain phylogeny-associated data. These imported data as <code>S4</code> objects can be visualized directly using <a href="http://bioconductor.org/packages/ggtree">ggtree</a>. Figure <a href="chapter5.html#fig:ggtreeNHX">5.1</a> demonstrates a tree annotated using the information (species classification, duplication event and bootstrap value) stored in NHX file. <em>PHYLDOG</em> and <em>RevBayes</em> output NHX files that can be parsed by <a href="http://bioconductor.org/packages/treeio">treeio</a> and visualized by <a href="http://bioconductor.org/packages/ggtree">ggtree</a> with annotation using their inference data.</p>
<p>Furthermore, the evolutionary data from the inference of <em>BEAST</em>, <em>MrBayes</em> and <em>RevBayes</em>, <em>d<sub>N</sub>/d<sub>S</sub></em> values inferred by <em>CodeML</em>, ancestral sequences inferred by <em>HyPhy</em>, <em>CodeML</em> or <em>BaseML</em> and short read placement by <em>EPA</em> and <em>pplacer</em> can be used to annotate the tree directly.</p>



<div class="sourceCode" id="cb167"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb167-1"><a href="chapter5.html#cb167-1"></a>file &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata/BEAST&quot;</span>, <span class="st">&quot;beast_mcc.tree&quot;</span>, <span class="dt">package=</span><span class="st">&quot;treeio&quot;</span>)</span>
<span id="cb167-2"><a href="chapter5.html#cb167-2"></a>beast &lt;-<span class="st"> </span><span class="kw">read.beast</span>(file)</span>
<span id="cb167-3"><a href="chapter5.html#cb167-3"></a><span class="kw">ggtree</span>(beast, <span class="kw">aes</span>(<span class="dt">color=</span>rate))  <span class="op">+</span></span>
<span id="cb167-4"><a href="chapter5.html#cb167-4"></a><span class="st">    </span><span class="kw">geom_range</span>(<span class="dt">range=</span><span class="st">&#39;length_0.95_HPD&#39;</span>, <span class="dt">color=</span><span class="st">&#39;red&#39;</span>, <span class="dt">alpha=</span>.<span class="dv">6</span>, <span class="dt">size=</span><span class="dv">2</span>) <span class="op">+</span></span>
<span id="cb167-5"><a href="chapter5.html#cb167-5"></a><span class="st">    </span><span class="kw">geom_nodelab</span>(<span class="kw">aes</span>(<span class="dt">x=</span>branch, <span class="dt">label=</span><span class="kw">round</span>(posterior, <span class="dv">2</span>)), <span class="dt">vjust=</span><span class="op">-</span>.<span class="dv">5</span>, <span class="dt">size=</span><span class="dv">3</span>) <span class="op">+</span></span>
<span id="cb167-6"><a href="chapter5.html#cb167-6"></a><span class="st">    </span><span class="kw">scale_color_continuous</span>(<span class="dt">low=</span><span class="st">&quot;darkgreen&quot;</span>, <span class="dt">high=</span><span class="st">&quot;red&quot;</span>) <span class="op">+</span></span>
<span id="cb167-7"><a href="chapter5.html#cb167-7"></a><span class="st">    </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="kw">c</span>(.<span class="dv">1</span>, <span class="fl">.8</span>))</span></code></pre></div>
<div class="figure"><span id="fig:beast"></span>
<img src="treedata_files/figure-html/beast-1.png" alt="Annotating BEAST tree with length_95%_HPD and posterior. Branch length credible intervals (95% HPD) were displayed as red horizontal bars and clade posterior values were shown on the middle of branches." width="672"  />
<p class="caption">
Figure 5.7: <strong>Annotating <em>BEAST</em> tree with <em>length_95%_HPD</em> and posterior.</strong> Branch length credible intervals (95% HPD) were displayed as red horizontal bars and clade posterior values were shown on the middle of branches.
</p>
</div>

<p>In Figure <a href="chapter5.html#fig:beast">5.7</a>, the tree was visualized and annotated with posterior &gt; 0.9 and demonstrated length uncertainty (95% Highest Posterior Density (HPD) interval).</p>
<p>Ancestral sequences inferred by <em>HyPhy</em> can be parsed using <a href="chapter1.html#parsing-hyphy-output"><em>treeio</em></a>, whereas the substitutions along each tree branch was automatically computed and stored inside the phylogenetic tree object (<em>i.e.</em>, <code>S4</code> object). The <em>ggtree</em> can utilize this information in the object to annotate the tree, as demonstrated in Figure <a href="chapter5.html#fig:hyphy">5.8</a>.</p>



<div class="sourceCode" id="cb168"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb168-1"><a href="chapter5.html#cb168-1"></a>nwk &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata/HYPHY&quot;</span>, <span class="st">&quot;labelledtree.tree&quot;</span>, </span>
<span id="cb168-2"><a href="chapter5.html#cb168-2"></a>                   <span class="dt">package=</span><span class="st">&quot;treeio&quot;</span>)</span>
<span id="cb168-3"><a href="chapter5.html#cb168-3"></a>ancseq &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata/HYPHY&quot;</span>, <span class="st">&quot;ancseq.nex&quot;</span>, </span>
<span id="cb168-4"><a href="chapter5.html#cb168-4"></a>                      <span class="dt">package=</span><span class="st">&quot;treeio&quot;</span>)</span>
<span id="cb168-5"><a href="chapter5.html#cb168-5"></a>tipfas &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;pa.fas&quot;</span>, <span class="dt">package=</span><span class="st">&quot;treeio&quot;</span>)</span>
<span id="cb168-6"><a href="chapter5.html#cb168-6"></a>hy &lt;-<span class="st"> </span><span class="kw">read.hyphy</span>(nwk, ancseq, tipfas)</span>
<span id="cb168-7"><a href="chapter5.html#cb168-7"></a><span class="kw">ggtree</span>(hy) <span class="op">+</span><span class="st"> </span></span>
<span id="cb168-8"><a href="chapter5.html#cb168-8"></a><span class="st">  </span><span class="kw">geom_text</span>(<span class="kw">aes</span>(<span class="dt">x=</span>branch, <span class="dt">label=</span>AA_subs), <span class="dt">size=</span><span class="dv">2</span>, </span>
<span id="cb168-9"><a href="chapter5.html#cb168-9"></a>            <span class="dt">vjust=</span><span class="op">-</span>.<span class="dv">3</span>, <span class="dt">color=</span><span class="st">&quot;firebrick&quot;</span>)</span></code></pre></div>
<div class="figure"><span id="fig:hyphy"></span>
<img src="treedata_files/figure-html/hyphy-1.png" alt="Annotating tree with amino acid substitution determined by ancestral sequences inferred by HYPHY. Amino acid substitutions were displayed on the middle of branches." width="748.8"  />
<p class="caption">
Figure 5.8: <strong>Annotating tree with amino acid substitution determined by ancestral sequences inferred by HYPHY.</strong> Amino acid substitutions were displayed on the middle of branches.
</p>
</div>

<p><em>PAML</em>s <em>BaseML</em> and <em>CodeML</em> can be also used to infer ancestral sequences, whereas <em>CodeML</em> can infer selection pressure. After parsing this information using <a href="chapter1.html#parsing-paml-output"><em>treeio</em></a>, <em>ggtree</em> can integrate this information into the same tree structure and used for annotation as illustrated in Figure <a href="chapter5.html#fig:codeml">5.9</a>.</p>



<div class="sourceCode" id="cb169"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb169-1"><a href="chapter5.html#cb169-1"></a>rstfile &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata/PAML_Codeml&quot;</span>, <span class="st">&quot;rst&quot;</span>, </span>
<span id="cb169-2"><a href="chapter5.html#cb169-2"></a>                       <span class="dt">package=</span><span class="st">&quot;treeio&quot;</span>)</span>
<span id="cb169-3"><a href="chapter5.html#cb169-3"></a>mlcfile &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata/PAML_Codeml&quot;</span>, <span class="st">&quot;mlc&quot;</span>, </span>
<span id="cb169-4"><a href="chapter5.html#cb169-4"></a>                       <span class="dt">package=</span><span class="st">&quot;treeio&quot;</span>)</span>
<span id="cb169-5"><a href="chapter5.html#cb169-5"></a>ml &lt;-<span class="st"> </span><span class="kw">read.codeml</span>(rstfile, mlcfile)</span>
<span id="cb169-6"><a href="chapter5.html#cb169-6"></a><span class="kw">ggtree</span>(ml, <span class="kw">aes</span>(<span class="dt">color=</span>dN_vs_dS), <span class="dt">branch.length=</span><span class="st">&#39;dN_vs_dS&#39;</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb169-7"><a href="chapter5.html#cb169-7"></a><span class="st">  </span><span class="kw">scale_color_continuous</span>(<span class="dt">name=</span><span class="st">&#39;dN/dS&#39;</span>, <span class="dt">limits=</span><span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">1.5</span>),</span>
<span id="cb169-8"><a href="chapter5.html#cb169-8"></a>                         <span class="dt">oob=</span>scales<span class="op">::</span>squish,</span>
<span id="cb169-9"><a href="chapter5.html#cb169-9"></a>                         <span class="dt">low=</span><span class="st">&#39;darkgreen&#39;</span>, <span class="dt">high=</span><span class="st">&#39;red&#39;</span>) <span class="op">+</span></span>
<span id="cb169-10"><a href="chapter5.html#cb169-10"></a><span class="st">  </span><span class="kw">geom_text</span>(<span class="kw">aes</span>(<span class="dt">x=</span>branch, <span class="dt">label=</span>AA_subs), </span>
<span id="cb169-11"><a href="chapter5.html#cb169-11"></a>            <span class="dt">vjust=</span><span class="op">-</span>.<span class="dv">5</span>, <span class="dt">color=</span><span class="st">&#39;steelblue&#39;</span>, <span class="dt">size=</span><span class="dv">2</span>) <span class="op">+</span></span>
<span id="cb169-12"><a href="chapter5.html#cb169-12"></a><span class="st">  </span><span class="kw">theme_tree2</span>(<span class="dt">legend.position=</span><span class="kw">c</span>(.<span class="dv">9</span>, <span class="fl">.3</span>))</span></code></pre></div>
<div class="figure"><span id="fig:codeml"></span>
<img src="treedata_files/figure-html/codeml-1.png" alt="Annotating tree with animo acid substitution and dN/dS inferred by CodeML. Branches were rescaled and colored by dN/dS values and amino acid substitutions were displayed on the middle of branches." width="672"  />
<p class="caption">
Figure 5.9: <strong>Annotating tree with animo acid substitution and <em>d<sub>N</sub>/d<sub>S</sub></em> inferred by <em>CodeML</em>.</strong> Branches were rescaled and colored by <em>d<sub>N</sub>/d<sub>S</sub></em> values and amino acid substitutions were displayed on the middle of branches.
</p>
</div>

<p>Not only all the tree data that parsed by <a href="http://bioconductor.org/packages/treeio">treeio</a> can be used to visualize and annotate phylogenetic tree using <code>ggtree</code>, but also other tree and tree-like objects defined in R community are supported. The <a href="http://bioconductor.org/packages/ggtree">ggtree</a> plays an unique role in R ecosystem to facilitate phylogenetic analysis and it can be easily integrated into other packages and pipelines. For more details, please refer to <a href="chapter9.html#chapter9">chapter 9</a>. In addition to direct support of tree objects, <a href="http://bioconductor.org/packages/ggtree">ggtree</a> also allow users to plot tree with different types of external data (see also <a href="chapter7.html#chapter7">chapter 7</a> and <span class="citation">(Yu et al. <a href="#ref-yu_two_2018" role="doc-biblioref">2018</a>)</span>).</p>
</div>
</div>
<div id="summary-4" class="section level2">
<h2><span class="header-section-number">5.4</span> Summary</h2>
<p><a href="http://bioconductor.org/packages/ggtree">ggtree</a> implements grammar of graphics for annotating phylogenetic trees. Users can use ggplot2 syntax to combine different annotation layers to produce complex tree annotation. If you are familiar with <a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a>, tree annotation with high level of customization can be intuitive and flexible using <a href="http://bioconductor.org/packages/ggtree">ggtree</a>.</p>

</div>
</div>
<h3>References</h3>
<div id="refs" class="references">
<div id="ref-yu_two_2018">
<p>Yu, Guangchuang, Tommy Tsan-Yuk Lam, Huachen Zhu, and Yi Guan. 2018. Two Methods for Mapping and Visualizing Associated Data on Phylogeny Using Ggtree. <em>Molecular Biology and Evolution</em> 35 (12): 30413. <a href="https://doi.org/10.1093/molbev/msy194">https://doi.org/10.1093/molbev/msy194</a>.</p>
</div>
<div id="ref-yu_ggtree:_2017">
<p>Yu, Guangchuang, David K. Smith, Huachen Zhu, Yi Guan, and Tommy Tsan-Yuk Lam. 2017. Ggtree: An R Package for Visualization and Annotation of Phylogenetic Trees with Their Covariates and Other Associated Data. <em>Methods in Ecology and Evolution</em> 8 (1): 2836. <a href="https://doi.org/10.1111/2041-210X.12628">https://doi.org/10.1111/2041-210X.12628</a>.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="chapter4.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="chapter6.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"download": ["treedata.pdf", "treedata.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
