<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Chapter 1 Importing tree with data | Data Integration, Manipulation and Visualization of Phylogenetic Trees</title>
  <meta name="description" content="Chapter 1 Importing tree with data | Data Integration, Manipulation and Visualization of Phylogenetic Trees">
  <meta name="generator" content="bookdown  and GitBook 2.6.7">

  <meta property="og:title" content="Chapter 1 Importing tree with data | Data Integration, Manipulation and Visualization of Phylogenetic Trees" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 1 Importing tree with data | Data Integration, Manipulation and Visualization of Phylogenetic Trees" />
  
  
  

<meta name="author" content="Guangchuang Yu">



  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="index.html">
<link rel="next" href="manipulating-tree-with-data.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(title);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="importing-tree-with-data.html"><a href="importing-tree-with-data.html"><i class="fa fa-check"></i><b>1</b> Importing tree with data</a><ul>
<li class="chapter" data-level="1.1" data-path="importing-tree-with-data.html"><a href="importing-tree-with-data.html#introduction"><i class="fa fa-check"></i><b>1.1</b> Introduction</a></li>
<li class="chapter" data-level="1.2" data-path="importing-tree-with-data.html"><a href="importing-tree-with-data.html#getting-tree-data-from-evolutionary-analysis-result"><i class="fa fa-check"></i><b>1.2</b> Getting tree data from evolutionary analysis result</a><ul>
<li class="chapter" data-level="1.2.1" data-path="importing-tree-with-data.html"><a href="importing-tree-with-data.html#parsing-beast-output"><i class="fa fa-check"></i><b>1.2.1</b> Parsing BEAST output</a></li>
<li class="chapter" data-level="1.2.2" data-path="importing-tree-with-data.html"><a href="importing-tree-with-data.html#parsing-mrbayes-output"><i class="fa fa-check"></i><b>1.2.2</b> Parsing MrBayes output</a></li>
<li class="chapter" data-level="1.2.3" data-path="importing-tree-with-data.html"><a href="importing-tree-with-data.html#parsing-paml-output"><i class="fa fa-check"></i><b>1.2.3</b> Parsing PAML output</a></li>
<li class="chapter" data-level="1.2.4" data-path="importing-tree-with-data.html"><a href="importing-tree-with-data.html#parsing-hyphy-output"><i class="fa fa-check"></i><b>1.2.4</b> Parsing HyPhy output</a></li>
<li class="chapter" data-level="1.2.5" data-path="importing-tree-with-data.html"><a href="importing-tree-with-data.html#parsing-r8s-output"><i class="fa fa-check"></i><b>1.2.5</b> Parsing r8s output</a></li>
<li class="chapter" data-level="1.2.6" data-path="importing-tree-with-data.html"><a href="importing-tree-with-data.html#parsing-output-of-raxml-bootstraping-analysis"><i class="fa fa-check"></i><b>1.2.6</b> Parsing output of RAxML bootstraping analysis</a></li>
<li class="chapter" data-level="1.2.7" data-path="importing-tree-with-data.html"><a href="importing-tree-with-data.html#parsing-nhx-tree"><i class="fa fa-check"></i><b>1.2.7</b> Parsing NHX tree</a></li>
<li class="chapter" data-level="1.2.8" data-path="importing-tree-with-data.html"><a href="importing-tree-with-data.html#parsing-phylip-tree"><i class="fa fa-check"></i><b>1.2.8</b> Parsing Phylip tree</a></li>
<li class="chapter" data-level="1.2.9" data-path="importing-tree-with-data.html"><a href="importing-tree-with-data.html#parsing-epa-and-pplacer-output"><i class="fa fa-check"></i><b>1.2.9</b> Parsing EPA and pplacer output</a></li>
<li class="chapter" data-level="1.2.10" data-path="importing-tree-with-data.html"><a href="importing-tree-with-data.html#jtree"><i class="fa fa-check"></i><b>1.2.10</b> Parsing jtree format</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="importing-tree-with-data.html"><a href="importing-tree-with-data.html#linking-external-data-to-phylogeny"><i class="fa fa-check"></i><b>1.3</b> Linking external data to phylogeny</a></li>
<li class="chapter" data-level="1.4" data-path="importing-tree-with-data.html"><a href="importing-tree-with-data.html#combining-tree-data"><i class="fa fa-check"></i><b>1.4</b> Combining tree data</a></li>
<li class="chapter" data-level="1.5" data-path="importing-tree-with-data.html"><a href="importing-tree-with-data.html#getting-information-from-treedata-object"><i class="fa fa-check"></i><b>1.5</b> Getting information from <em>treedata</em> object</a></li>
<li class="chapter" data-level="1.6" data-path="importing-tree-with-data.html"><a href="importing-tree-with-data.html#manipulating-tree-data-using-tidytree"><i class="fa fa-check"></i><b>1.6</b> Manipulating tree data using <em>tidytree</em></a></li>
<li class="chapter" data-level="1.7" data-path="importing-tree-with-data.html"><a href="importing-tree-with-data.html#visualizing-tree-data-with-ggtree"><i class="fa fa-check"></i><b>1.7</b> Visualizing tree data with <em>ggtree</em></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="manipulating-tree-with-data.html"><a href="manipulating-tree-with-data.html"><i class="fa fa-check"></i><b>2</b> Manipulating tree with data</a><ul>
<li class="chapter" data-level="2.1" data-path="manipulating-tree-with-data.html"><a href="manipulating-tree-with-data.html#convert-tree-object-to-tidy-data-frame-and-vice-versa"><i class="fa fa-check"></i><b>2.1</b> Convert tree object to tidy data frame and vice versa</a><ul>
<li class="chapter" data-level="2.1.1" data-path="manipulating-tree-with-data.html"><a href="manipulating-tree-with-data.html#phylo-object"><i class="fa fa-check"></i><b>2.1.1</b> <code>phylo</code> object</a></li>
<li class="chapter" data-level="2.1.2" data-path="manipulating-tree-with-data.html"><a href="manipulating-tree-with-data.html#treedata-object"><i class="fa fa-check"></i><b>2.1.2</b> treedata object</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="manipulating-tree-with-data.html"><a href="manipulating-tree-with-data.html#access-related-nodes"><i class="fa fa-check"></i><b>2.2</b> Access related nodes</a></li>
<li class="chapter" data-level="2.3" data-path="manipulating-tree-with-data.html"><a href="manipulating-tree-with-data.html#grouping-taxa"><i class="fa fa-check"></i><b>2.3</b> Grouping taxa</a><ul>
<li class="chapter" data-level="2.3.1" data-path="manipulating-tree-with-data.html"><a href="manipulating-tree-with-data.html#groupclade"><i class="fa fa-check"></i><b>2.3.1</b> groupClade</a></li>
<li class="chapter" data-level="2.3.2" data-path="manipulating-tree-with-data.html"><a href="manipulating-tree-with-data.html#groupotu"><i class="fa fa-check"></i><b>2.3.2</b> groupOTU</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="exporting-tree-with-data.html"><a href="exporting-tree-with-data.html"><i class="fa fa-check"></i><b>3</b> Exporting tree with data</a><ul>
<li class="chapter" data-level="3.1" data-path="exporting-tree-with-data.html"><a href="exporting-tree-with-data.html#introduction-1"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="exporting-tree-with-data.html"><a href="exporting-tree-with-data.html#exporting-tree-data-to-beast-nexus-format"><i class="fa fa-check"></i><b>3.2</b> Exporting tree data to BEAST Nexus format</a><ul>
<li class="chapter" data-level="3.2.1" data-path="exporting-tree-with-data.html"><a href="exporting-tree-with-data.html#exportingconverting-software-output"><i class="fa fa-check"></i><b>3.2.1</b> Exporting/converting software output</a></li>
<li class="chapter" data-level="3.2.2" data-path="exporting-tree-with-data.html"><a href="exporting-tree-with-data.html#combining-tree-with-external-data"><i class="fa fa-check"></i><b>3.2.2</b> Combining tree with external data</a></li>
<li class="chapter" data-level="3.2.3" data-path="exporting-tree-with-data.html"><a href="exporting-tree-with-data.html#merging-tree-data-from-different-sources"><i class="fa fa-check"></i><b>3.2.3</b> Merging tree data from different sources</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="importing-tree-with-data.html"><a href="importing-tree-with-data.html#jtree"><i class="fa fa-check"></i><b>3.3</b> Exporting tree data to <em>jtree</em> format</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="phylogenetic-tree-visualization.html"><a href="phylogenetic-tree-visualization.html"><i class="fa fa-check"></i><b>4</b> Phylogenetic Tree Visualization</a><ul>
<li class="chapter" data-level="4.1" data-path="phylogenetic-tree-visualization.html"><a href="phylogenetic-tree-visualization.html#viewing-a-phylogenetic-tree-with-ggtree"><i class="fa fa-check"></i><b>4.1</b> Viewing a phylogenetic tree with <em>ggtree</em></a></li>
<li class="chapter" data-level="4.2" data-path="phylogenetic-tree-visualization.html"><a href="phylogenetic-tree-visualization.html#layout"><i class="fa fa-check"></i><b>4.2</b> Layout</a><ul>
<li class="chapter" data-level="4.2.1" data-path="phylogenetic-tree-visualization.html"><a href="phylogenetic-tree-visualization.html#phylogram-and-cladogram"><i class="fa fa-check"></i><b>4.2.1</b> Phylogram and Cladogram</a></li>
<li class="chapter" data-level="4.2.2" data-path="phylogenetic-tree-visualization.html"><a href="phylogenetic-tree-visualization.html#time-scaled-tree"><i class="fa fa-check"></i><b>4.2.2</b> Time-scaled tree</a></li>
<li class="chapter" data-level="4.2.3" data-path="phylogenetic-tree-visualization.html"><a href="phylogenetic-tree-visualization.html#two-dimensional-tree"><i class="fa fa-check"></i><b>4.2.3</b> Two dimensional tree</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="phylogenetic-tree-visualization.html"><a href="phylogenetic-tree-visualization.html#displaying-tree-scale-evolution-distance"><i class="fa fa-check"></i><b>4.3</b> Displaying tree scale (evolution distance)</a></li>
<li class="chapter" data-level="4.4" data-path="phylogenetic-tree-visualization.html"><a href="phylogenetic-tree-visualization.html#displaying-nodestips"><i class="fa fa-check"></i><b>4.4</b> Displaying nodes/tips</a></li>
<li class="chapter" data-level="4.5" data-path="phylogenetic-tree-visualization.html"><a href="phylogenetic-tree-visualization.html#displaying-labels"><i class="fa fa-check"></i><b>4.5</b> Displaying labels</a></li>
<li class="chapter" data-level="4.6" data-path="phylogenetic-tree-visualization.html"><a href="phylogenetic-tree-visualization.html#update-tree-view-with-a-new-tree"><i class="fa fa-check"></i><b>4.6</b> Update tree view with a new tree</a></li>
<li class="chapter" data-level="4.7" data-path="phylogenetic-tree-visualization.html"><a href="phylogenetic-tree-visualization.html#theme"><i class="fa fa-check"></i><b>4.7</b> Theme</a></li>
<li class="chapter" data-level="4.8" data-path="phylogenetic-tree-visualization.html"><a href="phylogenetic-tree-visualization.html#visualize-a-list-of-trees"><i class="fa fa-check"></i><b>4.8</b> Visualize a list of trees</a></li>
<li class="chapter" data-level="4.9" data-path="phylogenetic-tree-visualization.html"><a href="phylogenetic-tree-visualization.html#rescale-tree"><i class="fa fa-check"></i><b>4.9</b> Rescale tree</a></li>
<li class="chapter" data-level="4.10" data-path="phylogenetic-tree-visualization.html"><a href="phylogenetic-tree-visualization.html#zoom-on-a-portion-of-tree"><i class="fa fa-check"></i><b>4.10</b> Zoom on a portion of tree</a></li>
<li class="chapter" data-level="4.11" data-path="phylogenetic-tree-visualization.html"><a href="phylogenetic-tree-visualization.html#color-tree"><i class="fa fa-check"></i><b>4.11</b> Color tree</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="phylogenetic-tree-annotation.html"><a href="phylogenetic-tree-annotation.html"><i class="fa fa-check"></i><b>5</b> Phylogenetic Tree Annotation</a><ul>
<li class="chapter" data-level="5.1" data-path="phylogenetic-tree-annotation.html"><a href="phylogenetic-tree-annotation.html#annotate-clades"><i class="fa fa-check"></i><b>5.1</b> Annotate clades</a><ul>
<li class="chapter" data-level="5.1.1" data-path="phylogenetic-tree-annotation.html"><a href="phylogenetic-tree-annotation.html#annotate-clades-for-unrooted-tree"><i class="fa fa-check"></i><b>5.1.1</b> Annotate clades for unrooted tree</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="phylogenetic-tree-annotation.html"><a href="phylogenetic-tree-annotation.html#labelling-associated-taxa-monophyletic-polyphyletic-or-paraphyletic"><i class="fa fa-check"></i><b>5.2</b> Labelling associated taxa (Monophyletic, Polyphyletic or Paraphyletic)</a></li>
<li class="chapter" data-level="5.3" data-path="phylogenetic-tree-annotation.html"><a href="phylogenetic-tree-annotation.html#highlight-clades"><i class="fa fa-check"></i><b>5.3</b> Highlight clades</a><ul>
<li class="chapter" data-level="5.3.1" data-path="phylogenetic-tree-annotation.html"><a href="phylogenetic-tree-annotation.html#highlight-balances"><i class="fa fa-check"></i><b>5.3.1</b> Highlight balances</a></li>
<li class="chapter" data-level="5.3.2" data-path="phylogenetic-tree-annotation.html"><a href="phylogenetic-tree-annotation.html#highlight-clades-for-unrooted-tree"><i class="fa fa-check"></i><b>5.3.2</b> Highlight clades for unrooted tree</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="phylogenetic-tree-annotation.html"><a href="phylogenetic-tree-annotation.html#taxa-connection"><i class="fa fa-check"></i><b>5.4</b> Taxa connection</a></li>
<li class="chapter" data-level="5.5" data-path="phylogenetic-tree-annotation.html"><a href="phylogenetic-tree-annotation.html#tree-annotation-with-output-from-evolution-software"><i class="fa fa-check"></i><b>5.5</b> Tree annotation with output from evolution software</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="visual-exploration-of-phylogenetic-tree.html"><a href="visual-exploration-of-phylogenetic-tree.html"><i class="fa fa-check"></i><b>6</b> Visual Exploration of Phylogenetic Tree</a><ul>
<li class="chapter" data-level="6.1" data-path="visual-exploration-of-phylogenetic-tree.html"><a href="visual-exploration-of-phylogenetic-tree.html#internal-node-number"><i class="fa fa-check"></i><b>6.1</b> Internal node number</a></li>
<li class="chapter" data-level="6.2" data-path="visual-exploration-of-phylogenetic-tree.html"><a href="visual-exploration-of-phylogenetic-tree.html#view-clade"><i class="fa fa-check"></i><b>6.2</b> View Clade</a></li>
<li class="chapter" data-level="6.3" data-path="visual-exploration-of-phylogenetic-tree.html"><a href="visual-exploration-of-phylogenetic-tree.html#group-clades"><i class="fa fa-check"></i><b>6.3</b> Group Clades</a></li>
<li class="chapter" data-level="6.4" data-path="visual-exploration-of-phylogenetic-tree.html"><a href="visual-exploration-of-phylogenetic-tree.html#group-otus"><i class="fa fa-check"></i><b>6.4</b> Group OTUs</a></li>
<li class="chapter" data-level="6.5" data-path="visual-exploration-of-phylogenetic-tree.html"><a href="visual-exploration-of-phylogenetic-tree.html#collapse-clade"><i class="fa fa-check"></i><b>6.5</b> Collapse clade</a></li>
<li class="chapter" data-level="6.6" data-path="visual-exploration-of-phylogenetic-tree.html"><a href="visual-exploration-of-phylogenetic-tree.html#expand-collapsed-clade"><i class="fa fa-check"></i><b>6.6</b> Expand collapsed clade</a></li>
<li class="chapter" data-level="6.7" data-path="visual-exploration-of-phylogenetic-tree.html"><a href="visual-exploration-of-phylogenetic-tree.html#scale-clade"><i class="fa fa-check"></i><b>6.7</b> Scale clade</a></li>
<li class="chapter" data-level="6.8" data-path="visual-exploration-of-phylogenetic-tree.html"><a href="visual-exploration-of-phylogenetic-tree.html#rotate-clade"><i class="fa fa-check"></i><b>6.8</b> Rotate clade</a></li>
<li class="chapter" data-level="6.9" data-path="visual-exploration-of-phylogenetic-tree.html"><a href="visual-exploration-of-phylogenetic-tree.html#flip-clade"><i class="fa fa-check"></i><b>6.9</b> Flip clade</a></li>
<li class="chapter" data-level="6.10" data-path="visual-exploration-of-phylogenetic-tree.html"><a href="visual-exploration-of-phylogenetic-tree.html#open-tree"><i class="fa fa-check"></i><b>6.10</b> Open tree</a></li>
<li class="chapter" data-level="6.11" data-path="visual-exploration-of-phylogenetic-tree.html"><a href="visual-exploration-of-phylogenetic-tree.html#rotate-tree"><i class="fa fa-check"></i><b>6.11</b> Rotate tree</a></li>
<li class="chapter" data-level="6.12" data-path="visual-exploration-of-phylogenetic-tree.html"><a href="visual-exploration-of-phylogenetic-tree.html#interactive-tree-manipulation"><i class="fa fa-check"></i><b>6.12</b> Interactive tree manipulation</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="plotting-tree-with-data.html"><a href="plotting-tree-with-data.html"><i class="fa fa-check"></i><b>7</b> Plotting tree with data</a><ul>
<li class="chapter" data-level="7.0.1" data-path="plotting-tree-with-data.html"><a href="plotting-tree-with-data.html#the-operator"><i class="fa fa-check"></i><b>7.0.1</b> The <code>%&lt;+%</code> operator</a></li>
<li class="chapter" data-level="7.1" data-path="plotting-tree-with-data.html"><a href="plotting-tree-with-data.html#visualize-tree-with-associated-matrix"><i class="fa fa-check"></i><b>7.1</b> Visualize tree with associated matrix</a></li>
<li class="chapter" data-level="7.2" data-path="plotting-tree-with-data.html"><a href="plotting-tree-with-data.html#visualize-tree-with-multiple-sequence-alignment"><i class="fa fa-check"></i><b>7.2</b> Visualize tree with multiple sequence alignment</a></li>
<li class="chapter" data-level="7.3" data-path="plotting-tree-with-data.html"><a href="plotting-tree-with-data.html#plot-tree-with-associated-data"><i class="fa fa-check"></i><b>7.3</b> Plot tree with associated data</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="annotating-tree-with-silhouette-images-and-sub-plots.html"><a href="annotating-tree-with-silhouette-images-and-sub-plots.html"><i class="fa fa-check"></i><b>8</b> Annotating Tree with Silhouette Images and Sub-plots</a><ul>
<li class="chapter" data-level="8.1" data-path="annotating-tree-with-silhouette-images-and-sub-plots.html"><a href="annotating-tree-with-silhouette-images-and-sub-plots.html#annotating-tree-with-images"><i class="fa fa-check"></i><b>8.1</b> Annotating Tree with Images</a></li>
<li class="chapter" data-level="8.2" data-path="annotating-tree-with-silhouette-images-and-sub-plots.html"><a href="annotating-tree-with-silhouette-images-and-sub-plots.html#annotating-tree-with-phylopic"><i class="fa fa-check"></i><b>8.2</b> Annotating Tree with Phylopic</a></li>
<li class="chapter" data-level="8.3" data-path="annotating-tree-with-silhouette-images-and-sub-plots.html"><a href="annotating-tree-with-silhouette-images-and-sub-plots.html#annotating-tree-with-sub-plots"><i class="fa fa-check"></i><b>8.3</b> Annotating Tree with Sub-plots</a><ul>
<li class="chapter" data-level="8.3.1" data-path="annotating-tree-with-silhouette-images-and-sub-plots.html"><a href="annotating-tree-with-silhouette-images-and-sub-plots.html#annotate-with-bar-charts"><i class="fa fa-check"></i><b>8.3.1</b> Annotate with bar charts</a></li>
<li class="chapter" data-level="8.3.2" data-path="annotating-tree-with-silhouette-images-and-sub-plots.html"><a href="annotating-tree-with-silhouette-images-and-sub-plots.html#annotate-with-pie-charts"><i class="fa fa-check"></i><b>8.3.2</b> Annotate with pie charts</a></li>
<li class="chapter" data-level="8.3.3" data-path="annotating-tree-with-silhouette-images-and-sub-plots.html"><a href="annotating-tree-with-silhouette-images-and-sub-plots.html#annotate-with-other-types-of-charts"><i class="fa fa-check"></i><b>8.3.3</b> Annotate with other types of charts</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="annotating-tree-with-silhouette-images-and-sub-plots.html"><a href="annotating-tree-with-silhouette-images-and-sub-plots.html#have-fun-with-phylomoji"><i class="fa fa-check"></i><b>8.4</b> Have Fun with Phylomoji</a><ul>
<li class="chapter" data-level="8.4.1" data-path="annotating-tree-with-silhouette-images-and-sub-plots.html"><a href="annotating-tree-with-silhouette-images-and-sub-plots.html#emoji-in-cirularfan-layout-tree"><i class="fa fa-check"></i><b>8.4.1</b> emoji in cirular/fan layout tree</a></li>
<li class="chapter" data-level="8.4.2" data-path="annotating-tree-with-silhouette-images-and-sub-plots.html"><a href="annotating-tree-with-silhouette-images-and-sub-plots.html#emoji-to-label-clades"><i class="fa fa-check"></i><b>8.4.2</b> emoji to label clades</a></li>
<li class="chapter" data-level="8.4.3" data-path="annotating-tree-with-silhouette-images-and-sub-plots.html"><a href="annotating-tree-with-silhouette-images-and-sub-plots.html#apple-color-emoji"><i class="fa fa-check"></i><b>8.4.3</b> Apple Color Emoji</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="ggtree-extensions.html"><a href="ggtree-extensions.html"><i class="fa fa-check"></i><b>9</b> ggtree extensions</a></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Data Integration, Manipulation and Visualization of Phylogenetic Trees</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="importing-tree-with-data" class="section level1">
<h1><span class="header-section-number">Chapter 1</span> Importing tree with data</h1>
<div id="introduction" class="section level2">
<h2><span class="header-section-number">1.1</span> Introduction</h2>
<p>Phylogenetic trees are commonly used to present evolutionary relationships of
species. Information associated with taxon species/strains may be further
analyzed in the context of the evolutionary history depicted by the phylogenetic
tree. For example, host information of the influenza virus strains in the tree
could be studied to understand host range of a virus linage. Moreover, such
meta-data (<em>e.g.</em>, isolation host, time, location, <em>etc.</em>) directly associated
with taxon strains are also often subjected to further evolutionary or
comparative phylogenetic models and analyses, to infer their dynamics associated
with the evolutionary or transmission processes of the virus. All these
meta-data or other phenotypic or experimental data are stored either as the
annotation data associated with the nodes or branches, and are often produced in
inconsistent format by different analysis programs.</p>
<p>Getting trees in to R is still limited. Newick and Nexus can be imported by
several packages, including <a href="https://CRAN.R-project.org/package=ape">ape</a>, <a href="https://CRAN.R-project.org/package=phylobase">phylobase</a>. NeXML
format can be parsed by <a href="https://CRAN.R-project.org/package=RNeXML">RNeXML</a>. However, analysis results from
widely used software packages in this field are not well
supported. SIMMAP output can be parsed by <a href="https://CRAN.R-project.org/package=phyext2">phyext2</a> and <a href="https://CRAN.R-project.org/package=phytools">phytools</a>.
Although <a href="http://www.christophheibl.de/Rpackages.html">PHYLOCH</a> can
import BEAST and MrBayes output, only internal node attributes were parsed and
tip attributes were ignore. Many other software outputs are mainly required
programming expertise to import the tree with associated data. Linking external
data, including experimental and clinical data, to phylogeny is another obstacle
for evolution biologists.</p>
<p>The <a href="https://bioconductor.org/packages/treeio/">treeio</a> package defines base
classes and functions for phylogenetic tree input and output. It is an
infrastructure that enables evolutionary evidences that inferred by commonly
used software packages to be used in <code>R</code>. For instance, <em>d<sub>N</sub>/d<sub>S</sub></em> values or
ancestral sequences inferred
by <a href="http://abacus.gene.ucl.ac.uk/software/paml.html">CODEML</a> <span class="citation">(Yang <a href="#ref-yang_paml_2007">2007</a>)</span>,
clade support values (posterior) inferred
by <a href="http://beast2.org/">BEAST</a> <span class="citation">(Bouckaert et al. <a href="#ref-bouckaert_beast_2014">2014</a>)</span> and short read placement
by
<a href="http://sco.h-its.org/exelixis/web/software/epa/index.html">EPA</a> <span class="citation">(Berger, Krompass, and Stamatakis <a href="#ref-berger_EPA_2011">2011</a>)</span>
and <a href="http://matsen.fhcrc.org/pplacer/">pplacer</a> <span class="citation">(Matsen, Kodner, and Armbrust <a href="#ref-matsen_pplacer_2010">2010</a>)</span>. These
evolutionary evidences can be further analyzed in <code>R</code> and used to annotate
phylogenetic tree using <a href="https://bioconductor.org/packages/ggtree/">ggtree</a>
<span class="citation">(Yu et al. <a href="#ref-yu_ggtree:_2017">2017</a>)</span>. The growth of analysis tools and models available introduces
a challenge to integrate different varieties of data and analysis results from
different sources for an integral analysis on the the same phylogenetic tree
background. The <a href="https://bioconductor.org/packages/treeio/">treeio</a> package
provides a <code>merge_tree</code> function to allow combining tree data obtained from
different sources. In addition, <a href="https://bioconductor.org/packages/treeio/">treeio</a>
also enables external data to be linked to phylogenetic tree structure.</p>
</div>
<div id="getting-tree-data-from-evolutionary-analysis-result" class="section level2">
<h2><span class="header-section-number">1.2</span> Getting tree data from evolutionary analysis result</h2>
<p>To fill the gap that most of the tree formats or software outputs cannot be
easily parsed with the same
software/platform, <a href="https://bioconductor.org/packages/treeio/">treeio</a>
implemented several functions for parsing various tree file formats and outputs
of common evolutionary analysis software. Not only the tree structure can be
parsed but also the associated data and evolutionary inferences, including NHX
annotation, clock rate inferences (from <a href="http://beast2.org/">BEAST</a>
or <a href="http://ginger.ucdavis.edu/r8s">r8s</a> <span class="citation">(Sanderson <a href="#ref-sanderson_r8s:_2003">2003</a>)</span> programs),
snynonymous and non-synonymous substitutions (from CodeML), and ancestral
sequence construction (from
<a href="https://veg.github.io/hyphy-site/">HyPhy</a>,
<a href="http://abacus.gene.ucl.ac.uk/software/paml.html">BaseML</a>
or <a href="http://abacus.gene.ucl.ac.uk/software/paml.html">CodeML</a>), <em>etc.</em>.</p>
<p>Currently, <a href="https://bioconductor.org/packages/treeio/">treeio</a> is able to read
the following file formats: Newick, Nexus, New Hampshire eXtended format (NHX),
jplace and Phylip as well as the data outputs from the following analysis programs:
<a href="http://beast2.org/">BEAST</a>,
<a href="http://sco.h-its.org/exelixis/web/software/epa/index.html">EPA</a>,
<a href="https://veg.github.io/hyphy-site/">HyPhy</a>,
<a href="http://nbisweden.github.io/MrBayes/">MrBayes</a>,
<a href="http://abacus.gene.ucl.ac.uk/software/paml.html">PAML</a>,
<a href="https://pbil.univ-lyon1.fr/software/phyldog/">PHYLDOG</a>,
<a href="http://matsen.fhcrc.org/pplacer/">pplacer</a>,
<a href="http://ginger.ucdavis.edu/r8s">r8s</a>,
<a href="http://evomics.org/learning/phylogenetics/raxml/">RAxML</a> and
<a href="https://revbayes.github.io/intro.html">RevBayes</a>.</p>
<table>
<caption><span id="tab:treeio-function">Table 1.1: </span>Parser functions defined in treeio</caption>
<thead>
<tr class="header">
<th align="left">Parser function</th>
<th align="left">Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">read.beast</td>
<td align="left">parsing output of BEAST</td>
</tr>
<tr class="even">
<td align="left">read.codeml</td>
<td align="left">parsing output of CodeML (rst and mlc files)</td>
</tr>
<tr class="odd">
<td align="left">read.codeml_mlc</td>
<td align="left">parsing mlc file (output of CodeML)</td>
</tr>
<tr class="even">
<td align="left">read.hyphy</td>
<td align="left">parsing output of HYPHY</td>
</tr>
<tr class="odd">
<td align="left">read.jplace</td>
<td align="left">parsing jplace file including output of EPA and pplacer</td>
</tr>
<tr class="even">
<td align="left">read.mrbayes</td>
<td align="left">parsing output of MrBayes</td>
</tr>
<tr class="odd">
<td align="left">read.newick</td>
<td align="left">parsing newick string, with ability to parse node label as support values</td>
</tr>
<tr class="even">
<td align="left">read.nhx</td>
<td align="left">parsing NHX file including output of PHYLDOG and RevBayes</td>
</tr>
<tr class="odd">
<td align="left">read.paml_rst</td>
<td align="left">parsing rst file (output of BaseML or CodeML)</td>
</tr>
<tr class="even">
<td align="left">read.phylip</td>
<td align="left">parsing phylip file (phylip alignment + newick string)</td>
</tr>
<tr class="odd">
<td align="left">read.r8s</td>
<td align="left">parsing output of r8s</td>
</tr>
<tr class="even">
<td align="left">read.raxml</td>
<td align="left">parsing output of RAxML</td>
</tr>
</tbody>
</table>
<p>After parsing, storage of the tree structure with associated data is made
through a S4 class, treedata, defined in the <a href="https://bioconductor.org/packages/treeio/">treeio</a> package. These parsed data
are mapped to the tree branches and nodes inside <code>treedata</code> object, so that they
can be efficiently used to visually annotate the tree
using <a href="https://bioconductor.org/packages/ggtree/">ggtree</a> package <span class="citation">(Yu et al. <a href="#ref-yu_ggtree:_2017">2017</a>)</span>.
<a href="https://bioconductor.org/packages/treeio/">treeio</a> provides functions to merge these phylogeny-associated data for
comparison and further analysis.</p>
<div id="parsing-beast-output" class="section level3">
<h3><span class="header-section-number">1.2.1</span> Parsing BEAST output</h3>
<pre class="sourceCode r"><code class="sourceCode r">file &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata/BEAST&quot;</span>, <span class="st">&quot;beast_mcc.tree&quot;</span>, <span class="dt">package=</span><span class="st">&quot;treeio&quot;</span>)
beast &lt;-<span class="st"> </span><span class="kw">read.beast</span>(file)
beast</code></pre>
<pre><code>## &#39;treedata&#39; S4 object that stored information of
##  &#39;C:/Users/YGC/Library/R/library/treeio/extdata/BEAST/beast_mcc.tree&#39;.
## 
## ...@ phylo: 
## Phylogenetic tree with 15 tips and 14 internal nodes.
## 
## Tip labels:
##  A_1995, B_1996, C_1995, D_1987, E_1996, F_1997, ...
## 
## Rooted; includes branch lengths.
## 
## with the following features available:
##  &#39;height&#39;,   &#39;height_0.95_HPD&#39;,  &#39;height_median&#39;,
##  &#39;height_range&#39;, &#39;length&#39;,   &#39;length_0.95_HPD&#39;,
##  &#39;length_median&#39;,    &#39;length_range&#39;, &#39;posterior&#39;,    &#39;rate&#39;,
##  &#39;rate_0.95_HPD&#39;,    &#39;rate_median&#39;,  &#39;rate_range&#39;.</code></pre>
<p>Since <em><code>%</code></em> is not a valid character in <em><code>names</code></em>, all the feature names that contain <em><code>x%</code></em> will convert to <em><code>0.x</code></em>. For example, <em><code>length_95%_HPD</code></em> will be changed to <em><code>length_0.95_HPD</code></em>.</p>
<p>The <em><code>get.fields</code></em> method return all available features that can be used for
annotation.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">get.fields</span>(beast)</code></pre>
<pre><code>##  [1] &quot;height&quot;          &quot;height_0.95_HPD&quot; &quot;height_median&quot;  
##  [4] &quot;height_range&quot;    &quot;length&quot;          &quot;length_0.95_HPD&quot;
##  [7] &quot;length_median&quot;   &quot;length_range&quot;    &quot;posterior&quot;      
## [10] &quot;rate&quot;            &quot;rate_0.95_HPD&quot;   &quot;rate_median&quot;    
## [13] &quot;rate_range&quot;</code></pre>
</div>
<div id="parsing-mrbayes-output" class="section level3">
<h3><span class="header-section-number">1.2.2</span> Parsing MrBayes output</h3>
<pre class="sourceCode r"><code class="sourceCode r">file &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata/MrBayes&quot;</span>, <span class="st">&quot;Gq_nxs.tre&quot;</span>, <span class="dt">package=</span><span class="st">&quot;treeio&quot;</span>)
<span class="kw">read.mrbayes</span>(file)</code></pre>
<pre><code>## &#39;treedata&#39; S4 object that stored information of
##  &#39;C:/Users/YGC/Library/R/library/treeio/extdata/MrBayes/Gq_nxs.tre&#39;.
## 
## ...@ phylo: 
## Phylogenetic tree with 12 tips and 10 internal nodes.
## 
## Tip labels:
##  B_h, B_s, G_d, G_k, G_q, G_s, ...
## 
## Unrooted; includes branch lengths.
## 
## with the following features available:
##  &#39;length_0.95HPD&#39;,   &#39;length_mean&#39;,  &#39;length_median&#39;,    &#39;prob&#39;,
##  &#39;prob_percent&#39;, &#39;prob_range&#39;,   &#39;prob_stddev&#39;,  &#39;prob+-sd&#39;.</code></pre>
</div>
<div id="parsing-paml-output" class="section level3">
<h3><span class="header-section-number">1.2.3</span> Parsing PAML output</h3>
<p>The <code>read.paml_rst</code> function can parse <em>rst</em> file
from <a href="http://abacus.gene.ucl.ac.uk/software/paml.html">BASEML</a>
and <a href="http://abacus.gene.ucl.ac.uk/software/paml.html">CODEML</a>. The only
difference is the space in the sequences.
For <a href="http://abacus.gene.ucl.ac.uk/software/paml.html">BASEML</a>, each ten bases
are separated by one space, while
for <a href="http://abacus.gene.ucl.ac.uk/software/paml.html">CODEML</a>, each three bases
(triplet) are separated by one space.</p>
<pre class="sourceCode r"><code class="sourceCode r">brstfile &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata/PAML_Baseml&quot;</span>, <span class="st">&quot;rst&quot;</span>, <span class="dt">package=</span><span class="st">&quot;treeio&quot;</span>)
brst &lt;-<span class="st"> </span><span class="kw">read.paml_rst</span>(brstfile)
brst</code></pre>
<pre><code>## &#39;treedata&#39; S4 object that stored information of
##  &#39;C:/Users/YGC/Library/R/library/treeio/extdata/PAML_Baseml/rst&#39;.
## 
## ...@ phylo: 
## Phylogenetic tree with 15 tips and 13 internal nodes.
## 
## Tip labels:
##  A, B, C, D, E, F, ...
## Node labels:
##  16, 17, 18, 19, 20, 21, ...
## 
## Unrooted; includes branch lengths.
## 
## with the following features available:
##  &#39;subs&#39;, &#39;AA_subs&#39;.</code></pre>
<p>Similarly, we can parse the <em>rst</em> file from <a href="http://abacus.gene.ucl.ac.uk/software/paml.html">CODEML</a>.</p>
<pre class="sourceCode r"><code class="sourceCode r">crstfile &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata/PAML_Codeml&quot;</span>, <span class="st">&quot;rst&quot;</span>, <span class="dt">package=</span><span class="st">&quot;treeio&quot;</span>)
<span class="co">## type can be one of &quot;Marginal&quot; or &quot;Joint&quot;</span>
crst &lt;-<span class="st"> </span><span class="kw">read.paml_rst</span>(crstfile, <span class="dt">type =</span> <span class="st">&quot;Joint&quot;</span>)
crst</code></pre>
<pre><code>## &#39;treedata&#39; S4 object that stored information of
##  &#39;C:/Users/YGC/Library/R/library/treeio/extdata/PAML_Codeml/rst&#39;.
## 
## ...@ phylo: 
## Phylogenetic tree with 15 tips and 13 internal nodes.
## 
## Tip labels:
##  A, B, C, D, E, F, ...
## Node labels:
##  16, 17, 18, 19, 20, 21, ...
## 
## Unrooted; includes branch lengths.
## 
## with the following features available:
##  &#39;subs&#39;, &#39;AA_subs&#39;.</code></pre>
<p>Ancestral sequences inferred by <a href="http://abacus.gene.ucl.ac.uk/software/paml.html">BASEML</a>
or <a href="http://abacus.gene.ucl.ac.uk/software/paml.html">CODEML</a> via marginal or
joint ML reconstruction methods will be stored in the S4 object and mapped to
tree nodes. <a href="https://bioconductor.org/packages/treeio/">treeio</a> will automatically determine the substitutions between the
sequences at the both ends of each branch. Amino acid substitution will also be
determined by translating nucleotide sequences to amino acid sequences. These
computed substitutions will also be stored in the S4 object.</p>
<p><a href="http://abacus.gene.ucl.ac.uk/software/paml.html">CODEML</a> infers selection
pressure and estimated <em>d<sub>N</sub>/d<sub>S</sub></em>, <em>d<sub>N</sub></em> and <em>d<sub>S</sub></em>. These information are
stored in output file <em>mlc</em>, which can be parsed by <code>read.codeml_mlc</code> function.</p>
<pre class="sourceCode r"><code class="sourceCode r">mlcfile &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata/PAML_Codeml&quot;</span>, <span class="st">&quot;mlc&quot;</span>, <span class="dt">package=</span><span class="st">&quot;treeio&quot;</span>)
mlc &lt;-<span class="st"> </span><span class="kw">read.codeml_mlc</span>(mlcfile)
mlc</code></pre>
<pre><code>## &#39;treedata&#39; S4 object that stored information of
##  &#39;C:/Users/YGC/Library/R/library/treeio/extdata/PAML_Codeml/mlc&#39;.
## 
## ...@ phylo: 
## Phylogenetic tree with 15 tips and 13 internal nodes.
## 
## Tip labels:
##  A, B, C, D, E, F, ...
## Node labels:
##  16, 17, 18, 19, 20, 21, ...
## 
## Unrooted; includes branch lengths.
## 
## with the following features available:
##  &#39;t&#39;,    &#39;N&#39;,    &#39;S&#39;,    &#39;dN_vs_dS&#39;, &#39;dN&#39;,   &#39;dS&#39;,   &#39;N_x_dN&#39;,   &#39;S_x_dS&#39;.</code></pre>
<p>In previous session, we separately parsed <em>rst</em> and <em>mlc</em> files. However, they
can also be parsed together using <code>read.codeml</code> function.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="co">## tree can be one of &quot;rst&quot; or &quot;mlc&quot; to specify</span>
<span class="co">## using tree from which file as base tree in the object</span>
ml &lt;-<span class="st"> </span><span class="kw">read.codeml</span>(crstfile, mlcfile, <span class="dt">tree =</span> <span class="st">&quot;mlc&quot;</span>)
ml</code></pre>
<pre><code>## &#39;treedata&#39; S4 object that stored information of
##  &#39;C:/Users/YGC/Library/R/library/treeio/extdata/PAML_Codeml/rst&#39;,
##  &#39;C:/Users/YGC/Library/R/library/treeio/extdata/PAML_Codeml/mlc&#39;.
## 
## ...@ phylo: 
## Phylogenetic tree with 15 tips and 13 internal nodes.
## 
## Tip labels:
##  A, B, C, D, E, F, ...
## Node labels:
##  16, 17, 18, 19, 20, 21, ...
## 
## Unrooted; includes branch lengths.
## 
## with the following features available:
##  &#39;subs&#39;, &#39;AA_subs&#39;,  &#39;t&#39;,    &#39;N&#39;,    &#39;S&#39;,    &#39;dN_vs_dS&#39;, &#39;dN&#39;,   &#39;dS&#39;,
##  &#39;N_x_dN&#39;,   &#39;S_x_dS&#39;.</code></pre>
<p>All the features in both <em>rst</em> and <em>mlc</em> files were imported into a single S4
object and hence are available for further annotation and visualization. For
example, we can annotate and display both <em>d<sub>N</sub>/d<sub>S</sub></em> (from <em>mlc</em> file) and
amino acid substitutions (derived from <em>rst</em> file) on the same phylogenetic tree.</p>
</div>
<div id="parsing-hyphy-output" class="section level3">
<h3><span class="header-section-number">1.2.4</span> Parsing HyPhy output</h3>
<p>Ancestral sequences inferred by <a href="https://veg.github.io/hyphy-site/">HyPhy</a> are
stored in the Nexus output file, which contains the tree topology and ancestral
sequences. To parse this data file, users can use the <code>read.hyphy.seq</code> function.</p>
<pre class="sourceCode r"><code class="sourceCode r">ancseq &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata/HYPHY&quot;</span>, <span class="st">&quot;ancseq.nex&quot;</span>, <span class="dt">package=</span><span class="st">&quot;treeio&quot;</span>)
<span class="kw">read.hyphy.seq</span>(ancseq)</code></pre>
<pre><code>## 13 DNA sequences in binary format stored in a list.
## 
## All sequences of same length: 2148 
## 
## Labels:
## Node1
## Node2
## Node3
## Node4
## Node5
## Node12
## ...
## 
## Base composition:
##     a     c     g     t 
## 0.335 0.208 0.237 0.220 
## (Total: 27.92 kb)</code></pre>
<p>To map the sequences on the tree, user shall also provide an
internal-node-labelled tree. If users want to determine substitution, they need
also provide tip sequences.</p>
<pre class="sourceCode r"><code class="sourceCode r">nwk &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata/HYPHY&quot;</span>, <span class="st">&quot;labelledtree.tree&quot;</span>, <span class="dt">package=</span><span class="st">&quot;treeio&quot;</span>)
tipfas &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;pa.fas&quot;</span>, <span class="dt">package=</span><span class="st">&quot;treeio&quot;</span>)
hy &lt;-<span class="st"> </span><span class="kw">read.hyphy</span>(nwk, ancseq, tipfas)
hy</code></pre>
<pre><code>## &#39;treedata&#39; S4 object that stored information of
##  &#39;C:/Users/YGC/Library/R/library/treeio/extdata/HYPHY/labelledtree.tree&#39;.
## 
## ...@ phylo: 
## Phylogenetic tree with 15 tips and 13 internal nodes.
## 
## Tip labels:
##  K, N, D, L, J, G, ...
## Node labels:
##  Node1, Node2, Node3, Node4, Node5, Node12, ...
## 
## Unrooted; includes branch lengths.
## 
## with the following features available:
##  &#39;subs&#39;, &#39;AA_subs&#39;.</code></pre>
</div>
<div id="parsing-r8s-output" class="section level3">
<h3><span class="header-section-number">1.2.5</span> Parsing r8s output</h3>
<p><a href="http://loco.biosci.arizona.edu/r8s/">r8s</a> uses parametric, semiparametric and
nonparametric methods to relax molecular clock to allow better estimations of
divergence times and evolution rates <span class="citation">(@ Sanderson <a href="#ref-sanderson_r8s:_2003">2003</a>)</span>. It outputs three
trees in log file, namely <em>TREE</em>, <em>RATO</em> and <em>PHYLO</em> for time tree, rate tree
and absolute substitution tree respectively.</p>
<p>Time tree is scaled by divergence time, rate tree is scaled by substitution rate
and absolute substitution tree is scaled by absolute number of substitution.
After parsing the file, all these three trees are stored in a <em>multiPhylo</em> object.</p>
<pre class="sourceCode r"><code class="sourceCode r">r8s &lt;-<span class="st"> </span><span class="kw">read.r8s</span>(<span class="kw">system.file</span>(<span class="st">&quot;extdata/r8s&quot;</span>, <span class="st">&quot;H3_r8s_output.log&quot;</span>, <span class="dt">package=</span><span class="st">&quot;treeio&quot;</span>))
r8s</code></pre>
<pre><code>## 3 phylogenetic trees</code></pre>
</div>
<div id="parsing-output-of-raxml-bootstraping-analysis" class="section level3">
<h3><span class="header-section-number">1.2.6</span> Parsing output of RAxML bootstraping analysis</h3>
<p><a href="http://evomics.org/learning/phylogenetics/raxml/">RAxML</a> bootstraping analysis
output a Newick tree text that is not standard as it stores bootstrap values
inside square brackets after branch lengths. This file usually cannot be parsed
by traditional Newick parser, such as <code>ape::read.tree</code>. The function
<code>read.raxml</code> can read such file and stored the bootstrap as an additional
features, which can be used to display on the tree or used to color tree
branches, <em>etc.</em>.</p>
<pre class="sourceCode r"><code class="sourceCode r">raxml_file &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata/RAxML&quot;</span>, <span class="st">&quot;RAxML_bipartitionsBranchLabels.H3&quot;</span>, <span class="dt">package=</span><span class="st">&quot;treeio&quot;</span>)
raxml &lt;-<span class="st"> </span><span class="kw">read.raxml</span>(raxml_file)
raxml</code></pre>
<pre><code>## &#39;treedata&#39; S4 object that stored information of
##  &#39;C:/Users/YGC/Library/R/library/treeio/extdata/RAxML/RAxML_bipartitionsBranchLabels.H3&#39;.
## 
## ...@ phylo: 
## Phylogenetic tree with 64 tips and 62 internal nodes.
## 
## Tip labels:
##  A_Hokkaido_M1_2014_H3N2_2014, A_Czech_Republic_1_2014_H3N2_2014, FJ532080_A_California_09_2008_H3N2_2008, EU199359_A_Pennsylvania_05_2007_H3N2_2007, EU857080_A_Hong_Kong_CUHK69904_2006_H3N2_2006, EU857082_A_Hong_Kong_CUHK7047_2005_H3N2_2005, ...
## 
## Unrooted; includes branch lengths.
## 
## with the following features available:
##  &#39;bootstrap&#39;.</code></pre>
</div>
<div id="parsing-nhx-tree" class="section level3">
<h3><span class="header-section-number">1.2.7</span> Parsing NHX tree</h3>
<p>NHX (New Hampshire eXtended) format is an extension of Newick by introducing NHX
tags. NHX is commonly used in phylogenetics software
(including
<a href="http://pbil.univ-lyon1.fr/software/phyldog/">PHYLDOG</a> <span class="citation">(Boussau et al. <a href="#ref-boussau_genome-scale_2013">2013</a>)</span>,
<a href="http://revbayes.github.io/intro.html">RevBayes</a> <span class="citation">(HÃ¶hna et al. <a href="#ref-hohna_probabilistic_2014">2014</a>)</span>)
for storing statistical inferences. The following codes imported a NHX tree with
associated data inferred by PHYLDOG.</p>
<pre class="sourceCode r"><code class="sourceCode r">nhxfile &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata/NHX&quot;</span>, <span class="st">&quot;phyldog.nhx&quot;</span>, <span class="dt">package=</span><span class="st">&quot;treeio&quot;</span>)
nhx &lt;-<span class="st"> </span><span class="kw">read.nhx</span>(nhxfile)
nhx</code></pre>
<pre><code>## &#39;treedata&#39; S4 object that stored information of
##  &#39;C:/Users/YGC/Library/R/library/treeio/extdata/NHX/phyldog.nhx&#39;.
## 
## ...@ phylo: 
## Phylogenetic tree with 16 tips and 15 internal nodes.
## 
## Tip labels:
##  Prayidae_D27SS7@2825365, Kephyes_ovata@2606431, Chuniphyes_multidentata@1277217, Apolemia_sp_@1353964, Bargmannia_amoena@263997, Bargmannia_elongata@946788, ...
## 
## Rooted; includes branch lengths.
## 
## with the following features available:
##  &#39;Ev&#39;,   &#39;S&#39;,    &#39;ND&#39;.</code></pre>
</div>
<div id="parsing-phylip-tree" class="section level3">
<h3><span class="header-section-number">1.2.8</span> Parsing Phylip tree</h3>
<p>Phylip format contains multiple sequence alignment of taxa in Phylip sequence
format with corresponding Newick tree text that was built from taxon sequences.
Sequence alignment can be sorted based on the tree structure and displayed at
the right hand side of the tree using <a href="https://bioconductor.org/packages/ggtree/">ggtree</a> <span class="citation">(Yu et al. <a href="#ref-yu_ggtree:_2017">2017</a>)</span>.</p>
<pre class="sourceCode r"><code class="sourceCode r">phyfile &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;sample.phy&quot;</span>, <span class="dt">package=</span><span class="st">&quot;treeio&quot;</span>)
phylip &lt;-<span class="st"> </span><span class="kw">read.phylip</span>(phyfile)
phylip</code></pre>
<pre><code>## &#39;treedata&#39; S4 object that stored information of
##  &#39;C:/Users/YGC/Library/R/library/treeio/extdata/sample.phy&#39;.
## 
## ...@ phylo: 
## Phylogenetic tree with 15 tips and 13 internal nodes.
## 
## Tip labels:
##  K, N, D, L, J, G, ...
## 
## Unrooted; no branch lengths.</code></pre>
</div>
<div id="parsing-epa-and-pplacer-output" class="section level3">
<h3><span class="header-section-number">1.2.9</span> Parsing EPA and pplacer output</h3>
<p><a href="http://sco.h-its.org/exelixis/web/software/epa/index.html">EPA</a>
<span class="citation">(Berger, Krompass, and Stamatakis <a href="#ref-berger_EPA_2011">2011</a>)</span> and <a href="http://matsen.fhcrc.org/pplacer/">PPLACER</a>
<span class="citation">(Matsen, Kodner, and Armbrust <a href="#ref-matsen_pplacer_2010">2010</a>)</span> have common output file format, <code>jplace</code>, which can be
parsed by <code>read.jplace()</code> function.</p>
<pre class="sourceCode r"><code class="sourceCode r">jpf &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata/EPA.jplace&quot;</span>,  <span class="dt">package=</span><span class="st">&quot;treeio&quot;</span>)
jp &lt;-<span class="st"> </span><span class="kw">read.jplace</span>(jpf)
<span class="kw">print</span>(jp)</code></pre>
<pre><code>## &#39;treedata&#39; S4 object that stored information of
##  &#39;C:/Users/YGC/Library/R/library/treeio/extdata/EPA.jplace&#39;.
## 
## ...@ phylo: 
## Phylogenetic tree with 493 tips and 492 internal nodes.
## 
## Tip labels:
##  CIR000447A, CIR000479, CIR000078, CIR000083, CIR000070, CIR000060, ...
## 
## Rooted; includes branch lengths.
## 
## with the following features available:
##  &#39;nplace&#39;.</code></pre>
<p>The number of evolutionary placement on each branch will be calculated and
stored as the <em>nplace</em> feature, which can be mapped to line size and/or color
using <a href="https://bioconductor.org/packages/ggtree/">ggtree</a> <span class="citation">(Yu et al. <a href="#ref-yu_ggtree:_2017">2017</a>)</span>.</p>
</div>
<div id="jtree" class="section level3">
<h3><span class="header-section-number">1.2.10</span> Parsing jtree format</h3>
<p>The <em>jtree</em> is a JSON based format that was defined in
this <a href="https://bioconductor.org/packages/treeio/">treeio</a> package to support <a href="Exporter.html#jtree">tree
data inter change</a>.
Phylogenetic tree with associated data can be exported to a single <em>jtree</em>
file using <code>write.jtree</code> function. The <em>jtree</em> can be easily parsed using any
JSON parser. The <em>jtree</em> format contains three keys: tree, data and metadata.
The tree value contains tree text extended from Newick tree format by putting
the edge number in curly braces after branch length. The data value contains
node/branch-specific data, while metadata value contains additional meta information.</p>
<pre class="sourceCode r"><code class="sourceCode r">jtree_file &lt;-<span class="st"> </span><span class="kw">tempfile</span>(<span class="dt">fileext =</span> <span class="st">&#39;.jtree&#39;</span>)
<span class="kw">write.jtree</span>(beast, <span class="dt">file =</span> jtree_file)
<span class="kw">read.jtree</span>(<span class="dt">file =</span> jtree_file)</code></pre>
<pre><code>## &#39;treedata&#39; S4 object that stored information of
##  &#39;C:\Users\YGC\AppData\Local\Temp\RtmpARpTPX\file2f4845ea2b46.jtree&#39;.
## 
## ...@ phylo: 
## Phylogenetic tree with 15 tips and 14 internal nodes.
## 
## Tip labels:
##  K_2013, N_2010, D_1987, L_1980, J_1983, G_1992, ...
## 
## Rooted; includes branch lengths.
## 
## with the following features available:
##  &#39;height&#39;,   &#39;height_0.95_HPD&#39;,  &#39;height_range&#39;, &#39;length&#39;,
##  &#39;length_0.95_HPD&#39;,  &#39;length_median&#39;,    &#39;length_range&#39;, &#39;rate&#39;,
##  &#39;rate_0.95_HPD&#39;,    &#39;rate_median&#39;,  &#39;rate_range&#39;,
##  &#39;height_median&#39;,    &#39;posterior&#39;.</code></pre>
</div>
</div>
<div id="linking-external-data-to-phylogeny" class="section level2">
<h2><span class="header-section-number">1.3</span> Linking external data to phylogeny</h2>
<p>In addition to analysis findings that are associated with the tree as we showed
above, there is a wide range of heterogeneous data, including phenotypic data,
experimental data and clinical data <em>etc.</em>, that need to be integrated and
linked to phylogeny. For example, in the study of viral evolution, tree nodes may
associated with epidemiological information, such as location, age and subtype.
Functional annotations may need to be mapped on gene trees for comparative
genomics studies. To facilitate data
integration, <a href="https://bioconductor.org/packages/treeio">treeio</a> provides
<code>full_join</code> method to link external data to phylogeny and stored in <code>treedata</code> object.</p>
<p>Here are examples of linking external data to a phylogenetic tree. After that,
we can use <a href="Exporter.html">exporter</a> to combine the tree and the data to a
single tree file. The data that mapped on the phylogenetic tree can also be used to visualize or
annotate the tree using <a href="https://bioconductor.org/packages/ggtree/">ggtree</a>
<span class="citation">(Yu et al. <a href="#ref-yu_ggtree:_2017">2017</a>)</span>.</p>
<pre class="sourceCode r"><code class="sourceCode r">x &lt;-<span class="st"> </span><span class="kw">data_frame</span>(<span class="dt">label =</span> <span class="kw">as.phylo</span>(beast)<span class="op">$</span>tip.label, <span class="dt">trait =</span> <span class="kw">rnorm</span>(<span class="kw">Ntip</span>(beast)))
<span class="kw">full_join</span>(beast, x, <span class="dt">by=</span><span class="st">&quot;label&quot;</span>)</code></pre>
<pre><code>## &#39;treedata&#39; S4 object that stored information of
##  &#39;C:/Users/YGC/Library/R/library/treeio/extdata/BEAST/beast_mcc.tree&#39;.
## 
## ...@ phylo: 
## Phylogenetic tree with 15 tips and 14 internal nodes.
## 
## Tip labels:
##  A_1995, B_1996, C_1995, D_1987, E_1996, F_1997, ...
## 
## Rooted; includes branch lengths.
## 
## with the following features available:
##  &#39;height&#39;,   &#39;height_0.95_HPD&#39;,  &#39;height_median&#39;,
##  &#39;height_range&#39;, &#39;length&#39;,   &#39;length_0.95_HPD&#39;,
##  &#39;length_median&#39;,    &#39;length_range&#39;, &#39;posterior&#39;,    &#39;rate&#39;,
##  &#39;rate_0.95_HPD&#39;,    &#39;rate_median&#39;,  &#39;rate_range&#39;,   &#39;trait&#39;.</code></pre>
<pre class="sourceCode r"><code class="sourceCode r">N &lt;-<span class="st"> </span><span class="kw">Nnode2</span>(beast)
y &lt;-<span class="st"> </span><span class="kw">data_frame</span>(<span class="dt">node =</span> <span class="dv">1</span><span class="op">:</span>N, <span class="dt">fake_trait =</span> <span class="kw">rnorm</span>(N), <span class="dt">another_trait =</span> <span class="kw">runif</span>(N))
<span class="kw">full_join</span>(beast, y, <span class="dt">by=</span><span class="st">&quot;node&quot;</span>)</code></pre>
<pre><code>## &#39;treedata&#39; S4 object that stored information of
##  &#39;C:/Users/YGC/Library/R/library/treeio/extdata/BEAST/beast_mcc.tree&#39;.
## 
## ...@ phylo: 
## Phylogenetic tree with 15 tips and 14 internal nodes.
## 
## Tip labels:
##  A_1995, B_1996, C_1995, D_1987, E_1996, F_1997, ...
## 
## Rooted; includes branch lengths.
## 
## with the following features available:
##  &#39;height&#39;,   &#39;height_0.95_HPD&#39;,  &#39;height_median&#39;,
##  &#39;height_range&#39;, &#39;length&#39;,   &#39;length_0.95_HPD&#39;,
##  &#39;length_median&#39;,    &#39;length_range&#39;, &#39;posterior&#39;,    &#39;rate&#39;,
##  &#39;rate_0.95_HPD&#39;,    &#39;rate_median&#39;,  &#39;rate_range&#39;,   &#39;fake_trait&#39;,
##  &#39;another_trait&#39;.</code></pre>
</div>
<div id="combining-tree-data" class="section level2">
<h2><span class="header-section-number">1.4</span> Combining tree data</h2>
<p>The <a href="https://bioconductor.org/packages/treeio/">treeio</a> package serves as an
infrastructure that enables various types of phylogenetic data inferred from
common analysis programs to be imported and used in R. For instance <em>d<sub>N</sub>/d<sub>S</sub></em>
or ancestral sequences estimated
by <a href="http://abacus.gene.ucl.ac.uk/software/paml.html">CODEML</a>,
and clade support values (posterior) inferred
by <a href="http://beast2.org/">BEAST</a>/<a href="http://nbisweden.github.io/MrBayes/">MrBayes</a>.
In addition, <a href="https://bioconductor.org/packages/treeio/">treeio</a> package
supports linking external data to phylogeny. It brings these external
phylogenetic data (either from software output or exteranl sources) to the R
community and make it available for further analysis in R.
Furthermore, <a href="https://bioconductor.org/packages/treeio/">treeio</a> can combine
multiple phylogenetic trees together into one with their node/branch-specific
attribute data. Essentially, as a result, one such attribute (<em>e.g.</em>,
substitution rate) can be mapped to another attribute (<em>e.g.</em>, <em>d<sub>N</sub>/d<sub>S</sub></em>) of
the same node/branch for comparison and further computations.</p>
<p>A previously published data set, seventy-six H3 hemagglutinin gene sequences of
a lineage containing swine and human influenza A viruses
<span class="citation">(Liang et al. <a href="#ref-liang_expansion_2014">2014</a>)</span>, was here to demonstrate the utilities of comparing
evolutionary statistics inferred by different software. The dataset was
re-analyzed by <a href="http://beast2.org/">BEAST</a> for timescale estimation
and <a href="http://abacus.gene.ucl.ac.uk/software/paml.html">CODEML</a> for synonymous and
non-synonymous substitution estimation. In this example, we first parsed the
outputs from <a href="http://beast2.org/">BEAST</a> using <code>read.beast</code> and
from <a href="http://abacus.gene.ucl.ac.uk/software/paml.html">CODEML</a> using
<code>read.codeml</code> into two <code>treedata</code> objects. Then the two objects containing
separate sets of node/branch-specific data were merged via the <code>merge_tree</code> function.</p>
<pre class="sourceCode r"><code class="sourceCode r">beast_file &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;examples/MCC_FluA_H3.tree&quot;</span>, <span class="dt">package=</span><span class="st">&quot;ggtree&quot;</span>)
rst_file &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;examples/rst&quot;</span>, <span class="dt">package=</span><span class="st">&quot;ggtree&quot;</span>)
mlc_file &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;examples/mlc&quot;</span>, <span class="dt">package=</span><span class="st">&quot;ggtree&quot;</span>)
beast_tree &lt;-<span class="st"> </span><span class="kw">read.beast</span>(beast_file)
codeml_tree &lt;-<span class="st"> </span><span class="kw">read.codeml</span>(rst_file, mlc_file)

merged_tree &lt;-<span class="st"> </span><span class="kw">merge_tree</span>(beast_tree, codeml_tree)
merged_tree</code></pre>
<pre><code>## &#39;treedata&#39; S4 object that stored information of
##  &#39;C:/Users/YGC/Library/R/library/ggtree/examples/MCC_FluA_H3.tree&#39;,
##  &#39;C:/Users/YGC/Library/R/library/ggtree/examples/rst&#39;,
##  &#39;C:/Users/YGC/Library/R/library/ggtree/examples/mlc&#39;.
## 
## ...@ phylo: 
## Phylogenetic tree with 76 tips and 75 internal nodes.
## 
## Tip labels:
##  A/Hokkaido/30-1-a/2013, A/New_York/334/2004, A/New_York/463/2005, A/New_York/452/1999, A/New_York/238/2005, A/New_York/523/1998, ...
## 
## Rooted; includes branch lengths.
## 
## with the following features available:
##  &#39;height&#39;,   &#39;height_0.95_HPD&#39;,  &#39;height_median&#39;,
##  &#39;height_range&#39;, &#39;length&#39;,   &#39;length_0.95_HPD&#39;,
##  &#39;length_median&#39;,    &#39;length_range&#39;, &#39;posterior&#39;,    &#39;rate&#39;,
##  &#39;rate_0.95_HPD&#39;,    &#39;rate_median&#39;,  &#39;rate_range&#39;,   &#39;subs&#39;,
##  &#39;AA_subs&#39;,  &#39;t&#39;,    &#39;N&#39;,    &#39;S&#39;,    &#39;dN_vs_dS&#39;, &#39;dN&#39;,   &#39;dS&#39;,   &#39;N_x_dN&#39;,
##  &#39;S_x_dS&#39;.</code></pre>
<p>After merging the <code>beast_tree</code> and <code>codeml_tree</code> objects, all
node/branch-specific data imported from <a href="http://beast2.org/">BEAST</a>
and <a href="http://abacus.gene.ucl.ac.uk/software/paml.html">CODEML</a> output files are
all available in the <code>merged_tree</code> object. The tree object was converted to
tidy data frame using <a href="https://cran.r-project.org/package=tidytree">tidytree</a>
package and visualized as hexbin scatterplot of <em>d<sub>N</sub>/d<sub>S</sub></em>, <em>d<sub>N</sub></em> and <em>d<sub>S</sub></em> inferred
by <a href="http://abacus.gene.ucl.ac.uk/software/paml.html">CODEML</a> versus <em>rate</em>
(substitution rate in unit of substitutions/site/year) inferred
by <a href="http://beast2.org/">BEAST</a> on the same branches.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(tidytree)
<span class="kw">library</span>(ggplot2)

<span class="kw">as_tibble</span>(merged_tree) <span class="op">%&gt;%</span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw">select</span>(dN_vs_dS, dN, dS, rate) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">subset</span>(dN_vs_dS <span class="op">&gt;=</span><span class="dv">0</span> <span class="op">&amp;</span><span class="st"> </span>dN_vs_dS <span class="op">&lt;=</span><span class="st"> </span><span class="fl">1.5</span>) <span class="op">%&gt;%</span>
<span class="st">    </span>tidyr<span class="op">::</span><span class="kw">gather</span>(type, value, dN_vs_dS<span class="op">:</span>dS) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(rate, value)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_hex</span>() <span class="op">+</span>
<span class="st">    </span><span class="kw">facet_wrap</span>(<span class="op">~</span><span class="kw">factor</span>(type, <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&#39;dN_vs_dS&#39;</span>, <span class="st">&#39;dN&#39;</span>, <span class="st">&#39;dS&#39;</span>)),
               <span class="dt">scale=</span><span class="st">&#39;free_y&#39;</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">ylab</span>(<span class="ot">NULL</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-20-1.png" width="864" /></p>
<p>Using <code>merge_tree</code>, we are able to compare analysis results using identical
model from different software packages or different models using different or
identical software. It also allows users to integrate different analysis finding
from different software packages. Merging tree data is not restricted to
software findings, associating external data to analysis findings is also
granted. The <code>merge_tree</code> function is chainable and allows several tree objects
to be merged into one.</p>
<pre class="sourceCode r"><code class="sourceCode r">phylo &lt;-<span class="st"> </span><span class="kw">as.phylo</span>(beast_tree)
N &lt;-<span class="st"> </span><span class="kw">Nnode2</span>(phylo)
d &lt;-<span class="st"> </span><span class="kw">data_frame</span>(<span class="dt">node =</span> <span class="dv">1</span><span class="op">:</span>N, <span class="dt">fake_trait =</span> <span class="kw">rnorm</span>(N), <span class="dt">another_trait =</span> <span class="kw">runif</span>(N))
fake_tree &lt;-<span class="st"> </span><span class="kw">treedata</span>(<span class="dt">phylo =</span> phylo, <span class="dt">data =</span> d)
triple_tree &lt;-<span class="st"> </span><span class="kw">merge_tree</span>(merged_tree, fake_tree)
triple_tree</code></pre>
<pre><code>## &#39;treedata&#39; S4 object that stored information of
##  &#39;C:/Users/YGC/Library/R/library/ggtree/examples/MCC_FluA_H3.tree&#39;,
##  &#39;C:/Users/YGC/Library/R/library/ggtree/examples/rst&#39;,
##  &#39;C:/Users/YGC/Library/R/library/ggtree/examples/mlc&#39;.
## 
## ...@ phylo: 
## Phylogenetic tree with 76 tips and 75 internal nodes.
## 
## Tip labels:
##  A/Hokkaido/30-1-a/2013, A/New_York/334/2004, A/New_York/463/2005, A/New_York/452/1999, A/New_York/238/2005, A/New_York/523/1998, ...
## 
## Rooted; includes branch lengths.
## 
## with the following features available:
##  &#39;height&#39;,   &#39;height_0.95_HPD&#39;,  &#39;height_median&#39;,
##  &#39;height_range&#39;, &#39;length&#39;,   &#39;length_0.95_HPD&#39;,
##  &#39;length_median&#39;,    &#39;length_range&#39;, &#39;posterior&#39;,    &#39;rate&#39;,
##  &#39;rate_0.95_HPD&#39;,    &#39;rate_median&#39;,  &#39;rate_range&#39;,   &#39;subs&#39;,
##  &#39;AA_subs&#39;,  &#39;t&#39;,    &#39;N&#39;,    &#39;S&#39;,    &#39;dN_vs_dS&#39;, &#39;dN&#39;,   &#39;dS&#39;,   &#39;N_x_dN&#39;,
##  &#39;S_x_dS&#39;,   &#39;fake_trait&#39;,   &#39;another_trait&#39;.</code></pre>
<p>The <code>triple_tree</code> object showed above contains analysis results obtained from <a href="http://beast2.org/">BEAST</a>
and <a href="http://abacus.gene.ucl.ac.uk/software/paml.html">CODEML</a>, and evolutionary
trait from external sources. All these information can be used to annotate the
tree using <a href="https://bioconductor.org/packages/ggtree/">ggtree</a> <span class="citation">(Yu et al. <a href="#ref-yu_ggtree:_2017">2017</a>)</span>.</p>
</div>
<div id="getting-information-from-treedata-object" class="section level2">
<h2><span class="header-section-number">1.5</span> Getting information from <em>treedata</em> object</h2>
<p>After the tree was imported, users may want to extract information that stored
in the <code>treedata</code> object. <a href="http://bioconductor.org/packages/treeio">treeio</a> provides several accessor
methods to extract tree structure, features/attributes that stored in the object
and their corresponding values.</p>
<p>The <code>get.tree</code> or <code>as.phylo</code> methods can convert the <code>treedata</code> object to
<code>phylo</code> object which is the fundamental tree object in the R community and
many packages work with <code>phylo</code> object.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="co"># or get.tree</span>
<span class="kw">as.phylo</span>(beast_tree)</code></pre>
<pre><code>## 
## Phylogenetic tree with 76 tips and 75 internal nodes.
## 
## Tip labels:
##  A/Hokkaido/30-1-a/2013, A/New_York/334/2004, A/New_York/463/2005, A/New_York/452/1999, A/New_York/238/2005, A/New_York/523/1998, ...
## 
## Rooted; includes branch lengths.</code></pre>
<p>The <code>get.fields</code> method return a vector of features/attributes that stored in
the object and associated with the phylogeny.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">get.fields</span>(beast_tree)</code></pre>
<pre><code>##  [1] &quot;height&quot;          &quot;height_0.95_HPD&quot; &quot;height_median&quot;  
##  [4] &quot;height_range&quot;    &quot;length&quot;          &quot;length_0.95_HPD&quot;
##  [7] &quot;length_median&quot;   &quot;length_range&quot;    &quot;posterior&quot;      
## [10] &quot;rate&quot;            &quot;rate_0.95_HPD&quot;   &quot;rate_median&quot;    
## [13] &quot;rate_range&quot;</code></pre>
<p>The <code>get.data</code> method return a tibble of all the associated data.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">get.data</span>(beast_tree)</code></pre>
<pre><code>## # A tibble: 151 x 14
##    height height_0.95_HPD height_median height_range length
##     &lt;dbl&gt; &lt;list&gt;                  &lt;dbl&gt; &lt;list&gt;        &lt;dbl&gt;
##  1   19   &lt;dbl [2]&gt;                19   &lt;dbl [2]&gt;     2.34 
##  2   17   &lt;dbl [2]&gt;                17   &lt;dbl [2]&gt;     1.18 
##  3   14   &lt;dbl [2]&gt;                14   &lt;dbl [2]&gt;     0.966
##  4   12   &lt;dbl [2]&gt;                12   &lt;dbl [2]&gt;     1.87 
##  5    9   &lt;dbl [2]&gt;                 9   &lt;dbl [2]&gt;     2.93 
##  6   10   &lt;dbl [2]&gt;                10   &lt;dbl [2]&gt;     0.827
##  7   10   &lt;dbl [2]&gt;                10   &lt;dbl [2]&gt;     0.834
##  8   10.8 &lt;dbl [2]&gt;                10.8 &lt;dbl [2]&gt;     0.233
##  9    9   &lt;dbl [2]&gt;                 9   &lt;dbl [2]&gt;     1.28 
## 10    9   &lt;dbl [2]&gt;                 9   &lt;dbl [2]&gt;     0.414
## # ... with 141 more rows, and 9 more variables:
## #   length_0.95_HPD &lt;list&gt;, length_median &lt;dbl&gt;,
## #   length_range &lt;list&gt;, posterior &lt;dbl&gt;, rate &lt;dbl&gt;,
## #   rate_0.95_HPD &lt;list&gt;, rate_median &lt;dbl&gt;,
## #   rate_range &lt;list&gt;, node &lt;int&gt;</code></pre>
<p>If users are only interesting a subset of the features/attributes return by
<code>get.fields</code>, they can extract the information from the output of <code>get.data</code> or
directly subset the data by <code>[</code> or <code>[[</code>.</p>
<pre class="sourceCode r"><code class="sourceCode r">beast_tree[, <span class="kw">c</span>(<span class="st">&quot;node&quot;</span>, <span class="st">&quot;height&quot;</span>)]</code></pre>
<pre><code>## # A tibble: 151 x 2
##     node height
##    &lt;int&gt;  &lt;dbl&gt;
##  1    10   19  
##  2     9   17  
##  3    36   14  
##  4    31   12  
##  5    29    9  
##  6    28   10  
##  7    39   10  
##  8    90   10.8
##  9    16    9  
## 10     2    9  
## # ... with 141 more rows</code></pre>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(beast_tree[[<span class="st">&quot;height_median&quot;</span>]])</code></pre>
<pre><code>## [1] 19 17 14 12  9 10</code></pre>
</div>
<div id="manipulating-tree-data-using-tidytree" class="section level2">
<h2><span class="header-section-number">1.6</span> Manipulating tree data using <em>tidytree</em></h2>
<p>All the tree data parsed/merged
by <a href="https://bioconductor.org/packages/treeio/">treeio</a> can be converted to tidy
data frame using the <a href="https://cran.r-project.org/package=tidytree">tidytree</a>
package. The <a href="https://cran.r-project.org/package=tidytree">tidytree</a> package
provides tidy interfaces to manipulate tree with associated data. For instances,
external data can be linked to phylogeny or evolutionary data obtained from
different sources can be merged using tidyverse verbs. After the tree data was
manipulated, it can be converted back to <code>treedata</code> object and <a href="Exporter.html">exported to a
single tree file</a>, further analyzed in R or visualized using <a href="https://bioconductor.org/packages/ggtree/">ggtree</a> <span class="citation">(Yu et al. <a href="#ref-yu_ggtree:_2017">2017</a>)</span>.</p>
<p>For more details, please refer to the <a href="https://cran.r-project.org/web/packages/tidytree/vignettes/tidytree.html">tidytree package vignette</a>.</p>
</div>
<div id="visualizing-tree-data-with-ggtree" class="section level2">
<h2><span class="header-section-number">1.7</span> Visualizing tree data with <em>ggtree</em></h2>
<p><a href="https://bioconductor.org/packages/treeio/">treeio</a> is seamlessly integrated
into the <a href="https://bioconductor.org/packages/ggtree/">ggtree</a> <span class="citation">(Yu et al. <a href="#ref-yu_ggtree:_2017">2017</a>)</span>
package and all the information either directly imported or linking from
external sources can be used to visualize and annotate the tree.</p>

</div>
</div>
<h3>References</h3>
<div id="refs" class="references">
<div id="ref-yang_paml_2007">
<p>Yang, Ziheng. 2007. âPAML 4: Phylogenetic Analysis by Maximum Likelihood.â <em>Molecular Biology and Evolution</em> 24 (8): 1586â91. <a href="https://doi.org/10.1093/molbev/msm088">https://doi.org/10.1093/molbev/msm088</a>.</p>
</div>
<div id="ref-bouckaert_beast_2014">
<p>Bouckaert, Remco, Joseph Heled, Denise KÃ¼hnert, Tim Vaughan, Chieh-Hsi Wu, Dong Xie, Marc A. Suchard, Andrew Rambaut, and Alexei J. Drummond. 2014. âBEAST 2: A Software Platform for Bayesian Evolutionary Analysis.â <em>PLoS Comput Biol</em> 10 (4): e1003537. <a href="https://doi.org/10.1371/journal.pcbi.1003537">https://doi.org/10.1371/journal.pcbi.1003537</a>.</p>
</div>
<div id="ref-berger_EPA_2011">
<p>Berger, Simon A., Denis Krompass, and Alexandros Stamatakis. 2011. âPerformance, Accuracy, and Web Server for Evolutionary Placement of Short Sequence Reads Under Maximum Likelihood.â <em>Systematic Biology</em> 60 (3): 291â302. <a href="https://doi.org/10.1093/sysbio/syr010">https://doi.org/10.1093/sysbio/syr010</a>.</p>
</div>
<div id="ref-matsen_pplacer_2010">
<p>Matsen, Frederick A, Robin B Kodner, and E Virginia Armbrust. 2010. âPplacer: Linear Time Maximum-Likelihood and Bayesian Phylogenetic Placement of Sequences onto a Fixed Reference Tree.â <em>BMC Bioinformatics</em> 11 (1): 538. <a href="https://doi.org/10.1186/1471-2105-11-538">https://doi.org/10.1186/1471-2105-11-538</a>.</p>
</div>
<div id="ref-yu_ggtree:_2017">
<p>Yu, Guangchuang, David K. Smith, Huachen Zhu, Yi Guan, and Tommy Tsan-Yuk Lam. 2017. âGgtree: An R Package for Visualization and Annotation of Phylogenetic Trees with Their Covariates and Other Associated Data.â <em>Methods in Ecology and Evolution</em> 8 (1): 28â36. <a href="https://doi.org/10.1111/2041-210X.12628">https://doi.org/10.1111/2041-210X.12628</a>.</p>
</div>
<div id="ref-sanderson_r8s:_2003">
<p>Sanderson, Michael J. 2003. âR8s: Inferring Absolute Rates of Molecular Evolution and Divergence Times in the Absence of a Molecular Clock.â <em>Bioinformatics</em> 19 (2): 301â2. <a href="https://doi.org/10.1093/bioinformatics/19.2.301">https://doi.org/10.1093/bioinformatics/19.2.301</a>.</p>
</div>
<div id="ref-boussau_genome-scale_2013">
<p>Boussau, Bastien, Gergely J. SzÃ¶llÅsi, Laurent Duret, Manolo Gouy, Eric Tannier, and Vincent Daubin. 2013. âGenome-Scale Coestimation of Species and Gene Trees.â <em>Genome Research</em> 23 (2): 323â30. <a href="https://doi.org/10.1101/gr.141978.112">https://doi.org/10.1101/gr.141978.112</a>.</p>
</div>
<div id="ref-hohna_probabilistic_2014">
<p>HÃ¶hna, Sebastian, Tracy A. Heath, Bastien Boussau, Michael J. Landis, Fredrik Ronquist, and John P. Huelsenbeck. 2014. âProbabilistic Graphical Model Representation in Phylogenetics.â <em>Systematic Biology</em> 63 (5): 753â71. <a href="https://doi.org/10.1093/sysbio/syu039">https://doi.org/10.1093/sysbio/syu039</a>.</p>
</div>
<div id="ref-liang_expansion_2014">
<p>Liang, Huyi, Tommy Tsan-Yuk Lam, Xiaohui Fan, Xinchun Chen, Yu Zeng, Ji Zhou, Lian Duan, et al. 2014. âExpansion of Genotypic Diversity and Establishment of 2009 H1N1 Pandemic-Origin Internal Genes in Pigs in China.â <em>Journal of Virology</em>, July, JVI.01327â14. <a href="https://doi.org/10.1128/JVI.01327-14">https://doi.org/10.1128/JVI.01327-14</a>.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="manipulating-tree-with-data.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"download": ["treedata.pdf"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
